---
title: "CD4_SCS_CCA_V2_5"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)

# pallet dark2 colors
### "#1B9E77", "#D95F02", "#7570B3", "#E7298A", "#66A61E", "#E6AB02"
```

```{r cpmbined CCA on all 3 rounds of data M1, M2, M3, M4, M6}
## this is the final analysis; the data were analyzed using cell ranger V2 and Seurat V2
library(Seurat)
packageVersion('Seurat') # 2.3.3
packageVersion('monocle') # 2.14

# Enter commands in R (or R studio, if installed)
# Install the devtools package from Hadley Wickham
#install.packages('devtools')
# Replace '2.3.0' with your desired version
#devtools::install_version(package = 'Seurat', version = package_version('2.3.3'))

#Sys.setenv(TAR= "/bin/tar")

library(Seurat)

#remove.packages('Seurat')
library(tidyverse)
library(usethis)
library(devtools)
#library(GGally)
library(scales)
library(monocle)
#install.packages("VennDiagram")
#library(VennDiagram)
library(pheatmap)
#install.packages("ggupset")
library(ggupset)



library(ggplot2)
#library(reticulate)
#library(cowplot)
#library(viridis)
library(Biostrings)
library(ggpubr)
library(ggbuildr)
#install.packages("seqinr")
library(seqinr)
library(pheatmap)
library(Startrac)

setwd("/media/runa/Elements/Jem_revision/Seurat/")
getwd()

```

```{r}
# load and set objects
rm(list=ls())

CD4_M1 <- Read10X(data.dir = "/media/runa/Elements/Jem_revision/Seurat/GP66_M1_R1_V2/")
CD4_M1 <- CreateSeuratObject (raw.data = CD4_M1, project = "CD4_M1")
CD4_M1@meta.data$mice <- "M1"


CD4_M2 <- Read10X(data.dir = "/media/runa/Elements/Jem_revision/Seurat/GP66_M2_R1_V2/")
CD4_M2 <- CreateSeuratObject(raw.data = CD4_M2, project = "CD4_M2")
CD4_M2@meta.data$mice <- "M2"


CD4_M3 <- Read10X(data.dir = "/media/runa/Elements/Jem_revision/Seurat/GP66_M1_R2_V2/filtered_gene_bc_matrices/mm10/")
CD4_M3 <- CreateSeuratObject(raw.data = CD4_M3, project = "CD4_M3")
CD4_M3@meta.data$mice <- "M3"


CD4_M4 <- Read10X(data.dir = "/media/runa/Elements/Jem_revision/Seurat/GP66_M2_R2_V2/filtered_gene_bc_matrices/mm10/")
CD4_M4 <- CreateSeuratObject(raw.data = CD4_M4, project = "CD4_M4")
CD4_M4@meta.data$mice <- "M4"


#CD4_M5 <- Read10X(data.dir = "/media/runa/Elements/Jem_revision/Seurat/GP66_M1_R3_V2/filtered_gene_bc_matrices/mm10/")
#CD4_M5 <- CreateSeuratObject (raw.data = CD4_M5, project = "CD4_M5")
#CD4_M5@meta.data$mice <- "M5"


CD4_M6 <- Read10X(data.dir = "/media/runa/Elements/Jem_revision/Seurat/GP66_M2_R3_V2/filtered_gene_bc_matrices/mm10/")
CD4_M6 <- CreateSeuratObject(raw.data = CD4_M6, project = "CD4_M6")
CD4_M6@meta.data$mice <- "M6"


# Rename cells by adding a prefix to prevent duplicates when integrating data sets, does not work in Seurat V2

CD4_M1 <- RenameCells(CD4_M1, add.cell.id = "CD4_M1")
CD4_M2 <- RenameCells(CD4_M2, add.cell.id = "CD4_M2")
CD4_M3 <- RenameCells(CD4_M3, add.cell.id = "CD4_M3")
CD4_M4 <- RenameCells(CD4_M4, add.cell.id = "CD4_M4")
#CD4_M5 <- RenameCells(CD4_M5, add.cell.id = "CD4_M5")
CD4_M6 <- RenameCells(CD4_M6, add.cell.id = "CD4_M6")

```


```{r}
# Calculate percentage of mitochondrial genes, store in percent.mito
# Store percent.mito data for each gene in new column
mito.genes <- grep("mt-", rownames(CD4_M1@data), value = T)
percent.mito <- Matrix::colSums(CD4_M1@raw.data[mito.genes, ]) / Matrix::colSums(CD4_M1@raw.data)
CD4_M1 <- AddMetaData(CD4_M1, percent.mito, "percent.mito")
rm(mito.genes); rm(percent.mito)

mito.genes <- grep("mt-", rownames(CD4_M2@data), value = T)
percent.mito <- Matrix::colSums(CD4_M2@raw.data[mito.genes, ]) / Matrix::colSums(CD4_M2@raw.data)
CD4_M2 <- AddMetaData(CD4_M2, percent.mito, "percent.mito")
rm(mito.genes); rm(percent.mito)

mito.genes <- grep("mt-", rownames(CD4_M3@data), value = T)
percent.mito <- Matrix::colSums(CD4_M3@raw.data[mito.genes, ]) / Matrix::colSums(CD4_M3@raw.data)
CD4_M3 <- AddMetaData(CD4_M3, percent.mito, "percent.mito")
rm(mito.genes); rm(percent.mito)

mito.genes <- grep("mt-", rownames(CD4_M4@data), value = T)
percent.mito <- Matrix::colSums(CD4_M4@raw.data[mito.genes, ]) / Matrix::colSums(CD4_M4@raw.data)
CD4_M4 <- AddMetaData(CD4_M4, percent.mito, "percent.mito")
rm(mito.genes); rm(percent.mito)

#mito.genes <- grep("mt-", rownames(CD4_M5@data), value = T)
#percent.mito <- Matrix::colSums(CD4_M5@raw.data[mito.genes, ]) / Matrix::colSums(CD4_M5@raw.data)
#CD4_M5 <- AddMetaData(CD4_M5, percent.mito, "percent.mito")
#rm(mito.genes); rm(percent.mito)

mito.genes <- grep("mt-", rownames(CD4_M6@data), value = T)
percent.mito <- Matrix::colSums(CD4_M6@raw.data[mito.genes, ]) / Matrix::colSums(CD4_M6@raw.data)
CD4_M6 <- AddMetaData(CD4_M6, percent.mito, "percent.mito")
rm(mito.genes); rm(percent.mito)

```



```{r}
png("summary_R1_V2/01_wt1_vln_pre_filt.png", width = 8, height = 6, units = "in", res = 300)
VlnPlot(CD4_M1, c("nGene", "nUMI", "percent.mito"), nCol = 3, point.size.use = 0.2, size.x.use = 0)
dev.off()

CD4_M1 <- FilterCells(CD4_M1, subset.names = c("nGene","nUMI","percent.mito"),
                       low.thresholds = c(800,-Inf,-Inf), 
                       high.thresholds = c(6000,40000,0.1))

table(CD4_M1@meta.data$orig.ident) # 2228

png("summary_R1_V2/01_wt2_vln_pre_filt.png", width = 8, height = 6, units = "in", res = 300)
VlnPlot(CD4_M2, c("nGene", "nUMI", "percent.mito"), nCol = 3, point.size.use = 0.2, size.x.use = 0)
dev.off()

CD4_M2 <- FilterCells(CD4_M2, subset.names = c("nGene","nUMI","percent.mito"),
                       low.thresholds = c(800,-Inf,-Inf), 
                       high.thresholds = c(6000,40000,0.1))

table(CD4_M2@meta.data$orig.ident) #1949

png("summary_R1_V2/01_wt1_vln_pre_filt.png", width = 8, height = 6, units = "in", res = 300)
VlnPlot(CD4_M3, c("nGene", "nUMI", "percent.mito"), nCol = 3, point.size.use = 0.2, size.x.use = 0)
dev.off()

CD4_M3 <- FilterCells(CD4_M3, subset.names = c("nGene","nUMI","percent.mito"),
                       low.thresholds = c(800,-Inf,-Inf), 
                       high.thresholds = c(5000,30000,0.1)) # prev 40000

table(CD4_M3@meta.data$orig.ident) #3924

png("summary_R1_V2/01_wt1_vln_pre_filt.png", width = 8, height = 6, units = "in", res = 300)
VlnPlot(CD4_M4, c("nGene", "nUMI", "percent.mito"), nCol = 3, point.size.use = 0.2, size.x.use = 0)
dev.off()

CD4_M4 <- FilterCells(CD4_M4, subset.names = c("nGene","nUMI","percent.mito"),
                       low.thresholds = c(800,-Inf,-Inf), 
                       high.thresholds = c(5000,35000,0.1)) # prev 40000

table(CD4_M4@meta.data$orig.ident) # 5355

png("summary_R1_V2/01_wt1_vln_pre_filt.png", width = 8, height = 6, units = "in", res = 300)
#VlnPlot(CD4_M5, c("nGene", "nUMI", "percent.mito"), nCol = 3, point.size.use = 0.2, size.x.use = 0)
dev.off()

#CD4_M5 <- FilterCells(CD4_M5, subset.names = c("nGene","nUMI","percent.mito"),
                    #   low.thresholds = c(800,-Inf,-Inf), 
                    #   high.thresholds = c(5000,25000,0.1)) # prev 40000


#table(CD4_M5@meta.data$orig.ident) # 5978


png("summary_R1_V2/01_wt1_vln_pre_filt.png", width = 8, height = 6, units = "in", res = 300)
VlnPlot(CD4_M6, c("nGene", "nUMI", "percent.mito"), nCol = 3, point.size.use = 0.2, size.x.use = 0)
dev.off()

CD4_M6 <- FilterCells(CD4_M6, subset.names = c("nGene","nUMI","percent.mito"),
                       low.thresholds = c(800,-Inf,-Inf), 
                       high.thresholds = c(5000,32000,0.1)) # prev 40000

table(CD4_M6@meta.data$orig.ident) # 4251
```


```{r}
## Normalize and Scale data for each mice b4 CCA

CD4_M1 <- NormalizeData(CD4_M1, normalization.method = "LogNormalize", scale.factor = 10000)

# Assign cell cycle scores to all cells
cc.genes <- readLines(con = "regev_lab_cell_cycle_genes.txt")

cc.genes <- paste(substring(cc.genes, 1, 1), tolower(substring(cc.genes, 2)), sep = "", collapse = " ") %>% strsplit(split = " ") %>% unlist()
s.genes <- cc.genes[1:43]
g2m.genes <- cc.genes[44:97]
s.genes
g2m.genes
CD4_M1 <- CellCycleScoring(CD4_M1, s.genes = s.genes, g2m.genes = g2m.genes, 
                         set.ident = TRUE)


# Scale data; regressing UMI means diff gene exp will not be based on copies of UMI
CD4_M1 <- ScaleData(CD4_M1, vars.to.regress = c("nUMI","percent.mito", "S.Score", "G2M.Score"), display.progress = T)

#Regressing out cell cycle score duting data scaling

#CD4_M1 <- ScaleData(object = CD4_M1, vars.to.regress = c("S.Score", "G2M.Score"), 
                  #  display.progress = TRUE)


## M2

CD4_M2 <- NormalizeData(CD4_M2, normalization.method = "LogNormalize", scale.factor = 10000)

# Assign cell cycle scores to all cells
CD4_M2 <- CellCycleScoring(CD4_M2, s.genes = s.genes, g2m.genes = g2m.genes, 
                         set.ident = TRUE)

# Scale data; regressing UMI means diff gene exp will not be based on copies of UMI
#CD4_M2 <- ScaleData(CD4_M2, vars.to.regress = c("nUMI","percent.mito"))

#Regressing out cell cycle score duting data scaling

CD4_M2 <- ScaleData(object = CD4_M2, vars.to.regress = c("nUMI","percent.mito", "S.Score", "G2M.Score"), 
                    display.progress = TRUE)

## M3

CD4_M3 <- NormalizeData(CD4_M3, normalization.method = "LogNormalize", scale.factor = 10000)

# Assign cell cycle scores to all cells
CD4_M3 <- CellCycleScoring(CD4_M3, s.genes = s.genes, g2m.genes = g2m.genes, 
                         set.ident = TRUE)

# Scale data; regressing UMI means diff gene exp will not be based on copies of UMI
#CD4_M3 <- ScaleData(CD4_M3, vars.to.regress = c("nUMI","percent.mito"))

#Regressing out cell cycle score duting data scaling

CD4_M3 <- ScaleData(object = CD4_M3, vars.to.regress = c("nUMI","percent.mito", "S.Score", "G2M.Score"), 
                    display.progress = TRUE)

## M4

CD4_M4 <- NormalizeData(CD4_M4, normalization.method = "LogNormalize", scale.factor = 10000)

# Assign cell cycle scores to all cells
CD4_M4 <- CellCycleScoring(CD4_M4, s.genes = s.genes, g2m.genes = g2m.genes, 
                         set.ident = TRUE)

# Scale data; regressing UMI means diff gene exp will not be based on copies of UMI
#CD4_M4 <- ScaleData(CD4_M4, vars.to.regress = c("nUMI","percent.mito"))

#Regressing out cell cycle score duting data scaling

CD4_M4 <- ScaleData(object = CD4_M4, vars.to.regress = c("nUMI","percent.mito", "S.Score", "G2M.Score"), 
                    display.progress = TRUE)

## M5

#CD4_M5 <- NormalizeData(CD4_M5, normalization.method = "LogNormalize", scale.factor = 10000)

# Assign cell cycle scores to all cells
#CD4_M5 <- CellCycleScoring(CD4_M5, s.genes = s.genes, g2m.genes = g2m.genes, 
 #                        set.ident = TRUE)

# Scale data; regressing UMI means diff gene exp will not be based on copies of UMI
#CD4_M5 <- ScaleData(CD4_M5, vars.to.regress = c("nUMI","percent.mito"))

#Regressing out cell cycle score duting data scaling

#CD4_M5 <- ScaleData(object = CD4_M5, vars.to.regress = c("S.Score", "G2M.Score"), 
 #                   display.progress = TRUE)



## M6

CD4_M6 <- NormalizeData(CD4_M6, normalization.method = "LogNormalize", scale.factor = 10000)

# Assign cell cycle scores to all cells
CD4_M6 <- CellCycleScoring(CD4_M6, s.genes = s.genes, g2m.genes = g2m.genes, 
                         set.ident = TRUE)

# Scale data; regressing UMI means diff gene exp will not be based on copies of UMI
#CD4_M6 <- ScaleData(CD4_M6, vars.to.regress = c("nUMI","percent.mito"))

#Regressing out cell cycle score duting data scaling

CD4_M6 <- ScaleData(object = CD4_M6, vars.to.regress = c("nUMI","percent.mito", "S.Score", "G2M.Score"), 
                    display.progress = TRUE)


```


```{r CCA}
# Gene selection for input to CCA
CD4_M1 <- FindVariableGenes(CD4_M1, do.plot = F, top.genes = 2000)
CD4_M2 <- FindVariableGenes(CD4_M2, do.plot = F, top.genes = 2000)
CD4_M3 <- FindVariableGenes(CD4_M3, do.plot = F, top.genes = 2000)
CD4_M4 <- FindVariableGenes(CD4_M4, do.plot = F, top.genes = 2000)
#CD4_M5 <- FindVariableGenes(CD4_M5, do.plot = F, top.genes = 2000)
CD4_M6 <- FindVariableGenes(CD4_M6, do.plot = F, top.genes = 2000)



g.1 <- head(rownames(CD4_M1@hvg.info), 2000)
g.2 <- head(rownames(CD4_M2@hvg.info), 2000)
g.3 <- head(rownames(CD4_M3@hvg.info), 2000)
g.4 <- head(rownames(CD4_M4@hvg.info), 2000)
#g.5 <- head(rownames(CD4_M5@hvg.info), 2000)
g.6 <- head(rownames(CD4_M6@hvg.info), 2000)


genes.use <- unique(c(g.1, g.2, g.3, g.4, g.6))
genes.use <- intersect(genes.use, rownames(CD4_M1@scale.data))
genes.use <- intersect(genes.use, rownames(CD4_M2@scale.data))
genes.use <- intersect(genes.use, rownames(CD4_M3@scale.data))
genes.use <- intersect(genes.use, rownames(CD4_M4@scale.data))
#genes.use <- intersect(genes.use, rownames(CD4_M5@scale.data))
genes.use <- intersect(genes.use, rownames(CD4_M6@scale.data))

length(genes.use)

mice.list <- list(CD4_M1, CD4_M2, CD4_M3, CD4_M4, CD4_M6)



### filter out TCR and BCR genes from genes.use list

## loading a previous CCA object to do this

immune.combined <- readRDS("summary_6M_CCA_V2/immune_combined_6M_ccregress.RDS")


# Regress TCR genes from integrated assay to remove their influence on PCA/UMAP
genes.use <- setdiff(genes.use,
              grep(
    paste(c("Trav", "Traj", "Trac", "Trbv", "Trbd", "Trbj", "Trbc"), collapse = "|"),
    rownames(immune.combined@data),
    value = T)) 

genes.use <- setdiff(genes.use,
              grep(
    paste(c("Igh", "Igk", "Igl"), collapse = "|"),
    rownames(immune.combined@data),
    value = T)) 

## reduced from 4413 to 4180

rm(immune.combined)


```


```{r}
#immune.combined <- RunCCA(CD4_M1, CD4_M2, CD4_M3, CD4_M4, CD4_M6, genes.use = genes.use, num.cc = 30)

immune.combined <- RunMultiCCA(object.list = mice.list ,genes.use = genes.use, num.cc = 30)

saveRDS(immune.combined, file = "summary_5M_CCA_V2/immune_combined_5M.RDS")

length(immune.combined@var.genes) # 3083


```



```{r}
## Run CCA again

#immune.combined <- RunMultiCCA(object.list = mice.list, genes.use = genes.use, num.cc = 30)

## regress out CC genes

#immune.combined <- CellCycleScoring(immune.combined, s.genes = s.genes, g2m.genes = g2m.genes, 
 #                        set.ident = TRUE)

#immune.combined <- ScaleData(object = immune.combined, vars.to.regress = c("S.Score", "G2M.Score"), 
 #                   display.progress = TRUE)

#saveRDS(immune.combined, file = "summary_6M_CCA_V2/immune_combined_6M_ccregress.RDS")


```



```{r}
# visualize results of CCA plot CC1 versus CC2 and look at a violin plot
p1 <- DimPlot(object = immune.combined, reduction.use = "cca", group.by = "mice", 
    pt.size = 0.5, do.return = TRUE)
p2 <- VlnPlot(object = immune.combined, features.plot = "Cd4", group.by = "mice", 
    do.return = TRUE)
plot_grid(p1, p2)
```


```{r}
PrintDim(object = immune.combined, reduction.type = "cca", dims.print = 1:10, 
    genes.print = 10)

```


```{r decide on number of CCs to use}
p3 <- MetageneBicorPlot(immune.combined, grouping.var = "mice", dims.eval = 1:30, 
    display.progress = FALSE) # this step took quite a long time ~ 15 to 20 min

```


```{r}
DimHeatmap(object = immune.combined, reduction.type = "cca", cells.use = 500, 
    dim.use = 1:9, do.balanced = TRUE)
DimHeatmap(object = immune.combined, reduction.type = "cca", cells.use = 500, 
    dim.use = 10:19, do.balanced = TRUE)

```

```{r}
#Now we align the CCA subspaces, which returns a new dimensional reduction called cca.aligned that can be used for downstream analyses such as clustering.
immune.combined <- AlignSubspace(immune.combined, reduction.type = "cca", grouping.var = "mice", 
    dims.align = 1:20)

saveRDS(immune.combined, file = "summary_5M_CCA_V2/immune_combined_5M.RDS")

```



```{r}
# t-SNE and Clustering
immune.combined <- RunTSNE(immune.combined, reduction.use = "cca.aligned", dims.use = 1:20, 
    do.fast = T)

immune.combined <- FindClusters(immune.combined, reduction.type = "cca.aligned", 
    resolution = 0.5, dims.use = 1:20, force.recalc = TRUE)

```




```{r}
# Visualization
p1 <- TSNEPlot(immune.combined, do.return = T, pt.size = 0.3, group.by = "mice")
p2 <- TSNEPlot(immune.combined, do.label = T, do.return = T, pt.size = 0.3)
plot_grid(p1, p2)
```


```{r}
table(immune.combined@meta.data$mice,immune.combined@ident) 

#        0    1    2    3    4    5    6    7    8    9   10   11
#  M1  697  304  488  262  120  167   71   36   34   38   39   33
#  M2  548  316  356  202  151  125   36   33   53   48   42   39
#  M3  911  526  720  594  263  229  173   91   65  117  108  127
#  M4 1718  853  585  655  299  479  181  106  147  126  111   95
#  M6 1062  771  496  582  369  179  111  200  161  120  118   82
  

table(immune.combined@ident)

#   0    1    2    3    4    5    6    7    8    9   10   11 
#4936 2770 2645 2295 1202 1179  572  466  460  449  418  376 
```


```{r}
TSNEPlot(immune.combined, do.label = T)
FeaturePlot(immune.combined, c("Cxcr6","Cxcr5","Ccr7","Foxp3","Lag3","Ly6c2", "Irf8", "Ascl2", "Cxcl10"), pt.size = 0.5,
            nCol = 3, cols.use = c("grey", "red"))

VlnPlot(immune.combined, c("Cxcr6","Cxcr5","Ccr7","Foxp3","Lag3","Ly6c2", "Irf8", "Ascl2", "Cxcl10"))
VlnPlot(immune.combined, c("S.Score", "G2M.Score"))
DotPlot(immune.combined, c("Cxcr6","Cxcr5","Ccr7","Foxp3","Lag3","Ly6c2", "Irf8", "Ascl2", "Cxcl10", "Il21", "Batf"))

```


```{r}

## look at the markers for all lsusters

markers <- FindAllMarkers(immune.combined, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.15)
#top40 <- markers %>% group_by(cluster) %>% top_n(n = 40, wt = avg_logFC)
top20 <- markers %>% group_by(cluster) %>% top_n(n = 20, wt = avg_logFC)

colnames(immune.combined@meta.data)

## 0 Tcmp; 1,2, Th1; 3,4 TFH; 5,10 high CC; 9 Treg; 6 Isg CD4; 7 Tcmp like; 8 cont; 11 slamf7 CD4

## 0 Tcmp, 1,5 TFH, 2,3 Th1, 4,10 CC gene high, 6 Isg hi CD4, 7 cont, 8 slamf7hi CD4, 9 Treg, 11 Tcmp like 

```

```{r}

## check module score

##Th1
Geneset_Th1 <- readLines("Genesets/Th1_gene.txt", warn = FALSE) # Remove first two lines, which are headers
Geneset_Th1 <- Geneset_Th1[which(Geneset_Th1 %in% rownames(immune.combined@data))] # Restrict to genes present in our data
Geneset_Th1 <- list(Geneset_Th1)
immune.combined <- AddModuleScore(immune.combined, genes.list = Geneset_Th1, enrich.name = "Geneset_Th1")

## in the meta data it adds "1" by the end of the name of the Geneset_apoptosis

#png("summary/18_CD4_Acute_chronic_Th1_genescore.png", width = 10, height = 8, units = "in", res = 300)
VlnPlot(immune.combined, c("Geneset_Th11"), point.size.use = 0)
#dev.off()


```

```{r}
## TFH
Geneset_Tfh <- readLines("Genesets/Tfh_gene.txt", warn = FALSE) # Remove first two lines, which are headers
Geneset_Tfh <- Geneset_Tfh[which(Geneset_Tfh %in% rownames(immune.combined@data))] # Restrict to genes present in our data
Geneset_Tfh <- list(Geneset_Tfh)
immune.combined <- AddModuleScore(immune.combined, genes.list = Geneset_Tfh, enrich.name = "Geneset_Tfh")

## in the meta data it adds "1" by the end of the name of the Geneset_apoptosis

#png("summary/19_CD4_Acute_chronic_TFH_genescore.png", width = 10, height = 8, units = "in", res = 300)
VlnPlot(immune.combined, c("Geneset_Tfh1"), point.size.use = 0)
#dev.off()
#VlnPlot(Acute_4_mice_edit3_1cell, features = "Geneset_Tfh1", pt.size = 0)
#compare_means(Geneset_Tfh1 ~ seurat_clusters, data = (Acute_6_mice_edit4@meta.data %>% select(Geneset_Tfh1, seurat_clusters)))

```


```{r}

### remove and subset based on CC score

saveRDS(immune.combined, file = "summary_5M_CCA_V2/immune_combined_5M.RDS")
immune.combined <- readRDS("summary_5M_CCA_V2/immune_combined_5M.RDS")

# remove clusters 4, 10, 6, 7, 8
colnames(immune.combined@meta.data)
immune.combined_edit1 <-SubsetData(immune.combined, ident.remove = c(4,6,7,8,10))
rm(immune.combined)

table(immune.combined_edit1@meta.data$orig.ident) # 14650

## re do clustering 

immune.combined_edit1 <- ScaleData(immune.combined_edit1)
immune.combined_edit1 <- FindVariableGenes(immune.combined_edit1)

DimHeatmap(object = immune.combined_edit1, reduction.type = "cca", cells.use = 500, 
    dim.use = 1:9, do.balanced = TRUE)
DimHeatmap(object = immune.combined_edit1, reduction.type = "cca", cells.use = 500, 
    dim.use = 10:19, do.balanced = TRUE)


# 1:20 looks good

#Now we align the CCA subspaces, which returns a new dimensional reduction called cca.aligned that can be used for downstream analyses such as clustering.
#immune.combined_edit1 <- AlignSubspace(immune.combined_edit1, reduction.type = "cca", grouping.var = "mice", 
    #dims.align = 1:20)

# t-SNE and Clustering
immune.combined_edit1 <- RunTSNE(immune.combined_edit1, reduction.use = "cca.aligned", dims.use = 1:20, 
    do.fast = T)

immune.combined_edit1 <- FindClusters(immune.combined_edit1, reduction.type = "cca.aligned", 
    resolution = 0.6, dims.use = 1:20, force.recalc = TRUE)

```



```{r}

# Visualization
p1 <- TSNEPlot(immune.combined_edit1, do.return = T, pt.size = 0.3, group.by = "mice")
p2 <- TSNEPlot(immune.combined_edit1, do.label = T, do.return = T, pt.size = 0.3)
plot_grid(p1, p2)



```


```{r}

TSNEPlot(immune.combined_edit1, do.label = T)
FeaturePlot(immune.combined_edit1, c("Cxcr6","Cxcr5","Ccr7","Foxp3","Lag3","Ly6c2", "Irf8", "Ascl2", "Cxcl10"), pt.size = 0.5,
            nCol = 3, cols.use = c("grey", "red"))

VlnPlot(immune.combined_edit1, c("Cxcr6","Cxcr5","Ccr7","Foxp3","Lag3","Ly6c2", "Irf8", "Ascl2", "Cxcl10"))
VlnPlot(immune.combined_edit1, c("S.Score", "G2M.Score"))
DotPlot(immune.combined_edit1, c("Cxcr6","Cxcr5","Ccr7","Foxp3","Lag3","Ly6c2", "Irf8", "Ascl2", "Cxcl10", "Il21", "Batf"))


```


```{r}

table(immune.combined_edit1@meta.data$mice,immune.combined_edit1@ident)

#        0    1    2    3    4    5    6
#  M1  658  309  496  270  173   38   45
#  M2  523  323  362  195  130   48   53
#  M3  871  537  742  590  237  117  130
#  M4 1632  886  612  655  495  126  105
#  M6 1026  789  499  580  191  120   87

## 0 TCmp, 1,4 TFH, 2,3 Th1, 5 Treg, 6 Tcmp like
## lower CC score S.Score <0.2 and G2M.Score <0.2
```


```{r}
## look at the markers for all lsusters

markers <- FindAllMarkers(immune.combined_edit1, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.15)
#top40 <- markers %>% group_by(cluster) %>% top_n(n = 40, wt = avg_logFC)
top20 <- markers %>% group_by(cluster) %>% top_n(n = 20, wt = avg_logFC)

colnames(immune.combined@meta.data)




```


```{r}

VlnPlot(immune.combined_edit1, c("Geneset_Th11"), point.size.use = 0)
VlnPlot(immune.combined_edit1, c("Geneset_Tfh1"), point.size.use = 0)

```

```{r}
## 0 TCmp, 1,4 TFH, 2,3 Th1, 5 Treg, 6 Tcmp like,; remove 6
## lower CC score S.Score <0.2 and G2M.Score <0.2
rm(immune.combined)

immune.combined_edit2 <-SubsetData(immune.combined_edit1, ident.remove = c(6))

immune.combined_edit2 <-SubsetData(immune.combined_edit2, subset.name = "S.Score", accept.high = 0.2)
immune.combined_edit2 <-SubsetData(immune.combined_edit2, subset.name = "G2M.Score", accept.high = 0.2)

VlnPlot(immune.combined_edit2, c("S.Score", "G2M.Score"))

table(immune.combined_edit2@meta.data$orig.ident) # 13636

## re do clustering 

immune.combined_edit2 <- ScaleData(immune.combined_edit2)
immune.combined_edit2 <- FindVariableGenes(immune.combined_edit2)

DimHeatmap(object = immune.combined_edit2, reduction.type = "cca", cells.use = 500, 
    dim.use = 1:9, do.balanced = TRUE)
DimHeatmap(object = immune.combined_edit2, reduction.type = "cca", cells.use = 500, 
    dim.use = 10:19, do.balanced = TRUE)

## skipping alignsubspace step

# t-SNE and Clustering
immune.combined_edit2 <- RunTSNE(immune.combined_edit2, reduction.use = "cca.aligned", dims.use = 1:15, 
    do.fast = T)

immune.combined_edit2 <- FindClusters(immune.combined_edit2, reduction.type = "cca.aligned", 
    resolution = 0.7, dims.use = 1:15, force.recalc = TRUE)

```



```{r}

# Visualization
p1 <- TSNEPlot(immune.combined_edit2, do.return = T, pt.size = 0.3, group.by = "mice")
p2 <- TSNEPlot(immune.combined_edit2, do.label = T, do.return = T, pt.size = 0.3)
plot_grid(p1, p2)


```

```{r}

TSNEPlot(immune.combined_edit2, do.label = T)
FeaturePlot(immune.combined_edit2, c("Cxcr6","Cxcr5","Ccr7","Foxp3","Lag3","Ly6c2", "Irf8", "Ascl2", "Cxcl10"), pt.size = 0.5,
            nCol = 3, cols.use = c("grey", "red"))

VlnPlot(immune.combined_edit2, c("Cxcr6","Cxcr5","Ccr7","Foxp3","Lag3","Ly6c2", "Irf8", "Ascl2", "Cxcl10"))
VlnPlot(immune.combined_edit2, c("S.Score", "G2M.Score"))
DotPlot(immune.combined_edit2, c("Cxcr6","Cxcr5","Ccr7","Foxp3","Lag3","Ly6c2", "Tbx21", "Ifng", "Irf8", "Ascl2", "Cxcl10", "Il21", "Selplg"), cols.use = c("grey", "red"))

```


```{r}
png(file = "summary_5M_CCA_V2/Th1_Modscore.png", width = 8, height = 6, units = "in", res = 300)
VlnPlot(immune.combined_edit2, c("Geneset_Th11"), point.size.use = 0) + geom_boxplot(notch = T, fill = "white", alpha = 0.3)
dev.off()

png(file = "summary_5M_CCA_V2/TFH_modulescore.png", width = 8, height = 6, units = "in", res = 300)
VlnPlot(immune.combined_edit2, c("Geneset_Tfh1"), point.size.use = 0) + geom_boxplot(notch = T, fill = "white", alpha = 0.3)
dev.off()

png(file = "summary_5M_CCA_V2/Tmem_modscore.png", width = 8, height = 6, units = "in", res = 300)
VlnPlot(immune.combined_edit2, c("Geneset_Tmem1"), point.size.use = 0) + geom_boxplot(notch = T, fill = "white", alpha = 0.3)
dev.off()


VlnPlot(immune.combined_edit2, c("Geneset_TCMP1"), point.size.use = 0) + geom_boxplot(notch = T, fill = "white", alpha = 0.3)

```


```{r}
table(immune.combined_edit2@ident)
#  0    1    2    3    4    5    6 
#3157 2603 2315 2166 1669 1385  341 
```

```{r}

### Tmem
Geneset_Tmem <- readLines("Genesets/Tmem_gene.txt", warn = FALSE) # Remove first two lines, which are headers
Geneset_Tmem <- Geneset_Tmem[which(Geneset_Tmem %in% rownames(immune.combined_edit2@data))] # Restrict to genes present in our data
Geneset_Tmem <- list(Geneset_Tmem)
immune.combined_edit2 <- AddModuleScore(immune.combined_edit2, genes.list = Geneset_Tmem, enrich.name = "Geneset_Tmem")

## in the meta data it adds "1" by the end of the name of the Geneset_apoptosis

#png("summary/21_CD4_Acute_chronic_Tmem_genescore.png", width = 10, height = 8, units = "in", res = 300)
VlnPlot(immune.combined_edit2, c("Geneset_Tmem1"), point.size.use = 0) + geom_boxplot(notch = T, fill = "white", alpha = 0.3)
#dev.off()

```

```{r}

Geneset_TCMP <- readLines("Genesets/Tcmp_gene.txt", warn = FALSE) # Remove first two lines, which are headers
Geneset_TCMP <- Geneset_TCMP[which(Geneset_TCMP %in% rownames(immune.combined_edit2@data))] # Restrict to genes present in our data
Geneset_TCMP <- list(Geneset_TCMP)
immune.combined_edit2 <- AddModuleScore(immune.combined_edit2, genes.list =  Geneset_TCMP, enrich.name = "Geneset_TCMP")

## in the meta data it adds "1" by the end of the name of the Geneset_apoptosis

#png("summary/20_CD4_Acute_Chronic_TCMP_genescore.png", width = 10, height = 8, units = "in", res = 300)
VlnPlot(immune.combined_edit2, c("Geneset_TCMP1"), point.size.use = 0) + geom_boxplot(notch = T, fill = "white", alpha = 0.3)
#dev.off()

```


```{r}
table(immune.combined_edit2@ident)
#  0    1    2    3    4    5    6 
#3157 2603 2315 2166 1669 1385  341

```


```{r}

### check gene on immune.combined_edit4

# rename clusters

current.cluster.ids <- c(0,1,2,3,4,5,6)
new.cluster.ids <- c("0_Tcmp", "1_Ly6cHi_Th1", "2_Pre_TFH", "3_Lag3hi_Th1", 
    "4_TFH1", "5_GC_TFH2", "6_Treg")

immune.combined_edit2@ident <- plyr::mapvalues(x = immune.combined_edit2@ident, from = current.cluster.ids, to = new.cluster.ids) ## does not work, will look later

png("summary_5M_CCA_V2/immComB_edit2_changesident_name.png", width = 8, height = 4, units = "in", res = 300)
TSNEPlot(immune.combined_edit2, do.label = T, label.size = 5, pt.size = 0.3)
dev.off()

png("summary_5M_CCA_V2/immComB_edit2_bymouse.png", width = 8, height = 4, units = "in", res = 300)
TSNEPlot(immune.combined_edit2, do.label = F, group.by = "mice")
dev.off()

FeaturePlot(immune.combined_edit2, c("Cxcr6","Cxcr5","Ccr7","Foxp3","Lag3","Ly6c2", "Irf8", "Ascl2", "Cxcl10"), pt.size = 0.5,
            nCol = 3, cols.use = c("grey", "red"))

VlnPlot(immune.combined_edit2, c("Cxcr6","Cxcr5","Ccr7","Foxp3","Lag3","Ly6c2", "Irf8", "Ascl2", "Cxcl10"))
VlnPlot(immune.combined_edit2, c("S.Score", "G2M.Score"))

png("summary_5M_CCA_V2/immComB_edit2_dotplot.png", width = 8, height = 4, units = "in", res = 300)
DotPlot(immune.combined_edit2, c("Cxcr6","Cxcr5","Ccr7","Foxp3","Lag3","Ly6c2", "Tbx21", "Ifng", "Irf8", "Ascl2", "Cxcl10", "Il21", "Selplg"), cols.use = c("grey", "red"))
dev.off()

```

```{r}
immune.combined_edit2@meta.data$cluster_ident <- immune.combined_edit2@ident


png("summary_5M_CCA_V2/immComB_edit2_clusterDist.png", width = 8, height = 4, units = "in", res = 300)
ggplot(immune.combined_edit2@meta.data, aes(x=mice, fill=cluster_ident)) + geom_bar(position = "fill") + 
  theme(panel.background = element_blank())
dev.off()
```


```{r}
saveRDS(immune.combined_edit2, file = "summary_5M_CCA_V2/immune.combined_edit2.RDS")

```


```{r add TCR data}

# load TCR data from R1 V2

rm(list=ls())

immune.combined_edit2 <- readRDS("summary_5M_CCA_V2/immune.combined_edit2.RDS")

CD4_M1_vdj <- read.csv("/media/runa/Elements/Jem_revision/Seurat/TCRseq/CD4_WT1_TCR_R1/filtered_contig_annotations.csv", stringsAsFactors = F)
CD4_M2_vdj <- read.csv("/media/runa/Elements/Jem_revision/Seurat/TCRseq/CD4_WT2_TCR_R1/filtered_contig_annotations.csv", stringsAsFactors = F)


CD4_M3_vdj <- read.csv("/media/runa/Elements/Jem_revision/Seurat/TCRseq/CD4_WT1_TCR_R2/filtered_contig_annotations.csv", stringsAsFactors = F)
CD4_M4_vdj <- read.csv("/media/runa/Elements/Jem_revision/Seurat/TCRseq/CD4_WT2_TCR_R2/filtered_contig_annotations.csv", stringsAsFactors = F)


#CD4_M5_vdj <- read.csv("/media/runa/Elements/Jem_revision/Seurat/TCRseq/CD4_WT1_TCR_R3/outs/filtered_contig_annotations.csv", stringsAsFactors = F)
CD4_M6_vdj <- read.csv("/media/runa/Elements/Jem_revision/Seurat/TCRseq/CD4_WT2_TCR_R3/outs/filtered_contig_annotations.csv", stringsAsFactors = F)


# Rename cells within the VDJ data to match the cell names in Seurat
CD4_M1_vdj$barcode <-
  substr(CD4_M1_vdj$barcode, 0, nchar(CD4_M1_vdj$barcode)) %>%
  paste0("CD4_M1_", .)
CD4_M2_vdj$barcode <-
  substr(CD4_M2_vdj$barcode, 0, nchar(CD4_M2_vdj$barcode)) %>%
  paste0("CD4_M2_", .)
CD4_M3_vdj$barcode <-
  substr(CD4_M3_vdj$barcode, 0, nchar(CD4_M3_vdj$barcode)) %>%
  paste0("CD4_M3_", .)
CD4_M4_vdj$barcode <-
  substr(CD4_M4_vdj$barcode, 0, nchar(CD4_M4_vdj$barcode)) %>%
  paste0("CD4_M4_", .)
#CD4_M5_vdj$barcode <-
  #substr(CD4_M5_vdj$barcode, 0, nchar(CD4_M5_vdj$barcode)) %>%
  #paste0("CD4_M5_", .)
CD4_M6_vdj$barcode <-
  substr(CD4_M6_vdj$barcode, 0, nchar(CD4_M6_vdj$barcode)) %>%
  paste0("CD4_M6_", .)

#Rename cells within the VDJ data to match the cell names in Seurat
CD4_M1_vdj$barcode <- substr(CD4_M1_vdj$barcode,  0, nchar(CD4_M1_vdj$barcode) - 2)
CD4_M2_vdj$barcode <- substr(CD4_M2_vdj$barcode,  0, nchar(CD4_M2_vdj$barcode) - 2)

CD4_M3_vdj$barcode <- substr(CD4_M3_vdj$barcode,  0, nchar(CD4_M3_vdj$barcode) - 2)
CD4_M4_vdj$barcode <- substr(CD4_M4_vdj$barcode,  0, nchar(CD4_M4_vdj$barcode) - 2)


#CD4_M5_vdj$barcode <- substr(CD4_M5_vdj$barcode,  0, nchar(CD4_M5_vdj$barcode) - 2)
CD4_M6_vdj$barcode <- substr(CD4_M6_vdj$barcode,  0, nchar(CD4_M6_vdj$barcode) - 2)

# Combine the data frames
Acute_vdj <- rbind(CD4_M1_vdj[, c("barcode", "length", "chain", "v_gene", "d_gene", "j_gene", "c_gene", "full_length", "productive", "cdr3", "cdr3_nt", "reads", "umis")],
                 CD4_M2_vdj[, c("barcode", "length", "chain", "v_gene", "d_gene", "j_gene", "c_gene", "full_length", "productive", "cdr3", "cdr3_nt", "reads", "umis")],
                 CD4_M3_vdj[, c("barcode", "length", "chain", "v_gene", "d_gene", "j_gene", "c_gene", "full_length", "productive", "cdr3", "cdr3_nt", "reads", "umis")],
                 CD4_M4_vdj[, c("barcode", "length", "chain", "v_gene", "d_gene", "j_gene", "c_gene", "full_length", "productive", "cdr3", "cdr3_nt", "reads", "umis")],
                 #CD4_M5_vdj[, c("barcode", "length", "chain", "v_gene", "d_gene", "j_gene", "c_gene", "full_length", "productive", "cdr3", "cdr3_nt", "reads", "umis")],
                 CD4_M6_vdj[, c("barcode", "length", "chain", "v_gene", "d_gene", "j_gene", "c_gene", "full_length", "productive", "cdr3", "cdr3_nt", "reads", "umis")])

Acute_vdj$barcode %>% length() # 60417
Acute_vdj$barcode %>% unique() %>% length()
# 19218


# Keep only TRA and TRB chains, remove entries without a CDR3 sequence, and only keep cells that are present within the Seurat data
Acute_vdj <- Acute_vdj[Acute_vdj$chain == "TRA" | Acute_vdj$chain == "TRB", ]
Acute_vdj <- Acute_vdj[Acute_vdj$cdr3_nt != "None", ]
Acute_vdj <- Acute_vdj[Acute_vdj$barcode %in% rownames(immune.combined_edit2@meta.data), ]
Acute_vdj$barcode %>% length()
Acute_vdj$barcode %>% unique() %>% length()
# Now down to 29763 TCR UMIs, of which 13456 are unique

# We need only one TRA and TRB chain per cell. Convert full_length and productive columns to factors that rank the entries
Acute_vdj$full_length <- as.factor(Acute_vdj$full_length)
Acute_vdj$productive <- as.factor(Acute_vdj$productive)


# Create empty dataframes to subet the 'best' TRA and TRB chains to.
tra_df <- data.frame()
trb_df <- data.frame()
sub_vdj <- data.frame()

# For each cell, rank each TRA and TRB entry to return the single best chain for each. Ties are broken by the number of UMIs that support them.
for(i in unique(Acute_vdj$barcode)){
  tra_df <- Acute_vdj[Acute_vdj$barcode == i & Acute_vdj$chain == "TRA", ]
  trb_df <- Acute_vdj[Acute_vdj$barcode == i & Acute_vdj$chain == "TRB", ]
  sub_vdj <- rbind(sub_vdj, tra_df[order(tra_df$full_length, tra_df$productive, tra_df$umis, decreasing = TRUE), ][1, ])
  sub_vdj <- rbind(sub_vdj, trb_df[order(trb_df$full_length, trb_df$productive, trb_df$umis, decreasing = TRUE), ][1, ])
}

# Remove NA entries from results
sub_vdj <- sub_vdj[!is.na(sub_vdj$barcode), ]

# Add the current cell identity to the meta.data
immune.combined_edit2@meta.data$current.ident <- immune.combined_edit2@ident

# Create a copy of the meta.data for merging results
### Check the index before subsetting, should be the number of columns in meta data. Do this in case you need to reset the meta data
colnames(immune.combined_edit2@meta.data)
rm(meta_data)
meta_data <- immune.combined_edit2@meta.data[1:18] # this is good to make this DF with columns first


# Merge metadata with VDJ TRA results
meta_data <- merge(meta_data, sub_vdj[sub_vdj$chain == "TRA", c("barcode", "v_gene", "d_gene", "j_gene", "c_gene", "full_length", "productive", "cdr3", "cdr3_nt")], by.x = "row.names", by.y = "barcode", all.x = TRUE)
# Rename columns from TRA to specify they're from TRA
colnames(meta_data)[which(colnames(meta_data) %in% c("v_gene", "d_gene", "j_gene", "c_gene", "full_length", "productive", "cdr3", "cdr3_nt"))] <-
  c("v_gene_TRA", "d_gene_TRA", "j_gene_TRA", "c_gene_TRA", "full_length_TRA", "productive_TRA", "cdr3_TRA", "cdr3_nt_TRA")

# Merge metadata wtih VDJ TRB results
meta_data <- merge(meta_data, sub_vdj[sub_vdj$chain == "TRB", c("barcode", "v_gene", "d_gene", "j_gene", "c_gene", "full_length", "productive", "cdr3", "cdr3_nt")], by.x = "Row.names", by.y = "barcode", all.x = TRUE)
# Rename columns from TRB to specify they're from TRB
colnames(meta_data)[which(colnames(meta_data) %in% c("v_gene", "d_gene", "j_gene", "c_gene", "full_length", "productive", "cdr3", "cdr3_nt"))] <-
  c("v_gene_TRB", "d_gene_TRB", "j_gene_TRB", "c_gene_TRB", "full_length_TRB", "productive_TRB", "cdr3_TRB", "cdr3_nt_TRB")

# Reorder meta_data table (with vdj results) to match Seurat's meta.data
meta_data <- meta_data[order(match(meta_data$Row.names, rownames(immune.combined_edit2@meta.data))), ]

# Add the best TRA and TRB chains to Seurat's meta.data
immune.combined_edit2@meta.data <- cbind(immune.combined_edit2@meta.data, meta_data[,c("v_gene_TRA", "d_gene_TRA", "j_gene_TRA", "c_gene_TRA", "full_length_TRA", "productive_TRA", "cdr3_TRA", "cdr3_nt_TRA", "v_gene_TRB", "d_gene_TRB", "j_gene_TRB", "c_gene_TRB", "full_length_TRB", "productive_TRB", "cdr3_TRB", "cdr3_nt_TRB")])

# Add TRA and TRB identifiers to the sequences and combine both chains to produce a unique CDR3 nucleotide identifier
immune.combined_edit2@meta.data$combined_cdr3_nt <- paste(
  paste("TRA", immune.combined_edit2@meta.data$cdr3_nt_TRA, sep = ":"),
  paste("TRB", immune.combined_edit2@meta.data$cdr3_nt_TRB, sep = ":"),
  sep = "_")
# Some cells don't have a TRA or TRB chain nucleotide sequence - Replace these entries with "No_CDR3"
immune.combined_edit2@meta.data$combined_cdr3_nt <- gsub("_TRB:NA|TRA:NA_|TRB:NA|TRA:NA", "", immune.combined_edit2@meta.data$combined_cdr3_nt)
immune.combined_edit2@meta.data$combined_cdr3_nt[nchar(immune.combined_edit2@meta.data$combined_cdr3_nt) == 0] <- "No CDR3"

# Combine CDR3 AAs
immune.combined_edit2@meta.data$combined_cdr3_AA <- paste(
  paste("TRA", immune.combined_edit2@meta.data$cdr3_TRA, sep = ":"),
  paste("TRB", immune.combined_edit2@meta.data$cdr3_TRB, sep = ":"),
  sep = "_")
# Some cells don't have a TRA or TRB chain amino acid sequence - Replace these entries with "No_CDR3"
immune.combined_edit2@meta.data$combined_cdr3_AA <- gsub("_TRB:NA|TRA:NA_|TRB:NA|TRA:NA", "", immune.combined_edit2@meta.data$combined_cdr3_AA)
immune.combined_edit2@meta.data$combined_cdr3_AA[nchar(immune.combined_edit2@meta.data$combined_cdr3_AA) == 0] <- "No CDR3"

which(immune.combined_edit2@meta.data$combined_cdr3_nt != "No CDR3") %>% length()
immune.combined_edit2@meta.data$combined_cdr3_nt %>% length()
# 13456 cells have either TRA, TRB, or both out of 13636 cells total

grep(pattern = "_", x = immune.combined_edit2@meta.data$combined_cdr3_nt) %>% length()
# _ serves as a marker for cells with a complete TRA_TRB sequence
# 11967 cells have both TRA and TRB out of 13636 cells total (88%)

immune.combined_edit2@meta.data$combined_cdr3_nt %>% unique() %>% length()
# 2511 clones with either TRA, TRB, or both

grep(pattern = "_", x = immune.combined_edit2@meta.data$combined_cdr3_nt, value = T) %>% unique() %>% length()
# 2154 clones with both TRA and TRB

saveRDS(immune.combined_edit2, file = "summary_5M_CCA_V2/immune.combined_edit2_TCR.RDS")


```


```{r}
# run Monocle

## do monocle by removing Treg
rm(list=ls())

immune.combined_edit2 <- readRDS("summary_5M_CCA_V2/immune.combined_edit2_TCR.RDS")

TSNEPlot(immune.combined_edit2, do.label = T)


###Monocle analysis
#Remove cluster 6 from data going to monocle

data_no_cl_6 <- SubsetData(immune.combined_edit2, ident.remove = "6_Treg")
rm(immune.combined_edit2)
#rm(data_no_cl_6)

#Keep the current cluster ident within the meta.data
data_no_cl_6@meta.data$current.ident <- as.character(data_no_cl_6@ident)
#rm(data_monocle)
#Import Seurat data into monocle
data_monocle <- importCDS(data_no_cl_6, import_all = FALSE)
data_monocle <- estimateSizeFactors(data_monocle)
data_monocle <- estimateDispersions(data_monocle)
data_monocle <- detectGenes(data_monocle, min_expr = 0.1)
expressed_genes <- row.names(subset(fData(data_monocle), num_cells_expressed >= 0.05 * ncol(data_monocle)))

#Find the top 100 markers per cluster
markers <- FindAllMarkers(data_no_cl_6, logfc.threshold = 0.1)


write.csv(markers, file = "summary_5M_CCA_V2/markers.csv")


markers <- subset(markers, p_val_adj < 0.05 & abs(avg_logFC) > 0.25)

rm(top_markers_per_cluster)

top_markers_per_cluster <- markers %>% group_by(cluster) %>% top_n(-130, p_val_adj)

# want to see what happens if I use all p value signofcoant markers
#markers <- subset(markers, p_val_adj < 0.05 & abs(avg_logFC) > 0.25)
#top_markers_per_cluster <- markers %>% group_by(cluster) # want to see what happens if I use all p value signofcoant markers

#top_markers_per_cluster <- markers %>% group_by(cluster) %>% top_n(-200, p_val_adj)
data_ordering_genes <- unique(top_markers_per_cluster$gene)

#Run monocle, regressing out mouse sample and the number of genes detected per cell
data_monocle <- setOrderingFilter(data_monocle, ordering_genes = data_ordering_genes)
plot_ordering_genes(data_monocle)
data_monocle <- reduceDimension(data_monocle, reduction_method = "DDRTree", residualModelFormulaStr = "~mice + num_genes_expressed", max_components = 2)
data_monocle <- orderCells(data_monocle)

saveRDS(data_monocle, file = "summary_5M_CCA_V2/data_monocle.RDS")

rm(markers)
data_monocle <- readRDS("summary_5M_CCA_V2/data_monocle.RDS")
markers <- read.csv("summary_5M_CCA_V2/markers.csv")


plot_cell_trajectory(data_monocle)
plot_cell_trajectory(data_monocle, color_by = "current.ident")
plot_cell_trajectory(data_monocle, color_by = "mice")
plot_cell_trajectory(data_monocle, color_by = "Pseudotime")
plot_complex_cell_trajectory(data_monocle, color_by = "current.ident") + facet_wrap("current.ident")
plot_cell_trajectory(data_monocle, color_by = "current.ident", cell_size = 0.2) + facet_wrap("current.ident") + scale_color_manual(values=c("#F35E5A", "#B68A04", "#45A900", "#17B682", "#16A7E6", "#9370FF", "#F740CE"))


data_monocle <- orderCells(data_monocle, root_state = 7) ## 110 gene, not selected by p values worked well


plot_cell_trajectory(data_monocle, cell_size = 0.2)
plot_cell_trajectory(data_monocle, color_by = "current.ident", cell_size = 0.2)
plot_cell_trajectory(data_monocle, color_by = "mice")
plot_cell_trajectory(data_monocle, color_by = "Pseudotime")
plot_complex_cell_trajectory(data_monocle, color_by = "current.ident", cell_size = 0.2) + facet_wrap("current.ident")
plot_cell_trajectory(data_monocle, color_by = "current.ident", cell_size = 0.2) + facet_wrap("current.ident") + scale_color_manual(values=c("#F35E5A", "#B68A04", "#45A900", "#17B682", "#16A7E6", "#9370FF", "#F740CE"))



#Create PNG files for some monocle plots
png(file = "summary_5M_CCA_V2/monocle_state_current_R1data.png", width = 8, height = 6, units = "in", res = 300)
print(plot_cell_trajectory(data_monocle, cell_size = 0.2))
dev.off()

png(file = "summary_5M_CCA_V2/monocle_original_current_R1data.png", width = 8, height = 6, units = "in", res = 300)
print(plot_cell_trajectory(data_monocle, color_by = "current.ident", cell_size = 0.2) + scale_color_manual(values=c("#F35E5A", "#B68A04", "#45A900", "#17B682", "#16A7E6", "#9370FF", "#F740CE")))
dev.off()

png(file = "summary_5M_CCA_V2/monocle_pseudotime_current_R1data.png", width = 8, height = 6, units = "in", res = 300)
print(plot_cell_trajectory(data_monocle, color_by = "Pseudotime", cell_size = 0.2) + theme(legend.text=element_text(size=6)))
dev.off()

png(file = "summary_5M_CCA_V2/monocle_TREE_current_R1data.png", width = 8, height = 6, units = "in", res = 300)
print(plot_complex_cell_trajectory(data_monocle, color_by = "current.ident", cell_size = 0.2) + facet_wrap("current.ident") + scale_color_manual(values=c("#F35E5A", "#B68A04", "#45A900", "#17B682", "#16A7E6", "#9370FF", "#F740CE")))
dev.off()

png(file = "summary_5M_CCA_V2/monocle_splitplot_current_R1data.png", width = 8, height = 6, units = "in", res = 300)
print(plot_cell_trajectory(data_monocle, color_by = "current.ident", cell_size = 0.2) + facet_wrap("current.ident") + scale_color_manual(values=c("#F35E5A", "#B68A04", "#45A900", "#17B682", "#16A7E6", "#9370FF", "#F740CE")))
dev.off()


#### repeat plots with Monocle made from object (CD4_R1_edit1) upon removal of 589
#Create PNG files for some monocle plots
png(file = "summary_R1_V2/monocle_state_current_R1data_remove589.png", width = 8, height = 6, units = "in", res = 300)
print(plot_cell_trajectory(data_monocle))
dev.off()

png(file = "summary_R1_V2/monocle_original_current_R1data_remove589.png", width = 8, height = 6, units = "in", res = 300)
print(plot_cell_trajectory(data_monocle, color_by = "current.ident") + scale_color_manual(values=c("#F35E5A", "#B68A04", "#45A900", "#17B682", "#16A7E6", "#9370FF", "#F740CE")))
dev.off()

png(file = "summary_R1_V2/monocle_pseudotime_current_R1data_remove589.png", width = 8, height = 6, units = "in", res = 300)
print(plot_cell_trajectory(data_monocle, color_by = "Pseudotime") + theme(legend.text=element_text(size=6)))
dev.off()

png(file = "summary_R1_V2/monocle_TREE_current_R1data_remove589.png", width = 8, height = 6, units = "in", res = 300)
print(plot_complex_cell_trajectory(data_monocle, color_by = "current.ident") + facet_wrap("current.ident") + scale_color_manual(values=c("#F35E5A", "#B68A04", "#45A900", "#17B682", "#16A7E6", "#9370FF", "#F740CE")))
dev.off()

png(file = "summary_R1_V2/monocle_splitplot_current_R1data_remove589.png", width = 8, height = 6, units = "in", res = 300)
print(plot_cell_trajectory(data_monocle, color_by = "current.ident") + facet_wrap("current.ident") + scale_color_manual(values=c("#F35E5A", "#B68A04", "#45A900", "#17B682", "#16A7E6", "#9370FF", "#F740CE")))
dev.off()

```

```{r}

rm(cell_dist)
cell_dist <- table(data_monocle@phenoData@data$State, data_monocle@phenoData@data$current.ident) %>% as.table()

write.csv(cell_dist, file = "summary_5M_CCA_V2/cell_dist.csv")

 
 #   0_Tcmp 1_Ly6cHi_Th1 2_Pre_TFH 3_Lag3hi_Th1 4_TFH1 5_GC_TFH2
#  1    868         2445       240         1932    324        66
#  2     77           12       120           33    119       115
#  3     13            2        18            6      8         9
#  4     12            0         8            2      3         2
#  5    789           13       951           31     71       239
#  6    321           23       675           82   1109       908
#  7   1077          108       303           80     35        46

cell_percent <- table(data_monocle@phenoData@data$State, data_monocle@phenoData@data$current.ident) %>% prop.table(margin = 2)* 100 

#      0_Tcmp     1_Ly6cHi_Th1 2_Pre_TFH   3_Lag3hi_Th1    4_TFH1   5_GC_TFH2
#  1 27.49445676  93.93008068 10.36717063  89.19667590 19.41282205  4.76534296
#  2  2.43902439   0.46100653  5.18358531   1.52354571  7.13001797  8.30324910
#  3  0.41178334   0.07683442  0.77753780   0.27700831  0.47932894  0.64981949
#  4  0.38010770   0.00000000  0.34557235   0.09233610  0.17974835  0.14440433
#  5 24.99208109   0.49942374 41.07991361   1.43120960  4.25404434 17.25631769
#  6 10.16788090   0.88359585 29.15766739   3.78578024 66.44697424 65.55956679
#  7 34.11466582   4.14905878 13.08855292   3.69344414  2.09706411  3.32129964

write.csv(cell_percent, file = "summary_5M_CCA_V2/cell_percent.csv")
```


```{r}
plot_cell_trajectory(data_monocle)
plot_cell_trajectory(data_monocle, color_by = "current.ident")
plot_cell_trajectory(data_monocle, color_by = "mice")
plot_cell_trajectory(data_monocle, color_by = "Pseudotime")
plot_complex_cell_trajectory(data_monocle, color_by = "current.ident") + facet_wrap("current.ident")
print(plot_cell_trajectory(data_monocle, color_by = "current.ident") + facet_wrap("current.ident") + scale_color_manual(values=c("#F35E5A", "#B68A04", "#45A900", "#17B682", "#16A7E6", "#9370FF", "#F740CE")))

```

```{r}
### try monocle by restricting 2 cells/clone
rm(list=ls())
rm(immune.combined_edit2)
immune.combined_edit2 <- readRDS("summary_5M_CCA_V2/immune.combined_edit2_TCR.RDS")

TSNEPlot(immune.combined_edit2)

length(immune.combined_edit2@ident) # 13636


# Make new DF that includes original identity
Acute_CD4_5mice_clones_by_mouse <- immune.combined_edit2@meta.data %>% select(combined_cdr3_nt, mice) %>% table() %>% as.data.frame()
# Remove "No CDR3" rows
Acute_CD4_5mice_clones_by_mouse <- Acute_CD4_5mice_clones_by_mouse[-which(Acute_CD4_5mice_clones_by_mouse$combined_cdr3_nt == "No CDR3"), ]
# table() double counted some cells but left the counts as 0; remove these
Acute_CD4_5mice_clones_by_mouse <- Acute_CD4_5mice_clones_by_mouse[which(Acute_CD4_5mice_clones_by_mouse$Freq > 0), ]
# Order in decreasing order
Acute_CD4_5mice_clones_by_mouse <- Acute_CD4_5mice_clones_by_mouse[order(Acute_CD4_5mice_clones_by_mouse$Freq, decreasing = T), ]

# Subset the first data frame to only include clones with both TRA and TRB
Acute_CD4_5mice_clones_by_mouse_TRA_TRB <- Acute_CD4_5mice_clones_by_mouse[grep(pattern = "_", x = Acute_CD4_5mice_clones_by_mouse$combined_cdr3_nt), ]

# calculate # of clones in each mouse with both chains

table(Acute_CD4_5mice_clones_by_mouse_TRA_TRB$mice)
#CD4_M1 CD4_M2 CD4_M3 CD4_M4 CD4_M5 CD4_M6 
 #  382    370    471    540    725    464 

#  M1  M2  M3  M4  M6 
# 386 372 423 496 485 

Acute_CD4_5mice_clones_by_mouse_TRA_TRB_2cell <- Acute_CD4_5mice_clones_by_mouse_TRA_TRB[Acute_CD4_5mice_clones_by_mouse_TRA_TRB$Freq >= 2, ]


Acute_CD4_5mice_clones_by_mouse_TRA_TRB$Freq %>% sum() # 11967
Acute_CD4_5mice_clones_by_mouse_TRA_TRB_2cell$Freq %>% sum() # 10491

table(Acute_CD4_5mice_clones_by_mouse_TRA_TRB_2cell$mice)
# M1  M2  M3  M4  M6 
#134 140 102 160 150 


## subset Seurat object based on 2 cell clone

immune.combined_edit2_2cell <- SubsetData(immune.combined_edit2, cells.use = rownames(immune.combined_edit2@meta.data)[which(immune.combined_edit2@meta.data$combined_cdr3_nt %in% Acute_CD4_5mice_clones_by_mouse_TRA_TRB_2cell$combined_cdr3_nt)])

saveRDS(immune.combined_edit2_2cell, file = "summary_5M_CCA_V2/immune.combined_edit2_2cell.RDS")
length(immune.combined_edit2_2cell@ident) # 10492, from 13636 cells 

TSNEPlot(immune.combined_edit2, do.label = T)
TSNEPlot(immune.combined_edit2_2cell, do.label = T)

```



```{r}
## DO monocle on this subsetted object with >= 2 cells

rm(list = ls())

immune.combined_edit2_2cell <- readRDS("summary_5M_CCA_V2/immune.combined_edit2_2cell.RDS")

png(file = "summary_5M_CCA_V2/TSNE_edit2_2cellplot.png", width = 8, height = 6, units = "in", res = 300)
TSNEPlot(immune.combined_edit2_2cell, do.label = T)
dev.off()


###Monocle analysis
#Remove cluster 6 from data going to monocle

data_no_cl_6 <- SubsetData(immune.combined_edit2_2cell, ident.remove = "6_Treg")

TSNEPlot(data_no_cl_6, do.label = T)

#rm(data_no_cl_6)

#Keep the current cluster ident within the meta.data
data_no_cl_6@meta.data$current.ident <- as.character(data_no_cl_6@ident)
#rm(data_monocle)
#Import Seurat data into monocle
data_monocle <- importCDS(data_no_cl_6, import_all = FALSE)
data_monocle <- estimateSizeFactors(data_monocle)
data_monocle <- estimateDispersions(data_monocle)
data_monocle <- detectGenes(data_monocle, min_expr = 0.1)
expressed_genes <- row.names(subset(fData(data_monocle), num_cells_expressed >= 0.05 * ncol(data_monocle)))

#Find the top 100 markers per cluster
markers <- FindAllMarkers(data_no_cl_6, logfc.threshold = 0.1)


write.csv(markers, file = "summary_5M_CCA_V2/markers_2cell.csv")


markers <- subset(markers, p_val_adj < 0.05 & abs(avg_logFC) > 0.25)

rm(top_markers_per_cluster)

top_markers_per_cluster <- markers %>% group_by(cluster) %>% top_n(-150, p_val_adj)

# want to see what happens if I use all p value signofcoant markers
#markers <- subset(markers, p_val_adj < 0.05 & abs(avg_logFC) > 0.25)
#top_markers_per_cluster <- markers %>% group_by(cluster) # want to see what happens if I use all p value signofcoant markers

#top_markers_per_cluster <- markers %>% group_by(cluster) %>% top_n(-200, p_val_adj)
data_ordering_genes <- unique(top_markers_per_cluster$gene)

#Run monocle, regressing out mouse sample and the number of genes detected per cell
data_monocle <- setOrderingFilter(data_monocle, ordering_genes = data_ordering_genes)
plot_ordering_genes(data_monocle)
data_monocle <- reduceDimension(data_monocle, reduction_method = "DDRTree", residualModelFormulaStr = "~mice + num_genes_expressed", max_components = 2)
data_monocle <- orderCells(data_monocle)

saveRDS(data_monocle, file = "summary_5M_CCA_V2/data_monocle_2cell.RDS") # 150 genes finalized one 

rm(markers)
data_monocle <- readRDS("summary_5M_CCA_V2/data_monocle_2cell.RDS")
markers <- read.csv("summary_5M_CCA_V2/markers_2cell.csv")


plot_cell_trajectory(data_monocle)
plot_cell_trajectory(data_monocle, color_by = "current.ident")
plot_cell_trajectory(data_monocle, color_by = "mice")
plot_cell_trajectory(data_monocle, color_by = "Pseudotime")
plot_complex_cell_trajectory(data_monocle, color_by = "current.ident") + facet_wrap("current.ident")
plot_cell_trajectory(data_monocle, color_by = "current.ident", cell_size = 0.2) + facet_wrap("current.ident") + scale_color_manual(values=c("#F35E5A", "#B68A04", "#45A900", "#17B682", "#16A7E6", "#9370FF", "#F740CE"))


data_monocle <- orderCells(data_monocle, root_state = 2) ## 150 gene, not selected by p values worked well


plot_cell_trajectory(data_monocle, cell_size = 0.2)
plot_cell_trajectory(data_monocle, color_by = "current.ident", cell_size = 0.2)
plot_cell_trajectory(data_monocle, color_by = "mice")
plot_cell_trajectory(data_monocle, color_by = "Pseudotime")
plot_complex_cell_trajectory(data_monocle, color_by = "current.ident", cell_size = 0.2) + facet_wrap("current.ident")
plot_cell_trajectory(data_monocle, color_by = "current.ident", cell_size = 0.2) + facet_wrap("current.ident") + scale_color_manual(values=c("#F35E5A", "#B68A04", "#45A900", "#17B682", "#16A7E6", "#9370FF", "#F740CE"))



#Create PNG files for some monocle plots
png(file = "summary_5M_CCA_V2/monocle_state_current_R1data_2cell.png", width = 8, height = 6, units = "in", res = 300)
print(plot_cell_trajectory(data_monocle, cell_size = 0.2))
dev.off()

png(file = "summary_5M_CCA_V2/monocle_original_current_R1data_2cell.png", width = 8, height = 6, units = "in", res = 300)
print(plot_cell_trajectory(data_monocle, color_by = "current.ident", cell_size = 0.2) + scale_color_manual(values=c("#F35E5A", "#B68A04", "#45A900", "#17B682", "#16A7E6", "#9370FF", "#F740CE")))
dev.off()

png(file = "summary_5M_CCA_V2/monocle_pseudotime_current_R1data_2cell.png", width = 8, height = 6, units = "in", res = 300)
print(plot_cell_trajectory(data_monocle, color_by = "Pseudotime", cell_size = 0.2) + theme(legend.text=element_text(size=6)))
dev.off()

png(file = "summary_5M_CCA_V2/monocle_TREE_current_R1data_2cell.png", width = 8, height = 6, units = "in", res = 300)
print(plot_complex_cell_trajectory(data_monocle, color_by = "current.ident", cell_size = 0.2) + facet_wrap("current.ident") + scale_color_manual(values=c("#F35E5A", "#B68A04", "#45A900", "#17B682", "#16A7E6", "#9370FF", "#F740CE")))
dev.off()

png(file = "summary_5M_CCA_V2/monocle_splitplot_current_R1data_2cell.png", width = 8, height = 6, units = "in", res = 300)
print(plot_cell_trajectory(data_monocle, color_by = "current.ident", cell_size = 0.2) + facet_wrap("current.ident") + scale_color_manual(values=c("#F35E5A", "#B68A04", "#45A900", "#17B682", "#16A7E6", "#9370FF", "#F740CE")))
dev.off()

#Create PDF files for some monocle plots
pdf(file = "summary_5mice/monocle_state_current_R1data_2cell.pdf", width = 8, height = 6, useDingbats = FALSE )
print(plot_cell_trajectory(data_monocle, cell_size = 0.2))
dev.off()

pdf(file = "summary_5mice/monocle_original_current_R1data_2cell.pdf", width = 8, height = 6, useDingbats = FALSE)
print(plot_cell_trajectory(data_monocle, color_by = "current.ident", cell_size = 0.2) + scale_color_manual(values=c("#F35E5A", "#B68A04", "#45A900", "#17B682", "#16A7E6", "#9370FF", "#F740CE")))
dev.off()

pdf(file = "summary_5mice/monocle_pseudotime_current_R1data_2cell.pdf", width = 8, height = 6, useDingbats = FALSE)
print(plot_cell_trajectory(data_monocle, color_by = "Pseudotime", cell_size = 0.2) + theme(legend.text=element_text(size=6)))
dev.off()

pdf(file = "summary_5mice/monocle_TREE_current_R1data_2cell.pdf", width = 8, height = 6, useDingbats = FALSE)
print(plot_complex_cell_trajectory(data_monocle, color_by = "current.ident", cell_size = 0.2) + facet_wrap("current.ident") + scale_color_manual(values=c("#F35E5A", "#B68A04", "#45A900", "#17B682", "#16A7E6", "#9370FF", "#F740CE")))
dev.off()

pdf(file = "summary_5mice/monocle_splitplot_current_R1data_2cell.pdf", width = 8, height = 6, useDingbats = FALSE)
print(plot_cell_trajectory(data_monocle, color_by = "current.ident", cell_size = 0.2) + facet_wrap("current.ident") + scale_color_manual(values=c("#F35E5A", "#B68A04", "#45A900", "#17B682", "#16A7E6", "#9370FF", "#F740CE")))
dev.off()




```


```{r}
rm(cell_dist)

cell_dist <- table(data_monocle@phenoData@data$State, data_monocle@phenoData@data$current.ident) %>% as.table()

write.csv(cell_dist, file = "summary_5M_CCA_V2/cell_dist_2cell.csv")

#     0_Tcmp 1_Ly6cHi_Th1 2_Pre_TFH 3_Lag3hi_Th1 4_TFH1 5_GC_TFH2
#  1    363         1729        76         1458    114        10
#  2    681           90       189          105     13        19
#  3    105           12        44           45     20         7
#  4   1022           28      1247           92    225       596
#  5    208           21       411          106    998       415

cell_percent <- table(data_monocle@phenoData@data$State, data_monocle@phenoData@data$current.ident) %>% prop.table(margin = 2)* 100 


#        0_Tcmp 1_Ly6cHi_Th1  2_Pre_TFH 3_Lag3hi_Th1     4_TFH1  5_GC_TFH2
#  1 15.2585120   91.9680851  3.8637519   80.7308970  8.3211679  0.9551098
#  2 28.6254729    4.7872340  9.6085409    5.8139535  0.9489051  1.8147087
#  3  4.4136192    0.6382979  2.2369090    2.4916944  1.4598540  0.6685769
#  4 42.9592266    1.4893617 63.3960346    5.0941307 16.4233577 56.9245463
#  5  8.7431694    1.1170213 20.8947636    5.8693245 72.8467153 39.6370583

write.csv(cell_percent, file = "summary_5M_CCA_V2/cell_percent_2cell.csv")


```

```{r get color names for seurat object}
rm(pdata)
p <- TSNEPlot(immune.combined_edit2_2cell, do.label = T) # Generate the tSNE plot, but save it as an object
pbuild <- ggplot2::ggplot_build(p) # Use ggplot_build to deconstruct the ggplot object
pdata <- pbuild$data[[1]]# Pull the data used for the plot

pdata <- unique(pdata) 
view(pdata)
pdata <- pdata[order(pdata$group, decreasing = T), ]

pdata <- unique(pdata$colour)
```


```{r cluster cell % across 5 mice}
pdf("summary_5mice/Figure2/CD4_5mice_2cell_edit2_TSNE_clusterdist_permice.pdf", width = 8, height = 6)
ggplot(immune.combined_edit2_2cell@meta.data, aes(x=mice, fill=cluster_ident)) + geom_bar(position = "fill") + 
  theme(panel.background = element_blank())
dev.off()

table(immune.combined_edit2_2cell@meta.data$mice, immune.combined_edit2_2cell@meta.data$cluster_ident) %>% prop.table(margin = 1) * 100


 #        0_Tcmp 1_Ly6cHi_Th1  2_Pre_TFH 3_Lag3hi_Th1     4_TFH1  5_GC_TFH2     6_Treg
#  M1 23.3555767   24.1182078 14.2993327   15.5386082 12.3927550  9.6282173  0.6673022
#  M2 24.9272551   21.9204656 17.2647915   13.8700291 12.2211445  9.1173618  0.6789525
#  M3 18.0842607   24.7217806 14.9443561   20.7869634 11.9634340  9.1812401  0.3179650
#  M4 26.6121973   13.1601984 18.2958856   16.1365626 12.9267581 12.4598774  0.4085206
#  M6 20.6561361   13.2847307 25.7594168   17.1729445 14.9858242  7.8574322  0.2835156
```

```{r make gene expression heatmap using immune combined 2 cell object }

immune.combined_edit2_2cell <- readRDS("summary_5M_CCA_V2/immune.combined_edit2_2cell.RDS") # this has Treg and all clones >= 2 cells


## make the heatmap for average gene expression 
TFH_markers <- c("Cxcr5","Bcl6" , "Ascl2", "Pdcd1", "Icos", "Il21", "Il4")
Th1_markers <- c("Cxcr6", "Tbx21", "Id2", "Gzmb", "Ifng","Ly6c2","Lag3")
Tcmp_markers <- c("Id3", "Ccr7", "Il7r", "Lef1", "Slamf6", "Bcl2", "Tcf7")
Treg_markers <- c("Foxp3", "Il2ra", "Il10", "Cd81", "Cd74", "Klrg1")

#As you can see the parameters changed between Seurat v2 and v3
#Here I just calculate the average expression, subset to only the RNA assay,
#And finally subset to the genes you've chosen
#matrix_to_plot <- AverageExpression(Acute_6_mice_edit4_2cell)
#matrix_to_plot <- matrix_to_plot$RNA
#matrix_to_plot <- matrix_to_plot[c(Tcmp_markers,Th1_markers,TFH_markers,Treg_markers), ]
#The following arguments are not used: genes.use'use.scale' is a deprecated argument, please use the 'slot' argument insteadError in data.use[features.assay, temp.cells, drop = FALSE] : 
 # no 'dimnames' attribute for array

#col_df <- data.frame(row.names = colnames(matrix_to_plot), cluster = colnames(matrix_to_plot))
#colVars <- list(cluster = setNames(c("#F8766D","#C49A00", "#53B400", "#00C094", "#00B6EB", "#A58AFF", "#FB61D7"),
           
                      #  c("0_Tcmp", "1_Ly6cHi_Th1", "2_Lag3Hi_Th1", "3_GC_TFH2", "4_TFH1", "5_Pre_TFH", "6_Treg")))]

#Everything can basically stay the same.  We didn't use the 'scale data' from Seurat
#So here we scale by row within pheatmap.
#pdf("summary_6mice/heatmap_with_new_ident_name_6mice.pdf", width = 10, height = 8)
#pheatmap(matrix_to_plot, scale = "row", cluster_cols = FALSE, cluster_rows = FALSE, show_colnames = FALSE, show_rownames = TRUE,
 #        annotation_col = col_df, annotation_colors = colVars, gaps_row = c(7, 14, 21)) #use pheatmap to plot the matrix like the command we worked on.
#dev.off()



################################ use following commands if using Seurat V2 ################################
rm(scale_matrix_to_plot)
scale_matrix_to_plot <- AverageExpression(immune.combined_edit2_2cell, genes.use = c(Tcmp_markers,Th1_markers,TFH_markers,Treg_markers), use.scale = TRUE)
scale_matrix_to_plot[scale_matrix_to_plot < -1] <- -1 #replace extremely low values with some minimum value
scale_matrix_to_plot[scale_matrix_to_plot > 1] <- 1 #Replace high values with some maximum
col_df <- data.frame(row.names = colnames(scale_matrix_to_plot), cluster = colnames(scale_matrix_to_plot))

colVars <- list(cluster = setNames(c("#F8766D","#C49A00", "#53B400", "#00C094", "#00B6EB", "#A58AFF", "#FB61D7"),
                                    c("0_Tcmp", "1_Ly6cHi_Th1", "2_Pre_TFH", "3_Lag3hi_Th1", "4_TFH1", "5_GC_TFH2", "6_Treg")))

pdf("summary_5mice/heatmap_with_new_ident_name_5mice.pdf", width = 10, height = 8)
pheatmap(scale_matrix_to_plot, scale = "none", cluster_cols = FALSE, cluster_rows = FALSE, show_colnames = FALSE, show_rownames = TRUE,
         annotation_col = col_df, annotation_colors = colVars, gaps_row = c(7, 14, 21), fontsize = 16) #use pheatmap to plot the matrix like the command we worked on.
dev.off()

TSNEPlot(immune.combined_edit2_2cell)
```

```{r this is for V2}

immune.combined_edit2_2cell <- readRDS("summary_5M_CCA_V2/immune.combined_edit2_2cell.RDS") # this has Treg and all clones >= 2 cells
immune.combined_edit2_2cell@meta.data$cluster_ident

# 0_Tcmp  1_Ly6cHi_Th1  2_Pre_TFH  3_Lag3hi_Th1  4_TFH1  5_GC_TFH2  6_Treg


TSNE_label_coords <- data.frame(tSNE_1 = rep(0, 7), tSNE_2 = rep(0, 7))
TSNE_label_coords[1,] <- c(15, 15)
TSNE_label_coords[2,] <- c(-5, 20)
TSNE_label_coords[3,] <- c(2, -10)
TSNE_label_coords[4,] <- c(-20, 0)
TSNE_label_coords[5,] <- c(-10, -30)
TSNE_label_coords[6,] <- c(20, -20)
TSNE_label_coords[7,] <- c(30, 25)

# Function to make a vector of default ggplot colors; to be used when making legend
gg_color_hue <- function(n) {
  hues = seq(15, 375, length = n + 1)
  hcl(h = hues, l = 65, c = 100)[1:n]
}

png("summary_5mice/00_TSNE_changedident.png", width = 10, height = 8, units = "in", res = 300)
# Use TSNEPlot to make the graph; do.return = T will give a ggplot object which we can manipulate
TSNEPlot(immune.combined_edit2_2cell, pt.size = 0.5, label.size = 7) + 
  theme(legend.text = element_text(size = 10)) +
   # Use annotate() to make labels, expression() to make superscripts and subscripts
  annotate("label", x = TSNE_label_coords[1,1], y = TSNE_label_coords[1,2], parse = T, label = as.character(expression(Tcmp)), size = 5, fontface = "bold") +
  annotate("label", x = TSNE_label_coords[2,1], y = TSNE_label_coords[2,2], parse = T, label = as.character(expression(Ly6c^Hi*"-Th1")), size = 5, fontface = "bold") +
  annotate("label", x = TSNE_label_coords[3,1], y = TSNE_label_coords[3,2], parse = T, label = as.character(expression("Pre-T"[FH])), size = 5, fontface = "bold") +
  annotate("label", x = TSNE_label_coords[4,1], y = TSNE_label_coords[4,2], parse = T, label = as.character(expression(Lag3^Hi*"-Th1")), size = 5, fontface = "bold") +
  annotate("label", x = TSNE_label_coords[5,1], y = TSNE_label_coords[5,2], parse = T, label = as.character(expression(T[FH]*1)), size = 5, fontface = "bold") +
  annotate("label", x = TSNE_label_coords[6,1], y = TSNE_label_coords[6,2], parse = T, label = as.character(expression("GC-"*T[FH]*2)), size = 5, fontface = "bold") +
  annotate("label", x = TSNE_label_coords[7,1], y = TSNE_label_coords[7,2], parse = T, label = as.character(expression(Treg)), size = 5, fontface = "bold") +
   scale_color_manual(values = gg_color_hue(7), labels = c(
    expression("0 "*Tcmp),
    expression("1 "*Ly6c^Hi*"-Th1"),
    expression("2 Pre T"[FH]),
    expression("3 "*Lag3^Hi*"-Th1"),
    expression("4 "*T[FH]*1),
    expression("5 GC-"*T[FH]*2),
    expression("6 Treg")
  )) +
  theme(legend.text.align = 0) # Aligns text to left side; for right side, use 1; for center, 0.5
dev.off()



pdf(file = "summary_5mice/01_TSNE_changedident.pdf",   width = 10, height = 8, useDingbats = FALSE) 
# Use TSNEPlot to make the graph; do.return = T will give a ggplot object which we can manipulate
TSNEPlot(immune.combined_edit2_2cell, pt.size = 0.8, label.size = 8) + 
  theme(legend.text = element_text(size = 10)) +
   # Use annotate() to make labels, expression() to make superscripts and subscripts
  annotate("label", x = TSNE_label_coords[1,1], y = TSNE_label_coords[1,2], parse = T, label = as.character(expression(Tcmp)), size = 5, fontface = "bold") +
  annotate("label", x = TSNE_label_coords[2,1], y = TSNE_label_coords[2,2], parse = T, label = as.character(expression(Ly6c^Hi*"-Th1")), size = 5, fontface = "bold") +
  annotate("label", x = TSNE_label_coords[3,1], y = TSNE_label_coords[3,2], parse = T, label = as.character(expression("Pre-T"[FH])), size = 5, fontface = "bold") +
  annotate("label", x = TSNE_label_coords[4,1], y = TSNE_label_coords[4,2], parse = T, label = as.character(expression(Lag3^Hi*"-Th1")), size = 5, fontface = "bold") +
  annotate("label", x = TSNE_label_coords[5,1], y = TSNE_label_coords[5,2], parse = T, label = as.character(expression(T[FH]*1)), size = 5, fontface = "bold") +
  annotate("label", x = TSNE_label_coords[6,1], y = TSNE_label_coords[6,2], parse = T, label = as.character(expression("GC-"*T[FH]*2)), size = 5, fontface = "bold") +
  annotate("label", x = TSNE_label_coords[7,1], y = TSNE_label_coords[7,2], parse = T, label = as.character(expression(Treg)), size = 5, fontface = "bold") +
   scale_color_manual(values = gg_color_hue(7), labels = c(
    expression("0 "*Tcmp),
    expression("1 "*Ly6c^Hi*"-Th1"),
    expression("2 Pre T"[FH]),
    expression("3 "*Lag3^Hi*"-Th1"),
    expression("4 "*T[FH]*1),
    expression("5 GC-"*T[FH]*2),
    expression("6 Treg")
  )) +
  theme(legend.text.align = 0) 
dev.off()

```


```{r make plots on TSNE object for main and supp figures}


pdf(file = "summary_5mice/Figure2/01_TSNE_bymice_S2.pdf",   width = 10, height = 8, useDingbats = FALSE) 
TSNEPlot(immune.combined_edit2_2cell,  group.by = "mice") + scale_color_brewer(palette = "Dark2")
dev.off()


pdf(file = "summary_5mice/Th1_Modscore.pdf", width = 8, height = 6)
VlnPlot(immune.combined_edit2_2cell, c("Geneset_Th11"), point.size.use = 0) + geom_boxplot(notch = T, fill = "white", alpha = 0.3)
dev.off()

pdf(file = "summary_5mice/TFH_modulescore.pdf", width = 8, height = 6)
VlnPlot(immune.combined_edit2_2cell, c("Geneset_Tfh1"), point.size.use = 0) + geom_boxplot(notch = T, fill = "white", alpha = 0.3)
dev.off()

pdf(file = "summary_5mice/Tmem_modscore.pdf", width = 8, height = 6)
VlnPlot(immune.combined_edit2_2cell, c("Geneset_Tmem1"), point.size.use = 0) + geom_boxplot(notch = T, fill = "white", alpha = 0.3)
dev.off()


# check overall genes one at a time using PDF
pdf("summary_5mice/CD4_GP66_CCR7.pdf", width = 10, height = 8)
VlnPlot(immune.combined_edit2_2cell, features = "Ccr7", point.size.use = 0.2)
dev.off()

pdf("summary_5mice/CD4_GP66_Cxcr6.pdf", width = 10, height = 8)
VlnPlot(immune.combined_edit2_2cell, features = "Cxcr6", point.size.use = 0.2)
dev.off()

pdf("summary_5mice/CD4_GP66_Cxcr5.pdf", width = 10, height = 8)
VlnPlot(immune.combined_edit2_2cell, features = "Cxcr5", point.size.use = 0.2)
dev.off()

pdf("summary_5mice/CD4_GP66_Foxp3.pdf", width = 10, height = 8)
VlnPlot(immune.combined_edit2_2cell, features = "Foxp3", point.size.use = 0.2)
dev.off()

pdf("summary_5mice/Figure2/CD4_GP66_Tbx21.pdf", width = 10, height = 8)
VlnPlot(immune.combined_edit2_2cell, features = "Tbx21", point.size.use = 0.2)
dev.off()

pdf("summary_5mice/Figure2/CD4_GP66_Id3.pdf", width = 10, height = 8)
VlnPlot(immune.combined_edit2_2cell, features = "Id3", point.size.use = 0.2)
dev.off()

pdf("summary_5mice/Figure2/CD4_GP66_Id2.pdf", width = 10, height = 8)
VlnPlot(immune.combined_edit2_2cell, features = "Id2", point.size.use = 0.2)
dev.off()

pdf("summary_5mice/Figure2/CD4_GP66_Il21.pdf", width = 10, height = 8)
VlnPlot(immune.combined_edit2_2cell, features = "Il21", point.size.use = 0.2)
dev.off()

pdf("summary_5mice/Figure2/CD4_GP66_Pdcd1.pdf", width = 10, height = 8)
VlnPlot(immune.combined_edit2_2cell, features = "Pdcd1", point.size.use = 0.2)
dev.off()

pdf("summary_5mice/Figure2/CD4_GP66_Ascl2.pdf", width = 10, height = 8)
VlnPlot(immune.combined_edit2_2cell, features = "Ascl2", point.size.use = 0.2)
dev.off()

pdf("summary_5mice/CC_score_S2B.pdf", width = 10, height = 8)
VlnPlot(immune.combined_edit2_2cell, c("S.Score", "G2M.Score"), point.size.use = 0.2 )
dev.off()


# "useDingbats = FALSE" helps the pdf file not to be changed in Adobe illustrator
genes <- c("Slamf6", "Tcf7","Cxcr6", "Gzmb", "Cxcr5","Bcl6", "Icos", "Foxp3", "Il2ra")

pdf("summary_5mice/Figure2/Lineage_gene_09272020_S2C.pdf", width = 8, height = 5, useDingbats = FALSE)
DotPlot(immune.combined_edit2_2cell, genes.plot = genes, cols.use = c("grey", "red"), plot.legend = T, do.return = T) + ggplot2::theme_classic()
dev.off()

#DotPlot(immune.combined_edit2_2cell, genes.plot = genes, cols.use = c("grey", "red"), plot.legend = T, do.return = T) + theme_minimal()



#colVars <- list(cluster = setNames(c("#F8766D","#C49A00", "#53B400", "#00C094", "#00B6EB", "#A58AFF", "#FB61D7"),
 #                                   c("0_Tcmp", "1_Ly6cHi_Th1", "2_Pre_TFH", "3_Lag3hi_Th1", "4_TFH1", "5_GC_TFH2", "6_Treg")))



## just want to subset out 2 Th1 clusters to be used in Dotplot ################
immune.combined_edit2_2cell_Th1 <- SubsetData(immune.combined_edit2_2cell, ident.use = c("1_Ly6cHi_Th1", "3_Lag3hi_Th1"))

genes_Th1 <- c("Ly6c2","Selplg", "S1pr1","S1pr5", "Cx3cr1", "Lag3", "Pdcd1", "Tnfrsf4", "Havcr2", "Slamf1")
pdf("summary_5mice/Figure2/Th1_DE_genes_S2D_09272020.pdf", width = 8, height = 5, useDingbats = FALSE)
DotPlot(immune.combined_edit2_2cell_Th1, genes.plot = genes_Th1, cols.use = c("grey", "red"), plot.legend = T, do.return = T) + ggplot2::theme_classic()
dev.off()


## just want to subset out 3 TFH clusters to be used in Dotplot ###############################
immune.combined_edit2_2cell_TFH <- SubsetData(immune.combined_edit2_2cell, ident.use = c("2_Pre_TFH", "4_TFH1", "5_GC_TFH2"))

genes_TFH <- c("Tcf7", "Sh2d1a", "Ascl2","Tox2", "Ifng", "Tbx21", "Cdk4", "Ran", "Cxcr5","Selplg")
pdf("summary_5mice/Figure2/TFH_DE_genes_S2E_09272020.pdf", width = 8, height = 5, useDingbats = FALSE)
DotPlot(immune.combined_edit2_2cell_TFH, genes.plot = genes_TFH, cols.use = c("grey", "red"), plot.legend = T, do.return = T) + ggplot2::theme_classic()
dev.off()




pdf("summary_5mice/CD4_GP66_Foxp3.pdf", width = 10, height = 8)
VlnPlot(immune.combined_edit2_2cell, features = c("Tbx21", "Il21", "Id2"), point.size.use = 0)

dev.off()


```


```{r}

immune.combined_edit2_2cell@meta.data$cluster_ident
png(file = "summary_5M_CCA_V2/Th1_Modscore.png", width = 8, height = 6, units = "in", res = 300)
VlnPlot(immune.combined_edit2_2cell, c("Geneset_Th11"), point.size.use = 0) + geom_boxplot(notch = T, fill = "white", alpha = 0.3)
# For manuscript
compare_means(Geneset_Th11 ~ cluster_ident, data = (immune.combined_edit2_2cell@meta.data %>% select(Geneset_Th11, cluster_ident)))
# p-values from Wilcoxon test

dev.off()

png(file = "summary_5M_CCA_V2/TFH_modulescore.png", width = 8, height = 6, units = "in", res = 300)
VlnPlot(immune.combined_edit2_2cell, c("Geneset_Tfh1"), point.size.use = 0) + geom_boxplot(notch = T, fill = "white", alpha = 0.3)
# For manuscript
compare_means(Geneset_Tfh1 ~ cluster_ident, data = (immune.combined_edit2_2cell@meta.data %>% select(Geneset_Tfh1, cluster_ident)))
# p-values from Wilcoxon test

dev.off()

png(file = "summary_5M_CCA_V2/Tmem_modscore.png", width = 8, height = 6, units = "in", res = 300)
VlnPlot(immune.combined_edit2_2cell, c("Geneset_Tmem1"), point.size.use = 0) + geom_boxplot(notch = T, fill = "white", alpha = 0.3)
# For manuscript
compare_means(Geneset_Tmem1 ~ cluster_ident, data = (immune.combined_edit2_2cell@meta.data %>% select(Geneset_Tmem1, cluster_ident)))
# p-values from Wilcoxon test

dev.off()


```

```{r}
## TFH
Geneset_Tfh <- readLines("Genesets/Tfh_gene.txt", warn = FALSE) # Remove first two lines, which are headers
Geneset_Tfh <- Geneset_Tfh[which(Geneset_Tfh %in% rownames(immune.combined_edit2_2cell@data))] # Restrict to genes present in our data
Geneset_Tfh <- list(Geneset_Tfh)
immune.combined <- AddModuleScore(immune.combined_edit2_2cell, genes.list = Geneset_Tfh, enrich.name = "Geneset_Tfh")

## in the meta data it adds "1" by the end of the name of the Geneset_apoptosis

#png("summary/19_CD4_Acute_chronic_TFH_genescore.png", width = 10, height = 8, units = "in", res = 300)
VlnPlot(immune.combined_edit2_2cell, c("Geneset_Tfh1"), point.size.use = 0)

```


```{r check oxphos pathways and CC score for TFH subsets}
#colVars <- list(cluster = setNames(c("#F8766D","#C49A00", "#53B400", "#00C094", "#00B6EB", "#A58AFF", "#FB61D7"),
 #                                   c("0_Tcmp", "1_Ly6cHi_Th1", "2_Pre_TFH", "3_Lag3hi_Th1", "4_TFH1", "5_GC_TFH2", "6_Treg")))


# Gene Set: GO_OXIDATIVE_PHOSPHORYLATION
Oxphos <- readLines("Genesets/GO_Oxidative_phosphorylation.txt")[c(-1, -2)] # Remove first two lines, which are headers
Oxphos <- paste0(
 substring(Oxphos, 1, 1),
 tolower(substring(Oxphos, 2))) # Change capitalization
Oxphos <- Oxphos[which(Oxphos %in% rownames(immune.combined_edit2_2cell@data))] # Restrict to genes present in our data
Oxphos <- list(Oxphos)
immune.combined_edit2_2cell <- AddModuleScore(immune.combined_edit2_2cell, genes.list = Oxphos, enrich.name = "Oxphos")


pdf("summary_5mice/Figure2/Geneset_Oxidative_Phosphorylation_S2F.pdf", width = 8, height = 6)
VlnPlot(immune.combined_edit2_2cell, c("Oxphos1"), ident.include = c("2_Pre_TFH", "4_TFH1", "5_GC_TFH2"), cols.use = c("#53B400", "#00B6EB", "#A58AFF"), point.size.use = 0) +
  geom_boxplot(notch = T, fill = "white", alpha = 0.3)
compare_means(Oxphos1 ~ cluster_ident, data = (immune.combined_edit2_2cell@meta.data %>% select(Oxphos1, cluster_ident)))
# p-values from Wilcoxon test


dev.off()

# Gene set GO_cell_cycle_process
CC_Process_GO <- readLines("Genesets/GO_cell_cycle_process.txt")[c(-1, -2)] # Remove first two lines, which are headers
CC_Process_GO <- paste0(
 substring(CC_Process_GO, 1, 1),
 tolower(substring(CC_Process_GO, 2))) # Change capitalization
CC_Process_GO <- CC_Process_GO[which(CC_Process_GO %in% rownames(immune.combined_edit2_2cell@data))] # Restrict to genes present in our data
CC_Process_GO <- list(CC_Process_GO)
immune.combined_edit2_2cell <- AddModuleScore(immune.combined_edit2_2cell, genes.list = CC_Process_GO, enrich.name = "CC_Process_GO")


pdf("summary_5mice/Figure2/Geneset_GO_CC_process_S2F.pdf", width = 8, height = 6)
VlnPlot(immune.combined_edit2_2cell, c("CC_Process_GO1"), ident.include = c("2_Pre_TFH", "4_TFH1", "5_GC_TFH2"), cols.use = c("#53B400", "#00B6EB", "#A58AFF"), point.size.use = 0) +
  geom_boxplot(notch = T, fill = "white", alpha = 0.3)
compare_means(CC_Process_GO1 ~ cluster_ident, data = (immune.combined_edit2_2cell@meta.data %>% select(CC_Process_GO1, cluster_ident)))

dev.off()

```

```{r make DF to use for heatmap, clone size dist and Motif}

rm(list=ls())

### subset immune.combined_2cell onject to remove Treg to make motif analysis DF  comparble with monocle analysis;

immune.combined_edit2_2cell <- readRDS("summary_5M_CCA_V2/immune.combined_edit2_2cell.RDS") # this has Treg and all clones >= 2 cells

length(unique(immune.combined_edit2_2cell@meta.data$combined_cdr3_nt)) # 682 clones 

## remove Treg

immune.combined_edit3_2cell <- SubsetData(immune.combined_edit2_2cell, ident.remove = "6_Treg")

length(unique(immune.combined_edit3_2cell@meta.data$combined_cdr3_nt)) # 670 clones, so there are 12 clones with >=2 cells for Treg cluster

saveRDS(immune.combined_edit3_2cell, file = "summary_5M_CCA_V2/immune.combined_edit3_2cell.RDS") # This has no Treg

```

```{r}
rm(list=ls())
## load seurat and monocle object to make DF with all mouse data to be used for Motif, heatmap and clone size dist
#rm(immune.combined_edit2_2cell)
immune.combined_edit3_2cell <- readRDS("summary_5M_CCA_V2/immune.combined_edit3_2cell.RDS")

data_monocle <- readRDS("summary_5M_CCA_V2/data_monocle_2cell.RDS")

## state 1 Th1; 2,3 Tcmp; 4,5 TFH


### add cell ident to monocle branches
data_monocle$Branch_3subset <- data_monocle$State %>% as.character()
data_monocle$Branch_3subset <- gsub(pattern = "4|5", replacement = "TFH", x = data_monocle$Branch_3subset)
data_monocle$Branch_3subset <- gsub(pattern = "2|3", replacement = "Tcmp", x = data_monocle$Branch_3subset)
data_monocle$Branch_3subset <- gsub(pattern = "1", replacement = "Th1", x = data_monocle$Branch_3subset)
data_monocle$Branch_3subset %>% unique()
#[1] "TFH"  "Th1"  "Tcmp"

```

```{r make DF to use for heatmap, clone size dist and Motif}

rm(data_clones_by_mouse_wide_binary)

# Make new DF that includes mouse identity
data_clones_by_mouse <- immune.combined_edit3_2cell@meta.data %>% select(combined_cdr3_nt, mice) %>% table() %>% as.data.frame()

### for motif analysis we need CDR3_TRA #############3
colnames(immune.combined_edit3_2cell@meta.data)
data_clones_by_mouse_TRA <- immune.combined_edit3_2cell@meta.data %>% select(combined_cdr3_nt, cdr3_TRA, mice) %>% table() %>% as.data.frame()
# Remove "No CDR3" rows
#data_clones_by_mouse <- data_clones_by_mouse[-which(data_clones_by_mouse$combined_cdr3_nt == "No CDR3"), ]

# table() double counted some cells but left the counts as 0; remove these
data_clones_by_mouse <- data_clones_by_mouse[which(data_clones_by_mouse$Freq > 0), ]
#data_clones_by_mouse_AA <- data_clones_by_mouse_AA[which(data_clones_by_mouse_AA$Freq > 0), ]
data_clones_by_mouse_TRA <- data_clones_by_mouse_TRA[which(data_clones_by_mouse_TRA$Freq > 0), ]

# data_clones_by_mouse <- data_clones_by_mouse[which(data_clones_by_mouse$Freq > 1), ]
# Order in decreasing order
data_clones_by_mouse <- data_clones_by_mouse[order(data_clones_by_mouse$Freq, decreasing = T), ]
data_clones_by_mouse <- data_clones_by_mouse[which(data_clones_by_mouse$Freq > 1), ]
## sometime some clones with 1 cell gets included, remove that




#data_clones_by_mouse_AA <- data_clones_by_mouse_AA[order(data_clones_by_mouse_AA$Freq, decreasing = T), ]
data_clones_by_mouse_TRA <- data_clones_by_mouse_TRA[order(data_clones_by_mouse_TRA$Freq, decreasing = T), ]
data_clones_by_mouse_TRA <- data_clones_by_mouse_TRA[which(data_clones_by_mouse_TRA$Freq > 1), ]
## sometime some clones with 1 cell gets included, remove that


length(unique(data_clones_by_mouse$combined_cdr3_nt)) # 669 clones, after removing those clones with 1 cell


# Convert to wide format data
data_clones_by_mouse_wide <- data_clones_by_mouse %>% spread(mice, Freq, fill = 0)

# Check overlap
data_clones_by_mouse_wide_binary <- data_clones_by_mouse_wide
for (i in 2:ncol(data_clones_by_mouse_wide_binary)) {
  data_clones_by_mouse_wide_binary[, i][which(data_clones_by_mouse_wide_binary[, i] > 0)] <- 1
}

which(rowSums(data_clones_by_mouse_wide_binary[, -1]) > 1) %>% length() # 4
data_clones_by_mouse_wide_binary[which(rowSums(data_clones_by_mouse_wide_binary[, -1]) > 1), ]
# 4 overlaps with both chains


data_clones_by_mouse[which(data_clones_by_mouse$combined_cdr3_nt == "TRA:TGTGCAGCAAGCATCTATGCCCAGGGATTAACCTTC_TRB:TGTGCCAGCGGTGATAACCAAGACACCCAGTACTTT"), ] # 6 cells (3in each) in M4 and M6

data_clones_by_mouse[which(data_clones_by_mouse$combined_cdr3_nt == "TRA:TGTGCAGCAAGCCCCAATACCAACAAAGTCGTCTTT_TRB:TGTGCCAGCGGTGATTGGGGGGTCCAAGACACCCAGTACTTT"), ] # 130 cells in M6 and 81 cells in M4


data_clones_by_mouse[which(data_clones_by_mouse$combined_cdr3_nt == "TRA:TGTGCAGCAAGTAATAACTATGCCCAGGGATTAACCTTC_TRB:TGTGCCAGCAGTGACAGGGGGCCCAACGAAAGATTATTTTTC"), ] # 65 cells in M3 and 20 cells in M1
###

data_clones_by_mouse[which(data_clones_by_mouse$combined_cdr3_nt == "TRA:TGTGCAGCAAGTGGGAACAACAATGCCCCACGATTT_TRB:TGTGCCAGCAGTGACCATCAAGACACCCAGTACTTT"), ] # 9 cells in M3 and 8 cells in M4
###




#which(rowSums(data_clones_by_mouse_wide_binary[, -1]) > 1) %>% length() # 0 overlap now

which(data_clones_by_mouse$mice== "M1") %>% length() # 131 
which(data_clones_by_mouse$mice == "M2") %>% length() # 138
which(data_clones_by_mouse$mice == "M3") %>% length() # 100
which(data_clones_by_mouse$mice == "M4") %>% length() # 155 

which(data_clones_by_mouse$mice == "M6") %>% length() # 149



## so total 523 clones with >= 2cell/clone
### MK - 529 clones in total, but unique is 526 clones


# Subset the first data frame to only include clones with both TRA and TRB
rm(data_clones_by_mouse_TRA_TRB)
data_clones_by_mouse_TRA_TRB <- data_clones_by_mouse[grep(pattern = "_", x = data_clones_by_mouse$combined_cdr3_nt), ]
#data_clones_by_mouse_AA_TRA_TRB <- data_clones_by_mouse_AA[grep(pattern = "_", x = data_clones_by_mouse_AA$combined_cdr3_nt), ]
### MK - edited here, the DF in the command should be data_clones_by_mouse, since this is what you're subsetting
### MK - Same DF as before because all these clones have both TRA and TRB

data_clones_by_mouse_AA_TRA_TRB <- data_clones_by_mouse_TRA[grep(pattern = "_", x = data_clones_by_mouse_TRA$combined_cdr3_nt), ]

length(unique(immune.combined_edit3_2cell@meta.data$combined_cdr3_nt)) # 670
length(unique(data_clones_by_mouse_TRA_TRB$combined_cdr3_nt)) # 669 ## this is because of the removal of 1 clone with 1 cell; showing 673 total if not unique, cause overlapped clones
length(unique(data_clones_by_mouse_AA_TRA_TRB$combined_cdr3_nt)) # 669


table(data_clones_by_mouse_TRA_TRB$mice)
# M1  M2  M3  M4  M6 
#131 138 100 155 149 


### MK - modified this for loop to account for both CDR3 NT sequence and orig.ident
# Make columns showing how many cells are in each cluster for each clone
### MK - Because these clusters aren't named, I'm adding "Cluster " in front to avoid issues with columns having numbers for names

#data_clones_by_mouse_TRA_TRB[, paste0("Cluster_", levels(Acute_6_mice_edit5_2cell@active.ident))] <- 0
#data_clones_by_mouse_TRA_TRB[, levels(Acute_6_mice_edit5_2cell@active.ident)] <- 0
#for (i in 1:nrow(Acute_6_mice_edit5_2cell@meta.data)) {
 # if (grepl(pattern = "_", x = Acute_6_mice_edit5_2cell$combined_cdr3_nt[i])) {
  #  data_clones_by_mouse_TRA_TRB[
   #  which(data_clones_by_mouse_TRA_TRB$combined_cdr3_nt == Acute_6_mice_edit5_2cell$combined_cdr3_nt[i] & #data_clones_by_mouse_TRA_TRB$orig.ident == Acute_6_mice_edit5_2cell$orig.ident[i]),
     # paste0("Cluster_", Acute_6_mice_edit5_2cell@active.ident[i])] <-
     # data_clones_by_mouse_TRA_TRB[
    #  which(data_clones_by_mouse_TRA_TRB$combined_cdr3_nt == Acute_6_mice_edit5_2cell$combined_cdr3_nt[i] & data_clones_by_mouse_TRA_TRB$orig.ident == Acute_6_mice_edit5_2cell$orig.ident[i]),
  #    paste0("Cluster_", Acute_6_mice_edit5_2cell@active.ident[i])] + 1
 # }
#}


### trying this cuase now the idents have changed cluster names so no need to have paste0 option
data_clones_by_mouse_TRA_TRB[, levels(immune.combined_edit3_2cell@ident)] <- 0
for (i in 1:nrow(immune.combined_edit3_2cell@meta.data)) {
  if (grepl(pattern = "_", x = immune.combined_edit3_2cell@meta.data$combined_cdr3_nt[i])) {
    data_clones_by_mouse_TRA_TRB [
      which(data_clones_by_mouse_TRA_TRB$combined_cdr3_nt == immune.combined_edit3_2cell@meta.data$combined_cdr3_nt[i] & data_clones_by_mouse_TRA_TRB$mice == immune.combined_edit3_2cell@meta.data$mice[i]),
      as.character(immune.combined_edit3_2cell@ident[i])] <-
      data_clones_by_mouse_TRA_TRB [
      which(data_clones_by_mouse_TRA_TRB$combined_cdr3_nt == immune.combined_edit3_2cell@meta.data$combined_cdr3_nt[i] & data_clones_by_mouse_TRA_TRB$mice == immune.combined_edit3_2cell@meta.data$mice[i]),
       as.character(immune.combined_edit3_2cell@ident[i])] + 1
  }
}


## do same thing on data_clones_by_mouse_AA_TRA_TRB for motif analysis###

data_clones_by_mouse_AA_TRA_TRB[, levels(immune.combined_edit3_2cell@ident)] <- 0
for (i in 1:nrow(immune.combined_edit3_2cell@meta.data)) {
  if (grepl(pattern = "_", x = immune.combined_edit3_2cell@meta.data$combined_cdr3_nt[i])) {
   data_clones_by_mouse_AA_TRA_TRB[
     which(data_clones_by_mouse_AA_TRA_TRB$combined_cdr3_nt == immune.combined_edit3_2cell@meta.data$combined_cdr3_nt[i] & data_clones_by_mouse_AA_TRA_TRB$mice == immune.combined_edit3_2cell@meta.data$mice[i]),
      as.character(immune.combined_edit3_2cell@ident[i])] <-
      data_clones_by_mouse_AA_TRA_TRB[
      which(data_clones_by_mouse_AA_TRA_TRB$combined_cdr3_nt == immune.combined_edit3_2cell@meta.data$combined_cdr3_nt[i] & data_clones_by_mouse_AA_TRA_TRB$mice == immune.combined_edit3_2cell@meta.data$mice[i]),
    as.character (immune.combined_edit3_2cell@ident[i])] + 1
 }
}



# Make columns that show what percent of each clone is in each cluster
data_clones_by_mouse_TRA_TRB[, paste0("Percent_", levels(immune.combined_edit3_2cell@ident))] <- 0
data_clones_by_mouse_TRA_TRB[, paste0("Percent_", levels(immune.combined_edit3_2cell@ident))] <-
  data_clones_by_mouse_TRA_TRB[, as.character(levels(immune.combined_edit3_2cell@ident))] / data_clones_by_mouse_TRA_TRB[, "Freq"] * 100
 
# Make columns that show what percent of each clone is in each cluster ### for motif analysis
data_clones_by_mouse_AA_TRA_TRB[, paste0("Percent_", levels(immune.combined_edit3_2cell@ident))] <- 0
data_clones_by_mouse_AA_TRA_TRB[, paste0("Percent_", levels(immune.combined_edit3_2cell@ident))] <-
 data_clones_by_mouse_AA_TRA_TRB[, as.character (levels(immune.combined_edit3_2cell@ident))] / data_clones_by_mouse_AA_TRA_TRB[, "Freq"] * 100

### MK - typo in line 1390, should be levels(Acute_4_mice_edit3_2cell@active.ident), not just Acute_4_mice_edit3_2cell@active.ident

### MK - Check to make sure percentages add up to 100 for each row
which(rowSums(data_clones_by_mouse_TRA_TRB[, 10:15]) == 100) %>% length()
### MK - This only gives 429, but if you check rowsums that supposedly don't add up to 100 manually, they all do add up to 100
which(rowSums(data_clones_by_mouse_TRA_TRB[, 10:15]) != 100) %>% length()
which(rowSums(data_clones_by_mouse_TRA_TRB[, 10:15]) != 100) %>% head()
rowSums(data_clones_by_mouse_TRA_TRB[10, 10:15])
rowSums(data_clones_by_mouse_TRA_TRB[20, 10:15])

### MK - This is probably due to a floating point arithmetic error
### MK - To avoid this, we can check if the rowsums are within numerical error of 100
which(100 - rowSums(data_clones_by_mouse_TRA_TRB[, 10:15]) < 0.00000000001) %>% length()
### MK - All rowsums are within 0.00000000001 of 100, so I think I'm correct in saying this is a numeric error caused by floating point operations





```


```{r}
# Merge Monocle state with Seurat metadata
# Branch_3subset is the same thing as state, but renamed for convenience
# First make sure the rownames are in the same order (they are)

all.equal(
  rownames(immune.combined_edit3_2cell@meta.data),
  rownames(data_monocle@phenoData@data))


# checking whether combined_cdr3nt match or not 
#x <- as.factor(data_clones_by_mouse_TRA_TRB$combined_cdr3_nt)
#y <- as.factor(Acute_4_mice_edit3_2cell$combined_cdr3_nt)

#all.equal(x,y) 

immune.combined_edit3_2cell@meta.data$Monocle_State <- data_monocle$Branch_3subset %>% factor()
### MK - need to convert this to a factor, currently a character vector

```


```{r}
# Now incorporate this data into the clone data frame
# Make columns showing how many cells are in each Monocle State for each clone
# remove the overlapped 2 clones from Acute_4_mice_edit3_2cell and data_monocle to match with data_clones_by_mouse_TRA_TRB


# Acute_4_mice_edit4_2cell <- subset(Acute_4_mice_edit3_2cell, cells = rownames(Acute_4_mice_edit3_2cell@meta.data)[which(Acute_4_mice_edit3_2cell$combined_cdr3_nt %in% data_clones_by_mouse_TRA_TRB$combined_cdr3_nt)])
# 
# dim(Acute_4_mice_edit4_2cell) # 8003
# 
# Acute_4_mice_edit4_2cell$Monocle_State <- data_monocle$Branch_3subset
# 
# data_monocle_edit <- subset(data_monocle, cells = rownames(data_monocle@phenoData@data)[which(data_monocle@phenoData@data$combined_cdr3_nt %in% data_clones_by_mouse_TRA_TRB$combined_cdr3_nt)]) # this command does not work



## trying w/o getting rid of overlapped clones

data_clones_by_mouse_TRA_TRB[, paste0("Monocle_", levels(immune.combined_edit3_2cell@meta.data$Monocle_State))] <- 0
data_clones_by_mouse_AA_TRA_TRB[, paste0("Monocle_", levels(immune.combined_edit3_2cell@meta.data$Monocle_State))] <- 0


### MK - modified this for loop to account for both CDR3 NT sequence and orig.ident
for (i in 1:nrow(immune.combined_edit3_2cell@meta.data)) {
  data_clones_by_mouse_TRA_TRB[
    which(data_clones_by_mouse_TRA_TRB$combined_cdr3_nt == immune.combined_edit3_2cell@meta.data$combined_cdr3_nt[i] & data_clones_by_mouse_TRA_TRB$mice == immune.combined_edit3_2cell@meta.data$mice[i]),
    paste0("Monocle_", immune.combined_edit3_2cell@meta.data$Monocle_State[i])] <-
    data_clones_by_mouse_TRA_TRB[
      which(data_clones_by_mouse_TRA_TRB$combined_cdr3_nt == immune.combined_edit3_2cell@meta.data$combined_cdr3_nt[i] & data_clones_by_mouse_TRA_TRB$mice == immune.combined_edit3_2cell@meta.data$mice[i]),
      paste0("Monocle_", immune.combined_edit3_2cell@meta.data$Monocle_State[i])] + 1
}


### repeat this for data_clones_by_mouse_AA_TRA_TRB ### for motif analysis
for (i in 1:nrow(immune.combined_edit3_2cell@meta.data)) {
  data_clones_by_mouse_AA_TRA_TRB[
    which(data_clones_by_mouse_AA_TRA_TRB$combined_cdr3_nt == immune.combined_edit3_2cell@meta.data$combined_cdr3_nt[i] & data_clones_by_mouse_AA_TRA_TRB$mice == immune.combined_edit3_2cell@meta.data$mice[i]),
    paste0("Monocle_", immune.combined_edit3_2cell@meta.data$Monocle_State[i])] <-
    data_clones_by_mouse_AA_TRA_TRB[
      which(data_clones_by_mouse_AA_TRA_TRB$combined_cdr3_nt == immune.combined_edit3_2cell@meta.data$combined_cdr3_nt[i] & data_clones_by_mouse_AA_TRA_TRB$mice == immune.combined_edit3_2cell@meta.data$mice[i]),
      paste0("Monocle_", immune.combined_edit3_2cell@meta.data$Monocle_State[i])] + 1
}


# Make sure the counting worked properly - Good
which(data_clones_by_mouse_TRA_TRB$Freq == rowSums(data_clones_by_mouse_TRA_TRB[, paste0("Monocle_", levels(immune.combined_edit3_2cell@meta.data$Monocle_State))])) %>% length() # 673

data_clones_by_mouse_TRA_TRB$Freq %>% length() 
### 673


# Make columns that show what percent of each clone is in each Monocle Cell
### MK - changed data_CD4_2cell_edit to Acute_6_mice_edit5_2cell
data_clones_by_mouse_TRA_TRB[, paste0("Percent_Monocle_", levels(immune.combined_edit3_2cell@meta.data$Monocle_State))] <- 0
data_clones_by_mouse_TRA_TRB[, paste0("Percent_Monocle_", levels(immune.combined_edit3_2cell@meta.data$Monocle_State))] <-
    data_clones_by_mouse_TRA_TRB[, paste0("Monocle_", levels(immune.combined_edit3_2cell@meta.data$Monocle_State))] / data_clones_by_mouse_TRA_TRB$Freq * 100

data_clones_by_mouse_AA_TRA_TRB[, paste0("Percent_Monocle_", levels(immune.combined_edit3_2cell@meta.data$Monocle_State))] <- 0
data_clones_by_mouse_AA_TRA_TRB[, paste0("Percent_Monocle_", levels(immune.combined_edit3_2cell@meta.data$Monocle_State))] <-
 data_clones_by_mouse_AA_TRA_TRB[, paste0("Monocle_", levels(immune.combined_edit3_2cell@meta.data$Monocle_State))] / data_clones_by_mouse_AA_TRA_TRB$Freq * 100

### MK - Check to make sure percentages add up to 100 for each row
which(rowSums(data_clones_by_mouse_TRA_TRB[, 19:21]) == 100) %>% length()
### MK - This only gives 446, but if you check rowsums that supposedly don't add up to 100 manually, they all do add up to 100
which(rowSums(data_clones_by_mouse_TRA_TRB[, 19:21]) != 100) %>% length()
which(rowSums(data_clones_by_mouse_TRA_TRB[, 19:21]) != 100) %>% head()
rowSums(data_clones_by_mouse_TRA_TRB[7, 19:21])
rowSums(data_clones_by_mouse_TRA_TRB[15, 19:21])

### MK - This is probably due to a floating point arithmetic error
### MK - To avoid this, we can check if the rowsums are within numerical error of 100
which(100 - rowSums(data_clones_by_mouse_TRA_TRB[, 19:21]) < 0.00000000001) %>% length()
### MK - All rowsums are within 0.00000000001 of 100, so I think I'm correct in saying this is a numeric error caused by floating point operations

write.csv(data_clones_by_mouse_TRA_TRB, file = "summary_5M_CCA_V2/data_clones_by_mouse_TRA_TRB_5mice.csv")
write.csv(data_clones_by_mouse_AA_TRA_TRB, file = "summary_5M_CCA_V2/data_clones_by_mouse_AA_TRA_TRB_5mice.csv") 
# saveRDS(data_monocle, "data_CD4_2cell_Monocle.RDS")
### MK - Didn't save this RDS object

```

```{r}
## add a column with combined CDR3nt and orig ident 2gether to track the overlapped clones across mice


data_clones_by_mouse_AA_TRA_TRB$CDR3nt_Ident <- paste0(
  data_clones_by_mouse_AA_TRA_TRB$combined_cdr3_nt,
  data_clones_by_mouse_AA_TRA_TRB$mice,
  sep = "-"
)

write.csv(data_clones_by_mouse_AA_TRA_TRB, file = "summary_5M_CCA_V2/data_clones_by_mouse_AA_TRA_TRB_5mice.csv") 

# Add a clone_number row formatted as 001, 002, etc to keep track of clones by decreasing size - needed because R is bad at ordering things numerically


data_clones_by_mouse_TRA_TRB$Clone_number <- paste0(
  "Clone_",
  sprintf("%03d", 1:nrow(data_clones_by_mouse_TRA_TRB))
)

write.csv(data_clones_by_mouse_TRA_TRB, file = "summary_5M_CCA_V2/data_clones_by_mouse_TRA_TRB_5mice.csv")
```

```{r clone dist pie charts}

data_clones_by_mouse_TRA_TRB <- read.csv("summary_5M_CCA_V2/data_clones_by_mouse_TRA_TRB_5mice.csv")

## change cluster ident names

colnames(data_clones_by_mouse_TRA_TRB)[colnames(data_clones_by_mouse_TRA_TRB) == "X0_Tcmp"] <- "Cluster0_Tcmp"
colnames(data_clones_by_mouse_TRA_TRB)[colnames(data_clones_by_mouse_TRA_TRB) == "X1_Ly6cHi_Th1"] <- "Cluster1_Ly6cHi_Th1"
colnames(data_clones_by_mouse_TRA_TRB)[colnames(data_clones_by_mouse_TRA_TRB) == "X2_Pre_TFH"] <- "Cluster2_Pre_TFH"
colnames(data_clones_by_mouse_TRA_TRB)[colnames(data_clones_by_mouse_TRA_TRB) ==  "X3_Lag3hi_Th1"] <- "Cluster3_Lag3hi_Th1"
colnames(data_clones_by_mouse_TRA_TRB)[colnames(data_clones_by_mouse_TRA_TRB) ==   "X4_TFH1"] <- "Cluster4_TFH1"
colnames(data_clones_by_mouse_TRA_TRB)[colnames(data_clones_by_mouse_TRA_TRB) ==  "X5_GC_TFH2"] <- "Cluster5_GC_TFH2"


Clones_piechart_list <- data_clones_by_mouse_TRA_TRB %>% group_split(mice) %>% lapply(`[`, i = 1:50, j = )

Clones_piechart_list[[1]] %>% View
Clones_piechart_list[[1]]$Freq %>% sum() # 824
Clones_piechart_list[[1]]$Freq %>% length() # 50

# M1 - 824 cells
ggplot(Clones_piechart_list[[1]], aes(x = "", y = Freq, fill = Clone_number)) +
  geom_bar(width = 0.25, stat = "identity", color = "black") +
  geom_text(
    aes(
      y = Freq,
      label = ifelse(
        Freq > 20,
        paste0(
          round(Freq/824 * 100, 1),
          "%"),
        "")),
    position = position_stack(vjust = 0.6),
    size = 5) +
  theme_minimal() +
  theme(
    axis.title = element_blank(),
    axis.text = element_blank(),
    axis.ticks = element_blank(),
    axis.line = element_blank(),
    panel.grid = element_blank(),
    legend.position = "None")
ggsave("summary_5mice/Figure1/Figure1_M1_clones.pdf", dpi = 300, bg = "transparent", width = 5, height = 5)

Clones_piechart_list[[2]] %>% View
Clones_piechart_list[[2]]$Freq %>% sum() # 778
Clones_piechart_list[[2]]$Freq %>% length() # 50

# M2 - 778 cells
ggplot(Clones_piechart_list[[2]], aes(x = "", y = Freq, fill = Clone_number)) +
  geom_bar(width = 0.25, stat = "identity", color = "black") +
  geom_text(
    aes(
      y = Freq,
      label = ifelse(
        Freq > 20,
        paste0(
          round(Freq/778 * 100, 1),
          "%"),
        "")),
    position = position_stack(vjust = 0.5),
    size = 5) +
  theme_minimal() +
  theme(
    axis.title = element_blank(),
    axis.text = element_blank(),
    axis.ticks = element_blank(),
    axis.line = element_blank(),
    panel.grid = element_blank(),
    legend.position = "None")
ggsave("summary_5mice/Figure1/Figure1_M2_clones.pdf", dpi = 300, bg = "transparent", width = 5, height = 5)

Clones_piechart_list[[3]] %>% View
Clones_piechart_list[[3]]$Freq %>% sum() # 2376


# M3 - 2376 cells
ggplot(Clones_piechart_list[[3]], aes(x = "", y = Freq, fill = Clone_number)) +
  geom_bar(width = 0.25, stat = "identity", color = "black") +
  geom_text(
    aes(
      y = Freq,
      label = ifelse(
        Freq > 50,
        paste0(
          round(Freq/2376 * 100, 1),
          "%"),
        "")),
    position = position_stack(vjust = 0.5),
    size = 5) +
  theme_minimal() +
  theme(
    axis.title = element_blank(),
    axis.text = element_blank(),
    axis.ticks = element_blank(),
    axis.line = element_blank(),
    panel.grid = element_blank(),
    legend.position = "None")
ggsave("summary_5mice/Figure1/Figure1_M3_clones.pdf", dpi = 300, bg = "transparent", width = 5, height = 5)

Clones_piechart_list[[4]] %>% View
Clones_piechart_list[[4]]$Freq %>% sum() # 3105

# M4 - 3105 cells
ggplot(Clones_piechart_list[[4]], aes(x = "", y = Freq, fill = Clone_number)) +
  geom_bar(width = 0.25, stat = "identity", color = "black") +
  geom_text(
    aes(
      y = Freq,
      label = ifelse(
        Freq > 70,
        paste0(
          round(Freq/3105 * 100, 1),
          "%"),
        "")),
    position = position_stack(vjust = 0.5),
    size = 5) +
  theme_minimal() +
  theme(
    axis.title = element_blank(),
    axis.text = element_blank(),
    axis.ticks = element_blank(),
    axis.line = element_blank(),
    panel.grid = element_blank(),
    legend.position = "None")
ggsave("summary_5mice/Figure1/Figure1_M4_clones.pdf", dpi = 300, bg = "transparent", width = 5, height = 5)


Clones_piechart_list[[5]] %>% View
Clones_piechart_list[[5]]$Freq %>% sum() # 2139

# M6 - 2139 cells
ggplot(Clones_piechart_list[[5]], aes(x = "", y = Freq, fill = Clone_number)) +
  geom_bar(width = 0.25, stat = "identity", color = "black") +
  geom_text(
    aes(
      y = Freq,
      label = ifelse(
        Freq > 50,
        paste0(
          round(Freq/2139 * 100, 1),
          "%"),
        "")),
    position = position_stack(vjust = 0.5),
    size = 5) +
  theme_minimal() +
  theme(
    axis.title = element_blank(),
    axis.text = element_blank(),
    axis.ticks = element_blank(),
    axis.line = element_blank(),
    panel.grid = element_blank(),
    legend.position = "None")
ggsave("summary_5mice/Figure1/Figure1_M5_clones.pdf", dpi = 300, bg = "transparent", width = 5, height = 5)



```


```{r check sahred V and J gene usage for both chains among mice}

immune.combined_edit2 <- readRDS("summary_5M_CCA_V2/immune.combined_edit2.RDS")

TSNEPlot(immune.combined_edit2)

## add TCR data analyzed by cell ranger V2 comparable to Seurat R1 seq data. I did not use cell ranger V2 TCR data
## in main analysis cause the V2 analysis lesses # of cells
# for shared gene usage and chord diagrams we only car about the gene names to match them among mice so should be fine


# load TCR data from R1 R2 R3 analyzed by cell ranger V2

rm(list=ls())

immune.combined_edit2 <- readRDS("summary_5M_CCA_V2/immune.combined_edit2.RDS")

CD4_M1_vdj <- read.csv("/media/runa/Elements/Jem_revision/Seurat/TCRseq/CD4_WT1_TCR_R1/filtered_contig_annotations.csv", stringsAsFactors = F)
CD4_M2_vdj <- read.csv("/media/runa/Elements/Jem_revision/Seurat/TCRseq/CD4_WT2_TCR_R1/filtered_contig_annotations.csv", stringsAsFactors = F)


CD4_M3_vdj <- read.csv("/media/runa/Elements/Jem_revision/Seurat/TCRseq/M1_TCR_R2_V2/filtered_contig_annotations.csv", stringsAsFactors = F)
CD4_M4_vdj <- read.csv("/media/runa/Elements/Jem_revision/Seurat/TCRseq/M2_TCR_R2_V2/filtered_contig_annotations.csv", stringsAsFactors = F)


#CD4_M5_vdj <- read.csv("/media/runa/Elements/Jem_revision/Seurat/TCRseq/CD4_WT1_TCR_R3/outs/filtered_contig_annotations.csv", stringsAsFactors = F)
CD4_M6_vdj <- read.csv("/media/runa/Elements/Jem_revision/Seurat/TCRseq/M2_TCR_R3_V2/filtered_contig_annotations.csv", stringsAsFactors = F)


# Rename cells within the VDJ data to match the cell names in Seurat
CD4_M1_vdj$barcode <-
  substr(CD4_M1_vdj$barcode, 0, nchar(CD4_M1_vdj$barcode)) %>%
  paste0("CD4_M1_", .)
CD4_M2_vdj$barcode <-
  substr(CD4_M2_vdj$barcode, 0, nchar(CD4_M2_vdj$barcode)) %>%
  paste0("CD4_M2_", .)
CD4_M3_vdj$barcode <-
  substr(CD4_M3_vdj$barcode, 0, nchar(CD4_M3_vdj$barcode)) %>%
  paste0("CD4_M3_", .)
CD4_M4_vdj$barcode <-
  substr(CD4_M4_vdj$barcode, 0, nchar(CD4_M4_vdj$barcode)) %>%
  paste0("CD4_M4_", .)
#CD4_M5_vdj$barcode <-
  #substr(CD4_M5_vdj$barcode, 0, nchar(CD4_M5_vdj$barcode)) %>%
  #paste0("CD4_M5_", .)
CD4_M6_vdj$barcode <-
  substr(CD4_M6_vdj$barcode, 0, nchar(CD4_M6_vdj$barcode)) %>%
  paste0("CD4_M6_", .)

#Rename cells within the VDJ data to match the cell names in Seurat
CD4_M1_vdj$barcode <- substr(CD4_M1_vdj$barcode,  0, nchar(CD4_M1_vdj$barcode) - 2)
CD4_M2_vdj$barcode <- substr(CD4_M2_vdj$barcode,  0, nchar(CD4_M2_vdj$barcode) - 2)

CD4_M3_vdj$barcode <- substr(CD4_M3_vdj$barcode,  0, nchar(CD4_M3_vdj$barcode) - 2)
CD4_M4_vdj$barcode <- substr(CD4_M4_vdj$barcode,  0, nchar(CD4_M4_vdj$barcode) - 2)


#CD4_M5_vdj$barcode <- substr(CD4_M5_vdj$barcode,  0, nchar(CD4_M5_vdj$barcode) - 2)
CD4_M6_vdj$barcode <- substr(CD4_M6_vdj$barcode,  0, nchar(CD4_M6_vdj$barcode) - 2)

# Combine the data frames
Acute_vdj <- rbind(CD4_M1_vdj[, c("barcode", "length", "chain", "v_gene", "d_gene", "j_gene", "c_gene", "full_length", "productive", "cdr3", "cdr3_nt", "reads", "umis")],
                 CD4_M2_vdj[, c("barcode", "length", "chain", "v_gene", "d_gene", "j_gene", "c_gene", "full_length", "productive", "cdr3", "cdr3_nt", "reads", "umis")],
                 CD4_M3_vdj[, c("barcode", "length", "chain", "v_gene", "d_gene", "j_gene", "c_gene", "full_length", "productive", "cdr3", "cdr3_nt", "reads", "umis")],
                 CD4_M4_vdj[, c("barcode", "length", "chain", "v_gene", "d_gene", "j_gene", "c_gene", "full_length", "productive", "cdr3", "cdr3_nt", "reads", "umis")],
                 #CD4_M5_vdj[, c("barcode", "length", "chain", "v_gene", "d_gene", "j_gene", "c_gene", "full_length", "productive", "cdr3", "cdr3_nt", "reads", "umis")],
                 CD4_M6_vdj[, c("barcode", "length", "chain", "v_gene", "d_gene", "j_gene", "c_gene", "full_length", "productive", "cdr3", "cdr3_nt", "reads", "umis")])

Acute_vdj$barcode %>% length() # 66238
Acute_vdj$barcode %>% unique() %>% length()
# 19244


# Keep only TRA and TRB chains, remove entries without a CDR3 sequence, and only keep cells that are present within the Seurat data
Acute_vdj <- Acute_vdj[Acute_vdj$chain == "TRA" | Acute_vdj$chain == "TRB", ]
Acute_vdj <- Acute_vdj[Acute_vdj$cdr3_nt != "None", ]
Acute_vdj <- Acute_vdj[Acute_vdj$barcode %in% rownames(immune.combined_edit2@meta.data), ]
Acute_vdj$barcode %>% length()
Acute_vdj$barcode %>% unique() %>% length()
# Now down to 30549 TCR UMIs, of which 13462 are unique

# We need only one TRA and TRB chain per cell. Convert full_length and productive columns to factors that rank the entries
Acute_vdj$full_length <- as.factor(Acute_vdj$full_length)
Acute_vdj$productive <- as.factor(Acute_vdj$productive)


# Create empty dataframes to subet the 'best' TRA and TRB chains to.
tra_df <- data.frame()
trb_df <- data.frame()
sub_vdj <- data.frame()

# For each cell, rank each TRA and TRB entry to return the single best chain for each. Ties are broken by the number of UMIs that support them.
for(i in unique(Acute_vdj$barcode)){
  tra_df <- Acute_vdj[Acute_vdj$barcode == i & Acute_vdj$chain == "TRA", ]
  trb_df <- Acute_vdj[Acute_vdj$barcode == i & Acute_vdj$chain == "TRB", ]
  sub_vdj <- rbind(sub_vdj, tra_df[order(tra_df$full_length, tra_df$productive, tra_df$umis, decreasing = TRUE), ][1, ])
  sub_vdj <- rbind(sub_vdj, trb_df[order(trb_df$full_length, trb_df$productive, trb_df$umis, decreasing = TRUE), ][1, ])
}

# Remove NA entries from results
sub_vdj <- sub_vdj[!is.na(sub_vdj$barcode), ]

# Add the current cell identity to the meta.data
immune.combined_edit2@meta.data$current.ident <- immune.combined_edit2@ident

# Create a copy of the meta.data for merging results
### Check the index before subsetting, should be the number of columns in meta data. Do this in case you need to reset the meta data
colnames(immune.combined_edit2@meta.data)
rm(meta_data)
meta_data <- immune.combined_edit2@meta.data[1:18] # this is good to make this DF with columns first


# Merge metadata with VDJ TRA results
meta_data <- merge(meta_data, sub_vdj[sub_vdj$chain == "TRA", c("barcode", "v_gene", "d_gene", "j_gene", "c_gene", "full_length", "productive", "cdr3", "cdr3_nt")], by.x = "row.names", by.y = "barcode", all.x = TRUE)
# Rename columns from TRA to specify they're from TRA
colnames(meta_data)[which(colnames(meta_data) %in% c("v_gene", "d_gene", "j_gene", "c_gene", "full_length", "productive", "cdr3", "cdr3_nt"))] <-
  c("v_gene_TRA", "d_gene_TRA", "j_gene_TRA", "c_gene_TRA", "full_length_TRA", "productive_TRA", "cdr3_TRA", "cdr3_nt_TRA")

# Merge metadata wtih VDJ TRB results
meta_data <- merge(meta_data, sub_vdj[sub_vdj$chain == "TRB", c("barcode", "v_gene", "d_gene", "j_gene", "c_gene", "full_length", "productive", "cdr3", "cdr3_nt")], by.x = "Row.names", by.y = "barcode", all.x = TRUE)
# Rename columns from TRB to specify they're from TRB
colnames(meta_data)[which(colnames(meta_data) %in% c("v_gene", "d_gene", "j_gene", "c_gene", "full_length", "productive", "cdr3", "cdr3_nt"))] <-
  c("v_gene_TRB", "d_gene_TRB", "j_gene_TRB", "c_gene_TRB", "full_length_TRB", "productive_TRB", "cdr3_TRB", "cdr3_nt_TRB")

# Reorder meta_data table (with vdj results) to match Seurat's meta.data
meta_data <- meta_data[order(match(meta_data$Row.names, rownames(immune.combined_edit2@meta.data))), ]

# Add the best TRA and TRB chains to Seurat's meta.data
immune.combined_edit2@meta.data <- cbind(immune.combined_edit2@meta.data, meta_data[,c("v_gene_TRA", "d_gene_TRA", "j_gene_TRA", "c_gene_TRA", "full_length_TRA", "productive_TRA", "cdr3_TRA", "cdr3_nt_TRA", "v_gene_TRB", "d_gene_TRB", "j_gene_TRB", "c_gene_TRB", "full_length_TRB", "productive_TRB", "cdr3_TRB", "cdr3_nt_TRB")])

# Add TRA and TRB identifiers to the sequences and combine both chains to produce a unique CDR3 nucleotide identifier
immune.combined_edit2@meta.data$combined_cdr3_nt <- paste(
  paste("TRA", immune.combined_edit2@meta.data$cdr3_nt_TRA, sep = ":"),
  paste("TRB", immune.combined_edit2@meta.data$cdr3_nt_TRB, sep = ":"),
  sep = "_")
# Some cells don't have a TRA or TRB chain nucleotide sequence - Replace these entries with "No_CDR3"
immune.combined_edit2@meta.data$combined_cdr3_nt <- gsub("_TRB:NA|TRA:NA_|TRB:NA|TRA:NA", "", immune.combined_edit2@meta.data$combined_cdr3_nt)
immune.combined_edit2@meta.data$combined_cdr3_nt[nchar(immune.combined_edit2@meta.data$combined_cdr3_nt) == 0] <- "No CDR3"

# Combine CDR3 AAs
immune.combined_edit2@meta.data$combined_cdr3_AA <- paste(
  paste("TRA", immune.combined_edit2@meta.data$cdr3_TRA, sep = ":"),
  paste("TRB", immune.combined_edit2@meta.data$cdr3_TRB, sep = ":"),
  sep = "_")
# Some cells don't have a TRA or TRB chain amino acid sequence - Replace these entries with "No_CDR3"
immune.combined_edit2@meta.data$combined_cdr3_AA <- gsub("_TRB:NA|TRA:NA_|TRB:NA|TRA:NA", "", immune.combined_edit2@meta.data$combined_cdr3_AA)
immune.combined_edit2@meta.data$combined_cdr3_AA[nchar(immune.combined_edit2@meta.data$combined_cdr3_AA) == 0] <- "No CDR3"

which(immune.combined_edit2@meta.data$combined_cdr3_nt != "No CDR3") %>% length()
immune.combined_edit2@meta.data$combined_cdr3_nt %>% length()
# 13462 cells have either TRA, TRB, or both out of 13636 cells total

grep(pattern = "_", x = immune.combined_edit2@meta.data$combined_cdr3_nt) %>% length()
# _ serves as a marker for cells with a complete TRA_TRB sequence
# 11623 cells have both TRA and TRB out of 13636 cells total (88%)

immune.combined_edit2@meta.data$combined_cdr3_nt %>% unique() %>% length()
# 2584 clones with either TRA, TRB, or both

grep(pattern = "_", x = immune.combined_edit2@meta.data$combined_cdr3_nt, value = T) %>% unique() %>% length()
# 2166 clones with both TRA and TRB

saveRDS(immune.combined_edit2, file = "summary_5M_CCA_V2/immune.combined_edit2_TCR_CRV2.RDS")



```



```{r check shared V-J gene usage for both chains among 5 mice}
rm(list=ls())
immune.combined_edit2 <- readRDS("summary_5M_CCA_V2/immune.combined_edit2_TCR_CRV2.RDS")
# Make new DF to use TRAV-J gene usage across mice
Acute_CD4_5mice_clones_by_mouse_Alpha <- immune.combined_edit2@meta.data %>% select(combined_cdr3_nt, mice, v_gene_TRA, j_gene_TRA) %>% table() %>% as.data.frame()
# Remove "No CDR3" rows
Acute_CD4_5mice_clones_by_mouse_Alpha <- Acute_CD4_5mice_clones_by_mouse_Alpha[-which(Acute_CD4_5mice_clones_by_mouse_Alpha$combined_cdr3_nt == "No CDR3"), ]
# table() double counted some cells but left the counts as 0; remove these
Acute_CD4_5mice_clones_by_mouse_Alpha <- Acute_CD4_5mice_clones_by_mouse_Alpha[which(Acute_CD4_5mice_clones_by_mouse_Alpha$Freq > 0 ), ]
# Order in decreasing order
Acute_CD4_5mice_clones_by_mouse_Alpha <- Acute_CD4_5mice_clones_by_mouse_Alpha[order(Acute_CD4_5mice_clones_by_mouse_Alpha$Freq, decreasing = T), ]


# Subset the first data frame to only include clones with both TRA and TRB
Acute_CD4_5mice_clones_by_mouse_Alpha_TRA_TRB <- Acute_CD4_5mice_clones_by_mouse_Alpha[grep(pattern = "_", x = Acute_CD4_5mice_clones_by_mouse_Alpha$combined_cdr3_nt), ]

# calculate # of clones in each mouse with both chains

table(Acute_CD4_5mice_clones_by_mouse_Alpha_TRA_TRB$mice)
 # M1  M2  M3  M4  M6 
# 393 376 437 512 486 

write.csv(Acute_CD4_5mice_clones_by_mouse_Alpha_TRA_TRB, file= "summary_5mice/Figure1/File_used_for_TCR_AV-J_gene_usage_8cell.csv")
rm(Acute_CD4_6mice_clones_by_mouse_Alpha_TRA_TRB_filter)


# Make new DF to use TRBV-J gene usage across mice
Acute_CD4_5mice_clones_by_mouse_Beta <- immune.combined_edit2@meta.data %>% select(combined_cdr3_nt, mice, v_gene_TRB, j_gene_TRB) %>% table() %>% as.data.frame()
# Remove "No CDR3" rows
Acute_CD4_5mice_clones_by_mouse_Beta <- Acute_CD4_5mice_clones_by_mouse_Beta[-which(Acute_CD4_5mice_clones_by_mouse_Beta$combined_cdr3_nt == "No CDR3"), ]
# table() double counted some cells but left the counts as 0; remove these
Acute_CD4_5mice_clones_by_mouse_Beta <- Acute_CD4_5mice_clones_by_mouse_Beta[which(Acute_CD4_5mice_clones_by_mouse_Beta$Freq > 0 ), ]
# Order in decreasing order
Acute_CD4_5mice_clones_by_mouse_Beta <- Acute_CD4_5mice_clones_by_mouse_Beta[order(Acute_CD4_5mice_clones_by_mouse_Beta$Freq, decreasing = T), ]


# Subset the first data frame to only include clones with both TRA and TRB
Acute_CD4_5mice_clones_by_mouse_Beta_TRA_TRB <- Acute_CD4_5mice_clones_by_mouse_Beta[grep(pattern = "_", x = Acute_CD4_5mice_clones_by_mouse_Beta$combined_cdr3_nt), ]

# calculate # of clones in each mouse with both chains

table(Acute_CD4_5mice_clones_by_mouse_Beta_TRA_TRB$mice)
#  M1  M2  M3  M4  M6 
# 386 372 436 497 484 

```


```{r check shared V-J gene usage for both chains among 5 mice }
Acute_CD4_5mice_clones_by_mouse_Alpha_TRA_TRB_filter <- Acute_CD4_5mice_clones_by_mouse_Alpha_TRA_TRB[which(Acute_CD4_5mice_clones_by_mouse_Alpha_TRA_TRB$Freq > 8), ]
# TRA V genes
ggplot(Acute_CD4_5mice_clones_by_mouse_Alpha_TRA_TRB_filter, aes(x = v_gene_TRA, y = Freq, fill = mice)) +
  geom_bar(stat = "identity", position = "dodge") +
  theme_minimal() +
  theme(
    panel.grid.major.x = element_blank(),
    panel.grid.minor.x = element_blank(),
    panel.grid.major.y = element_blank(),
    panel.grid.minor.y = element_blank(),
    # axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5, margin = margin(t = -8)),
    axis.text.x = element_text(angle = 0, hjust = 0.5, vjust = 0.5, margin = margin(t = 0)),
    legend.position = "None") +
  xlab("TCR Alpha V genes") +
  ylab("Number of cells") +
  scale_fill_manual(
    name = "Sample",
    labels = c("M1", "M2", "M3", "M4", "M5"),
    values = c("#1B9E77", "#D95F02", "#7570B3", "#E7298A", "#66A61E")) +
  coord_flip() 

ggsave("summary_5mice/Figure1/TRA_V_genes_5mice.pdf", dpi = 300, width = 4 * 4, height = 3.5 * 3.5) # USE THIS ONE

# TRA J genes
ggplot(Acute_CD4_5mice_clones_by_mouse_Alpha_TRA_TRB_filter, aes(x = j_gene_TRA, y = Freq, fill = mice)) +
  geom_bar(stat = "identity", position = "dodge") +
  theme_minimal() +
  theme(
    panel.grid.major.x = element_blank(),
    panel.grid.minor.x = element_blank(),
    panel.grid.major.y = element_blank(),
    panel.grid.minor.y = element_blank(),
    # axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5, margin = margin(t = -8)),
    axis.text.x = element_text(angle = 0, hjust = 0.5, vjust = 0.5, margin = margin(t = 0)),
    legend.position = "None") +
  xlab("TCR Alpha J genes") +
  ylab("Number of cells") +
  scale_fill_manual(
    name = "Sample",
    labels = c("M1", "M2", "M3", "M4", "M5"),
    values = c("#1B9E77", "#D95F02", "#7570B3", "#E7298A", "#66A61E")) +
  coord_flip()
# ggsave("Chronic_CD8_2cell_pure/Figures/04_TRA_J_genes.pdf", dpi = 300, width = 3.5 * 2.5, height = 2 * 2.5)
ggsave("summary_5mice/Figure1/TRA_J_genes_5mice.pdf", dpi = 300, width = 4 * 4, height = 3.5 * 3.5) # USE THIS ONE


Acute_CD4_6mice_clones_by_mouse_Beta_TRA_TRB_filter <- Acute_CD4_6mice_clones_by_mouse_Beta_TRA_TRB[which(Acute_CD4_6mice_clones_by_mouse_Beta_TRA_TRB$Freq > 8), ]

# TRB V genes
ggplot(Acute_CD4_5mice_clones_by_mouse_Beta_TRA_TRB, aes(x = v_gene_TRB, y = Freq, fill = mice)) +
  geom_bar(stat = "identity", position = "dodge") +
  theme_minimal() +
  theme(
    panel.grid.major.x = element_blank(),
    panel.grid.minor.x = element_blank(),
    panel.grid.major.y = element_blank(),
    panel.grid.minor.y = element_blank(),
    # axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5, margin = margin(t = -8)),
    axis.text.x = element_text(angle = 0, hjust = 0.5, vjust = 0.5, margin = margin(t = 0)),
    legend.position = "None") +
  xlab("TCR Beta V genes") +
  ylab("Number of cells") +
  scale_fill_manual(
    name = "Sample",
    labels = c("M1", "M2", "M3", "M4", "M5"),
    values = c("#1B9E77", "#D95F02", "#7570B3", "#E7298A", "#66A61E")) +
  coord_flip()
# ggsave("Chronic_CD8_2cell_pure/Figures/04_TRB_V_genes.pdf", dpi = 300, width = 3.5 * 2.5, height = 2 * 2.5)
ggsave("summary_5mice/Figure1/TRB_V_genes_5mice.pdf", dpi = 300, width = 4 * 4, height = 3.5 * 3.5) # USE THIS ONE

# TRA J genes
ggplot(Acute_CD4_5mice_clones_by_mouse_Beta_TRA_TRB, aes(x = j_gene_TRB, y = Freq, fill = mice)) +
  geom_bar(stat = "identity", position = "dodge") +
  theme_minimal() +
  theme(
    panel.grid.major.x = element_blank(),
    panel.grid.minor.x = element_blank(),
    panel.grid.major.y = element_blank(),
    panel.grid.minor.y = element_blank(),
    # axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5, margin = margin(t = -8)),
    axis.text.x = element_text(angle = 0, hjust = 0.5, vjust = 0.5, margin = margin(t = 0)),
    legend.position = "None") +
  xlab("TCR Beta J genes") +
  ylab("Number of cells") +
  scale_fill_manual(
    name = "Sample",
    labels = c("M1", "M2", "M3", "M4", "M5"),
    values = c("#1B9E77", "#D95F02", "#7570B3", "#E7298A", "#66A61E")) +
  coord_flip()
# ggsave("Chronic_CD8_2cell_pure/Figures/04_TRB_J_genes.pdf", dpi = 300, width = 3.5 * 2.5, height = 2 * 2.5)
ggsave("summary_5mice/Figure1/TRB_J_genes_5mice.pdf", dpi = 300, width = 4 * 4, height = 3.5 * 3.5) # USE THIS ONE
```


```{r did not need this way to add cell name}
## add cell_name information needed for motif analysis

# now write a sapply where you match combined cdr3_nt seq with fullTCR_monocle df first and then extract any cell name from it 

#clones3Lin4_M1_TRA_fasta$cell_name <- sapply(clones3Lin4_M1_TRA_fasta$combined_cdr3_nt, FUN = function(x) unique(rownames(fullTCR_monocle)[fullTCR_monocle$combined_cdr3_nt %in% x]))

#rownames(data_monocle@phenoData@data)

#data_clones_by_mouse_AA_TRA_TRB$cell_name <- sapply(data_clones_by_mouse_AA_TRA_TRB$combined_cdr3_nt, FUN = function(x)
 # unique(rownames(data_monocle@phenoData@data)[data_monocle$combined_cdr3_nt %in% x]))

# Only keep first cell in cell name column 

#for (i in 1:nrow(data_clones_by_mouse_AA_TRA_TRB)) {
 # data_clones_by_mouse_AA_TRA_TRB[i, "cell_name"] <- data_clones_by_mouse_AA_TRA_TRB$cell_name[[i]][[1]]
#}

#data_clones_by_mouse_AA_TRA_TRB_DF <- as.data.frame(data_clones_by_mouse_AA_TRA_TRB)
#write.csv(data_clones_by_mouse_AA_TRA_TRB_DF, file = "data_clones_by_mouse_AA_TRA_TRB_DF.csv")
#class(data_clones_by_mouse_AA_TRA_TRB)

 #as.factor(data_clones_by_mouse_AA_TRA_TRB_DF$cell_name)
```


```{r}
### M2, M4 and M6 will be test mouse; M1, M3 will be test mouse [ cause we have excluded M5 since it was an outlier]
### not removing any overlapped clones
write.csv(data_clones_by_mouse_AA_TRA_TRB, file = "summary_5M_CCA_V2/data_clones_by_mouse_AA_TRA_TRB_5mice.csv") 

data_clones_by_mouse_AA_TRA_TRB <- read.csv("summary_5M_CCA_V2/data_clones_by_mouse_AA_TRA_TRB_5mice.csv")

length(data_clones_by_mouse_AA_TRA_TRB$combined_cdr3_nt)# 673


table(data_clones_by_mouse_AA_TRA_TRB$mice[which(data_clones_by_mouse_AA_TRA_TRB$Freq >= 4)])
#M1 M2 M3 M4 M6 
#63 70 59 72 88 
## separate out DF based on orig ident that has >= 4 cells using data_clones_by_mouse_AA_TRA_TRB_NOV
#rm(data_clones_by_mouse_AA_TRA_TRB_NOV_M1)

# CD4_M1 63 clones 
clone3lin4_M1 <- data_clones_by_mouse_AA_TRA_TRB[which(data_clones_by_mouse_AA_TRA_TRB$Freq >=4 & data_clones_by_mouse_AA_TRA_TRB$mice == "M1"), ] # 63



# subset data_clones_by_mouse_AA_TRA_TRB_NOV based on > 65% for Th1 and TFH
rm(clone3lin4_M1_Th1)
clone3lin4_M1_Th1_65 <- clone3lin4_M1[clone3lin4_M1$Percent_Monocle_Th1 >= 65, ] # 11
clone3lin4_M1_TFH_65 <- clone3lin4_M1[clone3lin4_M1$Percent_Monocle_TFH >= 65, ] # 20

# make seprate TRA_CDR3_AA file for each for motif analysis
clone3lin4_M1_Th1_65_TRA <- as.data.frame(clone3lin4_M1_Th1_65[, 2], drop = FALSE)
clone3lin4_M1_TFH_65_TRA <- as.data.frame(clone3lin4_M1_TFH_65[, 2], drop = FALSE)

write.csv(clone3lin4_M1_Th1_65_TRA, "Motif_5mice/clone3lin4_M1_Th1_65_TRA.csv")
write.csv(clone3lin4_M1_TFH_65_TRA, "Motif_5mice/clone3lin4_M1_TFH_65_TRA.csv")


# make a separate DF with cell name and CDr3TRA AA for all clones >= 4 cells from M1 which will be
# used to scan motif across all cells; will be shown as the cellular level ans statistics will be 
# done at cellular level; no of clones will be mentioned as before


clone3lin4_M1_TRA <- pData(data_monocle)[which(data_monocle$combined_cdr3_nt %in% clone3lin4_M1$combined_cdr3_nt), c("cdr3_TRA", "mice")]
view(clone3lin4_M1_TRA)

clone3lin4_M1_TRA$mice <- NULL

nrow(clone3lin4_M1_TRA) 
# 943

unique(clone3lin4_M1_TRA$cdr3_TRA) %>% length() # 59


M1_TRA_cln4_fasta <- clone3lin4_M1_TRA$cdr3_TRA
names(M1_TRA_cln4_fasta) <- rownames(clone3lin4_M1_TRA)
view(M1_TRA_cln4_fasta)
M1_TRA_cln4_fasta %>% length()
# 943 cells 

write.fasta(sequences = as.list(M1_TRA_cln4_fasta), file.out = "Motif_5mice/M1_TRA_cln4.fasta", names = as.list(names(M1_TRA_cln4_fasta)))


# 63 clones with cell name and CDR3_AA


# CD4_M2 70 clones
clone3lin4_M2 <- data_clones_by_mouse_AA_TRA_TRB[which(data_clones_by_mouse_AA_TRA_TRB$Freq >=4 & data_clones_by_mouse_AA_TRA_TRB$mice == "M2"), ] # 70



# subset data_clones_by_mouse_AA_TRA_TRB_NOV based on > 65% for Th1 and TFH

clone3lin4_M2_Th1_65 <- clone3lin4_M2[clone3lin4_M2$Percent_Monocle_Th1 >= 65, ] # 14
clone3lin4_M2_TFH_65 <- clone3lin4_M2[clone3lin4_M2$Percent_Monocle_TFH >= 65, ] # 33 # 32 more if setting > 65

# make seprate TRA_CDR3_AA file for each for motif analysis
clone3lin4_M2_Th1_65_TRA <- as.data.frame(clone3lin4_M2_Th1_65[, 2], drop = FALSE)
clone3lin4_M2_TFH_65_TRA <- as.data.frame(clone3lin4_M2_TFH_65[, 2], drop = FALSE)

write.csv(clone3lin4_M2_Th1_65_TRA, "Motif_5mice/clone3lin4_M2_Th1_65_TRA.csv")
write.csv(clone3lin4_M2_TFH_65_TRA, "Motif_5mice/clone3lin4_M2_TFH_65_TRA.csv")


# make a separate DF with cell name and CDr3TRA AA for all clones >= 4 cells from M1 which will be
# used to scan motif across all cells; will be shown as the cellular level ans statistics will be 
# done at cellular level; no of clones will be mentioned as before



clone3lin4_M2_TRA <- pData(data_monocle)[which(data_monocle$combined_cdr3_nt %in% clone3lin4_M2$combined_cdr3_nt), c("cdr3_TRA", "mice")]
view(clone3lin4_M2_TRA)

clone3lin4_M2_TRA$mice <- NULL

nrow(clone3lin4_M2_TRA) 
# 863

unique(clone3lin4_M2_TRA$cdr3_TRA) %>% length() # 66


M2_TRA_cln4_fasta <- clone3lin4_M2_TRA$cdr3_TRA
names(M2_TRA_cln4_fasta) <- rownames(clone3lin4_M2_TRA)
M2_TRA_cln4_fasta %>% length()
# 863 cells 

write.fasta(sequences = as.list(M2_TRA_cln4_fasta), file.out = "Motif_5mice/M2_TRA_cln4.fasta", names = as.list(names(M2_TRA_cln4_fasta)))


# CD4_M3 57 clones
rm(clone3lin4_M3)
clone3lin4_M3 <- data_clones_by_mouse_AA_TRA_TRB[which(data_clones_by_mouse_AA_TRA_TRB$Freq >=4 & data_clones_by_mouse_AA_TRA_TRB$mice == "M3"), ] # 59



# subset data_clones_by_mouse_AA_TRA_TRB_NOV based on > 65% for Th1 and TFH

clone3lin4_M3_Th1_65 <- clone3lin4_M3[clone3lin4_M3$Percent_Monocle_Th1 >= 65, ] # 7
clone3lin4_M3_TFH_65 <- clone3lin4_M3[clone3lin4_M3$Percent_Monocle_TFH >= 65, ] # 31 , 30 if setting > 65

# make seprate TRA_CDR3_AA file for each for motif analysis
clone3lin4_M3_Th1_65_TRA <- as.data.frame(clone3lin4_M3_Th1_65[, 2], drop = FALSE) # 7
clone3lin4_M3_TFH_65_TRA <- as.data.frame(clone3lin4_M3_TFH_65[, 2], drop = FALSE) # 31

write.csv(clone3lin4_M3_Th1_65_TRA, "Motif_5mice/clone3lin4_M3_Th1_65_TRA.csv")
write.csv(clone3lin4_M3_TFH_65_TRA, "Motif_5mice/clone3lin4_M3_TFH_65_TRA.csv")


# make a separate DF with cell name and CDr3TRA AA for all clones >= 4 cells from M1 which will be
# used to scan motif across all cells; will be shown as the cellular level ans statistics will be 
# done at cellular level; no of clones will be mentioned as before



clone3lin4_M3_TRA <- pData(data_monocle)[which(data_monocle$combined_cdr3_nt %in% clone3lin4_M3$combined_cdr3_nt), c("cdr3_TRA", "mice")]
view(clone3lin4_M3_TRA)## 7/8 cells are from M4

clone3lin4_M3_TRA$mice <- NULL

nrow(clone3lin4_M3_TRA) 
# 2446

unique(clone3lin4_M3_TRA$cdr3_TRA) %>% length() # 47


M3_TRA_cln4_fasta <- clone3lin4_M3_TRA$cdr3_TRA
names(M3_TRA_cln4_fasta) <- rownames(clone3lin4_M3_TRA)
M3_TRA_cln4_fasta %>% length()
# 2446 cells 

write.fasta(sequences = as.list(M3_TRA_cln4_fasta), file.out = "Motif_5mice/M3_TRA_cln4.fasta", names = as.list(names(M3_TRA_cln4_fasta)))



# CD4_M4 72 clones
clone3lin4_M4 <- data_clones_by_mouse_AA_TRA_TRB[which(data_clones_by_mouse_AA_TRA_TRB$Freq >=4 & data_clones_by_mouse_AA_TRA_TRB$mice == "M4"), ] # 72



# subset data_clones_by_mouse_AA_TRA_TRB_NOV based on > 65% for Th1 and TFH

clone3lin4_M4_Th1_65 <- clone3lin4_M4[clone3lin4_M4$Percent_Monocle_Th1 >= 65, ] # 11, 12 if setting > 65
clone3lin4_M4_TFH_65 <- clone3lin4_M4[clone3lin4_M4$Percent_Monocle_TFH >= 65, ] # 22

# make seprate TRA_CDR3_AA file for each for motif analysis
clone3lin4_M4_Th1_65_TRA <- as.data.frame(clone3lin4_M4_Th1_65[, 2], drop = FALSE)
clone3lin4_M4_TFH_65_TRA <- as.data.frame(clone3lin4_M4_TFH_65[, 2], drop = FALSE)

write.csv(clone3lin4_M4_Th1_65_TRA, "Motif_5mice/clone3lin4_M4_Th1_65_TRA.csv")
write.csv(clone3lin4_M4_TFH_65_TRA, "Motif_5mice/clone3lin4_M4_TFH_65_TRA.csv")


# make a separate DF with cell name and CDr3TRA AA for all clones >= 4 cells from M1 which will be
# used to scan motif across all cells; will be shown as the cellular level ans statistics will be 
# done at cellular level; no of clones will be mentioned as before

clone3lin4_M4_TRA <- pData(data_monocle)[which(data_monocle$combined_cdr3_nt %in% clone3lin4_M4$combined_cdr3_nt), c("cdr3_TRA", "mice")]
view(clone3lin4_M4_TRA)

clone3lin4_M4_TRA$mice <- NULL

nrow(clone3lin4_M4_TRA) 
# 3357

unique(clone3lin4_M4_TRA$cdr3_TRA) %>% length() # 58


M4_TRA_cln4_fasta <- clone3lin4_M4_TRA$cdr3_TRA
names(M4_TRA_cln4_fasta) <- rownames(clone3lin4_M4_TRA)
M4_TRA_cln4_fasta %>% length()
# 3357 cells 

write.fasta(sequences = as.list(M4_TRA_cln4_fasta), file.out = "Motif_5mice/M4_TRA_cln4.fasta", names = as.list(names(M4_TRA_cln4_fasta)))




# CD4_M6 88 clones
clone3lin4_M6 <- data_clones_by_mouse_AA_TRA_TRB[which(data_clones_by_mouse_AA_TRA_TRB$Freq >=4 & data_clones_by_mouse_AA_TRA_TRB$mice == "M6"), ] # 88



# subset data_clones_by_mouse_AA_TRA_TRB_NOV based on > 65% for Th1 and TFH

clone3lin4_M6_Th1_65 <- clone3lin4_M6[clone3lin4_M6$Percent_Monocle_Th1 >= 65, ] #19
clone3lin4_M6_TFH_65 <- clone3lin4_M6[clone3lin4_M6$Percent_Monocle_TFH >= 65, ] #22

# make seprate TRA_CDR3_AA file for each for motif analysis
clone3lin4_M6_Th1_65_TRA <- as.data.frame(clone3lin4_M6_Th1_65[, 2], drop = FALSE)
clone3lin4_M6_TFH_65_TRA <- as.data.frame(clone3lin4_M6_TFH_65[, 2], drop = FALSE)

write.csv(clone3lin4_M6_Th1_65_TRA, "Motif_5mice/clone3lin4_M6_Th1_65_TRA.csv")
write.csv(clone3lin4_M6_TFH_65_TRA, "Motif_5mice/clone3lin4_M6_TFH_65_TRA.csv")


# make a separate DF with cell name and CDr3TRA AA for all clones >= 4 cells from M1 which will be
# used to scan motif across all cells; will be shown as the cellular level ans statistics will be 
# done at cellular level; no of clones will be mentioned as before

clone3lin4_M6_TRA <- pData(data_monocle)[which(data_monocle$combined_cdr3_nt %in% clone3lin4_M6$combined_cdr3_nt), c("cdr3_TRA", "mice")]
view(clone3lin4_M6_TRA)

clone3lin4_M4_TRA$mice <- NULL

nrow(clone3lin4_M6_TRA) 
# 2405

unique(clone3lin4_M6_TRA$cdr3_TRA) %>% length() # 68


M6_TRA_cln4_fasta <- clone3lin4_M6_TRA$cdr3_TRA
names(M6_TRA_cln4_fasta) <- rownames(clone3lin4_M6_TRA)
M6_TRA_cln4_fasta %>% length()
# 2405 cells 

write.fasta(sequences = as.list(M6_TRA_cln4_fasta), file.out = "Motif_5mice/M6_TRA_cln4.fasta", names = as.list(names(M6_TRA_cln4_fasta)))





### now for scanning combine M2, M4 and M6 fasta file together for training; combine M1 and M3 fasta file for testing usage

M2M4M6_TRA_cln4_fasta <- c(M2_TRA_cln4_fasta, M4_TRA_cln4_fasta, M6_TRA_cln4_fasta)
M2M4M6_TRA_cln4_fasta %>% length() #6625

view(M2M4M6_TRA_cln4_fasta)

#write.fasta(sequences = as.list(M2M4_TRA_cln4_fasta), names = as.list(names(M2M4_TRA_cln4_fasta)), file.out = #"Motif/M2M4_TRA_cln4.fasta")

write.fasta(sequences = as.list(M2M4M6_TRA_cln4_fasta), file.out = "Motif_5mice/Train_M2M4M6_TRA_cln4.fasta", names = as.list(names(M2M4M6_TRA_cln4_fasta)))


## test file
M1M3_TRA_cln4_fasta <- c(M1_TRA_cln4_fasta, M3_TRA_cln4_fasta)
M1M3_TRA_cln4_fasta %>% length() #3389

#write.fasta(sequences = as.list(M1M3_TRA_cln4_fasta), names = as.list(names(M1M3_TRA_cln4_fasta)), file.out = #"Motif/test_M2M4_TRA_cln4.fasta")

write.fasta(sequences = as.list(M1M3_TRA_cln4_fasta), file.out = "Motif_5mice/Test_M1M3_TRA_cln4.fasta", names = as.list(names(M1M3_TRA_cln4_fasta)))

```



```{r}
## define the significant motifs for Th1 or TFH in train mouse M2, M4 and M6; read the glamscore.txt files here
# TFH motif 1

data_monocle <- readRDS("summary_5M_CCA_V2/data_monocle_2cell.RDS")
# TFH Motif1
rm(glam2_scores)
glam2_scores <- read.table(file = "/media/runa/Elements/Jem_revision/Seurat/Motif_5mice/Train/TFH/Motif1/TFH_Motif1.glam2scan/glam2scan_scores_Motif1.txt", header = FALSE)
#We're only interested in the cell name and score columns
glam2_scores <- glam2_scores[,c(1,6)]
#Take the top score for each cell
glam2_scores <- aggregate(.~V1, glam2_scores, FUN=max) # here multiple cells can have
#more than one score, with aggregate command we are saying in those cases just keep the max
#score
rownames(glam2_scores) <- glam2_scores$V1
# to match cell names with monocle data add 1 in cell name
glam2_scores$V1 <-
  paste0(glam2_scores$V1, "1")
rownames(glam2_scores) <- glam2_scores$V1

#Add glam score to each cell within monocle pData
pData(data_monocle)$glam_score <- sapply(rownames(pData(data_monocle)), FUN =
function(x) glam2_scores[x,]$V6)

View(pData(data_monocle))


#Observe the glam score distribution, find a high score for the cells 40 looks good here; this
#scoring will vary from motif to motif

pdf("summary_5mice/Figure4/TFH_Motif1_Train_mice_score.pdf", width = 8, height = 6, useDingbats = FALSE )
plot(density(pData(data_monocle)$glam_score[!is.na(pData(data_monocle)$glam_score)]),
main = "Motif score distribution_TRA_TFH_Motif1", xlab = "")
abline(v = 30, col = "Red")
dev.off()

#Make a true/false filtering column based on the glam score for plotting 
pData(data_monocle)$Present <- pData(data_monocle)$glam_score > 30

# Make a new dataframe to use for statistical analysis but not plotting the cells

# Only include cells from mouse 2 and mouse 4, separately
glam_stats_M2 <- data_monocle$Present[which(data_monocle$mice == "M2")] %>% as.data.frame()
colnames(glam_stats_M2) <- "Present"
# Add other columns from data_monocle
glam_stats_M2$State <- data_monocle$State[which(data_monocle$mice == "M2")]
glam_stats_M2$combined_cdr3_nt <- data_monocle$combined_cdr3_nt[which(data_monocle$mice == "M2")]
glam_stats_M2$cellID <- rownames(data_monocle@phenoData)[which(data_monocle$mice == "M2")]
# Remove any rows with NA (cells without a motif score, incluing cells that don't meet definition of a clone)
glam_stats_M2 <- glam_stats_M2[!is.na(glam_stats_M2$Present), ]

glam_stats_M4 <- data_monocle$Present[which(data_monocle$mice == "M4")] %>% as.data.frame()
colnames(glam_stats_M4) <- "Present"
# Add other columns from data_monocle
glam_stats_M4$State <- data_monocle$State[which(data_monocle$mice == "M4")]
glam_stats_M4$combined_cdr3_nt <- data_monocle$combined_cdr3_nt[which(data_monocle$mice == "M4")]
glam_stats_M4$cellID <- rownames(data_monocle@phenoData)[which(data_monocle$mice == "M4")]
# Remove any rows with NA (cells without a motif score, incluing cells that don't meet definition of a clone)
glam_stats_M4 <- glam_stats_M4[!is.na(glam_stats_M4$Present), ]

glam_stats_M6 <- data_monocle$Present[which(data_monocle$mice == "M6")] %>% as.data.frame()
colnames(glam_stats_M6) <- "Present"
# Add other columns from data_monocle
glam_stats_M6$State <- data_monocle$State[which(data_monocle$mice == "M6")]
glam_stats_M6$combined_cdr3_nt <- data_monocle$combined_cdr3_nt[which(data_monocle$mice == "M6")]
glam_stats_M6$cellID <- rownames(data_monocle@phenoData)[which(data_monocle$mice == "M6")]
# Remove any rows with NA (cells without a motif score, incluing cells that don't meet definition of a clone)
glam_stats_M6 <- glam_stats_M6[!is.na(glam_stats_M6$Present), ]



#Make NA values false
pData(data_monocle)$Present[is.na(pData(data_monocle)$Present)] <- FALSE
plot_obj <- plot_complex_cell_trajectory(data_monocle, color_by = "Present", cell_size = 1)
plot_obj$data <- plot_obj$data[order(plot_obj$data$Present),]
plot_obj + scale_color_manual(values=c("grey", "red")) + ggtitle("TRA_TFH_motif1_30_across_M2M4M6'\n Score > 30 \n ")

pdf("summary_5mice/Figure4/TFH_Motif1_Train_mice_TREE.pdf", width = 8, height = 6, useDingbats = FALSE)
plot_obj + scale_color_manual(values=c("grey", "red")) + ggtitle("TRA_TH1_motif1_M2M4M6.png'\n Score > 30 \n 'P value '")
dev.off()

#Create a table detailing the number of cells in each monocle state that have a good motif match
# The line below was from Robert's analysis, I'm replacing it because there was an error
# glam_score_table <- as.data.frame(table(pData(data_no_6_mono_cds)$State, pData(data_no_6_mono_cds)$Present))
rm(glam_score_table_M2)
rm(glam_score_table_M4)
rm(glam_score_table_M6)


glam_score_table_M2 <- table(glam_stats_M2$State, glam_stats_M2$Present) %>% as.data.frame()
#Combine true and false rows
glam_score_table_M2 <- cbind(glam_score_table_M2[1:5,], glam_score_table_M2[6:10,])
#Create names that make more sense
colnames(glam_score_table_M2) <- c("State", "F", "Freq_F", "State2", "T", "Freq_T")

#Remove unneeded columns
glam_score_table_M2 <- glam_score_table_M2[,c(1,3,6)]
#Create a percent column (percent of cells within each state with a good motif match)
glam_score_table_M2$pct <- glam_score_table_M2$Freq_T / (glam_score_table_M2$Freq_F + glam_score_table_M2$Freq_T)
#write.csv(glam_score_table_M2, "Motif/train/TFH/score_file/TRA_TFH_T65_motif1_M2_cln4_cells_across_M2.csv")
write.csv(glam_score_table_M2, "Motif_5mice/Train/Score/TRA_TFH_T65_motif1_M2_cln4_cells_across_M2_cells.csv")
getwd()
glam_score_table_M2$Freq_F %>% sum () # 795
glam_score_table_M2$Freq_T %>% sum() # 68

glam_stats_M2$combined_cdr3_nt[which(glam_stats_M2$Present == TRUE)] %>% table()

#TRA:TGTGCAGCAAGTCTCGATGGAGGCTATAAAGTGGTCTTT_TRB:TGTGCTAGCAGTTTATCCGGGACAGGGGACGCTGAGCAGTTCTTC 
#                                                                                           31 
#   TRA:TGTGCAGTGAGCGGGACTGGAGGCTATAAAGTGGTCTTT_TRB:TGTGCCAGCAGGACTGGGGGGGCAGGGGACACCCAGTACTTT 
#                                                                                           13 
#      TRA:TGTGCTCTGGGCCTGACTGGAGGCTATAAAGTGGTCTTT_TRB:TGCAGTGCAAGGGGGGCGGACACCGGGCAGCTCTACTTT 
 #                                                                                          24 

#Create a table detailing the number of cells in each monocle state that have a good motif match
# The line below was from Robert's analysis, I'm replacing it because there was an error
# glam_score_table <- as.data.frame(table(pData(data_no_6_mono_cds)$State, pData(data_no_6_mono_cds)$Present))

glam_score_table_M4 <- table(glam_stats_M4$State, glam_stats_M4$Present) %>% as.data.frame()
#Combine true and false rows
glam_score_table_M4 <- cbind(glam_score_table_M4[1:5,], glam_score_table_M4[6:10,]) ## this # depends on the # of monocle states
#Create names that make more sense
colnames(glam_score_table_M4) <- c("State", "F", "Freq_F", "State2", "T", "Freq_T")

#Remove unneeded columns
glam_score_table_M4 <- glam_score_table_M4[,c(1,3,6)]
#Create a percent column (percent of cells within each state with a good motif match)
glam_score_table_M4$pct <- glam_score_table_M4$Freq_T / (glam_score_table_M4$Freq_F + glam_score_table_M4$Freq_T)
#write.csv(glam_score_table_M4, "Motif/train/TFH/score_file/TRA_TFH_T65_motif1_M4_cln4_cells_across_M4.csv")
write.csv(glam_score_table_M4, "Motif_5mice/Train/Score/TRA_TFH_T65_motif1_M4_cln4_cells_across_M4_cells.csv")
getwd()
glam_score_table_M4$Freq_F %>% sum () # 2408
glam_score_table_M4$Freq_T %>% sum() # 810

glam_stats_M4$combined_cdr3_nt[which(glam_stats_M4$Present == TRUE)] %>% table()

#TRA:TGCTCAGCCCCCGGGACTGGAGGCTATAAAGTGGTCTTT_TRB:TGTGCCAGCAGCCAACAAAACACAGAAGTCTTCTTT 
 #                                                                                              5 
#TRA:TGTGCAGCAAGTTGGGGGACTGGAGGCTATAAAGTGGTCTTT_TRB:TGTGCTAGCAGTTTTGCTGGGGGGGCTTCCTATGAACAGTACTTC 
 #                                                                                            805 


#Create a table detailing the number of cells in each monocle state that have a good motif match
# The line below was from Robert's analysis, I'm replacing it because there was an error
# glam_score_table <- as.data.frame(table(pData(data_no_6_mono_cds)$State, pData(data_no_6_mono_cds)$Present))

glam_score_table_M6 <- table(glam_stats_M6$State, glam_stats_M6$Present) %>% as.data.frame()
#Combine true and false rows
glam_score_table_M6 <- cbind(glam_score_table_M6[1:5,], glam_score_table_M6[6:10,]) ## this # depends on the # of monocle states
#Create names that make more sense
colnames(glam_score_table_M6) <- c("State", "F", "Freq_F", "State2", "T", "Freq_T")

#Remove unneeded columns
glam_score_table_M6 <- glam_score_table_M6[,c(1,3,6)]
#Create a percent column (percent of cells within each state with a good motif match)
glam_score_table_M6$pct <- glam_score_table_M6$Freq_T / (glam_score_table_M6$Freq_F + glam_score_table_M6$Freq_T)
#write.csv(glam_score_table_M4, "Motif/train/TFH/score_file/TRA_TFH_T65_motif1_M4_cln4_cells_across_M4.csv")
write.csv(glam_score_table_M6, "Motif_5mice/Train/Score/TRA_TFH_T65_motif1_M6_cln4_cells_across_M6_cells.csv")
getwd()
glam_score_table_M6$Freq_F %>% sum () # 2324
glam_score_table_M6$Freq_T %>% sum() # NA


```




```{r}

# TFH motif 2

#data_monocle <- readRDS("summary_5M_CCA_V2/data_monocle_2cell.RDS")
# TFH Motif2
rm(glam2_scores)
glam2_scores <- read.table(file = "/media/runa/Elements/Jem_revision/Seurat/Motif_5mice/Train/TFH/Motif2/TFH_Motif2.glam2scan/glam2scan_scores_Motif2.txt", header = FALSE)
#We're only interested in the cell name and score columns
glam2_scores <- glam2_scores[,c(1,6)]
#Take the top score for each cell
glam2_scores <- aggregate(.~V1, glam2_scores, FUN=max) # here multiple cells can have
#more than one score, with aggregate command we are saying in those cases just keep the max
#score
rownames(glam2_scores) <- glam2_scores$V1
# to match cell names with monocle data add 1 in cell name
glam2_scores$V1 <-
  paste0(glam2_scores$V1, "1")
rownames(glam2_scores) <- glam2_scores$V1

#Add glam score to each cell within monocle pData
pData(data_monocle)$glam_score <- sapply(rownames(pData(data_monocle)), FUN =
function(x) glam2_scores[x,]$V6)

View(pData(data_monocle))


#Observe the glam score distribution, find a high score for the cells 40 looks good here; this
#scoring will vary from motif to motif

pdf("summary_5mice/Figure4/TFH_Motif2_Train_mice_score.pdf", width = 8, height = 6)
plot(density(pData(data_monocle)$glam_score[!is.na(pData(data_monocle)$glam_score)]),
main = "Motif score distribution_TRA_TFH_Motif2", xlab = "")
abline(v = 37, col = "Red")
dev.off()

#Make a true/false filtering column based on the glam score for plotting 
pData(data_monocle)$Present <- pData(data_monocle)$glam_score > 37

# Make a new dataframe to use for statistical analysis but not plotting the cells

rm(glam_stats_M2)
rm(glam_stats_M4)
rm(glam_stats_M6)

# Only include cells from mouse 2 and mouse 4, separately
glam_stats_M2 <- data_monocle$Present[which(data_monocle$mice == "M2")] %>% as.data.frame()
colnames(glam_stats_M2) <- "Present"
# Add other columns from data_monocle
glam_stats_M2$State <- data_monocle$State[which(data_monocle$mice == "M2")]
glam_stats_M2$combined_cdr3_nt <- data_monocle$combined_cdr3_nt[which(data_monocle$mice == "M2")]
glam_stats_M2$cellID <- rownames(data_monocle@phenoData)[which(data_monocle$mice == "M2")]
# Remove any rows with NA (cells without a motif score, incluing cells that don't meet definition of a clone)
glam_stats_M2 <- glam_stats_M2[!is.na(glam_stats_M2$Present), ]

glam_stats_M4 <- data_monocle$Present[which(data_monocle$mice == "M4")] %>% as.data.frame()
colnames(glam_stats_M4) <- "Present"
# Add other columns from data_monocle
glam_stats_M4$State <- data_monocle$State[which(data_monocle$mice == "M4")]
glam_stats_M4$combined_cdr3_nt <- data_monocle$combined_cdr3_nt[which(data_monocle$mice == "M4")]
glam_stats_M4$cellID <- rownames(data_monocle@phenoData)[which(data_monocle$mice == "M4")]
# Remove any rows with NA (cells without a motif score, incluing cells that don't meet definition of a clone)
glam_stats_M4 <- glam_stats_M4[!is.na(glam_stats_M4$Present), ]

glam_stats_M6 <- data_monocle$Present[which(data_monocle$mice == "M6")] %>% as.data.frame()
colnames(glam_stats_M6) <- "Present"
# Add other columns from data_monocle
glam_stats_M6$State <- data_monocle$State[which(data_monocle$mice == "M6")]
glam_stats_M6$combined_cdr3_nt <- data_monocle$combined_cdr3_nt[which(data_monocle$mice == "M6")]
glam_stats_M6$cellID <- rownames(data_monocle@phenoData)[which(data_monocle$mice == "M6")]
# Remove any rows with NA (cells without a motif score, incluing cells that don't meet definition of a clone)
glam_stats_M6 <- glam_stats_M6[!is.na(glam_stats_M6$Present), ]



#Make NA values false
pData(data_monocle)$Present[is.na(pData(data_monocle)$Present)] <- FALSE
plot_obj <- plot_complex_cell_trajectory(data_monocle, color_by = "Present", cell_size = 2)
plot_obj$data <- plot_obj$data[order(plot_obj$data$Present),]
plot_obj + scale_color_manual(values=c("grey", "red")) + ggtitle("TRA_TFH_motif2_37_across_M2M4M6'\n Score > 37 \n ")

pdf("summary_5mice/Figure4/TFH_Motif2_Train_mice_TREE.pdf", width = 8, height = 6, useDingbats = FALSE )
plot_obj + scale_color_manual(values=c("grey", "red")) + ggtitle("TRA_TH1_motif2_M2M4M6.png'\n Score > 37 \n 'P value '")
dev.off()

#Create a table detailing the number of cells in each monocle state that have a good motif match
# The line below was from Robert's analysis, I'm replacing it because there was an error
# glam_score_table <- as.data.frame(table(pData(data_no_6_mono_cds)$State, pData(data_no_6_mono_cds)$Present))
rm(glam_score_table_M2)
rm(glam_score_table_M4)
rm(glam_score_table_M6)

glam_score_table_M2 <- table(glam_stats_M2$State, glam_stats_M2$Present) %>% as.data.frame()
#Combine true and false rows
glam_score_table_M2 <- cbind(glam_score_table_M2[1:5,], glam_score_table_M2[6:10,])
#Create names that make more sense
colnames(glam_score_table_M2) <- c("State", "F", "Freq_F", "State2", "T", "Freq_T")

#Remove unneeded columns
glam_score_table_M2 <- glam_score_table_M2[,c(1,3,6)]
#Create a percent column (percent of cells within each state with a good motif match)
glam_score_table_M2$pct <- glam_score_table_M2$Freq_T / (glam_score_table_M2$Freq_F + glam_score_table_M2$Freq_T)
#write.csv(glam_score_table_M2, "Motif/train/TFH/score_file/TRA_TFH_T65_motif1_M2_cln4_cells_across_M2.csv")
write.csv(glam_score_table_M2, "Motif_5mice/Train/Score/TRA_TFH_T65_motif2_M2_cln4_cells_across_M2_cells.csv")
getwd()
glam_score_table_M2$Freq_F %>% sum () # 803
glam_score_table_M2$Freq_T %>% sum() # 60

glam_stats_M2$combined_cdr3_nt[which(glam_stats_M2$Present == TRUE)] %>% table()

#TRA:TGTGCAGCAAGTCCTGGAGGCAATAATAAGCTGACTTTT_TRB:TGTGCCTGGGTGGGTAGTGCAGAAACGCTGTATTTT 
#                                                                                              30 
#               TRA:TGTGCTCTGAGCCCCAACATGGGCTACAAACTTACCTTC_TRB:TGTGCCAGCAGCCAAGAGAGGGAACAGTACTTC 
#                                                                                               4 
#TRA:TGTGCTCTGGGGAATTCTGGAGGAAGCAATGCAAAGCTAACCTTC_TRB:TGTGCCTGGAGCGGGATGGGGGGGCAGGACACCCAGTACTTT 
 #                                                                                             26 

#Create a table detailing the number of cells in each monocle state that have a good motif match
# The line below was from Robert's analysis, I'm replacing it because there was an error
# glam_score_table <- as.data.frame(table(pData(data_no_6_mono_cds)$State, pData(data_no_6_mono_cds)$Present))

glam_score_table_M4 <- table(glam_stats_M4$State, glam_stats_M4$Present) %>% as.data.frame()
#Combine true and false rows
glam_score_table_M4 <- cbind(glam_score_table_M4[1:5,], glam_score_table_M4[6:10,]) ## this # depends on the # of monocle states
#Create names that make more sense
colnames(glam_score_table_M4) <- c("State", "F", "Freq_F", "State2", "T", "Freq_T")

#Remove unneeded columns
glam_score_table_M4 <- glam_score_table_M4[,c(1,3,6)]
#Create a percent column (percent of cells within each state with a good motif match)
glam_score_table_M4$pct <- glam_score_table_M4$Freq_T / (glam_score_table_M4$Freq_F + glam_score_table_M4$Freq_T)
#write.csv(glam_score_table_M4, "Motif/train/TFH/score_file/TRA_TFH_T65_motif1_M4_cln4_cells_across_M4.csv")
write.csv(glam_score_table_M4, "Motif_5mice/Train/Score/TRA_TFH_T65_motif2_M4_cln4_cells_across_M4_cells.csv")
getwd()
glam_score_table_M4$Freq_F %>% sum () # 3129
glam_score_table_M4$Freq_T %>% sum() # 89

glam_stats_M4$combined_cdr3_nt[which(glam_stats_M4$Present == TRUE)] %>% table()

  #TRA:TGTGCAGCAAGTCCTGGAGGCAATAATAAGCTGACTTTT_TRB:TGTGCCTGGAGTCTGGGGGCAGAAACGCTGTATTTT 
   #                                                                                           69 
   #TRA:TGTGCAGCAAGTCCTGGAGGCAATAATAAGCTGACTTTT_TRB:TGTGCTAGCAGTAGAGCGGGACTCGCAAACTCCGACTACACCTTC 
    #                                                                                          16 
#TRA:TGTGCTCTGACTAACAGTGCAGGGAACAAGCTAACTTTT_TRB:TGTGCCAGCGGTGATTTGGGGGGGGCGAACCAAGACACCCAGTACTTT 
 #                                                                                              4 


#Create a table detailing the number of cells in each monocle state that have a good motif match
# The line below was from Robert's analysis, I'm replacing it because there was an error
# glam_score_table <- as.data.frame(table(pData(data_no_6_mono_cds)$State, pData(data_no_6_mono_cds)$Present))

glam_score_table_M6 <- table(glam_stats_M6$State, glam_stats_M6$Present) %>% as.data.frame()
#Combine true and false rows
glam_score_table_M6 <- cbind(glam_score_table_M6[1:5,], glam_score_table_M6[6:10,]) ## this # depends on the # of monocle states
#Create names that make more sense
colnames(glam_score_table_M6) <- c("State", "F", "Freq_F", "State2", "T", "Freq_T")

#Remove unneeded columns
glam_score_table_M6 <- glam_score_table_M6[,c(1,3,6)]
#Create a percent column (percent of cells within each state with a good motif match)
glam_score_table_M6$pct <- glam_score_table_M6$Freq_T / (glam_score_table_M6$Freq_F + glam_score_table_M6$Freq_T)
#write.csv(glam_score_table_M4, "Motif/train/TFH/score_file/TRA_TFH_T65_motif1_M4_cln4_cells_across_M4.csv")
write.csv(glam_score_table_M6, "Motif_5mice/Train/Score/TRA_TFH_T65_motif2_M6_cln4_cells_across_M6_cells.csv")
getwd()
glam_score_table_M6$Freq_F %>% sum () # 2248
glam_score_table_M6$Freq_T %>% sum() # 76

glam_stats_M6$combined_cdr3_nt[which(glam_stats_M6$Present == TRUE)] %>% table()

#TRA:TGTGCAGCAAGTGGGGACTCGGGATACAACAAACTCACTTTT_TRB:TGTGCCAGCGGTGATGTGGGACAGGGTAAAGACACCCAGTACTTT 
 #                                                                                              8 
#   TRA:TGTGCAGCCCGTGGGGACTCGGGATACAACAAACTCACTTTT_TRB:TGTGCCTGGAGTCGGGACAATAACCAAGACACCCAGTACTTT 
 #                                                                                              7 
#TRA:TGTGCTCTGAGTGACTCGGGATACAACAAACTCACTTTT_TRB:TGTGCCAGCGGTGATGCGGGGGGGCCTAGTGCAGAAACGCTGTATTTT 
 #                                                                                             34 
#   TRA:TGTGCTCTGAGTGACTCGGGATACAACAAACTCACTTTT_TRB:TGTGCCAGCGGTGATTATGGGGCTAGTCAAAACACCTTGTACTTT 
 #                                                                                             23 
#TRA:TGTGCTCTTACTAACAGTGCAGGGAACAAGCTAACTTTT_TRB:TGTGCCAGCGGTGATGTGGGGGGGCGGGGCCAAGACACCCAGTACTTT 
 #                                                                                              4 


# Perform clone-level statistical analysis

```



```{r}
# TFH motif 3

#data_monocle <- readRDS("summary_5M_CCA_V2/data_monocle_2cell.RDS")
# TFH Motif3
rm(glam2_scores)
glam2_scores <- read.table(file = "/media/runa/Elements/Jem_revision/Seurat/Motif_5mice/Train/TFH/Motif3/TFH_Motif3.glam2scan/glam2scan_scores_Motif3.txt", header = FALSE)
#We're only interested in the cell name and score columns
glam2_scores <- glam2_scores[,c(1,6)]
#Take the top score for each cell
glam2_scores <- aggregate(.~V1, glam2_scores, FUN=max) # here multiple cells can have
#more than one score, with aggregate command we are saying in those cases just keep the max
#score
rownames(glam2_scores) <- glam2_scores$V1
# to match cell names with monocle data add 1 in cell name
glam2_scores$V1 <-
  paste0(glam2_scores$V1, "1")
rownames(glam2_scores) <- glam2_scores$V1

#Add glam score to each cell within monocle pData
pData(data_monocle)$glam_score <- sapply(rownames(pData(data_monocle)), FUN =
function(x) glam2_scores[x,]$V6)

View(pData(data_monocle))


#Observe the glam score distribution, find a high score for the cells 40 looks good here; this
#scoring will vary from motif to motif
plot(density(pData(data_monocle)$glam_score[!is.na(pData(data_monocle)$glam_score)]),
main = "Motif score distribution_TRA_TFH_Motif3", xlab = "")
abline(v = 37, col = "Red")

#Make a true/false filtering column based on the glam score for plotting 
pData(data_monocle)$Present <- pData(data_monocle)$glam_score > 37

# Make a new dataframe to use for statistical analysis but not plotting the cells

rm(glam_stats_M2)
rm(glam_stats_M4)
rm(glam_stats_M6)

# Only include cells from mouse 2 and mouse 4, separately
glam_stats_M2 <- data_monocle$Present[which(data_monocle$mice == "M2")] %>% as.data.frame()
colnames(glam_stats_M2) <- "Present"
# Add other columns from data_monocle
glam_stats_M2$State <- data_monocle$State[which(data_monocle$mice == "M2")]
glam_stats_M2$combined_cdr3_nt <- data_monocle$combined_cdr3_nt[which(data_monocle$mice == "M2")]
glam_stats_M2$cellID <- rownames(data_monocle@phenoData)[which(data_monocle$mice == "M2")]
# Remove any rows with NA (cells without a motif score, incluing cells that don't meet definition of a clone)
glam_stats_M2 <- glam_stats_M2[!is.na(glam_stats_M2$Present), ]

glam_stats_M4 <- data_monocle$Present[which(data_monocle$mice == "M4")] %>% as.data.frame()
colnames(glam_stats_M4) <- "Present"
# Add other columns from data_monocle
glam_stats_M4$State <- data_monocle$State[which(data_monocle$mice == "M4")]
glam_stats_M4$combined_cdr3_nt <- data_monocle$combined_cdr3_nt[which(data_monocle$mice == "M4")]
glam_stats_M4$cellID <- rownames(data_monocle@phenoData)[which(data_monocle$mice == "M4")]
# Remove any rows with NA (cells without a motif score, incluing cells that don't meet definition of a clone)
glam_stats_M4 <- glam_stats_M4[!is.na(glam_stats_M4$Present), ]

glam_stats_M6 <- data_monocle$Present[which(data_monocle$mice == "M6")] %>% as.data.frame()
colnames(glam_stats_M6) <- "Present"
# Add other columns from data_monocle
glam_stats_M6$State <- data_monocle$State[which(data_monocle$mice == "M6")]
glam_stats_M6$combined_cdr3_nt <- data_monocle$combined_cdr3_nt[which(data_monocle$mice == "M6")]
glam_stats_M6$cellID <- rownames(data_monocle@phenoData)[which(data_monocle$mice == "M6")]
# Remove any rows with NA (cells without a motif score, incluing cells that don't meet definition of a clone)
glam_stats_M6 <- glam_stats_M6[!is.na(glam_stats_M6$Present), ]



#Make NA values false
pData(data_monocle)$Present[is.na(pData(data_monocle)$Present)] <- FALSE
plot_obj <- plot_complex_cell_trajectory(data_monocle, color_by = "Present")
plot_obj$data <- plot_obj$data[order(plot_obj$data$Present),]
plot_obj + scale_color_manual(values=c("grey", "red")) + ggtitle("TRA_TFH_motif3_37_across_M2M4M6'\n Score > 37 \n ")

png(file = "summary_6mice/TRA_TH1_motif1_M2M4M6_cln4.png", width = 8, height = 6, units = "in", res = 300)
plot_obj + scale_color_manual(values=c("grey", "blue")) + ggtitle("TRA_TH1_motif3_M2M4M6.png'\n Score > 37 \n 'P value '")
#dev.off()

#Create a table detailing the number of cells in each monocle state that have a good motif match
# The line below was from Robert's analysis, I'm replacing it because there was an error
# glam_score_table <- as.data.frame(table(pData(data_no_6_mono_cds)$State, pData(data_no_6_mono_cds)$Present))
rm(glam_score_table_M2)
rm(glam_score_table_M4)
rm(glam_score_table_M6)
glam_score_table_M2 <- table(glam_stats_M2$State, glam_stats_M2$Present) %>% as.data.frame()
#Combine true and false rows
glam_score_table_M2 <- cbind(glam_score_table_M2[1:5,], glam_score_table_M2[6:10,])
#Create names that make more sense
colnames(glam_score_table_M2) <- c("State", "F", "Freq_F", "State2", "T", "Freq_T")

#Remove unneeded columns
glam_score_table_M2 <- glam_score_table_M2[,c(1,3,6)]
#Create a percent column (percent of cells within each state with a good motif match)
glam_score_table_M2$pct <- glam_score_table_M2$Freq_T / (glam_score_table_M2$Freq_F + glam_score_table_M2$Freq_T)
#write.csv(glam_score_table_M2, "Motif/train/TFH/score_file/TRA_TFH_T65_motif1_M2_cln4_cells_across_M2.csv")
write.csv(glam_score_table_M2, "Motif_5mice/Train/Score/TRA_TFH_T65_motif3_M2_cln4_cells_across_M2_cells.csv")
getwd()
glam_score_table_M2$Freq_F %>% sum () # 836
glam_score_table_M2$Freq_T %>% sum() # 27

#Create a table detailing the number of cells in each monocle state that have a good motif match
# The line below was from Robert's analysis, I'm replacing it because there was an error
# glam_score_table <- as.data.frame(table(pData(data_no_6_mono_cds)$State, pData(data_no_6_mono_cds)$Present))

glam_score_table_M4 <- table(glam_stats_M4$State, glam_stats_M4$Present) %>% as.data.frame()
#Combine true and false rows
glam_score_table_M4 <- cbind(glam_score_table_M4[1:5,], glam_score_table_M4[6:10,]) ## this # depends on the # of monocle states
#Create names that make more sense
colnames(glam_score_table_M4) <- c("State", "F", "Freq_F", "State2", "T", "Freq_T")

#Remove unneeded columns
glam_score_table_M4 <- glam_score_table_M4[,c(1,3,6)]
#Create a percent column (percent of cells within each state with a good motif match)
glam_score_table_M4$pct <- glam_score_table_M4$Freq_T / (glam_score_table_M4$Freq_F + glam_score_table_M4$Freq_T)
#write.csv(glam_score_table_M4, "Motif/train/TFH/score_file/TRA_TFH_T65_motif1_M4_cln4_cells_across_M4.csv")
write.csv(glam_score_table_M4, "Motif_5mice/Train/Score/TRA_TFH_T65_motif3_M4_cln4_cells_across_M4_cells.csv")
getwd()
glam_score_table_M4$Freq_F %>% sum () # 3204
glam_score_table_M4$Freq_T %>% sum() # 14


#Create a table detailing the number of cells in each monocle state that have a good motif match
# The line below was from Robert's analysis, I'm replacing it because there was an error
# glam_score_table <- as.data.frame(table(pData(data_no_6_mono_cds)$State, pData(data_no_6_mono_cds)$Present))

glam_score_table_M6 <- table(glam_stats_M6$State, glam_stats_M6$Present) %>% as.data.frame()
#Combine true and false rows
glam_score_table_M6 <- cbind(glam_score_table_M6[1:5,], glam_score_table_M6[6:10,]) ## this # depends on the # of monocle states
#Create names that make more sense
colnames(glam_score_table_M6) <- c("State", "F", "Freq_F", "State2", "T", "Freq_T")

#Remove unneeded columns
glam_score_table_M6 <- glam_score_table_M6[,c(1,3,6)]
#Create a percent column (percent of cells within each state with a good motif match)
glam_score_table_M6$pct <- glam_score_table_M6$Freq_T / (glam_score_table_M6$Freq_F + glam_score_table_M6$Freq_T)
#write.csv(glam_score_table_M4, "Motif/train/TFH/score_file/TRA_TFH_T65_motif1_M4_cln4_cells_across_M4.csv")
write.csv(glam_score_table_M6, "Motif_5mice/Train/Score/TRA_TFH_T65_motif3_M6_cln4_cells_across_M6_cells.csv")
getwd()
glam_score_table_M6$Freq_F %>% sum () # 2013
glam_score_table_M6$Freq_T %>% sum() # 311


# Perform clone-level statistical analysis

glam_stats_M2$combined_cdr3_nt[which(glam_stats_M2$Present == TRUE)] %>% table()


#         TRA:TGCTCAGCAAGTATGACTGGAGGCTATAAAGTGGTCTTT_TRB:TGTGCCAGCAGCAAGGTAAACACAGAAGTCTTCTTT 
 #                                                                                           4 
#TRA:TGTGCAGCAAGTCTCGATGGAGGCTATAAAGTGGTCTTT_TRB:TGTGCTAGCAGTTTATCCGGGACAGGGGACGCTGAGCAGTTCTTC 
    #                                                                                       30 
   #TRA:TGTGCAGTGAGCGGGACTGGAGGCTATAAAGTGGTCTTT_TRB:TGTGCCAGCAGGACTGGGGGGGCAGGGGACACCCAGTACTTT 
    #                                                                                       13 
     # TRA:TGTGCTCTGGGCCTGACTGGAGGCTATAAAGTGGTCTTT_TRB:TGCAGTGCAAGGGGGGCGGACACCGGGCAGCTCTACTTT 
       #                                                                                    21 

grep(pattern = "TRA:TGTGCAGCAAGTTGGGGGACTGGAGGCTATAAAGTGGTCTTT_TRB:TGTGCTAGCAGTTTTGCTGGGGGGGCTTCCTATGAACAGTACTTC", x = data_clones_by_mouse_AA_TRA_TRB_NOV$combined_cdr3_nt)
# 3
as.data.frame(data_clones_by_mouse_AA_TRA_TRB_NOV[1, ])
# 79, 0, 21
getwd()
glam_stats_M4$combined_cdr3_nt[which(glam_stats_M4$Present == TRUE)] %>% table()

        #  TRA:TGTGCAGCAAGCATGAGAACAGGAAACTACAAATACGTCTTT_TRB:TGTGCCAGCAGCGCCCAAACGCTGTATTTT 
         #                                                                                      5 
      #TRA:TGTGCAGCAAGTCTTAATACAGGAAACTACAAATACGTCTTT_TRB:TGTGCCTGGAATGGGGGGGGCTATGCTGAGCAGTTCTTC 
       #                                                                                       12 
#TRA:TGTGCAGCAAGTTGGGGGACTGGAGGCTATAAAGTGGTCTTT_TRB:TGTGCTAGCAGTTTTGCTGGGGGGGCTTCCTATGAACAGTACTTC 
 #                                                                                            813

```



```{r}
# TFH motif 4

#data_monocle <- readRDS("summary_5M_CCA_V2/data_monocle_2cell.RDS")
# TFH Motif4
rm(glam2_scores)
glam2_scores <- read.table(file = "/media/runa/Elements/Jem_revision/Seurat/Motif_5mice/Train/TFH/Motif4/TFH_Motif4.glam2scan/glam2scan_scores_Motif4.txt", header = FALSE)
#We're only interested in the cell name and score columns
glam2_scores <- glam2_scores[,c(1,6)]
#Take the top score for each cell
glam2_scores <- aggregate(.~V1, glam2_scores, FUN=max) # here multiple cells can have
#more than one score, with aggregate command we are saying in those cases just keep the max
#score
rownames(glam2_scores) <- glam2_scores$V1
# to match cell names with monocle data add 1 in cell name
glam2_scores$V1 <-
  paste0(glam2_scores$V1, "1")
rownames(glam2_scores) <- glam2_scores$V1

#Add glam score to each cell within monocle pData
pData(data_monocle)$glam_score <- sapply(rownames(pData(data_monocle)), FUN =
function(x) glam2_scores[x,]$V6)

View(pData(data_monocle))


#Observe the glam score distribution, find a high score for the cells 40 looks good here; this
#scoring will vary from motif to motif
plot(density(pData(data_monocle)$glam_score[!is.na(pData(data_monocle)$glam_score)]),
main = "Motif score distribution_TRA_TFH_Motif4", xlab = "")
abline(v = 40, col = "Red")

#Make a true/false filtering column based on the glam score for plotting 
pData(data_monocle)$Present <- pData(data_monocle)$glam_score > 40

# Make a new dataframe to use for statistical analysis but not plotting the cells

rm(glam_stats_M2)
rm(glam_stats_M4)
rm(glam_stats_M6)

# Only include cells from mouse 2 and mouse 4, separately
glam_stats_M2 <- data_monocle$Present[which(data_monocle$mice == "M2")] %>% as.data.frame()
colnames(glam_stats_M2) <- "Present"
# Add other columns from data_monocle
glam_stats_M2$State <- data_monocle$State[which(data_monocle$mice == "M2")]
glam_stats_M2$combined_cdr3_nt <- data_monocle$combined_cdr3_nt[which(data_monocle$mice == "M2")]
glam_stats_M2$cellID <- rownames(data_monocle@phenoData)[which(data_monocle$mice == "M2")]
# Remove any rows with NA (cells without a motif score, incluing cells that don't meet definition of a clone)
glam_stats_M2 <- glam_stats_M2[!is.na(glam_stats_M2$Present), ]

glam_stats_M4 <- data_monocle$Present[which(data_monocle$mice == "M4")] %>% as.data.frame()
colnames(glam_stats_M4) <- "Present"
# Add other columns from data_monocle
glam_stats_M4$State <- data_monocle$State[which(data_monocle$mice == "M4")]
glam_stats_M4$combined_cdr3_nt <- data_monocle$combined_cdr3_nt[which(data_monocle$mice == "M4")]
glam_stats_M4$cellID <- rownames(data_monocle@phenoData)[which(data_monocle$mice == "M4")]
# Remove any rows with NA (cells without a motif score, incluing cells that don't meet definition of a clone)
glam_stats_M4 <- glam_stats_M4[!is.na(glam_stats_M4$Present), ]

glam_stats_M6 <- data_monocle$Present[which(data_monocle$mice == "M6")] %>% as.data.frame()
colnames(glam_stats_M6) <- "Present"
# Add other columns from data_monocle
glam_stats_M6$State <- data_monocle$State[which(data_monocle$mice == "M6")]
glam_stats_M6$combined_cdr3_nt <- data_monocle$combined_cdr3_nt[which(data_monocle$mice == "M6")]
glam_stats_M6$cellID <- rownames(data_monocle@phenoData)[which(data_monocle$mice == "M6")]
# Remove any rows with NA (cells without a motif score, incluing cells that don't meet definition of a clone)
glam_stats_M6 <- glam_stats_M6[!is.na(glam_stats_M6$Present), ]



#Make NA values false
pData(data_monocle)$Present[is.na(pData(data_monocle)$Present)] <- FALSE
plot_obj <- plot_complex_cell_trajectory(data_monocle, color_by = "Present")
plot_obj$data <- plot_obj$data[order(plot_obj$data$Present),]
plot_obj + scale_color_manual(values=c("grey", "red")) + ggtitle("TRA_TFH_motif4_40_across_M2M4M6'\n Score > 40 \n ")

png(file = "summary_6mice/TRA_TH1_motif1_M2M4M6_cln4.png", width = 8, height = 6, units = "in", res = 300)
plot_obj + scale_color_manual(values=c("grey", "blue")) + ggtitle("TRA_TH1_motif3_M2M4M6.png'\n Score > 37 \n 'P value '")
#dev.off()

#Create a table detailing the number of cells in each monocle state that have a good motif match
# The line below was from Robert's analysis, I'm replacing it because there was an error
# glam_score_table <- as.data.frame(table(pData(data_no_6_mono_cds)$State, pData(data_no_6_mono_cds)$Present))
rm(glam_score_table_M2)
rm(glam_score_table_M4)
rm(glam_score_table_M6)
glam_score_table_M2 <- table(glam_stats_M2$State, glam_stats_M2$Present) %>% as.data.frame()
#Combine true and false rows
glam_score_table_M2 <- cbind(glam_score_table_M2[1:5,], glam_score_table_M2[6:10,])
#Create names that make more sense
colnames(glam_score_table_M2) <- c("State", "F", "Freq_F", "State2", "T", "Freq_T")

#Remove unneeded columns
glam_score_table_M2 <- glam_score_table_M2[,c(1,3,6)]
#Create a percent column (percent of cells within each state with a good motif match)
glam_score_table_M2$pct <- glam_score_table_M2$Freq_T / (glam_score_table_M2$Freq_F + glam_score_table_M2$Freq_T)
#write.csv(glam_score_table_M2, "Motif/train/TFH/score_file/TRA_TFH_T65_motif1_M2_cln4_cells_across_M2.csv")
write.csv(glam_score_table_M2, "Motif_5mice/Train/Score/TRA_TFH_T65_motif4_M2_cln4_cells_across_M2_cells.csv")
getwd()
glam_score_table_M2$Freq_F %>% sum () # 790
glam_score_table_M2$Freq_T %>% sum() # 73

#Create a table detailing the number of cells in each monocle state that have a good motif match
# The line below was from Robert's analysis, I'm replacing it because there was an error
# glam_score_table <- as.data.frame(table(pData(data_no_6_mono_cds)$State, pData(data_no_6_mono_cds)$Present))

glam_score_table_M4 <- table(glam_stats_M4$State, glam_stats_M4$Present) %>% as.data.frame()
#Combine true and false rows
glam_score_table_M4 <- cbind(glam_score_table_M4[1:5,], glam_score_table_M4[6:10,]) ## this # depends on the # of monocle states
#Create names that make more sense
colnames(glam_score_table_M4) <- c("State", "F", "Freq_F", "State2", "T", "Freq_T")

#Remove unneeded columns
glam_score_table_M4 <- glam_score_table_M4[,c(1,3,6)]
#Create a percent column (percent of cells within each state with a good motif match)
glam_score_table_M4$pct <- glam_score_table_M4$Freq_T / (glam_score_table_M4$Freq_F + glam_score_table_M4$Freq_T)
#write.csv(glam_score_table_M4, "Motif/train/TFH/score_file/TRA_TFH_T65_motif1_M4_cln4_cells_across_M4.csv")
write.csv(glam_score_table_M4, "Motif_5mice/Train/Score/TRA_TFH_T65_motif4_M4_cln4_cells_across_M4_cells.csv")
getwd()
glam_score_table_M4$Freq_F %>% sum () # 2950
glam_score_table_M4$Freq_T %>% sum() # 268


#Create a table detailing the number of cells in each monocle state that have a good motif match
# The line below was from Robert's analysis, I'm replacing it because there was an error
# glam_score_table <- as.data.frame(table(pData(data_no_6_mono_cds)$State, pData(data_no_6_mono_cds)$Present))

glam_score_table_M6 <- table(glam_stats_M6$State, glam_stats_M6$Present) %>% as.data.frame()
#Combine true and false rows
glam_score_table_M6 <- cbind(glam_score_table_M6[1:5,], glam_score_table_M6[6:10,]) ## this # depends on the # of monocle states
#Create names that make more sense
colnames(glam_score_table_M6) <- c("State", "F", "Freq_F", "State2", "T", "Freq_T")

#Remove unneeded columns
glam_score_table_M6 <- glam_score_table_M6[,c(1,3,6)]
#Create a percent column (percent of cells within each state with a good motif match)
glam_score_table_M6$pct <- glam_score_table_M6$Freq_T / (glam_score_table_M6$Freq_F + glam_score_table_M6$Freq_T)
#write.csv(glam_score_table_M4, "Motif/train/TFH/score_file/TRA_TFH_T65_motif1_M4_cln4_cells_across_M4.csv")
write.csv(glam_score_table_M6, "Motif_5mice/Train/Score/TRA_TFH_T65_motif4_M6_cln4_cells_across_M6_cells.csv")
getwd()
glam_score_table_M6$Freq_F %>% sum () # 2324
glam_score_table_M6$Freq_T %>% sum() # NA


# Perform clone-level statistical analysis

glam_stats_M2$combined_cdr3_nt[which(glam_stats_M2$Present == TRUE)] %>% table()


#         TRA:TGCTCAGCAAGTATGACTGGAGGCTATAAAGTGGTCTTT_TRB:TGTGCCAGCAGCAAGGTAAACACAGAAGTCTTCTTT 
 #                                                                                           4 
#TRA:TGTGCAGCAAGTCTCGATGGAGGCTATAAAGTGGTCTTT_TRB:TGTGCTAGCAGTTTATCCGGGACAGGGGACGCTGAGCAGTTCTTC 
    #                                                                                       30 
   #TRA:TGTGCAGTGAGCGGGACTGGAGGCTATAAAGTGGTCTTT_TRB:TGTGCCAGCAGGACTGGGGGGGCAGGGGACACCCAGTACTTT 
    #                                                                                       13 
     # TRA:TGTGCTCTGGGCCTGACTGGAGGCTATAAAGTGGTCTTT_TRB:TGCAGTGCAAGGGGGGCGGACACCGGGCAGCTCTACTTT 
       #                                                                                    21 

grep(pattern = "TRA:TGTGCAGCAAGTTGGGGGACTGGAGGCTATAAAGTGGTCTTT_TRB:TGTGCTAGCAGTTTTGCTGGGGGGGCTTCCTATGAACAGTACTTC", x = data_clones_by_mouse_AA_TRA_TRB_NOV$combined_cdr3_nt)
# 3
as.data.frame(data_clones_by_mouse_AA_TRA_TRB_NOV[1, ])
# 79, 0, 21
getwd()
glam_stats_M4$combined_cdr3_nt[which(glam_stats_M4$Present == TRUE)] %>% table()

        #  TRA:TGTGCAGCAAGCATGAGAACAGGAAACTACAAATACGTCTTT_TRB:TGTGCCAGCAGCGCCCAAACGCTGTATTTT 
         #                                                                                      5 
      #TRA:TGTGCAGCAAGTCTTAATACAGGAAACTACAAATACGTCTTT_TRB:TGTGCCTGGAATGGGGGGGGCTATGCTGAGCAGTTCTTC 
       #                                                                                       12 
#TRA:TGTGCAGCAAGTTGGGGGACTGGAGGCTATAAAGTGGTCTTT_TRB:TGTGCTAGCAGTTTTGCTGGGGGGGCTTCCTATGAACAGTACTTC 
 #                                                                                            813

```

```{r}
# TFH motif 5

data_monocle <- readRDS("summary_5M_CCA_V2/data_monocle_2cell.RDS")
# TFH Motif5
rm(glam2_scores)
glam2_scores <- read.table(file = "/media/runa/Elements/Jem_revision/Seurat/Motif_5mice/Train/TFH/Motif5/TFH_Motif5.glam2scan/glam2scan_scores_Motif5.txt", header = FALSE)
#We're only interested in the cell name and score columns
glam2_scores <- glam2_scores[,c(1,6)]
#Take the top score for each cell
glam2_scores <- aggregate(.~V1, glam2_scores, FUN=max) # here multiple cells can have
#more than one score, with aggregate command we are saying in those cases just keep the max
#score
rownames(glam2_scores) <- glam2_scores$V1
# to match cell names with monocle data add 1 in cell name
glam2_scores$V1 <-
  paste0(glam2_scores$V1, "1")
rownames(glam2_scores) <- glam2_scores$V1

#Add glam score to each cell within monocle pData
pData(data_monocle)$glam_score <- sapply(rownames(pData(data_monocle)), FUN =
function(x) glam2_scores[x,]$V6)

View(pData(data_monocle))


#Observe the glam score distribution, find a high score for the cells 40 looks good here; this
#scoring will vary from motif to motif

pdf("summary_5mice/Figure4/TFH_Motif5_Train_mice_score.pdf", width = 8, height = 6)
plot(density(pData(data_monocle)$glam_score[!is.na(pData(data_monocle)$glam_score)]),
main = "Motif score distribution_TRA_TFH_Motif5", xlab = "")
abline(v = 38, col = "Red")
dev.off()

#Make a true/false filtering column based on the glam score for plotting 
pData(data_monocle)$Present <- pData(data_monocle)$glam_score > 38

# Make a new dataframe to use for statistical analysis but not plotting the cells

rm(glam_stats_M2)
rm(glam_stats_M4)
rm(glam_stats_M6)

# Only include cells from mouse 2 and mouse 4, separately
glam_stats_M2 <- data_monocle$Present[which(data_monocle$mice == "M2")] %>% as.data.frame()
colnames(glam_stats_M2) <- "Present"
# Add other columns from data_monocle
glam_stats_M2$State <- data_monocle$State[which(data_monocle$mice == "M2")]
glam_stats_M2$combined_cdr3_nt <- data_monocle$combined_cdr3_nt[which(data_monocle$mice == "M2")]
glam_stats_M2$cellID <- rownames(data_monocle@phenoData)[which(data_monocle$mice == "M2")]
# Remove any rows with NA (cells without a motif score, incluing cells that don't meet definition of a clone)
glam_stats_M2 <- glam_stats_M2[!is.na(glam_stats_M2$Present), ]

glam_stats_M4 <- data_monocle$Present[which(data_monocle$mice == "M4")] %>% as.data.frame()
colnames(glam_stats_M4) <- "Present"
# Add other columns from data_monocle
glam_stats_M4$State <- data_monocle$State[which(data_monocle$mice == "M4")]
glam_stats_M4$combined_cdr3_nt <- data_monocle$combined_cdr3_nt[which(data_monocle$mice == "M4")]
glam_stats_M4$cellID <- rownames(data_monocle@phenoData)[which(data_monocle$mice == "M4")]
# Remove any rows with NA (cells without a motif score, incluing cells that don't meet definition of a clone)
glam_stats_M4 <- glam_stats_M4[!is.na(glam_stats_M4$Present), ]

glam_stats_M6 <- data_monocle$Present[which(data_monocle$mice == "M6")] %>% as.data.frame()
colnames(glam_stats_M6) <- "Present"
# Add other columns from data_monocle
glam_stats_M6$State <- data_monocle$State[which(data_monocle$mice == "M6")]
glam_stats_M6$combined_cdr3_nt <- data_monocle$combined_cdr3_nt[which(data_monocle$mice == "M6")]
glam_stats_M6$cellID <- rownames(data_monocle@phenoData)[which(data_monocle$mice == "M6")]
# Remove any rows with NA (cells without a motif score, incluing cells that don't meet definition of a clone)
glam_stats_M6 <- glam_stats_M6[!is.na(glam_stats_M6$Present), ]



#Make NA values false
pData(data_monocle)$Present[is.na(pData(data_monocle)$Present)] <- FALSE
plot_obj <- plot_complex_cell_trajectory(data_monocle, color_by = "Present", cell_size = 2)
plot_obj$data <- plot_obj$data[order(plot_obj$data$Present),]
plot_obj + scale_color_manual(values=c("grey", "red")) + ggtitle("TRA_TFH_motif5_38_across_M2M4M6'\n Score > 38 \n ")

pdf("summary_5mice/Figure4/TFH_Motif5_Train_mice_TREE.pdf", width = 8, height = 6, useDingbats = FALSE)
plot_obj + scale_color_manual(values=c("grey", "red")) + ggtitle("TRA_TFH_motif5_M2M4M6.png'\n Score > 38 \n 'P value 0.0001'")
dev.off()

#Create a table detailing the number of cells in each monocle state that have a good motif match
# The line below was from Robert's analysis, I'm replacing it because there was an error
# glam_score_table <- as.data.frame(table(pData(data_no_6_mono_cds)$State, pData(data_no_6_mono_cds)$Present))
rm(glam_score_table_M2)
rm(glam_score_table_M4)
rm(glam_score_table_M6)


glam_score_table_M2 <- table(glam_stats_M2$State, glam_stats_M2$Present) %>% as.data.frame()
#Combine true and false rows
glam_score_table_M2 <- cbind(glam_score_table_M2[1:5,], glam_score_table_M2[6:10,])
#Create names that make more sense
colnames(glam_score_table_M2) <- c("State", "F", "Freq_F", "State2", "T", "Freq_T")

#Remove unneeded columns
glam_score_table_M2 <- glam_score_table_M2[,c(1,3,6)]
#Create a percent column (percent of cells within each state with a good motif match)
glam_score_table_M2$pct <- glam_score_table_M2$Freq_T / (glam_score_table_M2$Freq_F + glam_score_table_M2$Freq_T)
#write.csv(glam_score_table_M2, "Motif/train/TFH/score_file/TRA_TFH_T65_motif1_M2_cln4_cells_across_M2.csv")
write.csv(glam_score_table_M2, "Motif_5mice/Train/Score/TRA_TFH_T65_motif5_M2_cln4_cells_across_M2_cells.csv")
getwd()
glam_score_table_M2$Freq_F %>% sum () # 840
glam_score_table_M2$Freq_T %>% sum() # 23

glam_stats_M2$combined_cdr3_nt[which(glam_stats_M2$Present == TRUE)] %>% table()

#TRA:TGTGCAGCAGGGGTTAATAACAACAATGCCCCACGATTT_TRB:TGTGCCAGCAGTGATCCGGGGGGGCAGGGTCAAAACACCTTGTACTTT 
 #                                                                                             23 

#Create a table detailing the number of cells in each monocle state that have a good motif match
# The line below was from Robert's analysis, I'm replacing it because there was an error
# glam_score_table <- as.data.frame(table(pData(data_no_6_mono_cds)$State, pData(data_no_6_mono_cds)$Present))

glam_score_table_M4 <- table(glam_stats_M4$State, glam_stats_M4$Present) %>% as.data.frame()
#Combine true and false rows
glam_score_table_M4 <- cbind(glam_score_table_M4[1:5,], glam_score_table_M4[6:10,]) ## this # depends on the # of monocle states
#Create names that make more sense
colnames(glam_score_table_M4) <- c("State", "F", "Freq_F", "State2", "T", "Freq_T")

#Remove unneeded columns
glam_score_table_M4 <- glam_score_table_M4[,c(1,3,6)]
#Create a percent column (percent of cells within each state with a good motif match)
glam_score_table_M4$pct <- glam_score_table_M4$Freq_T / (glam_score_table_M4$Freq_F + glam_score_table_M4$Freq_T)
#write.csv(glam_score_table_M4, "Motif/train/TFH/score_file/TRA_TFH_T65_motif1_M4_cln4_cells_across_M4.csv")
write.csv(glam_score_table_M4, "Motif_5mice/Train/Score/TRA_TFH_T65_motif5_M4_cln4_cells_across_M4_cells.csv")
getwd()
glam_score_table_M4$Freq_F %>% sum () # 3041
glam_score_table_M4$Freq_T %>% sum() # 177

glam_stats_M4$combined_cdr3_nt[which(glam_stats_M4$Present == TRUE)] %>% table()

 # TRA:TGTGCAGCAAGTGGCAACAACAATGCCCCACGATTT_TRB:TGTGCCAGCAGTGAACATCAAGACACCCAGTACTTT 
  #                                                                                   30 
  #    TRA:TGTGCAGCAAGTGGGAACAACAATGCCCCACGATTT_TRB:TGTGCCAGCAGTGACCATCAAGACACCCAGTACTTT 
   #                                                                                   8 
#TRA:TGTGCAGCAGGCTATAACAACAATGCCCCACGATTT_TRB:TGTGCCAGCGGTGATGCTGGGGACCAAGACACCCAGTACTTT 
 #                                                                                    38 
#TRA:TGTGCAGCAGGGTATAACAACAATGCCCCACGATTT_TRB:TGTGCCAGCGGTGATGCGGGGGACCAAGACACCCAGTACTTT 
 #                                                                                   101


#Create a table detailing the number of cells in each monocle state that have a good motif match
# The line below was from Robert's analysis, I'm replacing it because there was an error
# glam_score_table <- as.data.frame(table(pData(data_no_6_mono_cds)$State, pData(data_no_6_mono_cds)$Present))

glam_score_table_M6 <- table(glam_stats_M6$State, glam_stats_M6$Present) %>% as.data.frame()
#Combine true and false rows
glam_score_table_M6 <- cbind(glam_score_table_M6[1:5,], glam_score_table_M6[6:10,]) ## this # depends on the # of monocle states
#Create names that make more sense
colnames(glam_score_table_M6) <- c("State", "F", "Freq_F", "State2", "T", "Freq_T")

#Remove unneeded columns
glam_score_table_M6 <- glam_score_table_M6[,c(1,3,6)]
#Create a percent column (percent of cells within each state with a good motif match)
glam_score_table_M6$pct <- glam_score_table_M6$Freq_T / (glam_score_table_M6$Freq_F + glam_score_table_M6$Freq_T)
#write.csv(glam_score_table_M4, "Motif/train/TFH/score_file/TRA_TFH_T65_motif1_M4_cln4_cells_across_M4.csv")
write.csv(glam_score_table_M6, "Motif_5mice/Train/Score/TRA_TFH_T65_motif5_M6_cln4_cells_across_M6_cells.csv")
getwd()
glam_score_table_M6$Freq_F %>% sum () # 2245
glam_score_table_M6$Freq_T %>% sum() # 79

glam_stats_M6$combined_cdr3_nt[which(glam_stats_M6$Present == TRUE)] %>% table()

 #TRA:TGTGCAGCAGCGTATAACAACAATGCCCCACGATTT_TRB:TGTGCCAGCAGTGACTGGGGGTATGAACAGTACTTC 
  #                                                                                      60 
#TRA:TGTGCAGCAGGTTATAACAACAATGCCCCACGATTT_TRB:TGTGCCAGCGGTGATGCGGGACAGGGGAAGGATGAGCAGTTCTTC 
 #                                                                                       19 


```

```{r}

# TFH motif 6

#data_monocle <- readRDS("summary_5M_CCA_V2/data_monocle_2cell.RDS")
# TFH Motif6
rm(glam2_scores)
glam2_scores <- read.table(file = "/media/runa/Elements/Jem_revision/Seurat/Motif_5mice/Train/TFH/Motif6/TFH_Motif6.glam2scan/glam2scan_scores_Motif6.txt", header = FALSE)
#We're only interested in the cell name and score columns
glam2_scores <- glam2_scores[,c(1,6)]
#Take the top score for each cell
glam2_scores <- aggregate(.~V1, glam2_scores, FUN=max) # here multiple cells can have
#more than one score, with aggregate command we are saying in those cases just keep the max
#score
rownames(glam2_scores) <- glam2_scores$V1
# to match cell names with monocle data add 1 in cell name
glam2_scores$V1 <-
  paste0(glam2_scores$V1, "1")
rownames(glam2_scores) <- glam2_scores$V1

#Add glam score to each cell within monocle pData
pData(data_monocle)$glam_score <- sapply(rownames(pData(data_monocle)), FUN =
function(x) glam2_scores[x,]$V6)

View(pData(data_monocle))


#Observe the glam score distribution, find a high score for the cells 40 looks good here; this
#scoring will vary from motif to motif
plot(density(pData(data_monocle)$glam_score[!is.na(pData(data_monocle)$glam_score)]),
main = "Motif score distribution_TRA_TFH_Motif6", xlab = "")
abline(v = 35, col = "Red")

#Make a true/false filtering column based on the glam score for plotting 
pData(data_monocle)$Present <- pData(data_monocle)$glam_score > 35

# Make a new dataframe to use for statistical analysis but not plotting the cells

rm(glam_stats_M2)
rm(glam_stats_M4)
rm(glam_stats_M6)

# Only include cells from mouse 2 and mouse 4, separately
glam_stats_M2 <- data_monocle$Present[which(data_monocle$mice == "M2")] %>% as.data.frame()
colnames(glam_stats_M2) <- "Present"
# Add other columns from data_monocle
glam_stats_M2$State <- data_monocle$State[which(data_monocle$mice == "M2")]
glam_stats_M2$combined_cdr3_nt <- data_monocle$combined_cdr3_nt[which(data_monocle$mice == "M2")]
glam_stats_M2$cellID <- rownames(data_monocle@phenoData)[which(data_monocle$mice == "M2")]
# Remove any rows with NA (cells without a motif score, incluing cells that don't meet definition of a clone)
glam_stats_M2 <- glam_stats_M2[!is.na(glam_stats_M2$Present), ]

glam_stats_M4 <- data_monocle$Present[which(data_monocle$mice == "M4")] %>% as.data.frame()
colnames(glam_stats_M4) <- "Present"
# Add other columns from data_monocle
glam_stats_M4$State <- data_monocle$State[which(data_monocle$mice == "M4")]
glam_stats_M4$combined_cdr3_nt <- data_monocle$combined_cdr3_nt[which(data_monocle$mice == "M4")]
glam_stats_M4$cellID <- rownames(data_monocle@phenoData)[which(data_monocle$mice == "M4")]
# Remove any rows with NA (cells without a motif score, incluing cells that don't meet definition of a clone)
glam_stats_M4 <- glam_stats_M4[!is.na(glam_stats_M4$Present), ]

glam_stats_M6 <- data_monocle$Present[which(data_monocle$mice == "M6")] %>% as.data.frame()
colnames(glam_stats_M6) <- "Present"
# Add other columns from data_monocle
glam_stats_M6$State <- data_monocle$State[which(data_monocle$mice == "M6")]
glam_stats_M6$combined_cdr3_nt <- data_monocle$combined_cdr3_nt[which(data_monocle$mice == "M6")]
glam_stats_M6$cellID <- rownames(data_monocle@phenoData)[which(data_monocle$mice == "M6")]
# Remove any rows with NA (cells without a motif score, incluing cells that don't meet definition of a clone)
glam_stats_M6 <- glam_stats_M6[!is.na(glam_stats_M6$Present), ]



#Make NA values false
pData(data_monocle)$Present[is.na(pData(data_monocle)$Present)] <- FALSE
plot_obj <- plot_complex_cell_trajectory(data_monocle, color_by = "Present")
plot_obj$data <- plot_obj$data[order(plot_obj$data$Present),]
plot_obj + scale_color_manual(values=c("grey", "red")) + ggtitle("TRA_TFH_motif6_35_across_M2M4M6'\n Score > 35 \n ")

png(file = "summary_6mice/TRA_TH1_motif1_M2M4M6_cln4.png", width = 8, height = 6, units = "in", res = 300)
plot_obj + scale_color_manual(values=c("grey", "red")) + ggtitle("TRA_TH1_motif6_M2M4M6.png'\n Score > 37 \n 'P value '")
#dev.off()

#Create a table detailing the number of cells in each monocle state that have a good motif match
# The line below was from Robert's analysis, I'm replacing it because there was an error
# glam_score_table <- as.data.frame(table(pData(data_no_6_mono_cds)$State, pData(data_no_6_mono_cds)$Present))
rm(glam_score_table_M2)
rm(glam_score_table_M4)
rm(glam_score_table_M6)
glam_score_table_M2 <- table(glam_stats_M2$State, glam_stats_M2$Present) %>% as.data.frame()
#Combine true and false rows
glam_score_table_M2 <- cbind(glam_score_table_M2[1:5,], glam_score_table_M2[6:10,])
#Create names that make more sense
colnames(glam_score_table_M2) <- c("State", "F", "Freq_F", "State2", "T", "Freq_T")

#Remove unneeded columns
glam_score_table_M2 <- glam_score_table_M2[,c(1,3,6)]
#Create a percent column (percent of cells within each state with a good motif match)
glam_score_table_M2$pct <- glam_score_table_M2$Freq_T / (glam_score_table_M2$Freq_F + glam_score_table_M2$Freq_T)
#write.csv(glam_score_table_M2, "Motif/train/TFH/score_file/TRA_TFH_T65_motif1_M2_cln4_cells_across_M2.csv")
write.csv(glam_score_table_M2, "Motif_5mice/Train/Score/TRA_TFH_T65_motif6_M2_cln4_cells_across_M2_cells.csv")
getwd()
glam_score_table_M2$Freq_F %>% sum () # 756
glam_score_table_M2$Freq_T %>% sum() # 107

#Create a table detailing the number of cells in each monocle state that have a good motif match
# The line below was from Robert's analysis, I'm replacing it because there was an error
# glam_score_table <- as.data.frame(table(pData(data_no_6_mono_cds)$State, pData(data_no_6_mono_cds)$Present))

glam_score_table_M4 <- table(glam_stats_M4$State, glam_stats_M4$Present) %>% as.data.frame()
#Combine true and false rows
glam_score_table_M4 <- cbind(glam_score_table_M4[1:5,], glam_score_table_M4[6:10,]) ## this # depends on the # of monocle states
#Create names that make more sense
colnames(glam_score_table_M4) <- c("State", "F", "Freq_F", "State2", "T", "Freq_T")

#Remove unneeded columns
glam_score_table_M4 <- glam_score_table_M4[,c(1,3,6)]
#Create a percent column (percent of cells within each state with a good motif match)
glam_score_table_M4$pct <- glam_score_table_M4$Freq_T / (glam_score_table_M4$Freq_F + glam_score_table_M4$Freq_T)
#write.csv(glam_score_table_M4, "Motif/train/TFH/score_file/TRA_TFH_T65_motif1_M4_cln4_cells_across_M4.csv")
write.csv(glam_score_table_M4, "Motif_5mice/Train/Score/TRA_TFH_T65_motif6_M4_cln4_cells_across_M4_cells.csv")
getwd()
glam_score_table_M4$Freq_F %>% sum () # 3113
glam_score_table_M4$Freq_T %>% sum() # 105


#Create a table detailing the number of cells in each monocle state that have a good motif match
# The line below was from Robert's analysis, I'm replacing it because there was an error
# glam_score_table <- as.data.frame(table(pData(data_no_6_mono_cds)$State, pData(data_no_6_mono_cds)$Present))

glam_score_table_M6 <- table(glam_stats_M6$State, glam_stats_M6$Present) %>% as.data.frame()
#Combine true and false rows
glam_score_table_M6 <- cbind(glam_score_table_M6[1:5,], glam_score_table_M6[6:10,]) ## this # depends on the # of monocle states
#Create names that make more sense
colnames(glam_score_table_M6) <- c("State", "F", "Freq_F", "State2", "T", "Freq_T")

#Remove unneeded columns
glam_score_table_M6 <- glam_score_table_M6[,c(1,3,6)]
#Create a percent column (percent of cells within each state with a good motif match)
glam_score_table_M6$pct <- glam_score_table_M6$Freq_T / (glam_score_table_M6$Freq_F + glam_score_table_M6$Freq_T)
#write.csv(glam_score_table_M4, "Motif/train/TFH/score_file/TRA_TFH_T65_motif1_M4_cln4_cells_across_M4.csv")
write.csv(glam_score_table_M6, "Motif_5mice/Train/Score/TRA_TFH_T65_motif6_M6_cln4_cells_across_M6_cells.csv")
getwd()
glam_score_table_M6$Freq_F %>% sum () # 2285
glam_score_table_M6$Freq_T %>% sum() # 39


# Perform clone-level statistical analysis

glam_stats_M2$combined_cdr3_nt[which(glam_stats_M2$Present == TRUE)] %>% table()


#         TRA:TGCTCAGCAAGTATGACTGGAGGCTATAAAGTGGTCTTT_TRB:TGTGCCAGCAGCAAGGTAAACACAGAAGTCTTCTTT 
 #                                                                                           4 
#TRA:TGTGCAGCAAGTCTCGATGGAGGCTATAAAGTGGTCTTT_TRB:TGTGCTAGCAGTTTATCCGGGACAGGGGACGCTGAGCAGTTCTTC 
    #                                                                                       30 
   #TRA:TGTGCAGTGAGCGGGACTGGAGGCTATAAAGTGGTCTTT_TRB:TGTGCCAGCAGGACTGGGGGGGCAGGGGACACCCAGTACTTT 
    #                                                                                       13 
     # TRA:TGTGCTCTGGGCCTGACTGGAGGCTATAAAGTGGTCTTT_TRB:TGCAGTGCAAGGGGGGCGGACACCGGGCAGCTCTACTTT 
       #                                                                                    21 

grep(pattern = "TRA:TGTGCAGCAAGTTGGGGGACTGGAGGCTATAAAGTGGTCTTT_TRB:TGTGCTAGCAGTTTTGCTGGGGGGGCTTCCTATGAACAGTACTTC", x = data_clones_by_mouse_AA_TRA_TRB_NOV$combined_cdr3_nt)
# 3
as.data.frame(data_clones_by_mouse_AA_TRA_TRB_NOV[1, ])
# 79, 0, 21
getwd()
glam_stats_M4$combined_cdr3_nt[which(glam_stats_M4$Present == TRUE)] %>% table()

        #  TRA:TGTGCAGCAAGCATGAGAACAGGAAACTACAAATACGTCTTT_TRB:TGTGCCAGCAGCGCCCAAACGCTGTATTTT 
         #                                                                                      5 
      #TRA:TGTGCAGCAAGTCTTAATACAGGAAACTACAAATACGTCTTT_TRB:TGTGCCTGGAATGGGGGGGGCTATGCTGAGCAGTTCTTC 
       #                                                                                       12 
#TRA:TGTGCAGCAAGTTGGGGGACTGGAGGCTATAAAGTGGTCTTT_TRB:TGTGCTAGCAGTTTTGCTGGGGGGGCTTCCTATGAACAGTACTTC 
 #                                                                                            813

```


```{r}
# TFH motif 7

#data_monocle <- readRDS("summary_5M_CCA_V2/data_monocle_2cell.RDS")
# TFH Motif7
rm(glam2_scores)
glam2_scores <- read.table(file = "/media/runa/Elements/Jem_revision/Seurat/Motif_5mice/Train/TFH/Motif7/TFH_Motif7.glam2scan/glam2scan_scores_Motif7.txt", header = FALSE)
#We're only interested in the cell name and score columns
glam2_scores <- glam2_scores[,c(1,6)]
#Take the top score for each cell
glam2_scores <- aggregate(.~V1, glam2_scores, FUN=max) # here multiple cells can have
#more than one score, with aggregate command we are saying in those cases just keep the max
#score
rownames(glam2_scores) <- glam2_scores$V1
# to match cell names with monocle data add 1 in cell name
glam2_scores$V1 <-
  paste0(glam2_scores$V1, "1")
rownames(glam2_scores) <- glam2_scores$V1

#Add glam score to each cell within monocle pData
pData(data_monocle)$glam_score <- sapply(rownames(pData(data_monocle)), FUN =
function(x) glam2_scores[x,]$V6)

View(pData(data_monocle))


#Observe the glam score distribution, find a high score for the cells 40 looks good here; this
#scoring will vary from motif to motif
plot(density(pData(data_monocle)$glam_score[!is.na(pData(data_monocle)$glam_score)]),
main = "Motif score distribution_TRA_TFH_Motif7", xlab = "")
abline(v = 40, col = "Red")

#Make a true/false filtering column based on the glam score for plotting 
pData(data_monocle)$Present <- pData(data_monocle)$glam_score > 40

# Make a new dataframe to use for statistical analysis but not plotting the cells

rm(glam_stats_M2)
rm(glam_stats_M4)
rm(glam_stats_M6)

# Only include cells from mouse 2 and mouse 4, separately
glam_stats_M2 <- data_monocle$Present[which(data_monocle$mice == "M2")] %>% as.data.frame()
colnames(glam_stats_M2) <- "Present"
# Add other columns from data_monocle
glam_stats_M2$State <- data_monocle$State[which(data_monocle$mice == "M2")]
glam_stats_M2$combined_cdr3_nt <- data_monocle$combined_cdr3_nt[which(data_monocle$mice == "M2")]
glam_stats_M2$cellID <- rownames(data_monocle@phenoData)[which(data_monocle$mice == "M2")]
# Remove any rows with NA (cells without a motif score, incluing cells that don't meet definition of a clone)
glam_stats_M2 <- glam_stats_M2[!is.na(glam_stats_M2$Present), ]

glam_stats_M4 <- data_monocle$Present[which(data_monocle$mice == "M4")] %>% as.data.frame()
colnames(glam_stats_M4) <- "Present"
# Add other columns from data_monocle
glam_stats_M4$State <- data_monocle$State[which(data_monocle$mice == "M4")]
glam_stats_M4$combined_cdr3_nt <- data_monocle$combined_cdr3_nt[which(data_monocle$mice == "M4")]
glam_stats_M4$cellID <- rownames(data_monocle@phenoData)[which(data_monocle$mice == "M4")]
# Remove any rows with NA (cells without a motif score, incluing cells that don't meet definition of a clone)
glam_stats_M4 <- glam_stats_M4[!is.na(glam_stats_M4$Present), ]

glam_stats_M6 <- data_monocle$Present[which(data_monocle$mice == "M6")] %>% as.data.frame()
colnames(glam_stats_M6) <- "Present"
# Add other columns from data_monocle
glam_stats_M6$State <- data_monocle$State[which(data_monocle$mice == "M6")]
glam_stats_M6$combined_cdr3_nt <- data_monocle$combined_cdr3_nt[which(data_monocle$mice == "M6")]
glam_stats_M6$cellID <- rownames(data_monocle@phenoData)[which(data_monocle$mice == "M6")]
# Remove any rows with NA (cells without a motif score, incluing cells that don't meet definition of a clone)
glam_stats_M6 <- glam_stats_M6[!is.na(glam_stats_M6$Present), ]



#Make NA values false
pData(data_monocle)$Present[is.na(pData(data_monocle)$Present)] <- FALSE
plot_obj <- plot_complex_cell_trajectory(data_monocle, color_by = "Present")
plot_obj$data <- plot_obj$data[order(plot_obj$data$Present),]
plot_obj + scale_color_manual(values=c("grey", "red")) + ggtitle("TRA_TFH_motif7_40_across_M2M4M6'\n Score > 40 \n ")

png(file = "summary_6mice/TRA_TH1_motif1_M2M4M6_cln4.png", width = 8, height = 6, units = "in", res = 300)
plot_obj + scale_color_manual(values=c("grey", "red")) + ggtitle("TRA_TH1_motif6_M2M4M6.png'\n Score > 37 \n 'P value '")
#dev.off()

#Create a table detailing the number of cells in each monocle state that have a good motif match
# The line below was from Robert's analysis, I'm replacing it because there was an error
# glam_score_table <- as.data.frame(table(pData(data_no_6_mono_cds)$State, pData(data_no_6_mono_cds)$Present))
rm(glam_score_table_M2)
rm(glam_score_table_M4)
rm(glam_score_table_M6)
glam_score_table_M2 <- table(glam_stats_M2$State, glam_stats_M2$Present) %>% as.data.frame()
#Combine true and false rows
glam_score_table_M2 <- cbind(glam_score_table_M2[1:5,], glam_score_table_M2[6:10,])
#Create names that make more sense
colnames(glam_score_table_M2) <- c("State", "F", "Freq_F", "State2", "T", "Freq_T")

#Remove unneeded columns
glam_score_table_M2 <- glam_score_table_M2[,c(1,3,6)]
#Create a percent column (percent of cells within each state with a good motif match)
glam_score_table_M2$pct <- glam_score_table_M2$Freq_T / (glam_score_table_M2$Freq_F + glam_score_table_M2$Freq_T)
#write.csv(glam_score_table_M2, "Motif/train/TFH/score_file/TRA_TFH_T65_motif1_M2_cln4_cells_across_M2.csv")
write.csv(glam_score_table_M2, "Motif_5mice/Train/Score/TRA_TFH_T65_motif7_M2_cln4_cells_across_M2_cells.csv")
getwd()
glam_score_table_M2$Freq_F %>% sum () # 838
glam_score_table_M2$Freq_T %>% sum() # 25

#Create a table detailing the number of cells in each monocle state that have a good motif match
# The line below was from Robert's analysis, I'm replacing it because there was an error
# glam_score_table <- as.data.frame(table(pData(data_no_6_mono_cds)$State, pData(data_no_6_mono_cds)$Present))

glam_score_table_M4 <- table(glam_stats_M4$State, glam_stats_M4$Present) %>% as.data.frame()
#Combine true and false rows
glam_score_table_M4 <- cbind(glam_score_table_M4[1:5,], glam_score_table_M4[6:10,]) ## this # depends on the # of monocle states
#Create names that make more sense
colnames(glam_score_table_M4) <- c("State", "F", "Freq_F", "State2", "T", "Freq_T")

#Remove unneeded columns
glam_score_table_M4 <- glam_score_table_M4[,c(1,3,6)]
#Create a percent column (percent of cells within each state with a good motif match)
glam_score_table_M4$pct <- glam_score_table_M4$Freq_T / (glam_score_table_M4$Freq_F + glam_score_table_M4$Freq_T)
#write.csv(glam_score_table_M4, "Motif/train/TFH/score_file/TRA_TFH_T65_motif1_M4_cln4_cells_across_M4.csv")
write.csv(glam_score_table_M4, "Motif_5mice/Train/Score/TRA_TFH_T65_motif7_M4_cln4_cells_across_M4_cells.csv")
getwd()
glam_score_table_M4$Freq_F %>% sum () # 3187
glam_score_table_M4$Freq_T %>% sum() # 31


#Create a table detailing the number of cells in each monocle state that have a good motif match
# The line below was from Robert's analysis, I'm replacing it because there was an error
# glam_score_table <- as.data.frame(table(pData(data_no_6_mono_cds)$State, pData(data_no_6_mono_cds)$Present))

glam_score_table_M6 <- table(glam_stats_M6$State, glam_stats_M6$Present) %>% as.data.frame()
#Combine true and false rows
glam_score_table_M6 <- cbind(glam_score_table_M6[1:5,], glam_score_table_M6[6:10,]) ## this # depends on the # of monocle states
#Create names that make more sense
colnames(glam_score_table_M6) <- c("State", "F", "Freq_F", "State2", "T", "Freq_T")

#Remove unneeded columns
glam_score_table_M6 <- glam_score_table_M6[,c(1,3,6)]
#Create a percent column (percent of cells within each state with a good motif match)
glam_score_table_M6$pct <- glam_score_table_M6$Freq_T / (glam_score_table_M6$Freq_F + glam_score_table_M6$Freq_T)
#write.csv(glam_score_table_M4, "Motif/train/TFH/score_file/TRA_TFH_T65_motif1_M4_cln4_cells_across_M4.csv")
write.csv(glam_score_table_M6, "Motif_5mice/Train/Score/TRA_TFH_T65_motif7_M6_cln4_cells_across_M6_cells.csv")
getwd()
glam_score_table_M6$Freq_F %>% sum () # 2285
glam_score_table_M6$Freq_T %>% sum() # 39


# Perform clone-level statistical analysis

glam_stats_M2$combined_cdr3_nt[which(glam_stats_M2$Present == TRUE)] %>% table()


#         TRA:TGCTCAGCAAGTATGACTGGAGGCTATAAAGTGGTCTTT_TRB:TGTGCCAGCAGCAAGGTAAACACAGAAGTCTTCTTT 
 #                                                                                           4 
#TRA:TGTGCAGCAAGTCTCGATGGAGGCTATAAAGTGGTCTTT_TRB:TGTGCTAGCAGTTTATCCGGGACAGGGGACGCTGAGCAGTTCTTC 
    #                                                                                       30 
   #TRA:TGTGCAGTGAGCGGGACTGGAGGCTATAAAGTGGTCTTT_TRB:TGTGCCAGCAGGACTGGGGGGGCAGGGGACACCCAGTACTTT 
    #                                                                                       13 
     # TRA:TGTGCTCTGGGCCTGACTGGAGGCTATAAAGTGGTCTTT_TRB:TGCAGTGCAAGGGGGGCGGACACCGGGCAGCTCTACTTT 
       #                                                                                    21 

grep(pattern = "TRA:TGTGCAGCAAGTTGGGGGACTGGAGGCTATAAAGTGGTCTTT_TRB:TGTGCTAGCAGTTTTGCTGGGGGGGCTTCCTATGAACAGTACTTC", x = data_clones_by_mouse_AA_TRA_TRB_NOV$combined_cdr3_nt)
# 3
as.data.frame(data_clones_by_mouse_AA_TRA_TRB_NOV[1, ])
# 79, 0, 21
getwd()
glam_stats_M4$combined_cdr3_nt[which(glam_stats_M4$Present == TRUE)] %>% table()

        #  TRA:TGTGCAGCAAGCATGAGAACAGGAAACTACAAATACGTCTTT_TRB:TGTGCCAGCAGCGCCCAAACGCTGTATTTT 
         #                                                                                      5 
      #TRA:TGTGCAGCAAGTCTTAATACAGGAAACTACAAATACGTCTTT_TRB:TGTGCCTGGAATGGGGGGGGCTATGCTGAGCAGTTCTTC 
       #                                                                                       12 
#TRA:TGTGCAGCAAGTTGGGGGACTGGAGGCTATAAAGTGGTCTTT_TRB:TGTGCTAGCAGTTTTGCTGGGGGGGCTTCCTATGAACAGTACTTC 
 #                              
```

```{r}
# TFH motif 8

#data_monocle <- readRDS("summary_5M_CCA_V2/data_monocle_2cell.RDS")
# TFH Motif8
rm(glam2_scores)
glam2_scores <- read.table(file = "/media/runa/Elements/Jem_revision/Seurat/Motif_5mice/Train/TFH/Motif8/TFH_Motif8.glam2scan/glam2scan_scores_Motif8.txt", header = FALSE)
#We're only interested in the cell name and score columns
glam2_scores <- glam2_scores[,c(1,6)]
#Take the top score for each cell
glam2_scores <- aggregate(.~V1, glam2_scores, FUN=max) # here multiple cells can have
#more than one score, with aggregate command we are saying in those cases just keep the max
#score
rownames(glam2_scores) <- glam2_scores$V1
# to match cell names with monocle data add 1 in cell name
glam2_scores$V1 <-
  paste0(glam2_scores$V1, "1")
rownames(glam2_scores) <- glam2_scores$V1

#Add glam score to each cell within monocle pData
pData(data_monocle)$glam_score <- sapply(rownames(pData(data_monocle)), FUN =
function(x) glam2_scores[x,]$V6)

View(pData(data_monocle))


#Observe the glam score distribution, find a high score for the cells 40 looks good here; this
#scoring will vary from motif to motif
plot(density(pData(data_monocle)$glam_score[!is.na(pData(data_monocle)$glam_score)]),
main = "Motif score distribution_TRA_TFH_Motif8", xlab = "")
abline(v = 32, col = "Red")

#Make a true/false filtering column based on the glam score for plotting 
pData(data_monocle)$Present <- pData(data_monocle)$glam_score > 32

# Make a new dataframe to use for statistical analysis but not plotting the cells

rm(glam_stats_M2)
rm(glam_stats_M4)
rm(glam_stats_M6)

# Only include cells from mouse 2 and mouse 4, separately
glam_stats_M2 <- data_monocle$Present[which(data_monocle$mice == "M2")] %>% as.data.frame()
colnames(glam_stats_M2) <- "Present"
# Add other columns from data_monocle
glam_stats_M2$State <- data_monocle$State[which(data_monocle$mice == "M2")]
glam_stats_M2$combined_cdr3_nt <- data_monocle$combined_cdr3_nt[which(data_monocle$mice == "M2")]
glam_stats_M2$cellID <- rownames(data_monocle@phenoData)[which(data_monocle$mice == "M2")]
# Remove any rows with NA (cells without a motif score, incluing cells that don't meet definition of a clone)
glam_stats_M2 <- glam_stats_M2[!is.na(glam_stats_M2$Present), ]

glam_stats_M4 <- data_monocle$Present[which(data_monocle$mice == "M4")] %>% as.data.frame()
colnames(glam_stats_M4) <- "Present"
# Add other columns from data_monocle
glam_stats_M4$State <- data_monocle$State[which(data_monocle$mice == "M4")]
glam_stats_M4$combined_cdr3_nt <- data_monocle$combined_cdr3_nt[which(data_monocle$mice == "M4")]
glam_stats_M4$cellID <- rownames(data_monocle@phenoData)[which(data_monocle$mice == "M4")]
# Remove any rows with NA (cells without a motif score, incluing cells that don't meet definition of a clone)
glam_stats_M4 <- glam_stats_M4[!is.na(glam_stats_M4$Present), ]

glam_stats_M6 <- data_monocle$Present[which(data_monocle$mice == "M6")] %>% as.data.frame()
colnames(glam_stats_M6) <- "Present"
# Add other columns from data_monocle
glam_stats_M6$State <- data_monocle$State[which(data_monocle$mice == "M6")]
glam_stats_M6$combined_cdr3_nt <- data_monocle$combined_cdr3_nt[which(data_monocle$mice == "M6")]
glam_stats_M6$cellID <- rownames(data_monocle@phenoData)[which(data_monocle$mice == "M6")]
# Remove any rows with NA (cells without a motif score, incluing cells that don't meet definition of a clone)
glam_stats_M6 <- glam_stats_M6[!is.na(glam_stats_M6$Present), ]



#Make NA values false
pData(data_monocle)$Present[is.na(pData(data_monocle)$Present)] <- FALSE
plot_obj <- plot_complex_cell_trajectory(data_monocle, color_by = "Present")
plot_obj$data <- plot_obj$data[order(plot_obj$data$Present),]
plot_obj + scale_color_manual(values=c("grey", "red")) + ggtitle("TRA_TFH_motif8_32_across_M2M4M6'\n Score > 32 \n ")

png(file = "summary_6mice/TRA_TH1_motif1_M2M4M6_cln4.png", width = 8, height = 6, units = "in", res = 300)
plot_obj + scale_color_manual(values=c("grey", "red")) + ggtitle("TRA_TH1_motif6_M2M4M6.png'\n Score > 37 \n 'P value '")
#dev.off()

#Create a table detailing the number of cells in each monocle state that have a good motif match
# The line below was from Robert's analysis, I'm replacing it because there was an error
# glam_score_table <- as.data.frame(table(pData(data_no_6_mono_cds)$State, pData(data_no_6_mono_cds)$Present))
rm(glam_score_table_M2)
rm(glam_score_table_M4)
rm(glam_score_table_M6)
glam_score_table_M2 <- table(glam_stats_M2$State, glam_stats_M2$Present) %>% as.data.frame()
#Combine true and false rows
glam_score_table_M2 <- cbind(glam_score_table_M2[1:5,], glam_score_table_M2[6:10,])
#Create names that make more sense
colnames(glam_score_table_M2) <- c("State", "F", "Freq_F", "State2", "T", "Freq_T")

#Remove unneeded columns
glam_score_table_M2 <- glam_score_table_M2[,c(1,3,6)]
#Create a percent column (percent of cells within each state with a good motif match)
glam_score_table_M2$pct <- glam_score_table_M2$Freq_T / (glam_score_table_M2$Freq_F + glam_score_table_M2$Freq_T)
#write.csv(glam_score_table_M2, "Motif/train/TFH/score_file/TRA_TFH_T65_motif1_M2_cln4_cells_across_M2.csv")
write.csv(glam_score_table_M2, "Motif_5mice/Train/Score/TRA_TFH_T65_motif8_M2_cln4_cells_across_M2_cells.csv")
getwd()
glam_score_table_M2$Freq_F %>% sum () # 841
glam_score_table_M2$Freq_T %>% sum() # 22

#Create a table detailing the number of cells in each monocle state that have a good motif match
# The line below was from Robert's analysis, I'm replacing it because there was an error
# glam_score_table <- as.data.frame(table(pData(data_no_6_mono_cds)$State, pData(data_no_6_mono_cds)$Present))

glam_score_table_M4 <- table(glam_stats_M4$State, glam_stats_M4$Present) %>% as.data.frame()
#Combine true and false rows
glam_score_table_M4 <- cbind(glam_score_table_M4[1:5,], glam_score_table_M4[6:10,]) ## this # depends on the # of monocle states
#Create names that make more sense
colnames(glam_score_table_M4) <- c("State", "F", "Freq_F", "State2", "T", "Freq_T")

#Remove unneeded columns
glam_score_table_M4 <- glam_score_table_M4[,c(1,3,6)]
#Create a percent column (percent of cells within each state with a good motif match)
glam_score_table_M4$pct <- glam_score_table_M4$Freq_T / (glam_score_table_M4$Freq_F + glam_score_table_M4$Freq_T)
#write.csv(glam_score_table_M4, "Motif/train/TFH/score_file/TRA_TFH_T65_motif1_M4_cln4_cells_across_M4.csv")
write.csv(glam_score_table_M4, "Motif_5mice/Train/Score/TRA_TFH_T65_motif8_M4_cln4_cells_across_M4_cells.csv")
getwd()
glam_score_table_M4$Freq_F %>% sum () # 3201
glam_score_table_M4$Freq_T %>% sum() # 17


#Create a table detailing the number of cells in each monocle state that have a good motif match
# The line below was from Robert's analysis, I'm replacing it because there was an error
# glam_score_table <- as.data.frame(table(pData(data_no_6_mono_cds)$State, pData(data_no_6_mono_cds)$Present))

glam_score_table_M6 <- table(glam_stats_M6$State, glam_stats_M6$Present) %>% as.data.frame()
#Combine true and false rows
glam_score_table_M6 <- cbind(glam_score_table_M6[1:5,], glam_score_table_M6[6:10,]) ## this # depends on the # of monocle states
#Create names that make more sense
colnames(glam_score_table_M6) <- c("State", "F", "Freq_F", "State2", "T", "Freq_T")

#Remove unneeded columns
glam_score_table_M6 <- glam_score_table_M6[,c(1,3,6)]
#Create a percent column (percent of cells within each state with a good motif match)
glam_score_table_M6$pct <- glam_score_table_M6$Freq_T / (glam_score_table_M6$Freq_F + glam_score_table_M6$Freq_T)
#write.csv(glam_score_table_M4, "Motif/train/TFH/score_file/TRA_TFH_T65_motif1_M4_cln4_cells_across_M4.csv")
write.csv(glam_score_table_M6, "Motif_5mice/Train/Score/TRA_TFH_T65_motif8_M6_cln4_cells_across_M6_cells.csv")
getwd()
glam_score_table_M6$Freq_F %>% sum () # 2306
glam_score_table_M6$Freq_T %>% sum() # 18


# Perform clone-level statistical analysis

glam_stats_M2$combined_cdr3_nt[which(glam_stats_M2$Present == TRUE)] %>% table()


#         TRA:TGCTCAGCAAGTATGACTGGAGGCTATAAAGTGGTCTTT_TRB:TGTGCCAGCAGCAAGGTAAACACAGAAGTCTTCTTT 
 #                                                                                           4 
#TRA:TGTGCAGCAAGTCTCGATGGAGGCTATAAAGTGGTCTTT_TRB:TGTGCTAGCAGTTTATCCGGGACAGGGGACGCTGAGCAGTTCTTC 
    #                                                                                       30 
   #TRA:TGTGCAGTGAGCGGGACTGGAGGCTATAAAGTGGTCTTT_TRB:TGTGCCAGCAGGACTGGGGGGGCAGGGGACACCCAGTACTTT 
    #                                                                                       13 
     # TRA:TGTGCTCTGGGCCTGACTGGAGGCTATAAAGTGGTCTTT_TRB:TGCAGTGCAAGGGGGGCGGACACCGGGCAGCTCTACTTT 
       #                                                                                    21 

grep(pattern = "TRA:TGTGCAGCAAGTTGGGGGACTGGAGGCTATAAAGTGGTCTTT_TRB:TGTGCTAGCAGTTTTGCTGGGGGGGCTTCCTATGAACAGTACTTC", x = data_clones_by_mouse_AA_TRA_TRB_NOV$combined_cdr3_nt)
# 3
as.data.frame(data_clones_by_mouse_AA_TRA_TRB_NOV[1, ])
# 79, 0, 21
getwd()
glam_stats_M4$combined_cdr3_nt[which(glam_stats_M4$Present == TRUE)] %>% table()

        #  TRA:TGTGCAGCAAGCATGAGAACAGGAAACTACAAATACGTCTTT_TRB:TGTGCCAGCAGCGCCCAAACGCTGTATTTT 
         #                                                                                      5 
      #TRA:TGTGCAGCAAGTCTTAATACAGGAAACTACAAATACGTCTTT_TRB:TGTGCCTGGAATGGGGGGGGCTATGCTGAGCAGTTCTTC 
       #                                                                                       12 
#TRA:TGTGCAGCAAGTTGGGGGACTGGAGGCTATAAAGTGGTCTTT_TRB:TGTGCTAGCAGTTTTGCTGGGGGGGCTTCCTATGAACAGTACTTC 
 #                              
```

```{r}
# TFH motif 9

#data_monocle <- readRDS("summary_5M_CCA_V2/data_monocle_2cell.RDS")
# TFH Motif9
rm(glam2_scores)
glam2_scores <- read.table(file = "/media/runa/Elements/Jem_revision/Seurat/Motif_5mice/Train/TFH/Motif9/TFH_Motif9.glam2scan/glam2scan_scores_Motif9.txt", header = FALSE)
#We're only interested in the cell name and score columns
glam2_scores <- glam2_scores[,c(1,6)]
#Take the top score for each cell
glam2_scores <- aggregate(.~V1, glam2_scores, FUN=max) # here multiple cells can have
#more than one score, with aggregate command we are saying in those cases just keep the max
#score
rownames(glam2_scores) <- glam2_scores$V1
# to match cell names with monocle data add 1 in cell name
glam2_scores$V1 <-
  paste0(glam2_scores$V1, "1")
rownames(glam2_scores) <- glam2_scores$V1

#Add glam score to each cell within monocle pData
pData(data_monocle)$glam_score <- sapply(rownames(pData(data_monocle)), FUN =
function(x) glam2_scores[x,]$V6)

View(pData(data_monocle))


#Observe the glam score distribution, find a high score for the cells 40 looks good here; this
#scoring will vary from motif to motif
plot(density(pData(data_monocle)$glam_score[!is.na(pData(data_monocle)$glam_score)]),
main = "Motif score distribution_TRA_TFH_Motif9", xlab = "")
abline(v = 40, col = "Red")

#Make a true/false filtering column based on the glam score for plotting 
pData(data_monocle)$Present <- pData(data_monocle)$glam_score > 40

# Make a new dataframe to use for statistical analysis but not plotting the cells

rm(glam_stats_M2)
rm(glam_stats_M4)
rm(glam_stats_M6)

# Only include cells from mouse 2 and mouse 4, separately
glam_stats_M2 <- data_monocle$Present[which(data_monocle$mice == "M2")] %>% as.data.frame()
colnames(glam_stats_M2) <- "Present"
# Add other columns from data_monocle
glam_stats_M2$State <- data_monocle$State[which(data_monocle$mice == "M2")]
glam_stats_M2$combined_cdr3_nt <- data_monocle$combined_cdr3_nt[which(data_monocle$mice == "M2")]
glam_stats_M2$cellID <- rownames(data_monocle@phenoData)[which(data_monocle$mice == "M2")]
# Remove any rows with NA (cells without a motif score, incluing cells that don't meet definition of a clone)
glam_stats_M2 <- glam_stats_M2[!is.na(glam_stats_M2$Present), ]

glam_stats_M4 <- data_monocle$Present[which(data_monocle$mice == "M4")] %>% as.data.frame()
colnames(glam_stats_M4) <- "Present"
# Add other columns from data_monocle
glam_stats_M4$State <- data_monocle$State[which(data_monocle$mice == "M4")]
glam_stats_M4$combined_cdr3_nt <- data_monocle$combined_cdr3_nt[which(data_monocle$mice == "M4")]
glam_stats_M4$cellID <- rownames(data_monocle@phenoData)[which(data_monocle$mice == "M4")]
# Remove any rows with NA (cells without a motif score, incluing cells that don't meet definition of a clone)
glam_stats_M4 <- glam_stats_M4[!is.na(glam_stats_M4$Present), ]

glam_stats_M6 <- data_monocle$Present[which(data_monocle$mice == "M6")] %>% as.data.frame()
colnames(glam_stats_M6) <- "Present"
# Add other columns from data_monocle
glam_stats_M6$State <- data_monocle$State[which(data_monocle$mice == "M6")]
glam_stats_M6$combined_cdr3_nt <- data_monocle$combined_cdr3_nt[which(data_monocle$mice == "M6")]
glam_stats_M6$cellID <- rownames(data_monocle@phenoData)[which(data_monocle$mice == "M6")]
# Remove any rows with NA (cells without a motif score, incluing cells that don't meet definition of a clone)
glam_stats_M6 <- glam_stats_M6[!is.na(glam_stats_M6$Present), ]



#Make NA values false
pData(data_monocle)$Present[is.na(pData(data_monocle)$Present)] <- FALSE
plot_obj <- plot_complex_cell_trajectory(data_monocle, color_by = "Present")
plot_obj$data <- plot_obj$data[order(plot_obj$data$Present),]
plot_obj + scale_color_manual(values=c("grey", "red")) + ggtitle("TRA_TFH_motif9_40_across_M2M4M6'\n Score > 40 \n ")

png(file = "summary_6mice/TRA_TH1_motif1_M2M4M6_cln4.png", width = 8, height = 6, units = "in", res = 300)
plot_obj + scale_color_manual(values=c("grey", "red")) + ggtitle("TRA_TH1_motif6_M2M4M6.png'\n Score > 37 \n 'P value '")
#dev.off()

#Create a table detailing the number of cells in each monocle state that have a good motif match
# The line below was from Robert's analysis, I'm replacing it because there was an error
# glam_score_table <- as.data.frame(table(pData(data_no_6_mono_cds)$State, pData(data_no_6_mono_cds)$Present))
rm(glam_score_table_M2)
rm(glam_score_table_M4)
rm(glam_score_table_M6)

glam_score_table_M2 <- table(glam_stats_M2$State, glam_stats_M2$Present) %>% as.data.frame()
#Combine true and false rows
glam_score_table_M2 <- cbind(glam_score_table_M2[1:5,], glam_score_table_M2[6:10,])
#Create names that make more sense
colnames(glam_score_table_M2) <- c("State", "F", "Freq_F", "State2", "T", "Freq_T")

#Remove unneeded columns
glam_score_table_M2 <- glam_score_table_M2[,c(1,3,6)]
#Create a percent column (percent of cells within each state with a good motif match)
glam_score_table_M2$pct <- glam_score_table_M2$Freq_T / (glam_score_table_M2$Freq_F + glam_score_table_M2$Freq_T)
#write.csv(glam_score_table_M2, "Motif/train/TFH/score_file/TRA_TFH_T65_motif1_M2_cln4_cells_across_M2.csv")
write.csv(glam_score_table_M2, "Motif_5mice/Train/Score/TRA_TFH_T65_motif9_M2_cln4_cells_across_M2_cells.csv")
getwd()
glam_score_table_M2$Freq_F %>% sum () # 854
glam_score_table_M2$Freq_T %>% sum() # 9

#Create a table detailing the number of cells in each monocle state that have a good motif match
# The line below was from Robert's analysis, I'm replacing it because there was an error
# glam_score_table <- as.data.frame(table(pData(data_no_6_mono_cds)$State, pData(data_no_6_mono_cds)$Present))

glam_score_table_M4 <- table(glam_stats_M4$State, glam_stats_M4$Present) %>% as.data.frame()
#Combine true and false rows
glam_score_table_M4 <- cbind(glam_score_table_M4[1:5,], glam_score_table_M4[6:10,]) ## this # depends on the # of monocle states
#Create names that make more sense
colnames(glam_score_table_M4) <- c("State", "F", "Freq_F", "State2", "T", "Freq_T")

#Remove unneeded columns
glam_score_table_M4 <- glam_score_table_M4[,c(1,3,6)]
#Create a percent column (percent of cells within each state with a good motif match)
glam_score_table_M4$pct <- glam_score_table_M4$Freq_T / (glam_score_table_M4$Freq_F + glam_score_table_M4$Freq_T)
#write.csv(glam_score_table_M4, "Motif/train/TFH/score_file/TRA_TFH_T65_motif1_M4_cln4_cells_across_M4.csv")
write.csv(glam_score_table_M4, "Motif_5mice/Train/Score/TRA_TFH_T65_motif9_M4_cln4_cells_across_M4_cells.csv")
getwd()
glam_score_table_M4$Freq_F %>% sum () # 3103
glam_score_table_M4$Freq_T %>% sum() # 115


#Create a table detailing the number of cells in each monocle state that have a good motif match
# The line below was from Robert's analysis, I'm replacing it because there was an error
# glam_score_table <- as.data.frame(table(pData(data_no_6_mono_cds)$State, pData(data_no_6_mono_cds)$Present))

glam_score_table_M6 <- table(glam_stats_M6$State, glam_stats_M6$Present) %>% as.data.frame()
#Combine true and false rows
glam_score_table_M6 <- cbind(glam_score_table_M6[1:5,], glam_score_table_M6[6:10,]) ## this # depends on the # of monocle states
#Create names that make more sense
colnames(glam_score_table_M6) <- c("State", "F", "Freq_F", "State2", "T", "Freq_T")

#Remove unneeded columns
glam_score_table_M6 <- glam_score_table_M6[,c(1,3,6)]
#Create a percent column (percent of cells within each state with a good motif match)
glam_score_table_M6$pct <- glam_score_table_M6$Freq_T / (glam_score_table_M6$Freq_F + glam_score_table_M6$Freq_T)
#write.csv(glam_score_table_M4, "Motif/train/TFH/score_file/TRA_TFH_T65_motif1_M4_cln4_cells_across_M4.csv")
write.csv(glam_score_table_M6, "Motif_5mice/Train/Score/TRA_TFH_T65_motif9_M6_cln4_cells_across_M6_cells.csv")
getwd()
glam_score_table_M6$Freq_F %>% sum () # 2173
glam_score_table_M6$Freq_T %>% sum() # 151


# Perform clone-level statistical analysis

glam_stats_M2$combined_cdr3_nt[which(glam_stats_M2$Present == TRUE)] %>% table()


#         TRA:TGCTCAGCAAGTATGACTGGAGGCTATAAAGTGGTCTTT_TRB:TGTGCCAGCAGCAAGGTAAACACAGAAGTCTTCTTT 
 #                                                                                           4 
#TRA:TGTGCAGCAAGTCTCGATGGAGGCTATAAAGTGGTCTTT_TRB:TGTGCTAGCAGTTTATCCGGGACAGGGGACGCTGAGCAGTTCTTC 
    #                                                                                       30 
   #TRA:TGTGCAGTGAGCGGGACTGGAGGCTATAAAGTGGTCTTT_TRB:TGTGCCAGCAGGACTGGGGGGGCAGGGGACACCCAGTACTTT 
    #                                                                                       13 
     # TRA:TGTGCTCTGGGCCTGACTGGAGGCTATAAAGTGGTCTTT_TRB:TGCAGTGCAAGGGGGGCGGACACCGGGCAGCTCTACTTT 
       #                                                                                    21 

grep(pattern = "TRA:TGTGCAGCAAGTTGGGGGACTGGAGGCTATAAAGTGGTCTTT_TRB:TGTGCTAGCAGTTTTGCTGGGGGGGCTTCCTATGAACAGTACTTC", x = data_clones_by_mouse_AA_TRA_TRB_NOV$combined_cdr3_nt)
# 3
as.data.frame(data_clones_by_mouse_AA_TRA_TRB_NOV[1, ])
# 79, 0, 21
getwd()
glam_stats_M4$combined_cdr3_nt[which(glam_stats_M4$Present == TRUE)] %>% table()

        #  TRA:TGTGCAGCAAGCATGAGAACAGGAAACTACAAATACGTCTTT_TRB:TGTGCCAGCAGCGCCCAAACGCTGTATTTT 
         #                                                                                      5 
      #TRA:TGTGCAGCAAGTCTTAATACAGGAAACTACAAATACGTCTTT_TRB:TGTGCCTGGAATGGGGGGGGCTATGCTGAGCAGTTCTTC 
       #                                                                                       12 
#TRA:TGTGCAGCAAGTTGGGGGACTGGAGGCTATAAAGTGGTCTTT_TRB:TGTGCTAGCAGTTTTGCTGGGGGGGCTTCCTATGAACAGTACTTC 
 #                              
```


```{r}
# TFH motif 10

data_monocle <- readRDS("summary_5M_CCA_V2/data_monocle_2cell.RDS")
# TFH Motif10
rm(glam2_scores)
glam2_scores <- read.table(file = "/media/runa/Elements/Jem_revision/Seurat/Motif_5mice/Train/TFH/Motif10/TFH_Motif10.glam2scan/glam2scan_scores_Motif10.txt", header = FALSE)
#We're only interested in the cell name and score columns
glam2_scores <- glam2_scores[,c(1,6)]
#Take the top score for each cell
glam2_scores <- aggregate(.~V1, glam2_scores, FUN=max) # here multiple cells can have
#more than one score, with aggregate command we are saying in those cases just keep the max
#score
rownames(glam2_scores) <- glam2_scores$V1
# to match cell names with monocle data add 1 in cell name
glam2_scores$V1 <-
  paste0(glam2_scores$V1, "1")
rownames(glam2_scores) <- glam2_scores$V1

#Add glam score to each cell within monocle pData
pData(data_monocle)$glam_score <- sapply(rownames(pData(data_monocle)), FUN =
function(x) glam2_scores[x,]$V6)

View(pData(data_monocle))


#Observe the glam score distribution, find a high score for the cells 40 looks good here; this
#scoring will vary from motif to motif

pdf("summary_5mice/Figure4/TFH_Motif10_Train_mice_score.pdf", width = 8, height = 6)
plot(density(pData(data_monocle)$glam_score[!is.na(pData(data_monocle)$glam_score)]),
main = "Motif score distribution_TRA_TFH_Motif10", xlab = "")
abline(v = 40, col = "Red")
dev.off()

#Make a true/false filtering column based on the glam score for plotting 
pData(data_monocle)$Present <- pData(data_monocle)$glam_score > 40

# Make a new dataframe to use for statistical analysis but not plotting the cells

rm(glam_stats_M2)
rm(glam_stats_M4)
rm(glam_stats_M6)

# Only include cells from mouse 2 and mouse 4, separately
glam_stats_M2 <- data_monocle$Present[which(data_monocle$mice == "M2")] %>% as.data.frame()
colnames(glam_stats_M2) <- "Present"
# Add other columns from data_monocle
glam_stats_M2$State <- data_monocle$State[which(data_monocle$mice == "M2")]
glam_stats_M2$combined_cdr3_nt <- data_monocle$combined_cdr3_nt[which(data_monocle$mice == "M2")]
glam_stats_M2$cellID <- rownames(data_monocle@phenoData)[which(data_monocle$mice == "M2")]
# Remove any rows with NA (cells without a motif score, incluing cells that don't meet definition of a clone)
glam_stats_M2 <- glam_stats_M2[!is.na(glam_stats_M2$Present), ]

glam_stats_M4 <- data_monocle$Present[which(data_monocle$mice == "M4")] %>% as.data.frame()
colnames(glam_stats_M4) <- "Present"
# Add other columns from data_monocle
glam_stats_M4$State <- data_monocle$State[which(data_monocle$mice == "M4")]
glam_stats_M4$combined_cdr3_nt <- data_monocle$combined_cdr3_nt[which(data_monocle$mice == "M4")]
glam_stats_M4$cellID <- rownames(data_monocle@phenoData)[which(data_monocle$mice == "M4")]
# Remove any rows with NA (cells without a motif score, incluing cells that don't meet definition of a clone)
glam_stats_M4 <- glam_stats_M4[!is.na(glam_stats_M4$Present), ]

glam_stats_M6 <- data_monocle$Present[which(data_monocle$mice == "M6")] %>% as.data.frame()
colnames(glam_stats_M6) <- "Present"
# Add other columns from data_monocle
glam_stats_M6$State <- data_monocle$State[which(data_monocle$mice == "M6")]
glam_stats_M6$combined_cdr3_nt <- data_monocle$combined_cdr3_nt[which(data_monocle$mice == "M6")]
glam_stats_M6$cellID <- rownames(data_monocle@phenoData)[which(data_monocle$mice == "M6")]
# Remove any rows with NA (cells without a motif score, incluing cells that don't meet definition of a clone)
glam_stats_M6 <- glam_stats_M6[!is.na(glam_stats_M6$Present), ]



#Make NA values false
pData(data_monocle)$Present[is.na(pData(data_monocle)$Present)] <- FALSE
plot_obj <- plot_complex_cell_trajectory(data_monocle, color_by = "Present", cell_size = 1)
plot_obj$data <- plot_obj$data[order(plot_obj$data$Present),]
plot_obj + scale_color_manual(values=c("grey", "red")) + ggtitle("TRA_TFH_motif10_40_across_M2M4M6'\n Score > 40 \n ")

pdf("summary_5mice/Figure4/TFH_Motif10_Train_mice_TREE.pdf", width = 8, height = 6, useDingbats = FALSE)
plot_obj + scale_color_manual(values=c("grey", "red")) + ggtitle("TRA_TFH_motif10_M2M4M6.png'\n Score > 40 \n 'P value 0.0001'")
dev.off()

#Create a table detailing the number of cells in each monocle state that have a good motif match
# The line below was from Robert's analysis, I'm replacing it because there was an error
# glam_score_table <- as.data.frame(table(pData(data_no_6_mono_cds)$State, pData(data_no_6_mono_cds)$Present))
rm(glam_score_table_M2)
rm(glam_score_table_M4)
rm(glam_score_table_M6)

glam_score_table_M2 <- table(glam_stats_M2$State, glam_stats_M2$Present) %>% as.data.frame()
#Combine true and false rows
glam_score_table_M2 <- cbind(glam_score_table_M2[1:5,], glam_score_table_M2[6:10,])
#Create names that make more sense
colnames(glam_score_table_M2) <- c("State", "F", "Freq_F", "State2", "T", "Freq_T")

#Remove unneeded columns
glam_score_table_M2 <- glam_score_table_M2[,c(1,3,6)]
#Create a percent column (percent of cells within each state with a good motif match)
glam_score_table_M2$pct <- glam_score_table_M2$Freq_T / (glam_score_table_M2$Freq_F + glam_score_table_M2$Freq_T)
#write.csv(glam_score_table_M2, "Motif/train/TFH/score_file/TRA_TFH_T65_motif1_M2_cln4_cells_across_M2.csv")
write.csv(glam_score_table_M2, "Motif_5mice/Train/Score/TRA_TFH_T65_motif10_M2_cln4_cells_across_M2_cells.csv")
getwd()
glam_score_table_M2$Freq_F %>% sum () # 813
glam_score_table_M2$Freq_T %>% sum() # 50

glam_stats_M2$combined_cdr3_nt[which(glam_stats_M2$Present == TRUE)] %>% table()

#TRA:TGTGCAGCATTGTATGGGAGCAGTGGCAACAAGCTCATCTTT_TRB:TGTGCCTGGAGTCCGGGACAGCAAGACACCCAGTACTTT TRA:TGTGCTTTGGAAGGGGATGGGAGCAGTGGCAACAAGCTCATCTTT_TRB:TGTGCCTGGAGTCTAGAACTGTATGAACAGTACTTC 
 #                                                                                       43                                                                                          7 



#Create a table detailing the number of cells in each monocle state that have a good motif match
# The line below was from Robert's analysis, I'm replacing it because there was an error
# glam_score_table <- as.data.frame(table(pData(data_no_6_mono_cds)$State, pData(data_no_6_mono_cds)$Present))

glam_score_table_M4 <- table(glam_stats_M4$State, glam_stats_M4$Present) %>% as.data.frame()
#Combine true and false rows
glam_score_table_M4 <- cbind(glam_score_table_M4[1:5,], glam_score_table_M4[6:10,]) ## this # depends on the # of monocle states
#Create names that make more sense
colnames(glam_score_table_M4) <- c("State", "F", "Freq_F", "State2", "T", "Freq_T")

#Remove unneeded columns
glam_score_table_M4 <- glam_score_table_M4[,c(1,3,6)]
#Create a percent column (percent of cells within each state with a good motif match)
glam_score_table_M4$pct <- glam_score_table_M4$Freq_T / (glam_score_table_M4$Freq_F + glam_score_table_M4$Freq_T)
#write.csv(glam_score_table_M4, "Motif/train/TFH/score_file/TRA_TFH_T65_motif1_M4_cln4_cells_across_M4.csv")
write.csv(glam_score_table_M4, "Motif_5mice/Train/Score/TRA_TFH_T65_motif10_M4_cln4_cells_across_M4_cells.csv")
getwd()
glam_score_table_M4$Freq_F %>% sum () # 2776
glam_score_table_M4$Freq_T %>% sum() # 442

glam_stats_M4$combined_cdr3_nt[which(glam_stats_M4$Present == TRUE)] %>% table()

#TRA:TGTGCAGCAATCTATGGGAGCAGTGGCAACAAGCTCATCTTT_TRB:TGTACTTCTGTGCCAGCAGTGATTGGGGGGGGGTATAGTGCAGAAACGCTGTATTTT 
 #                                                                                                        233 
  #             TRA:TGTGCAGCAATCTATGGGAGCAGTGGCAACAAGCTCATCTTT_TRB:TGTACTTCTGTGCCAGGGGACAGCTTCTGTGAACAGTACTTC 
   #                                                                                                      173 
    #  TRA:TGTGCAGCAATCTATGGGAGCAGTGGCAACAAGCTCATCTTT_TRB:TGTGCCAGCAGTGATTGGGGGGGGGGGTATAGTGCAGAAACGCTGTATTTT 
     #                                                                                                     19 
      #   TRA:TGTGCAGCAATCTATGGGAGCAGTGGCAACAAGCTCATCTTT_TRB:TGTGCCAGCAGTGATTGGGGGGGGTATAGTGCAGAAACGCTGTATTTT 
       #                                                                                                    5 
        #             TRA:TGTGCAGCAATCTATGGGAGCAGTGGCAACAAGCTCATCTTT_TRB:TGTGCCTGGAGTCTATCGGGTTATGAACAGTACTTC 
         #                                                                                                  6 
          #  TRA:TGTGCAGCAATCTATGGGAGCAGTGGCAACAAGCTCATCTTT_TRB:TGTGCTAGCAGTTTTGCTGGGGGGGCTTCCTATGAACAGTACTTC 
           #                                                                                                6 


#Create a table detailing the number of cells in each monocle state that have a good motif match
# The line below was from Robert's analysis, I'm replacing it because there was an error
# glam_score_table <- as.data.frame(table(pData(data_no_6_mono_cds)$State, pData(data_no_6_mono_cds)$Present))

glam_score_table_M6 <- table(glam_stats_M6$State, glam_stats_M6$Present) %>% as.data.frame()
#Combine true and false rows
glam_score_table_M6 <- cbind(glam_score_table_M6[1:5,], glam_score_table_M6[6:10,]) ## this # depends on the # of monocle states
#Create names that make more sense
colnames(glam_score_table_M6) <- c("State", "F", "Freq_F", "State2", "T", "Freq_T")

#Remove unneeded columns
glam_score_table_M6 <- glam_score_table_M6[,c(1,3,6)]
#Create a percent column (percent of cells within each state with a good motif match)
glam_score_table_M6$pct <- glam_score_table_M6$Freq_T / (glam_score_table_M6$Freq_F + glam_score_table_M6$Freq_T)
#write.csv(glam_score_table_M4, "Motif/train/TFH/score_file/TRA_TFH_T65_motif1_M4_cln4_cells_across_M4.csv")
write.csv(glam_score_table_M6, "Motif_5mice/Train/Score/TRA_TFH_T65_motif10_M6_cln4_cells_across_M6_cells.csv")
getwd()
glam_score_table_M6$Freq_F %>% sum () # 2288
glam_score_table_M6$Freq_T %>% sum() # 36

glam_stats_M6$combined_cdr3_nt[which(glam_stats_M6$Present == TRUE)] %>% table()

#TRA:TGTGCAGCAACCTATGGGAGCAGTGGCAACAAGCTCATCTTT_TRB:TGTGCCAGCGGTGATCACAACTCTGGAAATACGCTCTATTTT 
 #                                                                                                          9 
  #          TRA:TGTGCAGCAACCTATGGGAGCAGTGGCAACAAGCTCATCTTT_TRB:TGTGCCAGCGGTGATTATGGGGCTAGTCAAAACACCTTGTACTTT 
   #                                                                                                        6 
    #                 TRA:TGTGCAGCAAGCTATGGGAGCAGTGGCAACAAGCTCATCTTT_TRB:TGTGCCTGGAGCTGGGGTGCAGAAACGCTGTATTTT 
     #                                                                                                      8 
      #         TRA:TGTGCAGCACTCTATGGGAGCAGTGGCAACAAGCTCATCTTT_TRB:TGTGCCTGGAGTCGTACTGGAAACCAAGACACCCAGTACTTT 
       #                                                                                                    5 
#TRA:TGTGCAGCATCGAATTATGGGAGCAGTGGCAACAAGCTCATCTTT_TRB:TGTGCCAGCAGTGATGTCCTGGGGGGGCGCTGGAGTGCAGAAACGCTGTATTTT 
 #                                                                                                          8 

```


```{r}
# TFH motif 11

#data_monocle <- readRDS("summary_5M_CCA_V2/data_monocle_2cell.RDS")
# TFH Motif11
rm(glam2_scores)
glam2_scores <- read.table(file = "/media/runa/Elements/Jem_revision/Seurat/Motif_5mice/Train/TFH/Motif11/TFH_Motif11.glam2scan/glam2scan_scores_Motif11.txt", header = FALSE)
#We're only interested in the cell name and score columns
glam2_scores <- glam2_scores[,c(1,6)]
#Take the top score for each cell
glam2_scores <- aggregate(.~V1, glam2_scores, FUN=max) # here multiple cells can have
#more than one score, with aggregate command we are saying in those cases just keep the max
#score
rownames(glam2_scores) <- glam2_scores$V1
# to match cell names with monocle data add 1 in cell name
glam2_scores$V1 <-
  paste0(glam2_scores$V1, "1")
rownames(glam2_scores) <- glam2_scores$V1

#Add glam score to each cell within monocle pData
pData(data_monocle)$glam_score <- sapply(rownames(pData(data_monocle)), FUN =
function(x) glam2_scores[x,]$V6)

View(pData(data_monocle))


#Observe the glam score distribution, find a high score for the cells 40 looks good here; this
#scoring will vary from motif to motif
plot(density(pData(data_monocle)$glam_score[!is.na(pData(data_monocle)$glam_score)]),
main = "Motif score distribution_TRA_TFH_Motif11", xlab = "")
abline(v = 35, col = "Red")

#Make a true/false filtering column based on the glam score for plotting 
pData(data_monocle)$Present <- pData(data_monocle)$glam_score > 35

# Make a new dataframe to use for statistical analysis but not plotting the cells

rm(glam_stats_M2)
rm(glam_stats_M4)
rm(glam_stats_M6)

# Only include cells from mouse 2 and mouse 4, separately
glam_stats_M2 <- data_monocle$Present[which(data_monocle$mice == "M2")] %>% as.data.frame()
colnames(glam_stats_M2) <- "Present"
# Add other columns from data_monocle
glam_stats_M2$State <- data_monocle$State[which(data_monocle$mice == "M2")]
glam_stats_M2$combined_cdr3_nt <- data_monocle$combined_cdr3_nt[which(data_monocle$mice == "M2")]
glam_stats_M2$cellID <- rownames(data_monocle@phenoData)[which(data_monocle$mice == "M2")]
# Remove any rows with NA (cells without a motif score, incluing cells that don't meet definition of a clone)
glam_stats_M2 <- glam_stats_M2[!is.na(glam_stats_M2$Present), ]

glam_stats_M4 <- data_monocle$Present[which(data_monocle$mice == "M4")] %>% as.data.frame()
colnames(glam_stats_M4) <- "Present"
# Add other columns from data_monocle
glam_stats_M4$State <- data_monocle$State[which(data_monocle$mice == "M4")]
glam_stats_M4$combined_cdr3_nt <- data_monocle$combined_cdr3_nt[which(data_monocle$mice == "M4")]
glam_stats_M4$cellID <- rownames(data_monocle@phenoData)[which(data_monocle$mice == "M4")]
# Remove any rows with NA (cells without a motif score, incluing cells that don't meet definition of a clone)
glam_stats_M4 <- glam_stats_M4[!is.na(glam_stats_M4$Present), ]

glam_stats_M6 <- data_monocle$Present[which(data_monocle$mice == "M6")] %>% as.data.frame()
colnames(glam_stats_M6) <- "Present"
# Add other columns from data_monocle
glam_stats_M6$State <- data_monocle$State[which(data_monocle$mice == "M6")]
glam_stats_M6$combined_cdr3_nt <- data_monocle$combined_cdr3_nt[which(data_monocle$mice == "M6")]
glam_stats_M6$cellID <- rownames(data_monocle@phenoData)[which(data_monocle$mice == "M6")]
# Remove any rows with NA (cells without a motif score, incluing cells that don't meet definition of a clone)
glam_stats_M6 <- glam_stats_M6[!is.na(glam_stats_M6$Present), ]



#Make NA values false
pData(data_monocle)$Present[is.na(pData(data_monocle)$Present)] <- FALSE
plot_obj <- plot_complex_cell_trajectory(data_monocle, color_by = "Present")
plot_obj$data <- plot_obj$data[order(plot_obj$data$Present),]
plot_obj + scale_color_manual(values=c("grey", "red")) + ggtitle("TRA_TFH_motif11_35_across_M2M4M6'\n Score > 35 \n ")

png(file = "summary_6mice/TRA_TH1_motif1_M2M4M6_cln4.png", width = 8, height = 6, units = "in", res = 300)
plot_obj + scale_color_manual(values=c("grey", "red")) + ggtitle("TRA_TH1_motif6_M2M4M6.png'\n Score > 37 \n 'P value '")
#dev.off()

#Create a table detailing the number of cells in each monocle state that have a good motif match
# The line below was from Robert's analysis, I'm replacing it because there was an error
# glam_score_table <- as.data.frame(table(pData(data_no_6_mono_cds)$State, pData(data_no_6_mono_cds)$Present))
rm(glam_score_table_M2)
rm(glam_score_table_M4)
rm(glam_score_table_M6)
glam_score_table_M2 <- table(glam_stats_M2$State, glam_stats_M2$Present) %>% as.data.frame()
#Combine true and false rows
glam_score_table_M2 <- cbind(glam_score_table_M2[1:5,], glam_score_table_M2[6:10,])
#Create names that make more sense
colnames(glam_score_table_M2) <- c("State", "F", "Freq_F", "State2", "T", "Freq_T")

#Remove unneeded columns
glam_score_table_M2 <- glam_score_table_M2[,c(1,3,6)]
#Create a percent column (percent of cells within each state with a good motif match)
glam_score_table_M2$pct <- glam_score_table_M2$Freq_T / (glam_score_table_M2$Freq_F + glam_score_table_M2$Freq_T)
#write.csv(glam_score_table_M2, "Motif/train/TFH/score_file/TRA_TFH_T65_motif1_M2_cln4_cells_across_M2.csv")
write.csv(glam_score_table_M2, "Motif_5mice/Train/Score/TRA_TFH_T65_motif11_M2_cln4_cells_across_M2_cells.csv")
getwd()
glam_score_table_M2$Freq_F %>% sum () # 811
glam_score_table_M2$Freq_T %>% sum() # 52

#Create a table detailing the number of cells in each monocle state that have a good motif match
# The line below was from Robert's analysis, I'm replacing it because there was an error
# glam_score_table <- as.data.frame(table(pData(data_no_6_mono_cds)$State, pData(data_no_6_mono_cds)$Present))

glam_score_table_M4 <- table(glam_stats_M4$State, glam_stats_M4$Present) %>% as.data.frame()
#Combine true and false rows
glam_score_table_M4 <- cbind(glam_score_table_M4[1:5,], glam_score_table_M4[6:10,]) ## this # depends on the # of monocle states
#Create names that make more sense
colnames(glam_score_table_M4) <- c("State", "F", "Freq_F", "State2", "T", "Freq_T")

#Remove unneeded columns
glam_score_table_M4 <- glam_score_table_M4[,c(1,3,6)]
#Create a percent column (percent of cells within each state with a good motif match)
glam_score_table_M4$pct <- glam_score_table_M4$Freq_T / (glam_score_table_M4$Freq_F + glam_score_table_M4$Freq_T)
#write.csv(glam_score_table_M4, "Motif/train/TFH/score_file/TRA_TFH_T65_motif1_M4_cln4_cells_across_M4.csv")
write.csv(glam_score_table_M4, "Motif_5mice/Train/Score/TRA_TFH_T65_motif11_M4_cln4_cells_across_M4_cells.csv")
getwd()
glam_score_table_M4$Freq_F %>% sum () # 3135
glam_score_table_M4$Freq_T %>% sum() # 83


#Create a table detailing the number of cells in each monocle state that have a good motif match
# The line below was from Robert's analysis, I'm replacing it because there was an error
# glam_score_table <- as.data.frame(table(pData(data_no_6_mono_cds)$State, pData(data_no_6_mono_cds)$Present))

glam_score_table_M6 <- table(glam_stats_M6$State, glam_stats_M6$Present) %>% as.data.frame()
#Combine true and false rows
glam_score_table_M6 <- cbind(glam_score_table_M6[1:5,], glam_score_table_M6[6:10,]) ## this # depends on the # of monocle states
#Create names that make more sense
colnames(glam_score_table_M6) <- c("State", "F", "Freq_F", "State2", "T", "Freq_T")

#Remove unneeded columns
glam_score_table_M6 <- glam_score_table_M6[,c(1,3,6)]
#Create a percent column (percent of cells within each state with a good motif match)
glam_score_table_M6$pct <- glam_score_table_M6$Freq_T / (glam_score_table_M6$Freq_F + glam_score_table_M6$Freq_T)
#write.csv(glam_score_table_M4, "Motif/train/TFH/score_file/TRA_TFH_T65_motif1_M4_cln4_cells_across_M4.csv")
write.csv(glam_score_table_M6, "Motif_5mice/Train/Score/TRA_TFH_T65_motif11_M6_cln4_cells_across_M6_cells.csv")
getwd()
glam_score_table_M6$Freq_F %>% sum () # 2248
glam_score_table_M6$Freq_T %>% sum() # 76


# Perform clone-level statistical analysis

glam_stats_M2$combined_cdr3_nt[which(glam_stats_M2$Present == TRUE)] %>% table()


#         TRA:TGCTCAGCAAGTATGACTGGAGGCTATAAAGTGGTCTTT_TRB:TGTGCCAGCAGCAAGGTAAACACAGAAGTCTTCTTT 
 #                                                                                           4 
#TRA:TGTGCAGCAAGTCTCGATGGAGGCTATAAAGTGGTCTTT_TRB:TGTGCTAGCAGTTTATCCGGGACAGGGGACGCTGAGCAGTTCTTC 
    #                                                                                       30 
   #TRA:TGTGCAGTGAGCGGGACTGGAGGCTATAAAGTGGTCTTT_TRB:TGTGCCAGCAGGACTGGGGGGGCAGGGGACACCCAGTACTTT 
    #                                                                                       13 
     # TRA:TGTGCTCTGGGCCTGACTGGAGGCTATAAAGTGGTCTTT_TRB:TGCAGTGCAAGGGGGGCGGACACCGGGCAGCTCTACTTT 
       #                                                                                    21 

grep(pattern = "TRA:TGTGCAGCAAGTTGGGGGACTGGAGGCTATAAAGTGGTCTTT_TRB:TGTGCTAGCAGTTTTGCTGGGGGGGCTTCCTATGAACAGTACTTC", x = data_clones_by_mouse_AA_TRA_TRB_NOV$combined_cdr3_nt)
# 3
as.data.frame(data_clones_by_mouse_AA_TRA_TRB_NOV[1, ])
# 79, 0, 21
getwd()
glam_stats_M4$combined_cdr3_nt[which(glam_stats_M4$Present == TRUE)] %>% table()

        #  TRA:TGTGCAGCAAGCATGAGAACAGGAAACTACAAATACGTCTTT_TRB:TGTGCCAGCAGCGCCCAAACGCTGTATTTT 
         #                                                                                      5 
      #TRA:TGTGCAGCAAGTCTTAATACAGGAAACTACAAATACGTCTTT_TRB:TGTGCCTGGAATGGGGGGGGCTATGCTGAGCAGTTCTTC 
       #                                                                                       12 
#TRA:TGTGCAGCAAGTTGGGGGACTGGAGGCTATAAAGTGGTCTTT_TRB:TGTGCTAGCAGTTTTGCTGGGGGGGCTTCCTATGAACAGTACTTC 
 #                              
```
```{r}
# TFH motif 14

#data_monocle <- readRDS("summary_5M_CCA_V2/data_monocle_2cell.RDS")
# TFH Motif14
rm(glam2_scores)
glam2_scores <- read.table(file = "/media/runa/Elements/Jem_revision/Seurat/Motif_5mice/Train/TFH/Motif14/TFH_Motif14.glam2scan/glam2scan_scores_Motif14.txt", header = FALSE)
#We're only interested in the cell name and score columns
glam2_scores <- glam2_scores[,c(1,6)]
#Take the top score for each cell
glam2_scores <- aggregate(.~V1, glam2_scores, FUN=max) # here multiple cells can have
#more than one score, with aggregate command we are saying in those cases just keep the max
#score
rownames(glam2_scores) <- glam2_scores$V1
# to match cell names with monocle data add 1 in cell name
glam2_scores$V1 <-
  paste0(glam2_scores$V1, "1")
rownames(glam2_scores) <- glam2_scores$V1

#Add glam score to each cell within monocle pData
pData(data_monocle)$glam_score <- sapply(rownames(pData(data_monocle)), FUN =
function(x) glam2_scores[x,]$V6)

View(pData(data_monocle))


#Observe the glam score distribution, find a high score for the cells 40 looks good here; this
#scoring will vary from motif to motif
plot(density(pData(data_monocle)$glam_score[!is.na(pData(data_monocle)$glam_score)]),
main = "Motif score distribution_TRA_TFH_Motif14", xlab = "")
abline(v = 30, col = "Red")

#Make a true/false filtering column based on the glam score for plotting 
pData(data_monocle)$Present <- pData(data_monocle)$glam_score > 30

# Make a new dataframe to use for statistical analysis but not plotting the cells

rm(glam_stats_M2)
rm(glam_stats_M4)
rm(glam_stats_M6)

# Only include cells from mouse 2 and mouse 4, separately
glam_stats_M2 <- data_monocle$Present[which(data_monocle$mice == "M2")] %>% as.data.frame()
colnames(glam_stats_M2) <- "Present"
# Add other columns from data_monocle
glam_stats_M2$State <- data_monocle$State[which(data_monocle$mice == "M2")]
glam_stats_M2$combined_cdr3_nt <- data_monocle$combined_cdr3_nt[which(data_monocle$mice == "M2")]
glam_stats_M2$cellID <- rownames(data_monocle@phenoData)[which(data_monocle$mice == "M2")]
# Remove any rows with NA (cells without a motif score, incluing cells that don't meet definition of a clone)
glam_stats_M2 <- glam_stats_M2[!is.na(glam_stats_M2$Present), ]

glam_stats_M4 <- data_monocle$Present[which(data_monocle$mice == "M4")] %>% as.data.frame()
colnames(glam_stats_M4) <- "Present"
# Add other columns from data_monocle
glam_stats_M4$State <- data_monocle$State[which(data_monocle$mice == "M4")]
glam_stats_M4$combined_cdr3_nt <- data_monocle$combined_cdr3_nt[which(data_monocle$mice == "M4")]
glam_stats_M4$cellID <- rownames(data_monocle@phenoData)[which(data_monocle$mice == "M4")]
# Remove any rows with NA (cells without a motif score, incluing cells that don't meet definition of a clone)
glam_stats_M4 <- glam_stats_M4[!is.na(glam_stats_M4$Present), ]

glam_stats_M6 <- data_monocle$Present[which(data_monocle$mice == "M6")] %>% as.data.frame()
colnames(glam_stats_M6) <- "Present"
# Add other columns from data_monocle
glam_stats_M6$State <- data_monocle$State[which(data_monocle$mice == "M6")]
glam_stats_M6$combined_cdr3_nt <- data_monocle$combined_cdr3_nt[which(data_monocle$mice == "M6")]
glam_stats_M6$cellID <- rownames(data_monocle@phenoData)[which(data_monocle$mice == "M6")]
# Remove any rows with NA (cells without a motif score, incluing cells that don't meet definition of a clone)
glam_stats_M6 <- glam_stats_M6[!is.na(glam_stats_M6$Present), ]



#Make NA values false
pData(data_monocle)$Present[is.na(pData(data_monocle)$Present)] <- FALSE
plot_obj <- plot_complex_cell_trajectory(data_monocle, color_by = "Present")
plot_obj$data <- plot_obj$data[order(plot_obj$data$Present),]
plot_obj + scale_color_manual(values=c("grey", "red")) + ggtitle("TRA_TFH_motif14_30_across_M2M4M6'\n Score > 30 \n ")

png(file = "summary_6mice/TRA_TH1_motif1_M2M4M6_cln4.png", width = 8, height = 6, units = "in", res = 300)
plot_obj + scale_color_manual(values=c("grey", "red")) + ggtitle("TRA_TH1_motif6_M2M4M6.png'\n Score > 37 \n 'P value '")
#dev.off()

#Create a table detailing the number of cells in each monocle state that have a good motif match
# The line below was from Robert's analysis, I'm replacing it because there was an error
# glam_score_table <- as.data.frame(table(pData(data_no_6_mono_cds)$State, pData(data_no_6_mono_cds)$Present))
rm(glam_score_table_M2)
rm(glam_score_table_M4)
rm(glam_score_table_M6)

glam_score_table_M2 <- table(glam_stats_M2$State, glam_stats_M2$Present) %>% as.data.frame()
#Combine true and false rows
glam_score_table_M2 <- cbind(glam_score_table_M2[1:5,], glam_score_table_M2[6:10,])
#Create names that make more sense
colnames(glam_score_table_M2) <- c("State", "F", "Freq_F", "State2", "T", "Freq_T")

#Remove unneeded columns
glam_score_table_M2 <- glam_score_table_M2[,c(1,3,6)]
#Create a percent column (percent of cells within each state with a good motif match)
glam_score_table_M2$pct <- glam_score_table_M2$Freq_T / (glam_score_table_M2$Freq_F + glam_score_table_M2$Freq_T)
#write.csv(glam_score_table_M2, "Motif/train/TFH/score_file/TRA_TFH_T65_motif1_M2_cln4_cells_across_M2.csv")
write.csv(glam_score_table_M2, "Motif_5mice/Train/Score/TRA_TFH_T65_motif14_M2_cln4_cells_across_M2_cells.csv")
getwd()
glam_score_table_M2$Freq_F %>% sum () # 863
glam_score_table_M2$Freq_T %>% sum() # NA

#Create a table detailing the number of cells in each monocle state that have a good motif match
# The line below was from Robert's analysis, I'm replacing it because there was an error
# glam_score_table <- as.data.frame(table(pData(data_no_6_mono_cds)$State, pData(data_no_6_mono_cds)$Present))

glam_score_table_M4 <- table(glam_stats_M4$State, glam_stats_M4$Present) %>% as.data.frame()
#Combine true and false rows
glam_score_table_M4 <- cbind(glam_score_table_M4[1:5,], glam_score_table_M4[6:10,]) ## this # depends on the # of monocle states
#Create names that make more sense
colnames(glam_score_table_M4) <- c("State", "F", "Freq_F", "State2", "T", "Freq_T")

#Remove unneeded columns
glam_score_table_M4 <- glam_score_table_M4[,c(1,3,6)]
#Create a percent column (percent of cells within each state with a good motif match)
glam_score_table_M4$pct <- glam_score_table_M4$Freq_T / (glam_score_table_M4$Freq_F + glam_score_table_M4$Freq_T)
#write.csv(glam_score_table_M4, "Motif/train/TFH/score_file/TRA_TFH_T65_motif1_M4_cln4_cells_across_M4.csv")
write.csv(glam_score_table_M4, "Motif_5mice/Train/Score/TRA_TFH_T65_motif14_M4_cln4_cells_across_M4_cells.csv")
getwd()
glam_score_table_M4$Freq_F %>% sum () # 3214
glam_score_table_M4$Freq_T %>% sum() # 4


#Create a table detailing the number of cells in each monocle state that have a good motif match
# The line below was from Robert's analysis, I'm replacing it because there was an error
# glam_score_table <- as.data.frame(table(pData(data_no_6_mono_cds)$State, pData(data_no_6_mono_cds)$Present))

glam_score_table_M6 <- table(glam_stats_M6$State, glam_stats_M6$Present) %>% as.data.frame()
#Combine true and false rows
glam_score_table_M6 <- cbind(glam_score_table_M6[1:5,], glam_score_table_M6[6:10,]) ## this # depends on the # of monocle states
#Create names that make more sense
colnames(glam_score_table_M6) <- c("State", "F", "Freq_F", "State2", "T", "Freq_T")

#Remove unneeded columns
glam_score_table_M6 <- glam_score_table_M6[,c(1,3,6)]
#Create a percent column (percent of cells within each state with a good motif match)
glam_score_table_M6$pct <- glam_score_table_M6$Freq_T / (glam_score_table_M6$Freq_F + glam_score_table_M6$Freq_T)
#write.csv(glam_score_table_M4, "Motif/train/TFH/score_file/TRA_TFH_T65_motif1_M4_cln4_cells_across_M4.csv")
write.csv(glam_score_table_M6, "Motif_5mice/Train/Score/TRA_TFH_T65_motif14_M6_cln4_cells_across_M6_cells.csv")
getwd()
glam_score_table_M6$Freq_F %>% sum () # 2310
glam_score_table_M6$Freq_T %>% sum() # 14


# Perform clone-level statistical analysis

glam_stats_M2$combined_cdr3_nt[which(glam_stats_M2$Present == TRUE)] %>% table()


#         TRA:TGCTCAGCAAGTATGACTGGAGGCTATAAAGTGGTCTTT_TRB:TGTGCCAGCAGCAAGGTAAACACAGAAGTCTTCTTT 
 #                                                                                           4 
#TRA:TGTGCAGCAAGTCTCGATGGAGGCTATAAAGTGGTCTTT_TRB:TGTGCTAGCAGTTTATCCGGGACAGGGGACGCTGAGCAGTTCTTC 
    #                                                                                       30 
   #TRA:TGTGCAGTGAGCGGGACTGGAGGCTATAAAGTGGTCTTT_TRB:TGTGCCAGCAGGACTGGGGGGGCAGGGGACACCCAGTACTTT 
    #                                                                                       13 
     # TRA:TGTGCTCTGGGCCTGACTGGAGGCTATAAAGTGGTCTTT_TRB:TGCAGTGCAAGGGGGGCGGACACCGGGCAGCTCTACTTT 
       #                                                                                    21 

grep(pattern = "TRA:TGTGCAGCAAGTTGGGGGACTGGAGGCTATAAAGTGGTCTTT_TRB:TGTGCTAGCAGTTTTGCTGGGGGGGCTTCCTATGAACAGTACTTC", x = data_clones_by_mouse_AA_TRA_TRB_NOV$combined_cdr3_nt)
# 3
as.data.frame(data_clones_by_mouse_AA_TRA_TRB_NOV[1, ])
# 79, 0, 21
getwd()
glam_stats_M4$combined_cdr3_nt[which(glam_stats_M4$Present == TRUE)] %>% table()

        #  TRA:TGTGCAGCAAGCATGAGAACAGGAAACTACAAATACGTCTTT_TRB:TGTGCCAGCAGCGCCCAAACGCTGTATTTT 
         #                                                                                      5 
      #TRA:TGTGCAGCAAGTCTTAATACAGGAAACTACAAATACGTCTTT_TRB:TGTGCCTGGAATGGGGGGGGCTATGCTGAGCAGTTCTTC 
       #                                                                                       12 
#TRA:TGTGCAGCAAGTTGGGGGACTGGAGGCTATAAAGTGGTCTTT_TRB:TGTGCTAGCAGTTTTGCTGGGGGGGCTTCCTATGAACAGTACTTC 
 #                              
```
```{r}
# TFH motif 15

#data_monocle <- readRDS("summary_5M_CCA_V2/data_monocle_2cell.RDS")
# TFH Motif15
rm(glam2_scores)
glam2_scores <- read.table(file = "/media/runa/Elements/Jem_revision/Seurat/Motif_5mice/Train/TFH/Motif15/TFH_Motif15.glam2scan/glam2scan_scores_Motif15.txt", header = FALSE)
#We're only interested in the cell name and score columns
glam2_scores <- glam2_scores[,c(1,6)]
#Take the top score for each cell
glam2_scores <- aggregate(.~V1, glam2_scores, FUN=max) # here multiple cells can have
#more than one score, with aggregate command we are saying in those cases just keep the max
#score
rownames(glam2_scores) <- glam2_scores$V1
# to match cell names with monocle data add 1 in cell name
glam2_scores$V1 <-
  paste0(glam2_scores$V1, "1")
rownames(glam2_scores) <- glam2_scores$V1

#Add glam score to each cell within monocle pData
pData(data_monocle)$glam_score <- sapply(rownames(pData(data_monocle)), FUN =
function(x) glam2_scores[x,]$V6)

View(pData(data_monocle))


#Observe the glam score distribution, find a high score for the cells 40 looks good here; this
#scoring will vary from motif to motif
plot(density(pData(data_monocle)$glam_score[!is.na(pData(data_monocle)$glam_score)]),
main = "Motif score distribution_TRA_TFH_Motif15", xlab = "")
abline(v = 30, col = "Red")

#Make a true/false filtering column based on the glam score for plotting 
pData(data_monocle)$Present <- pData(data_monocle)$glam_score > 30

# Make a new dataframe to use for statistical analysis but not plotting the cells

rm(glam_stats_M2)
rm(glam_stats_M4)
rm(glam_stats_M6)

# Only include cells from mouse 2 and mouse 4, separately
glam_stats_M2 <- data_monocle$Present[which(data_monocle$mice == "M2")] %>% as.data.frame()
colnames(glam_stats_M2) <- "Present"
# Add other columns from data_monocle
glam_stats_M2$State <- data_monocle$State[which(data_monocle$mice == "M2")]
glam_stats_M2$combined_cdr3_nt <- data_monocle$combined_cdr3_nt[which(data_monocle$mice == "M2")]
glam_stats_M2$cellID <- rownames(data_monocle@phenoData)[which(data_monocle$mice == "M2")]
# Remove any rows with NA (cells without a motif score, incluing cells that don't meet definition of a clone)
glam_stats_M2 <- glam_stats_M2[!is.na(glam_stats_M2$Present), ]

glam_stats_M4 <- data_monocle$Present[which(data_monocle$mice == "M4")] %>% as.data.frame()
colnames(glam_stats_M4) <- "Present"
# Add other columns from data_monocle
glam_stats_M4$State <- data_monocle$State[which(data_monocle$mice == "M4")]
glam_stats_M4$combined_cdr3_nt <- data_monocle$combined_cdr3_nt[which(data_monocle$mice == "M4")]
glam_stats_M4$cellID <- rownames(data_monocle@phenoData)[which(data_monocle$mice == "M4")]
# Remove any rows with NA (cells without a motif score, incluing cells that don't meet definition of a clone)
glam_stats_M4 <- glam_stats_M4[!is.na(glam_stats_M4$Present), ]

glam_stats_M6 <- data_monocle$Present[which(data_monocle$mice == "M6")] %>% as.data.frame()
colnames(glam_stats_M6) <- "Present"
# Add other columns from data_monocle
glam_stats_M6$State <- data_monocle$State[which(data_monocle$mice == "M6")]
glam_stats_M6$combined_cdr3_nt <- data_monocle$combined_cdr3_nt[which(data_monocle$mice == "M6")]
glam_stats_M6$cellID <- rownames(data_monocle@phenoData)[which(data_monocle$mice == "M6")]
# Remove any rows with NA (cells without a motif score, incluing cells that don't meet definition of a clone)
glam_stats_M6 <- glam_stats_M6[!is.na(glam_stats_M6$Present), ]



#Make NA values false
pData(data_monocle)$Present[is.na(pData(data_monocle)$Present)] <- FALSE
plot_obj <- plot_complex_cell_trajectory(data_monocle, color_by = "Present")
plot_obj$data <- plot_obj$data[order(plot_obj$data$Present),]
plot_obj + scale_color_manual(values=c("grey", "red")) + ggtitle("TRA_TFH_motif15_30_across_M2M4M6'\n Score > 30 \n ")

png(file = "summary_6mice/TRA_TH1_motif1_M2M4M6_cln4.png", width = 8, height = 6, units = "in", res = 300)
plot_obj + scale_color_manual(values=c("grey", "red")) + ggtitle("TRA_TH1_motif6_M2M4M6.png'\n Score > 37 \n 'P value '")
#dev.off()

#Create a table detailing the number of cells in each monocle state that have a good motif match
# The line below was from Robert's analysis, I'm replacing it because there was an error
# glam_score_table <- as.data.frame(table(pData(data_no_6_mono_cds)$State, pData(data_no_6_mono_cds)$Present))
rm(glam_score_table_M2)
rm(glam_score_table_M4)
rm(glam_score_table_M6)

glam_score_table_M2 <- table(glam_stats_M2$State, glam_stats_M2$Present) %>% as.data.frame()
#Combine true and false rows
glam_score_table_M2 <- cbind(glam_score_table_M2[1:5,], glam_score_table_M2[6:10,])
#Create names that make more sense
colnames(glam_score_table_M2) <- c("State", "F", "Freq_F", "State2", "T", "Freq_T")

#Remove unneeded columns
glam_score_table_M2 <- glam_score_table_M2[,c(1,3,6)]
#Create a percent column (percent of cells within each state with a good motif match)
glam_score_table_M2$pct <- glam_score_table_M2$Freq_T / (glam_score_table_M2$Freq_F + glam_score_table_M2$Freq_T)
#write.csv(glam_score_table_M2, "Motif/train/TFH/score_file/TRA_TFH_T65_motif1_M2_cln4_cells_across_M2.csv")
write.csv(glam_score_table_M2, "Motif_5mice/Train/Score/TRA_TFH_T65_motif15_M2_cln4_cells_across_M2_cells.csv")
getwd()
glam_score_table_M2$Freq_F %>% sum () # 863
glam_score_table_M2$Freq_T %>% sum() # NA

#Create a table detailing the number of cells in each monocle state that have a good motif match
# The line below was from Robert's analysis, I'm replacing it because there was an error
# glam_score_table <- as.data.frame(table(pData(data_no_6_mono_cds)$State, pData(data_no_6_mono_cds)$Present))

glam_score_table_M4 <- table(glam_stats_M4$State, glam_stats_M4$Present) %>% as.data.frame()
#Combine true and false rows
glam_score_table_M4 <- cbind(glam_score_table_M4[1:5,], glam_score_table_M4[6:10,]) ## this # depends on the # of monocle states
#Create names that make more sense
colnames(glam_score_table_M4) <- c("State", "F", "Freq_F", "State2", "T", "Freq_T")

#Remove unneeded columns
glam_score_table_M4 <- glam_score_table_M4[,c(1,3,6)]
#Create a percent column (percent of cells within each state with a good motif match)
glam_score_table_M4$pct <- glam_score_table_M4$Freq_T / (glam_score_table_M4$Freq_F + glam_score_table_M4$Freq_T)
#write.csv(glam_score_table_M4, "Motif/train/TFH/score_file/TRA_TFH_T65_motif1_M4_cln4_cells_across_M4.csv")
write.csv(glam_score_table_M4, "Motif_5mice/Train/Score/TRA_TFH_T65_motif15_M4_cln4_cells_across_M4_cells.csv")
getwd()
glam_score_table_M4$Freq_F %>% sum () # 3201
glam_score_table_M4$Freq_T %>% sum() # 17


#Create a table detailing the number of cells in each monocle state that have a good motif match
# The line below was from Robert's analysis, I'm replacing it because there was an error
# glam_score_table <- as.data.frame(table(pData(data_no_6_mono_cds)$State, pData(data_no_6_mono_cds)$Present))

glam_score_table_M6 <- table(glam_stats_M6$State, glam_stats_M6$Present) %>% as.data.frame()
#Combine true and false rows
glam_score_table_M6 <- cbind(glam_score_table_M6[1:5,], glam_score_table_M6[6:10,]) ## this # depends on the # of monocle states
#Create names that make more sense
colnames(glam_score_table_M6) <- c("State", "F", "Freq_F", "State2", "T", "Freq_T")

#Remove unneeded columns
glam_score_table_M6 <- glam_score_table_M6[,c(1,3,6)]
#Create a percent column (percent of cells within each state with a good motif match)
glam_score_table_M6$pct <- glam_score_table_M6$Freq_T / (glam_score_table_M6$Freq_F + glam_score_table_M6$Freq_T)
#write.csv(glam_score_table_M4, "Motif/train/TFH/score_file/TRA_TFH_T65_motif1_M4_cln4_cells_across_M4.csv")
write.csv(glam_score_table_M6, "Motif_5mice/Train/Score/TRA_TFH_T65_motif15_M6_cln4_cells_across_M6_cells.csv")
getwd()
glam_score_table_M6$Freq_F %>% sum () # 2314
glam_score_table_M6$Freq_T %>% sum() # 10


# Perform clone-level statistical analysis

glam_stats_M2$combined_cdr3_nt[which(glam_stats_M2$Present == TRUE)] %>% table()


#         TRA:TGCTCAGCAAGTATGACTGGAGGCTATAAAGTGGTCTTT_TRB:TGTGCCAGCAGCAAGGTAAACACAGAAGTCTTCTTT 
 #                                                                                           4 
#TRA:TGTGCAGCAAGTCTCGATGGAGGCTATAAAGTGGTCTTT_TRB:TGTGCTAGCAGTTTATCCGGGACAGGGGACGCTGAGCAGTTCTTC 
    #                                                                                       30 
   #TRA:TGTGCAGTGAGCGGGACTGGAGGCTATAAAGTGGTCTTT_TRB:TGTGCCAGCAGGACTGGGGGGGCAGGGGACACCCAGTACTTT 
    #                                                                                       13 
     # TRA:TGTGCTCTGGGCCTGACTGGAGGCTATAAAGTGGTCTTT_TRB:TGCAGTGCAAGGGGGGCGGACACCGGGCAGCTCTACTTT 
       #                                                                                    21 

grep(pattern = "TRA:TGTGCAGCAAGTTGGGGGACTGGAGGCTATAAAGTGGTCTTT_TRB:TGTGCTAGCAGTTTTGCTGGGGGGGCTTCCTATGAACAGTACTTC", x = data_clones_by_mouse_AA_TRA_TRB_NOV$combined_cdr3_nt)
# 3
as.data.frame(data_clones_by_mouse_AA_TRA_TRB_NOV[1, ])
# 79, 0, 21
getwd()
glam_stats_M4$combined_cdr3_nt[which(glam_stats_M4$Present == TRUE)] %>% table()

        #  TRA:TGTGCAGCAAGCATGAGAACAGGAAACTACAAATACGTCTTT_TRB:TGTGCCAGCAGCGCCCAAACGCTGTATTTT 
         #                                                                                      5 
      #TRA:TGTGCAGCAAGTCTTAATACAGGAAACTACAAATACGTCTTT_TRB:TGTGCCTGGAATGGGGGGGGCTATGCTGAGCAGTTCTTC 
       #                                                                                       12 
#TRA:TGTGCAGCAAGTTGGGGGACTGGAGGCTATAAAGTGGTCTTT_TRB:TGTGCTAGCAGTTTTGCTGGGGGGGCTTCCTATGAACAGTACTTC 
 #                              
```

```{r}
# TFH motif 16

#data_monocle <- readRDS("summary_5M_CCA_V2/data_monocle_2cell.RDS")
# TFH Motif16
rm(glam2_scores)
glam2_scores <- read.table(file = "/media/runa/Elements/Jem_revision/Seurat/Motif_5mice/Train/TFH/Motif16/TFH_Motif16.glam2scan/glam2scan_scores_Motif16.txt", header = FALSE)
#We're only interested in the cell name and score columns
glam2_scores <- glam2_scores[,c(1,6)]
#Take the top score for each cell
glam2_scores <- aggregate(.~V1, glam2_scores, FUN=max) # here multiple cells can have
#more than one score, with aggregate command we are saying in those cases just keep the max
#score
rownames(glam2_scores) <- glam2_scores$V1
# to match cell names with monocle data add 1 in cell name
glam2_scores$V1 <-
  paste0(glam2_scores$V1, "1")
rownames(glam2_scores) <- glam2_scores$V1

#Add glam score to each cell within monocle pData
pData(data_monocle)$glam_score <- sapply(rownames(pData(data_monocle)), FUN =
function(x) glam2_scores[x,]$V6)

View(pData(data_monocle))


#Observe the glam score distribution, find a high score for the cells 40 looks good here; this
#scoring will vary from motif to motif

pdf("summary_5mice/Figure4/TFH_Motif16_Train_mice_score.pdf", width = 8, height = 6)
plot(density(pData(data_monocle)$glam_score[!is.na(pData(data_monocle)$glam_score)]),
main = "Motif score distribution_TRA_TFH_Motif16", xlab = "")
abline(v = 38, col = "Red")
dev.off()

#Make a true/false filtering column based on the glam score for plotting 
pData(data_monocle)$Present <- pData(data_monocle)$glam_score > 38

# Make a new dataframe to use for statistical analysis but not plotting the cells

rm(glam_stats_M2)
rm(glam_stats_M4)
rm(glam_stats_M6)

# Only include cells from mouse 2 and mouse 4, separately
glam_stats_M2 <- data_monocle$Present[which(data_monocle$mice == "M2")] %>% as.data.frame()
colnames(glam_stats_M2) <- "Present"
# Add other columns from data_monocle
glam_stats_M2$State <- data_monocle$State[which(data_monocle$mice == "M2")]
glam_stats_M2$combined_cdr3_nt <- data_monocle$combined_cdr3_nt[which(data_monocle$mice == "M2")]
glam_stats_M2$cellID <- rownames(data_monocle@phenoData)[which(data_monocle$mice == "M2")]
# Remove any rows with NA (cells without a motif score, incluing cells that don't meet definition of a clone)
glam_stats_M2 <- glam_stats_M2[!is.na(glam_stats_M2$Present), ]

glam_stats_M4 <- data_monocle$Present[which(data_monocle$mice == "M4")] %>% as.data.frame()
colnames(glam_stats_M4) <- "Present"
# Add other columns from data_monocle
glam_stats_M4$State <- data_monocle$State[which(data_monocle$mice == "M4")]
glam_stats_M4$combined_cdr3_nt <- data_monocle$combined_cdr3_nt[which(data_monocle$mice == "M4")]
glam_stats_M4$cellID <- rownames(data_monocle@phenoData)[which(data_monocle$mice == "M4")]
# Remove any rows with NA (cells without a motif score, incluing cells that don't meet definition of a clone)
glam_stats_M4 <- glam_stats_M4[!is.na(glam_stats_M4$Present), ]

glam_stats_M6 <- data_monocle$Present[which(data_monocle$mice == "M6")] %>% as.data.frame()
colnames(glam_stats_M6) <- "Present"
# Add other columns from data_monocle
glam_stats_M6$State <- data_monocle$State[which(data_monocle$mice == "M6")]
glam_stats_M6$combined_cdr3_nt <- data_monocle$combined_cdr3_nt[which(data_monocle$mice == "M6")]
glam_stats_M6$cellID <- rownames(data_monocle@phenoData)[which(data_monocle$mice == "M6")]
# Remove any rows with NA (cells without a motif score, incluing cells that don't meet definition of a clone)
glam_stats_M6 <- glam_stats_M6[!is.na(glam_stats_M6$Present), ]



#Make NA values false
pData(data_monocle)$Present[is.na(pData(data_monocle)$Present)] <- FALSE
plot_obj <- plot_complex_cell_trajectory(data_monocle, color_by = "Present", cell_size = 2)
plot_obj$data <- plot_obj$data[order(plot_obj$data$Present),]
plot_obj + scale_color_manual(values=c("grey", "red")) + ggtitle("TRA_TFH_motif16_38_across_M2M4M6'\n Score > 38 \n ")

pdf("summary_5mice/Figure4/TFH_Motif16_Train_mice_TREE.pdf", width = 8, height = 6, useDingbats = FALSE)
plot_obj + scale_color_manual(values=c("grey", "red")) + ggtitle("TRA_TH1_motif16_M2M4M6.png'\n Score > 38 \n 'P value 0.03'")
dev.off()

#Create a table detailing the number of cells in each monocle state that have a good motif match
# The line below was from Robert's analysis, I'm replacing it because there was an error
# glam_score_table <- as.data.frame(table(pData(data_no_6_mono_cds)$State, pData(data_no_6_mono_cds)$Present))
rm(glam_score_table_M2)
rm(glam_score_table_M4)
rm(glam_score_table_M6)


glam_score_table_M2 <- table(glam_stats_M2$State, glam_stats_M2$Present) %>% as.data.frame()
#Combine true and false rows
glam_score_table_M2 <- cbind(glam_score_table_M2[1:5,], glam_score_table_M2[6:10,])
#Create names that make more sense
colnames(glam_score_table_M2) <- c("State", "F", "Freq_F", "State2", "T", "Freq_T")

#Remove unneeded columns
glam_score_table_M2 <- glam_score_table_M2[,c(1,3,6)]
#Create a percent column (percent of cells within each state with a good motif match)
glam_score_table_M2$pct <- glam_score_table_M2$Freq_T / (glam_score_table_M2$Freq_F + glam_score_table_M2$Freq_T)
#write.csv(glam_score_table_M2, "Motif/train/TFH/score_file/TRA_TFH_T65_motif1_M2_cln4_cells_across_M2.csv")
write.csv(glam_score_table_M2, "Motif_5mice/Train/Score/TRA_TFH_T65_motif16_M2_cln4_cells_across_M2_cells.csv")
getwd()
glam_score_table_M2$Freq_F %>% sum () # 863
glam_score_table_M2$Freq_T %>% sum() # NA

#Create a table detailing the number of cells in each monocle state that have a good motif match
# The line below was from Robert's analysis, I'm replacing it because there was an error
# glam_score_table <- as.data.frame(table(pData(data_no_6_mono_cds)$State, pData(data_no_6_mono_cds)$Present))

glam_score_table_M4 <- table(glam_stats_M4$State, glam_stats_M4$Present) %>% as.data.frame()
#Combine true and false rows
glam_score_table_M4 <- cbind(glam_score_table_M4[1:5,], glam_score_table_M4[6:10,]) ## this # depends on the # of monocle states
#Create names that make more sense
colnames(glam_score_table_M4) <- c("State", "F", "Freq_F", "State2", "T", "Freq_T")

#Remove unneeded columns
glam_score_table_M4 <- glam_score_table_M4[,c(1,3,6)]
#Create a percent column (percent of cells within each state with a good motif match)
glam_score_table_M4$pct <- glam_score_table_M4$Freq_T / (glam_score_table_M4$Freq_F + glam_score_table_M4$Freq_T)
#write.csv(glam_score_table_M4, "Motif/train/TFH/score_file/TRA_TFH_T65_motif1_M4_cln4_cells_across_M4.csv")
write.csv(glam_score_table_M4, "Motif_5mice/Train/Score/TRA_TFH_T65_motif16_M4_cln4_cells_across_M4_cells.csv")
getwd()
glam_score_table_M4$Freq_F %>% sum () # 3171
glam_score_table_M4$Freq_T %>% sum() # 47


glam_stats_M4$combined_cdr3_nt[which(glam_stats_M4$Present == TRUE)] %>% table()

#TRA:TGTGCAGCAAATCAAGGAGGGTCTGCGAAGCTCATCTTT_TRB:TGTGCCTGGAGTTCCGACAGTTCTGGAAATACGCTCTATTTT TRA:TGTGCAGCGAATCAAGGAGGGTCTGCGAAGCTCATCTTT_TRB:TGTGCCTGGAGTTCCGACAGTTCTGGAAATACGCTCTATTTT 
 #                                                                                       13                                                                                         34 


#Create a table detailing the number of cells in each monocle state that have a good motif match
# The line below was from Robert's analysis, I'm replacing it because there was an error
# glam_score_table <- as.data.frame(table(pData(data_no_6_mono_cds)$State, pData(data_no_6_mono_cds)$Present))

glam_score_table_M6 <- table(glam_stats_M6$State, glam_stats_M6$Present) %>% as.data.frame()
#Combine true and false rows
glam_score_table_M6 <- cbind(glam_score_table_M6[1:5,], glam_score_table_M6[6:10,]) ## this # depends on the # of monocle states
#Create names that make more sense
colnames(glam_score_table_M6) <- c("State", "F", "Freq_F", "State2", "T", "Freq_T")

#Remove unneeded columns
glam_score_table_M6 <- glam_score_table_M6[,c(1,3,6)]
#Create a percent column (percent of cells within each state with a good motif match)
glam_score_table_M6$pct <- glam_score_table_M6$Freq_T / (glam_score_table_M6$Freq_F + glam_score_table_M6$Freq_T)
#write.csv(glam_score_table_M4, "Motif/train/TFH/score_file/TRA_TFH_T65_motif1_M4_cln4_cells_across_M4.csv")
write.csv(glam_score_table_M6, "Motif_5mice/Train/Score/TRA_TFH_T65_motif16_M6_cln4_cells_across_M6_cells.csv")
getwd()
glam_score_table_M6$Freq_F %>% sum () # 2318
glam_score_table_M6$Freq_T %>% sum() # 6

glam_stats_M6$combined_cdr3_nt[which(glam_stats_M6$Present == TRUE)] %>% table()

#TRA:TGTGCAGCAAATCAAGGAGGGTCTGCGAAGCTCATCTTT_TRB:TGTGCCTGGAGTTCGGACAATTCTGGAAATACGCTCTATTTT 
 #                                                                                        6 



```

```{r check siginificant TFH motifs 1,2,5,10 and 16 in Test mice M1 and M3}
# TFH motif 1 in test mice M1 and M3

data_monocle <- readRDS("summary_5M_CCA_V2/data_monocle_2cell.RDS")
# TFH Motif1
rm(glam2_scores)
glam2_scores <- read.table(file = "/media/runa/Elements/Jem_revision/Seurat/Motif_5mice/Test/TFH/Motif1/TFH_Motif1.glam2scan/glam2scan_scores_Motif1.txt", header = FALSE)
#We're only interested in the cell name and score columns
glam2_scores <- glam2_scores[,c(1,6)]
#Take the top score for each cell
glam2_scores <- aggregate(.~V1, glam2_scores, FUN=max) # here multiple cells can have
#more than one score, with aggregate command we are saying in those cases just keep the max
#score
rownames(glam2_scores) <- glam2_scores$V1
# to match cell names with monocle data add 1 in cell name
glam2_scores$V1 <-
  paste0(glam2_scores$V1, "1")
rownames(glam2_scores) <- glam2_scores$V1

#Add glam score to each cell within monocle pData
pData(data_monocle)$glam_score <- sapply(rownames(pData(data_monocle)), FUN =
function(x) glam2_scores[x,]$V6)

View(pData(data_monocle))


#Observe the glam score distribution, find a high score for the cells 40 looks good here; this
#scoring will vary from motif to motif

pdf("summary_5mice/Figure4/TFH_Motif1_Test_mice_score.pdf", width = 8, height = 6)
plot(density(pData(data_monocle)$glam_score[!is.na(pData(data_monocle)$glam_score)]),
main = "Motif score distribution_TRA_TFH_Motif1 in Test mice", xlab = "")
abline(v = 30, col = "Red")
dev.off()

#Make a true/false filtering column based on the glam score for plotting 
pData(data_monocle)$Present <- pData(data_monocle)$glam_score > 30

# Make a new dataframe to use for statistical analysis but not plotting the cells

# Only include cells from mouse 2 and mouse 4, separately
glam_stats_M1 <- data_monocle$Present[which(data_monocle$mice == "M1")] %>% as.data.frame()
colnames(glam_stats_M1) <- "Present"
# Add other columns from data_monocle
glam_stats_M1$State <- data_monocle$State[which(data_monocle$mice == "M1")]
glam_stats_M1$combined_cdr3_nt <- data_monocle$combined_cdr3_nt[which(data_monocle$mice == "M1")]
glam_stats_M1$cellID <- rownames(data_monocle@phenoData)[which(data_monocle$mice == "M1")]
# Remove any rows with NA (cells without a motif score, incluing cells that don't meet definition of a clone)
glam_stats_M1 <- glam_stats_M1[!is.na(glam_stats_M1$Present), ]

glam_stats_M3 <- data_monocle$Present[which(data_monocle$mice == "M3")] %>% as.data.frame()
colnames(glam_stats_M3) <- "Present"
# Add other columns from data_monocle
glam_stats_M3$State <- data_monocle$State[which(data_monocle$mice == "M3")]
glam_stats_M3$combined_cdr3_nt <- data_monocle$combined_cdr3_nt[which(data_monocle$mice == "M3")]
glam_stats_M3$cellID <- rownames(data_monocle@phenoData)[which(data_monocle$mice == "M3")]
# Remove any rows with NA (cells without a motif score, incluing cells that don't meet definition of a clone)
glam_stats_M3 <- glam_stats_M3[!is.na(glam_stats_M3$Present), ]




#Make NA values false
pData(data_monocle)$Present[is.na(pData(data_monocle)$Present)] <- FALSE
plot_obj <- plot_complex_cell_trajectory(data_monocle, color_by = "Present", cell_size = 3)
plot_obj$data <- plot_obj$data[order(plot_obj$data$Present),]
plot_obj + scale_color_manual(values=c("grey", "red")) + ggtitle("TRA_TFH_motif1_30_across_M1M3'\n Score > 30 \n ")

pdf(file = "summary_5mice/Figure4/TRA_TFH_motif1_M1M3_cln4.pdf", width = 8, height = 6, useDingbats = FALSE)
plot_obj + scale_color_manual(values=c("grey", "red")) + ggtitle("TRA_TFH_motif1_M1M3.png'\n Score > 30 \n 'P value 0.013'")
dev.off()

#Create a table detailing the number of cells in each monocle state that have a good motif match
# The line below was from Robert's analysis, I'm replacing it because there was an error
# glam_score_table <- as.data.frame(table(pData(data_no_6_mono_cds)$State, pData(data_no_6_mono_cds)$Present))
rm(glam_score_table_M1)
rm(glam_score_table_M3)


glam_score_table_M1 <- table(glam_stats_M1$State, glam_stats_M1$Present) %>% as.data.frame()
#Combine true and false rows
glam_score_table_M1 <- cbind(glam_score_table_M1[1:5,], glam_score_table_M1[6:10,])
#Create names that make more sense
colnames(glam_score_table_M1) <- c("State", "F", "Freq_F", "State2", "T", "Freq_T")

#Remove unneeded columns
glam_score_table_M1 <- glam_score_table_M1[,c(1,3,6)]
#Create a percent column (percent of cells within each state with a good motif match)
glam_score_table_M1$pct <- glam_score_table_M1$Freq_T / (glam_score_table_M1$Freq_F + glam_score_table_M1$Freq_T)
#write.csv(glam_score_table_M2, "Motif/train/TFH/score_file/TRA_TFH_T65_motif1_M2_cln4_cells_across_M2.csv")
write.csv(glam_score_table_M1, "Motif_5mice/Test/TFH/Score/TRA_TFH_T65_motif1_M1_cln4_cells_across_M1_cells.csv")
getwd()
glam_score_table_M1$Freq_F %>% sum () # 842
glam_score_table_M1$Freq_T %>% sum() # 36

glam_stats_M1$combined_cdr3_nt[which(glam_stats_M1$Present == TRUE)] %>% table()

# TRA:TGCTCAGCATCTAGGGGAGGAGGCTATAAAGTGGTCTTT_TRB:TGTGCCAGCAGCACGACAGCCACAGAAGTCTTCTTT 
 #                                                                                        7 
#   TRA:TGTGCAGCAAGCATGAGGACTGGAGGCTATAAAGTGGTCTTT_TRB:TGTGCCAGCAGCCTCGGAGGAAACACCTTGTACTTT 
 #                                                                                        5 
#TRA:TGTGCAGCCGGGACTGGAGGCTATAAAGTGGTCTTT_TRB:TGTGCCTGGAGTCCGGACAGGGGGCGGAACACAGAAGTCTTCTTT 
 #                                                                                       11 
#TRA:TGTGCAGCCGGGACTGGAGGCTATAAAGTGGTCTTT_TRB:TGTGCTAGCAGTCCCGCCGGGACAGGGGACGAAAGATTATTTTTC 
    #                                                                                    13 

#Create a table detailing the number of cells in each monocle state that have a good motif match
# The line below was from Robert's analysis, I'm replacing it because there was an error
# glam_score_table <- as.data.frame(table(pData(data_no_6_mono_cds)$State, pData(data_no_6_mono_cds)$Present))

glam_score_table_M3 <- table(glam_stats_M3$State, glam_stats_M3$Present) %>% as.data.frame()
#Combine true and false rows
glam_score_table_M3 <- cbind(glam_score_table_M3[1:5,], glam_score_table_M3[6:10,]) ## this # depends on the # of monocle states
#Create names that make more sense
colnames(glam_score_table_M3) <- c("State", "F", "Freq_F", "State2", "T", "Freq_T")

#Remove unneeded columns
glam_score_table_M3 <- glam_score_table_M3[,c(1,3,6)]
#Create a percent column (percent of cells within each state with a good motif match)
glam_score_table_M3$pct <- glam_score_table_M3$Freq_T / (glam_score_table_M3$Freq_F + glam_score_table_M3$Freq_T)
#write.csv(glam_score_table_M4, "Motif/train/TFH/score_file/TRA_TFH_T65_motif1_M4_cln4_cells_across_M4.csv")
write.csv(glam_score_table_M3, "Motif_5mice/Test/TFH/Score/TRA_TFH_T65_motif1_M3_cln4_cells_across_M3_cells.csv")
getwd()
glam_score_table_M3$Freq_F %>% sum () # 2408
glam_score_table_M3$Freq_T %>% sum() # 9



# Perform clone-level statistical analysis


glam_stats_M3$combined_cdr3_nt[which(glam_stats_M3$Present == TRUE)] %>% table()

#TRA:TGTGCAGCAAGTGGCACAAATGCTTACAAAGTCATCTTT_TRB:TGTGCCTGGGGACAGGGAGGAGAAGTCTTCTTT 
 #                                                                               9 


```
```{r}
# TFH motif 2 in test mice M1 and M3

data_monocle <- readRDS("summary_5M_CCA_V2/data_monocle_2cell.RDS")
# TFH Motif2
rm(glam2_scores)
glam2_scores <- read.table(file = "/media/runa/Elements/Jem_revision/Seurat/Motif_5mice/Test/TFH/Motif2/TFH_Motif2.glam2scan/glam2scan_scores_Motif2.txt", header = FALSE)
#We're only interested in the cell name and score columns
glam2_scores <- glam2_scores[,c(1,6)]
#Take the top score for each cell
glam2_scores <- aggregate(.~V1, glam2_scores, FUN=max) # here multiple cells can have
#more than one score, with aggregate command we are saying in those cases just keep the max
#score
rownames(glam2_scores) <- glam2_scores$V1
# to match cell names with monocle data add 1 in cell name
glam2_scores$V1 <-
  paste0(glam2_scores$V1, "1")
rownames(glam2_scores) <- glam2_scores$V1

#Add glam score to each cell within monocle pData
pData(data_monocle)$glam_score <- sapply(rownames(pData(data_monocle)), FUN =
function(x) glam2_scores[x,]$V6)

View(pData(data_monocle))


#Observe the glam score distribution, find a high score for the cells 40 looks good here; this
#scoring will vary from motif to motif
pdf("summary_5mice/Figure4/TFH_Motif2_Test_mice_score.pdf", width = 8, height = 6)
plot(density(pData(data_monocle)$glam_score[!is.na(pData(data_monocle)$glam_score)]),
main = "Motif score distribution_TRA_TFH_Motif2 in Test mice", xlab = "")
abline(v = 37, col = "Red")
dev.off()

#Make a true/false filtering column based on the glam score for plotting 
pData(data_monocle)$Present <- pData(data_monocle)$glam_score > 37

# Make a new dataframe to use for statistical analysis but not plotting the cells

rm(glam_stats_M1)
rm(glam_stats_M3)

# Only include cells from mouse 2 and mouse 4, separately
glam_stats_M1 <- data_monocle$Present[which(data_monocle$mice == "M1")] %>% as.data.frame()
colnames(glam_stats_M1) <- "Present"
# Add other columns from data_monocle
glam_stats_M1$State <- data_monocle$State[which(data_monocle$mice == "M1")]
glam_stats_M1$combined_cdr3_nt <- data_monocle$combined_cdr3_nt[which(data_monocle$mice == "M1")]
glam_stats_M1$cellID <- rownames(data_monocle@phenoData)[which(data_monocle$mice == "M1")]
# Remove any rows with NA (cells without a motif score, incluing cells that don't meet definition of a clone)
glam_stats_M1 <- glam_stats_M1[!is.na(glam_stats_M1$Present), ]

glam_stats_M3 <- data_monocle$Present[which(data_monocle$mice == "M3")] %>% as.data.frame()
colnames(glam_stats_M3) <- "Present"
# Add other columns from data_monocle
glam_stats_M3$State <- data_monocle$State[which(data_monocle$mice == "M3")]
glam_stats_M3$combined_cdr3_nt <- data_monocle$combined_cdr3_nt[which(data_monocle$mice == "M3")]
glam_stats_M3$cellID <- rownames(data_monocle@phenoData)[which(data_monocle$mice == "M3")]
# Remove any rows with NA (cells without a motif score, incluing cells that don't meet definition of a clone)
glam_stats_M3 <- glam_stats_M3[!is.na(glam_stats_M3$Present), ]




#Make NA values false
pData(data_monocle)$Present[is.na(pData(data_monocle)$Present)] <- FALSE
plot_obj <- plot_complex_cell_trajectory(data_monocle, color_by = "Present", cell_size = 3)
plot_obj$data <- plot_obj$data[order(plot_obj$data$Present),]
plot_obj + scale_color_manual(values=c("grey", "red")) + ggtitle("TRA_TFH_motif2_37_across_M1M3'\n Score > 37 \n ")

pdf(file = "summary_5mice/Figure4/TRA_TFH_motif2_M1M3_cln4.pdf", width = 8, height = 6, useDingbats = FALSE)
plot_obj + scale_color_manual(values=c("grey", "red")) + ggtitle("TRA_TFH_motif2_M1M3.pdf'\n Score > 30 \n 'P value 0.18'")
dev.off()

#Create a table detailing the number of cells in each monocle state that have a good motif match
# The line below was from Robert's analysis, I'm replacing it because there was an error
# glam_score_table <- as.data.frame(table(pData(data_no_6_mono_cds)$State, pData(data_no_6_mono_cds)$Present))
rm(glam_score_table_M1)
rm(glam_score_table_M3)


glam_score_table_M1 <- table(glam_stats_M1$State, glam_stats_M1$Present) %>% as.data.frame()
#Combine true and false rows
glam_score_table_M1 <- cbind(glam_score_table_M1[1:5,], glam_score_table_M1[6:10,])
#Create names that make more sense
colnames(glam_score_table_M1) <- c("State", "F", "Freq_F", "State2", "T", "Freq_T")

#Remove unneeded columns
glam_score_table_M1 <- glam_score_table_M1[,c(1,3,6)]
#Create a percent column (percent of cells within each state with a good motif match)
glam_score_table_M1$pct <- glam_score_table_M1$Freq_T / (glam_score_table_M1$Freq_F + glam_score_table_M1$Freq_T)
#write.csv(glam_score_table_M2, "Motif/train/TFH/score_file/TRA_TFH_T65_motif1_M2_cln4_cells_across_M2.csv")
write.csv(glam_score_table_M1, "Motif_5mice/Test/TFH/Score/TRA_TFH_T65_motif2_M1_cln4_cells_across_M1_cells.csv")
getwd()
glam_score_table_M1$Freq_F %>% sum () # 878
glam_score_table_M1$Freq_T %>% sum() # NA

#Create a table detailing the number of cells in each monocle state that have a good motif match
# The line below was from Robert's analysis, I'm replacing it because there was an error
# glam_score_table <- as.data.frame(table(pData(data_no_6_mono_cds)$State, pData(data_no_6_mono_cds)$Present))

glam_score_table_M3 <- table(glam_stats_M3$State, glam_stats_M3$Present) %>% as.data.frame()
#Combine true and false rows
glam_score_table_M3 <- cbind(glam_score_table_M3[1:5,], glam_score_table_M3[6:10,]) ## this # depends on the # of monocle states
#Create names that make more sense
colnames(glam_score_table_M3) <- c("State", "F", "Freq_F", "State2", "T", "Freq_T")

#Remove unneeded columns
glam_score_table_M3 <- glam_score_table_M3[,c(1,3,6)]
#Create a percent column (percent of cells within each state with a good motif match)
glam_score_table_M3$pct <- glam_score_table_M3$Freq_T / (glam_score_table_M3$Freq_F + glam_score_table_M3$Freq_T)
#write.csv(glam_score_table_M4, "Motif/train/TFH/score_file/TRA_TFH_T65_motif1_M4_cln4_cells_across_M4.csv")
write.csv(glam_score_table_M3, "Motif_5mice/Test/TFH/Score/TRA_TFH_T65_motif2_M3_cln4_cells_across_M3_cells.csv")
getwd()
glam_score_table_M3$Freq_F %>% sum () # 2411
glam_score_table_M3$Freq_T %>% sum() # 6



# Perform clone-level statistical analysis

glam_stats_M3$combined_cdr3_nt[which(glam_stats_M3$Present == TRUE)] %>% table()

#TRA:TGTGCAGCAAGCGGGTCTGGAGGAAGCAATGCAAAGCTAACCTTC_TRB:TGTGCCAGCGGTGATGGCGGGCTTGAACAGTACTTC 
 #                                                                                        6 
```

```{r}

# TFH motif 5 in test mice M1 and M3

data_monocle <- readRDS("summary_5M_CCA_V2/data_monocle_2cell.RDS")
# TFH Motif5
rm(glam2_scores)
glam2_scores <- read.table(file = "/media/runa/Elements/Jem_revision/Seurat/Motif_5mice/Test/TFH/Motif5/TFH_Motif5.glam2scan/glam2scan_scores_Motif5.txt", header = FALSE)
#We're only interested in the cell name and score columns
glam2_scores <- glam2_scores[,c(1,6)]
#Take the top score for each cell
glam2_scores <- aggregate(.~V1, glam2_scores, FUN=max) # here multiple cells can have
#more than one score, with aggregate command we are saying in those cases just keep the max
#score
rownames(glam2_scores) <- glam2_scores$V1
# to match cell names with monocle data add 1 in cell name
glam2_scores$V1 <-
  paste0(glam2_scores$V1, "1")
rownames(glam2_scores) <- glam2_scores$V1

#Add glam score to each cell within monocle pData
pData(data_monocle)$glam_score <- sapply(rownames(pData(data_monocle)), FUN =
function(x) glam2_scores[x,]$V6)

View(pData(data_monocle))


#Observe the glam score distribution, find a high score for the cells 40 looks good here; this
#scoring will vary from motif to motif
pdf("summary_5mice/Figure4/TFH_Motif5_Test_mice_score.pdf", width = 8, height = 6)
plot(density(pData(data_monocle)$glam_score[!is.na(pData(data_monocle)$glam_score)]),
main = "Motif score distribution_TRA_TFH_Motif5 in Test mice", xlab = "")
abline(v = 38, col = "Red")
dev.off()

#Make a true/false filtering column based on the glam score for plotting 
pData(data_monocle)$Present <- pData(data_monocle)$glam_score > 38

# Make a new dataframe to use for statistical analysis but not plotting the cells

rm(glam_stats_M1)
rm(glam_stats_M3)

# Only include cells from mouse 2 and mouse 4, separately
glam_stats_M1 <- data_monocle$Present[which(data_monocle$mice == "M1")] %>% as.data.frame()
colnames(glam_stats_M1) <- "Present"
# Add other columns from data_monocle
glam_stats_M1$State <- data_monocle$State[which(data_monocle$mice == "M1")]
glam_stats_M1$combined_cdr3_nt <- data_monocle$combined_cdr3_nt[which(data_monocle$mice == "M1")]
glam_stats_M1$cellID <- rownames(data_monocle@phenoData)[which(data_monocle$mice == "M1")]
# Remove any rows with NA (cells without a motif score, incluing cells that don't meet definition of a clone)
glam_stats_M1 <- glam_stats_M1[!is.na(glam_stats_M1$Present), ]

glam_stats_M3 <- data_monocle$Present[which(data_monocle$mice == "M3")] %>% as.data.frame()
colnames(glam_stats_M3) <- "Present"
# Add other columns from data_monocle
glam_stats_M3$State <- data_monocle$State[which(data_monocle$mice == "M3")]
glam_stats_M3$combined_cdr3_nt <- data_monocle$combined_cdr3_nt[which(data_monocle$mice == "M3")]
glam_stats_M3$cellID <- rownames(data_monocle@phenoData)[which(data_monocle$mice == "M3")]
# Remove any rows with NA (cells without a motif score, incluing cells that don't meet definition of a clone)
glam_stats_M3 <- glam_stats_M3[!is.na(glam_stats_M3$Present), ]




#Make NA values false
pData(data_monocle)$Present[is.na(pData(data_monocle)$Present)] <- FALSE
plot_obj <- plot_complex_cell_trajectory(data_monocle, color_by = "Present", cell_size = 3)
plot_obj$data <- plot_obj$data[order(plot_obj$data$Present),]
plot_obj + scale_color_manual(values=c("grey", "red")) + ggtitle("TRA_TFH_motif5_38_across_M1M3'\n Score > 38 \n ")

pdf("summary_5mice/Figure4/TFH_Motif5_Test_mice_TREE.pdf", width = 8, height = 6, useDingbats = FALSE)
plot_obj + scale_color_manual(values=c("grey", "red")) + ggtitle("TRA_TFH_motif5_M1M3.pdf'\n Score > 38 \n 'P value 0.0001'")
dev.off()

#Create a table detailing the number of cells in each monocle state that have a good motif match
# The line below was from Robert's analysis, I'm replacing it because there was an error
# glam_score_table <- as.data.frame(table(pData(data_no_6_mono_cds)$State, pData(data_no_6_mono_cds)$Present))
rm(glam_score_table_M1)
rm(glam_score_table_M3)


glam_score_table_M1 <- table(glam_stats_M1$State, glam_stats_M1$Present) %>% as.data.frame()
#Combine true and false rows
glam_score_table_M1 <- cbind(glam_score_table_M1[1:5,], glam_score_table_M1[6:10,])
#Create names that make more sense
colnames(glam_score_table_M1) <- c("State", "F", "Freq_F", "State2", "T", "Freq_T")

#Remove unneeded columns
glam_score_table_M1 <- glam_score_table_M1[,c(1,3,6)]
#Create a percent column (percent of cells within each state with a good motif match)
glam_score_table_M1$pct <- glam_score_table_M1$Freq_T / (glam_score_table_M1$Freq_F + glam_score_table_M1$Freq_T)
#write.csv(glam_score_table_M2, "Motif/train/TFH/score_file/TRA_TFH_T65_motif1_M2_cln4_cells_across_M2.csv")
write.csv(glam_score_table_M1, "Motif_5mice/Test/TFH/Score/TRA_TFH_T65_motif5_M1_cln4_cells_across_M1_cells.csv")
getwd()
glam_score_table_M1$Freq_F %>% sum () # 809
glam_score_table_M1$Freq_T %>% sum() # 69

glam_stats_M1$combined_cdr3_nt[which(glam_stats_M1$Present == TRUE)] %>% table()

#TRA:TGTGCAGCAAGCTATAACAACAATGCCCCACGATTT_TRB:TGTAACTATGCTGAGCAGTTCTTC 
 #                                                                  69

#Create a table detailing the number of cells in each monocle state that have a good motif match
# The line below was from Robert's analysis, I'm replacing it because there was an error
# glam_score_table <- as.data.frame(table(pData(data_no_6_mono_cds)$State, pData(data_no_6_mono_cds)$Present))

glam_score_table_M3 <- table(glam_stats_M3$State, glam_stats_M3$Present) %>% as.data.frame()
#Combine true and false rows
glam_score_table_M3 <- cbind(glam_score_table_M3[1:5,], glam_score_table_M3[6:10,]) ## this # depends on the # of monocle states
#Create names that make more sense
colnames(glam_score_table_M3) <- c("State", "F", "Freq_F", "State2", "T", "Freq_T")

#Remove unneeded columns
glam_score_table_M3 <- glam_score_table_M3[,c(1,3,6)]
#Create a percent column (percent of cells within each state with a good motif match)
glam_score_table_M3$pct <- glam_score_table_M3$Freq_T / (glam_score_table_M3$Freq_F + glam_score_table_M3$Freq_T)
#write.csv(glam_score_table_M4, "Motif/train/TFH/score_file/TRA_TFH_T65_motif1_M4_cln4_cells_across_M4.csv")
write.csv(glam_score_table_M3, "Motif_5mice/Test/TFH/Score/TRA_TFH_T65_motif5_M3_cln4_cells_across_M3_cells.csv")
getwd()
glam_score_table_M3$Freq_F %>% sum () # 2307
glam_score_table_M3$Freq_T %>% sum() # 110


glam_stats_M3$combined_cdr3_nt[which(glam_stats_M3$Present == TRUE)] %>% table()

#TRA:TGTGCAGCAAGTACTAACAACAATGCCCCACGATTT_TRB:TGTGCCAGCAGTGAGGGGACAGGGGCAAACACCGGGCAGCTCTACTTT 
 #                                                                                          91 
#            TRA:TGTGCAGCAAGTGGGAACAACAATGCCCCACGATTT_TRB:TGTGCCAGCAGTGACCATCAAGACACCCAGTACTTT 
 #                                                                                           9 
#            TRA:TGTGCAGCAAGTGGTAACAACAATGCCCCACGATTT_TRB:TGTGCCAGCAGTGAACACCAAGACACCCAGTACTTT 
 #                                                                                          10 

```



```{r}

# TFH motif 10 in test mice M1 and M3

data_monocle <- readRDS("summary_5M_CCA_V2/data_monocle_2cell.RDS")
# TFH Motif 10
rm(glam2_scores)
glam2_scores <- read.table(file = "/media/runa/Elements/Jem_revision/Seurat/Motif_5mice/Test/TFH/Motif10/TFH_Motif10.glam2scan/glam2scan_scores_Motif10.txt", header = FALSE)
#We're only interested in the cell name and score columns
glam2_scores <- glam2_scores[,c(1,6)]
#Take the top score for each cell
glam2_scores <- aggregate(.~V1, glam2_scores, FUN=max) # here multiple cells can have
#more than one score, with aggregate command we are saying in those cases just keep the max
#score
rownames(glam2_scores) <- glam2_scores$V1
# to match cell names with monocle data add 1 in cell name
glam2_scores$V1 <-
  paste0(glam2_scores$V1, "1")
rownames(glam2_scores) <- glam2_scores$V1

#Add glam score to each cell within monocle pData
pData(data_monocle)$glam_score <- sapply(rownames(pData(data_monocle)), FUN =
function(x) glam2_scores[x,]$V6)

View(pData(data_monocle))


#Observe the glam score distribution, find a high score for the cells 40 looks good here; this
#scoring will vary from motif to motif

pdf("summary_5mice/Figure4/TFH_Motif10_Test_mice_score.pdf", width = 8, height = 6)
plot(density(pData(data_monocle)$glam_score[!is.na(pData(data_monocle)$glam_score)]),
main = "Motif score distribution_TRA_TFH_Motif10 in Test mice", xlab = "")
abline(v = 40, col = "Red")
dev.off()

#Make a true/false filtering column based on the glam score for plotting 
pData(data_monocle)$Present <- pData(data_monocle)$glam_score > 40

# Make a new dataframe to use for statistical analysis but not plotting the cells

rm(glam_stats_M1)
rm(glam_stats_M3)

# Only include cells from mouse 2 and mouse 4, separately
glam_stats_M1 <- data_monocle$Present[which(data_monocle$mice == "M1")] %>% as.data.frame()
colnames(glam_stats_M1) <- "Present"
# Add other columns from data_monocle
glam_stats_M1$State <- data_monocle$State[which(data_monocle$mice == "M1")]
glam_stats_M1$combined_cdr3_nt <- data_monocle$combined_cdr3_nt[which(data_monocle$mice == "M1")]
glam_stats_M1$cellID <- rownames(data_monocle@phenoData)[which(data_monocle$mice == "M1")]
# Remove any rows with NA (cells without a motif score, incluing cells that don't meet definition of a clone)
glam_stats_M1 <- glam_stats_M1[!is.na(glam_stats_M1$Present), ]

glam_stats_M3 <- data_monocle$Present[which(data_monocle$mice == "M3")] %>% as.data.frame()
colnames(glam_stats_M3) <- "Present"
# Add other columns from data_monocle
glam_stats_M3$State <- data_monocle$State[which(data_monocle$mice == "M3")]
glam_stats_M3$combined_cdr3_nt <- data_monocle$combined_cdr3_nt[which(data_monocle$mice == "M3")]
glam_stats_M3$cellID <- rownames(data_monocle@phenoData)[which(data_monocle$mice == "M3")]
# Remove any rows with NA (cells without a motif score, incluing cells that don't meet definition of a clone)
glam_stats_M3 <- glam_stats_M3[!is.na(glam_stats_M3$Present), ]




#Make NA values false
pData(data_monocle)$Present[is.na(pData(data_monocle)$Present)] <- FALSE
plot_obj <- plot_complex_cell_trajectory(data_monocle, color_by = "Present")
plot_obj$data <- plot_obj$data[order(plot_obj$data$Present),]
plot_obj + scale_color_manual(values=c("grey", "red")) + ggtitle("TRA_TFH_motif10_40_across_M1M3'\n Score > 40 \n ")

pdf("summary_5mice/Figure4/TFH_Motif10_Test_mice_TREE.pdf", width = 8, height = 6, useDingbats = FALSE)
plot_obj + scale_color_manual(values=c("grey", "red")) + ggtitle("TRA_TFH_motif10_M1M3.png'\n Score > 40 \n 'P value 0.0001'")
dev.off()

#Create a table detailing the number of cells in each monocle state that have a good motif match
# The line below was from Robert's analysis, I'm replacing it because there was an error
# glam_score_table <- as.data.frame(table(pData(data_no_6_mono_cds)$State, pData(data_no_6_mono_cds)$Present))
rm(glam_score_table_M1)
rm(glam_score_table_M3)


glam_score_table_M1 <- table(glam_stats_M1$State, glam_stats_M1$Present) %>% as.data.frame()
#Combine true and false rows
glam_score_table_M1 <- cbind(glam_score_table_M1[1:5,], glam_score_table_M1[6:10,])
#Create names that make more sense
colnames(glam_score_table_M1) <- c("State", "F", "Freq_F", "State2", "T", "Freq_T")

#Remove unneeded columns
glam_score_table_M1 <- glam_score_table_M1[,c(1,3,6)]
#Create a percent column (percent of cells within each state with a good motif match)
glam_score_table_M1$pct <- glam_score_table_M1$Freq_T / (glam_score_table_M1$Freq_F + glam_score_table_M1$Freq_T)
#write.csv(glam_score_table_M2, "Motif/train/TFH/score_file/TRA_TFH_T65_motif1_M2_cln4_cells_across_M2.csv")
write.csv(glam_score_table_M1, "Motif_5mice/Test/TFH/Score/TRA_TFH_T65_motif10_M1_cln4_cells_across_M1_cells.csv")
getwd()
glam_score_table_M1$Freq_F %>% sum () # 829
glam_score_table_M1$Freq_T %>% sum() # 49

glam_stats_M1$combined_cdr3_nt[which(glam_stats_M1$Present == TRUE)] %>% table()

#TRA:TGTGCAGCAACTTATGGGAGCAGTGGCAACAAGCTCATCTTT_TRB:TGCCAGCAGTCTGATCAGGGATATGAACAGTACTTC 
 #                                                                                          31 
#TRA:TGTGCAGCAAGTTATGGGAGCAGTGGCAACAAGCTCATCTTT_TRB:TGTGCCTGGAGTCCGACTGGGGGGGAAGACACCCAGTACTTT 
 #                                                                                           5 
#TRA:TGTGCAGCCCTTTATGGGAGCAGTGGCAACAAGCTCATCTTT_TRB:TGTGCCTGGAGTCCCACTGGGGGGCAAGACACCCAGTACTTT 
 #                                                                                          13 

#Create a table detailing the number of cells in each monocle state that have a good motif match
# The line below was from Robert's analysis, I'm replacing it because there was an error
# glam_score_table <- as.data.frame(table(pData(data_no_6_mono_cds)$State, pData(data_no_6_mono_cds)$Present))

glam_score_table_M3 <- table(glam_stats_M3$State, glam_stats_M3$Present) %>% as.data.frame()
#Combine true and false rows
glam_score_table_M3 <- cbind(glam_score_table_M3[1:5,], glam_score_table_M3[6:10,]) ## this # depends on the # of monocle states
#Create names that make more sense
colnames(glam_score_table_M3) <- c("State", "F", "Freq_F", "State2", "T", "Freq_T")

#Remove unneeded columns
glam_score_table_M3 <- glam_score_table_M3[,c(1,3,6)]
#Create a percent column (percent of cells within each state with a good motif match)
glam_score_table_M3$pct <- glam_score_table_M3$Freq_T / (glam_score_table_M3$Freq_F + glam_score_table_M3$Freq_T)
#write.csv(glam_score_table_M4, "Motif/train/TFH/score_file/TRA_TFH_T65_motif1_M4_cln4_cells_across_M4.csv")
write.csv(glam_score_table_M3, "Motif_5mice/Test/TFH/Score/TRA_TFH_T65_motif10_M3_cln4_cells_across_M3_cells.csv")
getwd()
glam_score_table_M3$Freq_F %>% sum () # 1862
glam_score_table_M3$Freq_T %>% sum() # 555

glam_stats_M3$combined_cdr3_nt[which(glam_stats_M3$Present == TRUE)] %>% table()

#TRA:TGTGCAGCAAACTATGGGAGCAGTGGCAACAAGCTCATCTTT_TRB:TGTGCCTGGAGCCCCGGGACAGGGGGCGAGGAACAGTACTTC 
 #                                                                                                 4 
#      TRA:TGTGCAGCAACTTATGGGAGCAGTGGCAACAAGCTCATCTTT_TRB:TGTGCCAGCAGTGATATGGGGGCTGACGAAAGATTATTTTTC 
 #                                                                                               271 
#TRA:TGTGCAGCAAGTTGGGGGAGCAGTGGCAACAAGCTCATCTTT_TRB:TGTGCTAGCAGTTTCCCCGGGACAACTAGTGCAGAAACGCTGTATTTT 
 #                                                                                               206 
  #       TRA:TGTGCAGCACTTTATGGGAGCAGTGGCAACAAGCTCATCTTT_TRB:TGTGCCTGGGCCCCGGGACAAACAGACACCCAGTACTTT 
   #                                                                                              55 
   #TRA:TGTGCAGCCACTTATGGGAGCAGTGGCAACAAGCTCATCTTT_TRB:TGTGCCAGCGGTGAACCGGGACAGGGGGCAGACACCCAGTACTTT 
       #                                                                                          14 
        #    TRA:TGTGCAGCCTCTTATGGGAGCAGTGGCAACAAGCTCATCTTT_TRB:TGTGCCTGGACTGGGGGGTTGGACACCCAGTACTTT 
         #                                                                                         5 
```



```{r}
# TFH motif 16 in test mice M1 and M3 

data_monocle <- readRDS("summary_5M_CCA_V2/data_monocle_2cell.RDS")
# TFH Motif 16
rm(glam2_scores)
glam2_scores <- read.table(file = "/media/runa/Elements/Jem_revision/Seurat/Motif_5mice/Test/TFH/Motif16/TFH_Motif16.glam2scan/glam2scan_scores_Motif16.txt", header = FALSE)
#We're only interested in the cell name and score columns
glam2_scores <- glam2_scores[,c(1,6)]
#Take the top score for each cell
glam2_scores <- aggregate(.~V1, glam2_scores, FUN=max) # here multiple cells can have
#more than one score, with aggregate command we are saying in those cases just keep the max
#score
rownames(glam2_scores) <- glam2_scores$V1
# to match cell names with monocle data add 1 in cell name
glam2_scores$V1 <-
  paste0(glam2_scores$V1, "1")
rownames(glam2_scores) <- glam2_scores$V1

#Add glam score to each cell within monocle pData
pData(data_monocle)$glam_score <- sapply(rownames(pData(data_monocle)), FUN =
function(x) glam2_scores[x,]$V6)

View(pData(data_monocle))


#Observe the glam score distribution, find a high score for the cells 40 looks good here; this
#scoring will vary from motif to motif
pdf("summary_5mice/Figure4/TFH_Motif16_Test_mice_score.pdf", width = 8, height = 6, useDingbats = FALSE)
plot(density(pData(data_monocle)$glam_score[!is.na(pData(data_monocle)$glam_score)]),
main = "Motif score distribution_TRA_TFH_Motif16 in Test mice", xlab = "")
abline(v = 38, col = "Red") ## this threshold is out of boundary for test mice data
dev.off()

#Make a true/false filtering column based on the glam score for plotting 
pData(data_monocle)$Present <- pData(data_monocle)$glam_score > 38

# Make a new dataframe to use for statistical analysis but not plotting the cells

rm(glam_stats_M1)
rm(glam_stats_M3)

# Only include cells from mouse 2 and mouse 4, separately
glam_stats_M1 <- data_monocle$Present[which(data_monocle$mice == "M1")] %>% as.data.frame()
colnames(glam_stats_M1) <- "Present"
# Add other columns from data_monocle
glam_stats_M1$State <- data_monocle$State[which(data_monocle$mice == "M1")]
glam_stats_M1$combined_cdr3_nt <- data_monocle$combined_cdr3_nt[which(data_monocle$mice == "M1")]
glam_stats_M1$cellID <- rownames(data_monocle@phenoData)[which(data_monocle$mice == "M1")]
# Remove any rows with NA (cells without a motif score, incluing cells that don't meet definition of a clone)
glam_stats_M1 <- glam_stats_M1[!is.na(glam_stats_M1$Present), ]

glam_stats_M3 <- data_monocle$Present[which(data_monocle$mice == "M3")] %>% as.data.frame()
colnames(glam_stats_M3) <- "Present"
# Add other columns from data_monocle
glam_stats_M3$State <- data_monocle$State[which(data_monocle$mice == "M3")]
glam_stats_M3$combined_cdr3_nt <- data_monocle$combined_cdr3_nt[which(data_monocle$mice == "M3")]
glam_stats_M3$cellID <- rownames(data_monocle@phenoData)[which(data_monocle$mice == "M3")]
# Remove any rows with NA (cells without a motif score, incluing cells that don't meet definition of a clone)
glam_stats_M3 <- glam_stats_M3[!is.na(glam_stats_M3$Present), ]




#Make NA values false
pData(data_monocle)$Present[is.na(pData(data_monocle)$Present)] <- FALSE
plot_obj <- plot_complex_cell_trajectory(data_monocle, color_by = "Present")
plot_obj$data <- plot_obj$data[order(plot_obj$data$Present),]
plot_obj + scale_color_manual(values=c("grey", "red")) + ggtitle("TRA_TFH_motif10_40_across_M1M3'\n Score > 40 \n ")

png(file = "summary_6mice/TRA_TFH_motif1_M2M4M6_cln4.png", width = 8, height = 6, units = "in", res = 300)
plot_obj + scale_color_manual(values=c("grey", "blue")) + ggtitle("TRA_TH1_motif16_M2M4M6.png'\n Score > 30 \n 'P value '")
#dev.off()

#Create a table detailing the number of cells in each monocle state that have a good motif match
# The line below was from Robert's analysis, I'm replacing it because there was an error
# glam_score_table <- as.data.frame(table(pData(data_no_6_mono_cds)$State, pData(data_no_6_mono_cds)$Present))
rm(glam_score_table_M1)
rm(glam_score_table_M3)


glam_score_table_M1 <- table(glam_stats_M1$State, glam_stats_M1$Present) %>% as.data.frame()
#Combine true and false rows
glam_score_table_M1 <- cbind(glam_score_table_M1[1:5,], glam_score_table_M1[6:10,])
#Create names that make more sense
colnames(glam_score_table_M1) <- c("State", "F", "Freq_F", "State2", "T", "Freq_T")

#Remove unneeded columns
glam_score_table_M1 <- glam_score_table_M1[,c(1,3,6)]
#Create a percent column (percent of cells within each state with a good motif match)
glam_score_table_M1$pct <- glam_score_table_M1$Freq_T / (glam_score_table_M1$Freq_F + glam_score_table_M1$Freq_T)
#write.csv(glam_score_table_M2, "Motif/train/TFH/score_file/TRA_TFH_T65_motif1_M2_cln4_cells_across_M2.csv")
write.csv(glam_score_table_M1, "Motif_5mice/Test/TFH/Score/TRA_TFH_T65_motif16_M1_cln4_cells_across_M1_cells.csv")
getwd()
glam_score_table_M1$Freq_F %>% sum () # 829
glam_score_table_M1$Freq_T %>% sum() # 49

#Create a table detailing the number of cells in each monocle state that have a good motif match
# The line below was from Robert's analysis, I'm replacing it because there was an error
# glam_score_table <- as.data.frame(table(pData(data_no_6_mono_cds)$State, pData(data_no_6_mono_cds)$Present))

glam_score_table_M3 <- table(glam_stats_M3$State, glam_stats_M3$Present) %>% as.data.frame()
#Combine true and false rows
glam_score_table_M3 <- cbind(glam_score_table_M3[1:5,], glam_score_table_M3[6:10,]) ## this # depends on the # of monocle states
#Create names that make more sense
colnames(glam_score_table_M3) <- c("State", "F", "Freq_F", "State2", "T", "Freq_T")

#Remove unneeded columns
glam_score_table_M3 <- glam_score_table_M3[,c(1,3,6)]
#Create a percent column (percent of cells within each state with a good motif match)
glam_score_table_M3$pct <- glam_score_table_M3$Freq_T / (glam_score_table_M3$Freq_F + glam_score_table_M3$Freq_T)
#write.csv(glam_score_table_M4, "Motif/train/TFH/score_file/TRA_TFH_T65_motif1_M4_cln4_cells_across_M4.csv")
write.csv(glam_score_table_M3, "Motif_5mice/Test/TFH/Score/TRA_TFH_T65_motif16_M3_cln4_cells_across_M3_cells.csv")
getwd()
glam_score_table_M3$Freq_F %>% sum () # 1862
glam_score_table_M3$Freq_T %>% sum() # 555



# Perform clone-level statistical analysis

glam_stats_M2$combined_cdr3_nt[which(glam_stats_M2$Present == TRUE)] %>% table()


#         TRA:TGCTCAGCAAGTATGACTGGAGGCTATAAAGTGGTCTTT_TRB:TGTGCCAGCAGCAAGGTAAACACAGAAGTCTTCTTT 
 #                                                                                           4 
#TRA:TGTGCAGCAAGTCTCGATGGAGGCTATAAAGTGGTCTTT_TRB:TGTGCTAGCAGTTTATCCGGGACAGGGGACGCTGAGCAGTTCTTC 
    #                                                                                       30 
   #TRA:TGTGCAGTGAGCGGGACTGGAGGCTATAAAGTGGTCTTT_TRB:TGTGCCAGCAGGACTGGGGGGGCAGGGGACACCCAGTACTTT 
    #                                                                                       13 
     # TRA:TGTGCTCTGGGCCTGACTGGAGGCTATAAAGTGGTCTTT_TRB:TGCAGTGCAAGGGGGGCGGACACCGGGCAGCTCTACTTT 
       #                                                                                    21 

grep(pattern = "TRA:TGTGCAGCAAGTTGGGGGACTGGAGGCTATAAAGTGGTCTTT_TRB:TGTGCTAGCAGTTTTGCTGGGGGGGCTTCCTATGAACAGTACTTC", x = data_clones_by_mouse_AA_TRA_TRB_NOV$combined_cdr3_nt)
# 3
as.data.frame(data_clones_by_mouse_AA_TRA_TRB_NOV[1, ])
# 79, 0, 21
getwd()
glam_stats_M4$combined_cdr3_nt[which(glam_stats_M4$Present == TRUE)] %>% table()

        #  TRA:TGTGCAGCAAGCATGAGAACAGGAAACTACAAATACGTCTTT_TRB:TGTGCCAGCAGCGCCCAAACGCTGTATTTT 
         #                                                                                      5 
      #TRA:TGTGCAGCAAGTCTTAATACAGGAAACTACAAATACGTCTTT_TRB:TGTGCCTGGAATGGGGGGGGCTATGCTGAGCAGTTCTTC 
       #                                                                                       12 
#TRA:TGTGCAGCAAGTTGGGGGACTGGAGGCTATAAAGTGGTCTTT_TRB:TGTGCTAGCAGTTTTGCTGGGGGGGCTTCCTATGAACAGTACTTC 
 #                                                                                            813
```


```{r Th1 motifs 1 to 11 in train mice M2, M4 and M6}
rm(list=ls())

# Th1 motif 1

data_monocle <- readRDS("summary_5M_CCA_V2/data_monocle_2cell.RDS")
# Th1 motif 1
rm(glam2_scores)
glam2_scores <- read.table(file = "/media/runa/Elements/Jem_revision/Seurat/Motif_5mice/Train/Th1/Motif1/Th1_Motif1.glam2scan/glam2scan_scores_Motif1.txt", header = FALSE)
#We're only interested in the cell name and score columns
glam2_scores <- glam2_scores[,c(1,6)]
#Take the top score for each cell
glam2_scores <- aggregate(.~V1, glam2_scores, FUN=max) # here multiple cells can have
#more than one score, with aggregate command we are saying in those cases just keep the max
#score
rownames(glam2_scores) <- glam2_scores$V1
# to match cell names with monocle data add 1 in cell name
glam2_scores$V1 <-
  paste0(glam2_scores$V1, "1")
rownames(glam2_scores) <- glam2_scores$V1

#Add glam score to each cell within monocle pData
pData(data_monocle)$glam_score <- sapply(rownames(pData(data_monocle)), FUN =
function(x) glam2_scores[x,]$V6)

View(pData(data_monocle))


#Observe the glam score distribution, find a high score for the cells 40 looks good here; this
#scoring will vary from motif to motif
plot(density(pData(data_monocle)$glam_score[!is.na(pData(data_monocle)$glam_score)]),
main = "Motif score distribution_TRA_Th1_Motif1", xlab = "")
abline(v = 36, col = "Red")

#Make a true/false filtering column based on the glam score for plotting 
pData(data_monocle)$Present <- pData(data_monocle)$glam_score > 36

# Make a new dataframe to use for statistical analysis but not plotting the cells

rm(glam_stats_M2)
rm(glam_stats_M4)
rm(glam_stats_M6)

# Only include cells from mouse 2 and mouse 4, separately
glam_stats_M2 <- data_monocle$Present[which(data_monocle$mice == "M2")] %>% as.data.frame()
colnames(glam_stats_M2) <- "Present"
# Add other columns from data_monocle
glam_stats_M2$State <- data_monocle$State[which(data_monocle$mice == "M2")]
glam_stats_M2$combined_cdr3_nt <- data_monocle$combined_cdr3_nt[which(data_monocle$mice == "M2")]
glam_stats_M2$cellID <- rownames(data_monocle@phenoData)[which(data_monocle$mice == "M2")]
# Remove any rows with NA (cells without a motif score, incluing cells that don't meet definition of a clone)
glam_stats_M2 <- glam_stats_M2[!is.na(glam_stats_M2$Present), ]

glam_stats_M4 <- data_monocle$Present[which(data_monocle$mice == "M4")] %>% as.data.frame()
colnames(glam_stats_M4) <- "Present"
# Add other columns from data_monocle
glam_stats_M4$State <- data_monocle$State[which(data_monocle$mice == "M4")]
glam_stats_M4$combined_cdr3_nt <- data_monocle$combined_cdr3_nt[which(data_monocle$mice == "M4")]
glam_stats_M4$cellID <- rownames(data_monocle@phenoData)[which(data_monocle$mice == "M4")]
# Remove any rows with NA (cells without a motif score, incluing cells that don't meet definition of a clone)
glam_stats_M4 <- glam_stats_M4[!is.na(glam_stats_M4$Present), ]

glam_stats_M6 <- data_monocle$Present[which(data_monocle$mice == "M6")] %>% as.data.frame()
colnames(glam_stats_M6) <- "Present"
# Add other columns from data_monocle
glam_stats_M6$State <- data_monocle$State[which(data_monocle$mice == "M6")]
glam_stats_M6$combined_cdr3_nt <- data_monocle$combined_cdr3_nt[which(data_monocle$mice == "M6")]
glam_stats_M6$cellID <- rownames(data_monocle@phenoData)[which(data_monocle$mice == "M6")]
# Remove any rows with NA (cells without a motif score, incluing cells that don't meet definition of a clone)
glam_stats_M6 <- glam_stats_M6[!is.na(glam_stats_M6$Present), ]



#Make NA values false
pData(data_monocle)$Present[is.na(pData(data_monocle)$Present)] <- FALSE
plot_obj <- plot_complex_cell_trajectory(data_monocle, color_by = "Present")
plot_obj$data <- plot_obj$data[order(plot_obj$data$Present),]
plot_obj + scale_color_manual(values=c("grey", "red")) + ggtitle("TRA_Th1_motif1_36_across_M2M4M6'\n Score > 36 \n ")

png(file = "summary_6mice/TRA_TH1_motif1_M2M4M6_cln4.png", width = 8, height = 6, units = "in", res = 300)
plot_obj + scale_color_manual(values=c("grey", "red")) + ggtitle("TRA_TH1_motif6_M2M4M6.png'\n Score > 37 \n 'P value '")
#dev.off()

#Create a table detailing the number of cells in each monocle state that have a good motif match
# The line below was from Robert's analysis, I'm replacing it because there was an error
# glam_score_table <- as.data.frame(table(pData(data_no_6_mono_cds)$State, pData(data_no_6_mono_cds)$Present))
rm(glam_score_table_M2)
rm(glam_score_table_M4)
rm(glam_score_table_M6)


glam_score_table_M2 <- table(glam_stats_M2$State, glam_stats_M2$Present) %>% as.data.frame()
#Combine true and false rows
glam_score_table_M2 <- cbind(glam_score_table_M2[1:5,], glam_score_table_M2[6:10,])
#Create names that make more sense
colnames(glam_score_table_M2) <- c("State", "F", "Freq_F", "State2", "T", "Freq_T")

#Remove unneeded columns
glam_score_table_M2 <- glam_score_table_M2[,c(1,3,6)]
#Create a percent column (percent of cells within each state with a good motif match)
glam_score_table_M2$pct <- glam_score_table_M2$Freq_T / (glam_score_table_M2$Freq_F + glam_score_table_M2$Freq_T)
#write.csv(glam_score_table_M2, "Motif/train/TFH/score_file/TRA_TFH_T65_motif1_M2_cln4_cells_across_M2.csv")
write.csv(glam_score_table_M2, "Motif_5mice/Train/Th1/Score/TRA_Th1_T65_motif1_M2_cln4_cells_across_M2_cells.csv")
getwd()
glam_score_table_M2$Freq_F %>% sum () # 805
glam_score_table_M2$Freq_T %>% sum() # 58

#Create a table detailing the number of cells in each monocle state that have a good motif match
# The line below was from Robert's analysis, I'm replacing it because there was an error
# glam_score_table <- as.data.frame(table(pData(data_no_6_mono_cds)$State, pData(data_no_6_mono_cds)$Present))

glam_score_table_M4 <- table(glam_stats_M4$State, glam_stats_M4$Present) %>% as.data.frame()
#Combine true and false rows
glam_score_table_M4 <- cbind(glam_score_table_M4[1:5,], glam_score_table_M4[6:10,]) ## this # depends on the # of monocle states
#Create names that make more sense
colnames(glam_score_table_M4) <- c("State", "F", "Freq_F", "State2", "T", "Freq_T")

#Remove unneeded columns
glam_score_table_M4 <- glam_score_table_M4[,c(1,3,6)]
#Create a percent column (percent of cells within each state with a good motif match)
glam_score_table_M4$pct <- glam_score_table_M4$Freq_T / (glam_score_table_M4$Freq_F + glam_score_table_M4$Freq_T)
#write.csv(glam_score_table_M4, "Motif/train/TFH/score_file/TRA_TFH_T65_motif1_M4_cln4_cells_across_M4.csv")
write.csv(glam_score_table_M4, "Motif_5mice/Train/Th1/Score/TRA_Th1_T65_motif1_M4_cln4_cells_across_M4_cells.csv")
getwd()
glam_score_table_M4$Freq_F %>% sum () # 3218
glam_score_table_M4$Freq_T %>% sum() # 31


#Create a table detailing the number of cells in each monocle state that have a good motif match
# The line below was from Robert's analysis, I'm replacing it because there was an error
# glam_score_table <- as.data.frame(table(pData(data_no_6_mono_cds)$State, pData(data_no_6_mono_cds)$Present))

glam_score_table_M6 <- table(glam_stats_M6$State, glam_stats_M6$Present) %>% as.data.frame()
#Combine true and false rows
glam_score_table_M6 <- cbind(glam_score_table_M6[1:5,], glam_score_table_M6[6:10,]) ## this # depends on the # of monocle states
#Create names that make more sense
colnames(glam_score_table_M6) <- c("State", "F", "Freq_F", "State2", "T", "Freq_T")

#Remove unneeded columns
glam_score_table_M6 <- glam_score_table_M6[,c(1,3,6)]
#Create a percent column (percent of cells within each state with a good motif match)
glam_score_table_M6$pct <- glam_score_table_M6$Freq_T / (glam_score_table_M6$Freq_F + glam_score_table_M6$Freq_T)
#write.csv(glam_score_table_M4, "Motif/train/TFH/score_file/TRA_TFH_T65_motif1_M4_cln4_cells_across_M4.csv")
write.csv(glam_score_table_M6, "Motif_5mice/Train/Th1/Score/TRA_Th1_T65_motif1_M6_cln4_cells_across_M6_cells.csv")
getwd()
glam_score_table_M6$Freq_F %>% sum () # 2324
glam_score_table_M6$Freq_T %>% sum() # NA


# Perform clone-level statistical analysis

glam_stats_M2$combined_cdr3_nt[which(glam_stats_M2$Present == TRUE)] %>% table()


#         TRA:TGCTCAGCAAGTATGACTGGAGGCTATAAAGTGGTCTTT_TRB:TGTGCCAGCAGCAAGGTAAACACAGAAGTCTTCTTT 
 #                                                                                           4 
#TRA:TGTGCAGCAAGTCTCGATGGAGGCTATAAAGTGGTCTTT_TRB:TGTGCTAGCAGTTTATCCGGGACAGGGGACGCTGAGCAGTTCTTC 
    #                                                                                       30 
   #TRA:TGTGCAGTGAGCGGGACTGGAGGCTATAAAGTGGTCTTT_TRB:TGTGCCAGCAGGACTGGGGGGGCAGGGGACACCCAGTACTTT 
    #                                                                                       13 
     # TRA:TGTGCTCTGGGCCTGACTGGAGGCTATAAAGTGGTCTTT_TRB:TGCAGTGCAAGGGGGGCGGACACCGGGCAGCTCTACTTT 
       #                                                                                    21 

grep(pattern = "TRA:TGTGCAGCAAGTTGGGGGACTGGAGGCTATAAAGTGGTCTTT_TRB:TGTGCTAGCAGTTTTGCTGGGGGGGCTTCCTATGAACAGTACTTC", x = data_clones_by_mouse_AA_TRA_TRB_NOV$combined_cdr3_nt)
# 3
as.data.frame(data_clones_by_mouse_AA_TRA_TRB_NOV[1, ])
# 79, 0, 21
getwd()
glam_stats_M4$combined_cdr3_nt[which(glam_stats_M4$Present == TRUE)] %>% table()

        #  TRA:TGTGCAGCAAGCATGAGAACAGGAAACTACAAATACGTCTTT_TRB:TGTGCCAGCAGCGCCCAAACGCTGTATTTT 
         #                                                                                      5 
      #TRA:TGTGCAGCAAGTCTTAATACAGGAAACTACAAATACGTCTTT_TRB:TGTGCCTGGAATGGGGGGGGCTATGCTGAGCAGTTCTTC 
       #                                                                                       12 
#TRA:TGTGCAGCAAGTTGGGGGACTGGAGGCTATAAAGTGGTCTTT_TRB:TGTGCTAGCAGTTTTGCTGGGGGGGCTTCCTATGAACAGTACTTC 
 #                              

```
```{r}
# Th1 motif 2

#data_monocle <- readRDS("summary_5M_CCA_V2/data_monocle_2cell.RDS")
# Th1 motif 2
rm(glam2_scores)
glam2_scores <- read.table(file = "/media/runa/Elements/Jem_revision/Seurat/Motif_5mice/Train/Th1/Motif2/Th1_Motif2.glam2scan/glam2scan_scores_Motif2.txt", header = FALSE)
#We're only interested in the cell name and score columns
glam2_scores <- glam2_scores[,c(1,6)]
#Take the top score for each cell
glam2_scores <- aggregate(.~V1, glam2_scores, FUN=max) # here multiple cells can have
#more than one score, with aggregate command we are saying in those cases just keep the max
#score
rownames(glam2_scores) <- glam2_scores$V1
# to match cell names with monocle data add 1 in cell name
glam2_scores$V1 <-
  paste0(glam2_scores$V1, "1")
rownames(glam2_scores) <- glam2_scores$V1

#Add glam score to each cell within monocle pData
pData(data_monocle)$glam_score <- sapply(rownames(pData(data_monocle)), FUN =
function(x) glam2_scores[x,]$V6)

View(pData(data_monocle))


#Observe the glam score distribution, find a high score for the cells 40 looks good here; this
#scoring will vary from motif to motif
plot(density(pData(data_monocle)$glam_score[!is.na(pData(data_monocle)$glam_score)]),
main = "Motif score distribution_TRA_Th1_Motif2", xlab = "")
abline(v = 28, col = "Red")

#Make a true/false filtering column based on the glam score for plotting 
pData(data_monocle)$Present <- pData(data_monocle)$glam_score > 28

# Make a new dataframe to use for statistical analysis but not plotting the cells

rm(glam_stats_M2)
rm(glam_stats_M4)
rm(glam_stats_M6)

# Only include cells from mouse 2 and mouse 4, separately
glam_stats_M2 <- data_monocle$Present[which(data_monocle$mice == "M2")] %>% as.data.frame()
colnames(glam_stats_M2) <- "Present"
# Add other columns from data_monocle
glam_stats_M2$State <- data_monocle$State[which(data_monocle$mice == "M2")]
glam_stats_M2$combined_cdr3_nt <- data_monocle$combined_cdr3_nt[which(data_monocle$mice == "M2")]
glam_stats_M2$cellID <- rownames(data_monocle@phenoData)[which(data_monocle$mice == "M2")]
# Remove any rows with NA (cells without a motif score, incluing cells that don't meet definition of a clone)
glam_stats_M2 <- glam_stats_M2[!is.na(glam_stats_M2$Present), ]

glam_stats_M4 <- data_monocle$Present[which(data_monocle$mice == "M4")] %>% as.data.frame()
colnames(glam_stats_M4) <- "Present"
# Add other columns from data_monocle
glam_stats_M4$State <- data_monocle$State[which(data_monocle$mice == "M4")]
glam_stats_M4$combined_cdr3_nt <- data_monocle$combined_cdr3_nt[which(data_monocle$mice == "M4")]
glam_stats_M4$cellID <- rownames(data_monocle@phenoData)[which(data_monocle$mice == "M4")]
# Remove any rows with NA (cells without a motif score, incluing cells that don't meet definition of a clone)
glam_stats_M4 <- glam_stats_M4[!is.na(glam_stats_M4$Present), ]

glam_stats_M6 <- data_monocle$Present[which(data_monocle$mice == "M6")] %>% as.data.frame()
colnames(glam_stats_M6) <- "Present"
# Add other columns from data_monocle
glam_stats_M6$State <- data_monocle$State[which(data_monocle$mice == "M6")]
glam_stats_M6$combined_cdr3_nt <- data_monocle$combined_cdr3_nt[which(data_monocle$mice == "M6")]
glam_stats_M6$cellID <- rownames(data_monocle@phenoData)[which(data_monocle$mice == "M6")]
# Remove any rows with NA (cells without a motif score, incluing cells that don't meet definition of a clone)
glam_stats_M6 <- glam_stats_M6[!is.na(glam_stats_M6$Present), ]



#Make NA values false
pData(data_monocle)$Present[is.na(pData(data_monocle)$Present)] <- FALSE
plot_obj <- plot_complex_cell_trajectory(data_monocle, color_by = "Present")
plot_obj$data <- plot_obj$data[order(plot_obj$data$Present),]
plot_obj + scale_color_manual(values=c("grey", "red")) + ggtitle("TRA_Th1_motif2_28_across_M2M4M6'\n Score > 28 \n ")

png(file = "summary_6mice/TRA_TH1_motif1_M2M4M6_cln4.png", width = 8, height = 6, units = "in", res = 300)
plot_obj + scale_color_manual(values=c("grey", "red")) + ggtitle("TRA_TH1_motif6_M2M4M6.png'\n Score > 37 \n 'P value '")
#dev.off()

#Create a table detailing the number of cells in each monocle state that have a good motif match
# The line below was from Robert's analysis, I'm replacing it because there was an error
# glam_score_table <- as.data.frame(table(pData(data_no_6_mono_cds)$State, pData(data_no_6_mono_cds)$Present))
rm(glam_score_table_M2)
rm(glam_score_table_M4)
rm(glam_score_table_M6)


glam_score_table_M2 <- table(glam_stats_M2$State, glam_stats_M2$Present) %>% as.data.frame()
#Combine true and false rows
glam_score_table_M2 <- cbind(glam_score_table_M2[1:5,], glam_score_table_M2[6:10,])
#Create names that make more sense
colnames(glam_score_table_M2) <- c("State", "F", "Freq_F", "State2", "T", "Freq_T")

#Remove unneeded columns
glam_score_table_M2 <- glam_score_table_M2[,c(1,3,6)]
#Create a percent column (percent of cells within each state with a good motif match)
glam_score_table_M2$pct <- glam_score_table_M2$Freq_T / (glam_score_table_M2$Freq_F + glam_score_table_M2$Freq_T)
#write.csv(glam_score_table_M2, "Motif/train/TFH/score_file/TRA_TFH_T65_motif1_M2_cln4_cells_across_M2.csv")
write.csv(glam_score_table_M2, "Motif_5mice/Train/Th1/Score/TRA_Th1_T65_motif2_M2_cln4_cells_across_M2_cells.csv")
getwd()
glam_score_table_M2$Freq_F %>% sum () # 814
glam_score_table_M2$Freq_T %>% sum() # 49

#Create a table detailing the number of cells in each monocle state that have a good motif match
# The line below was from Robert's analysis, I'm replacing it because there was an error
# glam_score_table <- as.data.frame(table(pData(data_no_6_mono_cds)$State, pData(data_no_6_mono_cds)$Present))

glam_score_table_M4 <- table(glam_stats_M4$State, glam_stats_M4$Present) %>% as.data.frame()
#Combine true and false rows
glam_score_table_M4 <- cbind(glam_score_table_M4[1:5,], glam_score_table_M4[6:10,]) ## this # depends on the # of monocle states
#Create names that make more sense
colnames(glam_score_table_M4) <- c("State", "F", "Freq_F", "State2", "T", "Freq_T")

#Remove unneeded columns
glam_score_table_M4 <- glam_score_table_M4[,c(1,3,6)]
#Create a percent column (percent of cells within each state with a good motif match)
glam_score_table_M4$pct <- glam_score_table_M4$Freq_T / (glam_score_table_M4$Freq_F + glam_score_table_M4$Freq_T)
#write.csv(glam_score_table_M4, "Motif/train/TFH/score_file/TRA_TFH_T65_motif1_M4_cln4_cells_across_M4.csv")
write.csv(glam_score_table_M4, "Motif_5mice/Train/Th1/Score/TRA_Th1_T65_motif2_M4_cln4_cells_across_M4_cells.csv")
getwd()
glam_score_table_M4$Freq_F %>% sum () # 3175
glam_score_table_M4$Freq_T %>% sum() # 43


#Create a table detailing the number of cells in each monocle state that have a good motif match
# The line below was from Robert's analysis, I'm replacing it because there was an error
# glam_score_table <- as.data.frame(table(pData(data_no_6_mono_cds)$State, pData(data_no_6_mono_cds)$Present))

glam_score_table_M6 <- table(glam_stats_M6$State, glam_stats_M6$Present) %>% as.data.frame()
#Combine true and false rows
glam_score_table_M6 <- cbind(glam_score_table_M6[1:5,], glam_score_table_M6[6:10,]) ## this # depends on the # of monocle states
#Create names that make more sense
colnames(glam_score_table_M6) <- c("State", "F", "Freq_F", "State2", "T", "Freq_T")

#Remove unneeded columns
glam_score_table_M6 <- glam_score_table_M6[,c(1,3,6)]
#Create a percent column (percent of cells within each state with a good motif match)
glam_score_table_M6$pct <- glam_score_table_M6$Freq_T / (glam_score_table_M6$Freq_F + glam_score_table_M6$Freq_T)
#write.csv(glam_score_table_M4, "Motif/train/TFH/score_file/TRA_TFH_T65_motif1_M4_cln4_cells_across_M4.csv")
write.csv(glam_score_table_M6, "Motif_5mice/Train/Th1/Score/TRA_Th1_T65_motif2_M6_cln4_cells_across_M6_cells.csv")
getwd()
glam_score_table_M6$Freq_F %>% sum () # 1917
glam_score_table_M6$Freq_T %>% sum() # 407


# Perform clone-level statistical analysis

glam_stats_M2$combined_cdr3_nt[which(glam_stats_M2$Present == TRUE)] %>% table()


#         TRA:TGCTCAGCAAGTATGACTGGAGGCTATAAAGTGGTCTTT_TRB:TGTGCCAGCAGCAAGGTAAACACAGAAGTCTTCTTT 
 #                                                                                           4 
#TRA:TGTGCAGCAAGTCTCGATGGAGGCTATAAAGTGGTCTTT_TRB:TGTGCTAGCAGTTTATCCGGGACAGGGGACGCTGAGCAGTTCTTC 
    #                                                                                       30 
   #TRA:TGTGCAGTGAGCGGGACTGGAGGCTATAAAGTGGTCTTT_TRB:TGTGCCAGCAGGACTGGGGGGGCAGGGGACACCCAGTACTTT 
    #                                                                                       13 
     # TRA:TGTGCTCTGGGCCTGACTGGAGGCTATAAAGTGGTCTTT_TRB:TGCAGTGCAAGGGGGGCGGACACCGGGCAGCTCTACTTT 
       #                                                                                    21 

grep(pattern = "TRA:TGTGCAGCAAGTTGGGGGACTGGAGGCTATAAAGTGGTCTTT_TRB:TGTGCTAGCAGTTTTGCTGGGGGGGCTTCCTATGAACAGTACTTC", x = data_clones_by_mouse_AA_TRA_TRB_NOV$combined_cdr3_nt)
# 3
as.data.frame(data_clones_by_mouse_AA_TRA_TRB_NOV[1, ])
# 79, 0, 21
getwd()
glam_stats_M4$combined_cdr3_nt[which(glam_stats_M4$Present == TRUE)] %>% table()

        #  TRA:TGTGCAGCAAGCATGAGAACAGGAAACTACAAATACGTCTTT_TRB:TGTGCCAGCAGCGCCCAAACGCTGTATTTT 
         #                                                                                      5 
      #TRA:TGTGCAGCAAGTCTTAATACAGGAAACTACAAATACGTCTTT_TRB:TGTGCCTGGAATGGGGGGGGCTATGCTGAGCAGTTCTTC 
       #                                                                                       12 
#TRA:TGTGCAGCAAGTTGGGGGACTGGAGGCTATAAAGTGGTCTTT_TRB:TGTGCTAGCAGTTTTGCTGGGGGGGCTTCCTATGAACAGTACTTC 
 #                              

```
```{r}
# Th1 motif 3

#data_monocle <- readRDS("summary_5M_CCA_V2/data_monocle_2cell.RDS")
# Th1 motif 3
rm(glam2_scores)
glam2_scores <- read.table(file = "/media/runa/Elements/Jem_revision/Seurat/Motif_5mice/Train/Th1/Motif3/Th1_Motif3.glam2scan/glam2scan_scores_Motif3.txt", header = FALSE)
#We're only interested in the cell name and score columns
glam2_scores <- glam2_scores[,c(1,6)]
#Take the top score for each cell
glam2_scores <- aggregate(.~V1, glam2_scores, FUN=max) # here multiple cells can have
#more than one score, with aggregate command we are saying in those cases just keep the max
#score
rownames(glam2_scores) <- glam2_scores$V1
# to match cell names with monocle data add 1 in cell name
glam2_scores$V1 <-
  paste0(glam2_scores$V1, "1")
rownames(glam2_scores) <- glam2_scores$V1

#Add glam score to each cell within monocle pData
pData(data_monocle)$glam_score <- sapply(rownames(pData(data_monocle)), FUN =
function(x) glam2_scores[x,]$V6)

View(pData(data_monocle))


#Observe the glam score distribution, find a high score for the cells 40 looks good here; this
#scoring will vary from motif to motif
plot(density(pData(data_monocle)$glam_score[!is.na(pData(data_monocle)$glam_score)]),
main = "Motif score distribution_TRA_Th1_Motif3", xlab = "")
abline(v = 38, col = "Red")

#Make a true/false filtering column based on the glam score for plotting 
pData(data_monocle)$Present <- pData(data_monocle)$glam_score > 38

# Make a new dataframe to use for statistical analysis but not plotting the cells

rm(glam_stats_M2)
rm(glam_stats_M4)
rm(glam_stats_M6)

# Only include cells from mouse 2 and mouse 4, separately
glam_stats_M2 <- data_monocle$Present[which(data_monocle$mice == "M2")] %>% as.data.frame()
colnames(glam_stats_M2) <- "Present"
# Add other columns from data_monocle
glam_stats_M2$State <- data_monocle$State[which(data_monocle$mice == "M2")]
glam_stats_M2$combined_cdr3_nt <- data_monocle$combined_cdr3_nt[which(data_monocle$mice == "M2")]
glam_stats_M2$cellID <- rownames(data_monocle@phenoData)[which(data_monocle$mice == "M2")]
# Remove any rows with NA (cells without a motif score, incluing cells that don't meet definition of a clone)
glam_stats_M2 <- glam_stats_M2[!is.na(glam_stats_M2$Present), ]

glam_stats_M4 <- data_monocle$Present[which(data_monocle$mice == "M4")] %>% as.data.frame()
colnames(glam_stats_M4) <- "Present"
# Add other columns from data_monocle
glam_stats_M4$State <- data_monocle$State[which(data_monocle$mice == "M4")]
glam_stats_M4$combined_cdr3_nt <- data_monocle$combined_cdr3_nt[which(data_monocle$mice == "M4")]
glam_stats_M4$cellID <- rownames(data_monocle@phenoData)[which(data_monocle$mice == "M4")]
# Remove any rows with NA (cells without a motif score, incluing cells that don't meet definition of a clone)
glam_stats_M4 <- glam_stats_M4[!is.na(glam_stats_M4$Present), ]

glam_stats_M6 <- data_monocle$Present[which(data_monocle$mice == "M6")] %>% as.data.frame()
colnames(glam_stats_M6) <- "Present"
# Add other columns from data_monocle
glam_stats_M6$State <- data_monocle$State[which(data_monocle$mice == "M6")]
glam_stats_M6$combined_cdr3_nt <- data_monocle$combined_cdr3_nt[which(data_monocle$mice == "M6")]
glam_stats_M6$cellID <- rownames(data_monocle@phenoData)[which(data_monocle$mice == "M6")]
# Remove any rows with NA (cells without a motif score, incluing cells that don't meet definition of a clone)
glam_stats_M6 <- glam_stats_M6[!is.na(glam_stats_M6$Present), ]



#Make NA values false
pData(data_monocle)$Present[is.na(pData(data_monocle)$Present)] <- FALSE
plot_obj <- plot_complex_cell_trajectory(data_monocle, color_by = "Present")
plot_obj$data <- plot_obj$data[order(plot_obj$data$Present),]
plot_obj + scale_color_manual(values=c("grey", "red")) + ggtitle("TRA_Th1_motif3_38_across_M2M4M6'\n Score > 38 \n ")

png(file = "summary_6mice/TRA_TH1_motif1_M2M4M6_cln4.png", width = 8, height = 6, units = "in", res = 300)
plot_obj + scale_color_manual(values=c("grey", "red")) + ggtitle("TRA_TH1_motif6_M2M4M6.png'\n Score > 37 \n 'P value '")
#dev.off()

#Create a table detailing the number of cells in each monocle state that have a good motif match
# The line below was from Robert's analysis, I'm replacing it because there was an error
# glam_score_table <- as.data.frame(table(pData(data_no_6_mono_cds)$State, pData(data_no_6_mono_cds)$Present))
rm(glam_score_table_M2)
rm(glam_score_table_M4)
rm(glam_score_table_M6)


glam_score_table_M2 <- table(glam_stats_M2$State, glam_stats_M2$Present) %>% as.data.frame()
#Combine true and false rows
glam_score_table_M2 <- cbind(glam_score_table_M2[1:5,], glam_score_table_M2[6:10,])
#Create names that make more sense
colnames(glam_score_table_M2) <- c("State", "F", "Freq_F", "State2", "T", "Freq_T")

#Remove unneeded columns
glam_score_table_M2 <- glam_score_table_M2[,c(1,3,6)]
#Create a percent column (percent of cells within each state with a good motif match)
glam_score_table_M2$pct <- glam_score_table_M2$Freq_T / (glam_score_table_M2$Freq_F + glam_score_table_M2$Freq_T)
#write.csv(glam_score_table_M2, "Motif/train/TFH/score_file/TRA_TFH_T65_motif1_M2_cln4_cells_across_M2.csv")
write.csv(glam_score_table_M2, "Motif_5mice/Train/Th1/Score/TRA_Th1_T65_motif3_M2_cln4_cells_across_M2_cells.csv")
getwd()
glam_score_table_M2$Freq_F %>% sum () # 858
glam_score_table_M2$Freq_T %>% sum() # 5

#Create a table detailing the number of cells in each monocle state that have a good motif match
# The line below was from Robert's analysis, I'm replacing it because there was an error
# glam_score_table <- as.data.frame(table(pData(data_no_6_mono_cds)$State, pData(data_no_6_mono_cds)$Present))

glam_score_table_M4 <- table(glam_stats_M4$State, glam_stats_M4$Present) %>% as.data.frame()
#Combine true and false rows
glam_score_table_M4 <- cbind(glam_score_table_M4[1:5,], glam_score_table_M4[6:10,]) ## this # depends on the # of monocle states
#Create names that make more sense
colnames(glam_score_table_M4) <- c("State", "F", "Freq_F", "State2", "T", "Freq_T")

#Remove unneeded columns
glam_score_table_M4 <- glam_score_table_M4[,c(1,3,6)]
#Create a percent column (percent of cells within each state with a good motif match)
glam_score_table_M4$pct <- glam_score_table_M4$Freq_T / (glam_score_table_M4$Freq_F + glam_score_table_M4$Freq_T)
#write.csv(glam_score_table_M4, "Motif/train/TFH/score_file/TRA_TFH_T65_motif1_M4_cln4_cells_across_M4.csv")
write.csv(glam_score_table_M4, "Motif_5mice/Train/Th1/Score/TRA_Th1_T65_motif3_M4_cln4_cells_across_M4_cells.csv")
getwd()
glam_score_table_M4$Freq_F %>% sum () # 3218
glam_score_table_M4$Freq_T %>% sum() # NA


#Create a table detailing the number of cells in each monocle state that have a good motif match
# The line below was from Robert's analysis, I'm replacing it because there was an error
# glam_score_table <- as.data.frame(table(pData(data_no_6_mono_cds)$State, pData(data_no_6_mono_cds)$Present))

glam_score_table_M6 <- table(glam_stats_M6$State, glam_stats_M6$Present) %>% as.data.frame()
#Combine true and false rows
glam_score_table_M6 <- cbind(glam_score_table_M6[1:5,], glam_score_table_M6[6:10,]) ## this # depends on the # of monocle states
#Create names that make more sense
colnames(glam_score_table_M6) <- c("State", "F", "Freq_F", "State2", "T", "Freq_T")

#Remove unneeded columns
glam_score_table_M6 <- glam_score_table_M6[,c(1,3,6)]
#Create a percent column (percent of cells within each state with a good motif match)
glam_score_table_M6$pct <- glam_score_table_M6$Freq_T / (glam_score_table_M6$Freq_F + glam_score_table_M6$Freq_T)
#write.csv(glam_score_table_M4, "Motif/train/TFH/score_file/TRA_TFH_T65_motif1_M4_cln4_cells_across_M4.csv")
write.csv(glam_score_table_M6, "Motif_5mice/Train/Th1/Score/TRA_Th1_T65_motif3_M6_cln4_cells_across_M6_cells.csv")
getwd()
glam_score_table_M6$Freq_F %>% sum () # 2294
glam_score_table_M6$Freq_T %>% sum() # 30


# Perform clone-level statistical analysis

glam_stats_M2$combined_cdr3_nt[which(glam_stats_M2$Present == TRUE)] %>% table()


#         TRA:TGCTCAGCAAGTATGACTGGAGGCTATAAAGTGGTCTTT_TRB:TGTGCCAGCAGCAAGGTAAACACAGAAGTCTTCTTT 
 #                                                                                           4 
#TRA:TGTGCAGCAAGTCTCGATGGAGGCTATAAAGTGGTCTTT_TRB:TGTGCTAGCAGTTTATCCGGGACAGGGGACGCTGAGCAGTTCTTC 
    #                                                                                       30 
   #TRA:TGTGCAGTGAGCGGGACTGGAGGCTATAAAGTGGTCTTT_TRB:TGTGCCAGCAGGACTGGGGGGGCAGGGGACACCCAGTACTTT 
    #                                                                                       13 
     # TRA:TGTGCTCTGGGCCTGACTGGAGGCTATAAAGTGGTCTTT_TRB:TGCAGTGCAAGGGGGGCGGACACCGGGCAGCTCTACTTT 
       #                                                                                    21 

grep(pattern = "TRA:TGTGCAGCAAGTTGGGGGACTGGAGGCTATAAAGTGGTCTTT_TRB:TGTGCTAGCAGTTTTGCTGGGGGGGCTTCCTATGAACAGTACTTC", x = data_clones_by_mouse_AA_TRA_TRB_NOV$combined_cdr3_nt)
# 3
as.data.frame(data_clones_by_mouse_AA_TRA_TRB_NOV[1, ])
# 79, 0, 21
getwd()
glam_stats_M4$combined_cdr3_nt[which(glam_stats_M4$Present == TRUE)] %>% table()

        #  TRA:TGTGCAGCAAGCATGAGAACAGGAAACTACAAATACGTCTTT_TRB:TGTGCCAGCAGCGCCCAAACGCTGTATTTT 
         #                                                                                      5 
      #TRA:TGTGCAGCAAGTCTTAATACAGGAAACTACAAATACGTCTTT_TRB:TGTGCCTGGAATGGGGGGGGCTATGCTGAGCAGTTCTTC 
       #                                                                                       12 
#TRA:TGTGCAGCAAGTTGGGGGACTGGAGGCTATAAAGTGGTCTTT_TRB:TGTGCTAGCAGTTTTGCTGGGGGGGCTTCCTATGAACAGTACTTC 
 #                              

```

```{r}
# Th1 motif 4

#data_monocle <- readRDS("summary_5M_CCA_V2/data_monocle_2cell.RDS")
# Th1 motif 4
rm(glam2_scores)
glam2_scores <- read.table(file = "/media/runa/Elements/Jem_revision/Seurat/Motif_5mice/Train/Th1/Motif4/Th1_Motif4.glam2scan/glam2scan_scores_Motif4.txt", header = FALSE)
#We're only interested in the cell name and score columns
glam2_scores <- glam2_scores[,c(1,6)]
#Take the top score for each cell
glam2_scores <- aggregate(.~V1, glam2_scores, FUN=max) # here multiple cells can have
#more than one score, with aggregate command we are saying in those cases just keep the max
#score
rownames(glam2_scores) <- glam2_scores$V1
# to match cell names with monocle data add 1 in cell name
glam2_scores$V1 <-
  paste0(glam2_scores$V1, "1")
rownames(glam2_scores) <- glam2_scores$V1

#Add glam score to each cell within monocle pData
pData(data_monocle)$glam_score <- sapply(rownames(pData(data_monocle)), FUN =
function(x) glam2_scores[x,]$V6)

View(pData(data_monocle))


#Observe the glam score distribution, find a high score for the cells 40 looks good here; this
#scoring will vary from motif to motif
plot(density(pData(data_monocle)$glam_score[!is.na(pData(data_monocle)$glam_score)]),
main = "Motif score distribution_TRA_Th1_Motif4", xlab = "")
abline(v = 30, col = "Red")

#Make a true/false filtering column based on the glam score for plotting 
pData(data_monocle)$Present <- pData(data_monocle)$glam_score > 30

# Make a new dataframe to use for statistical analysis but not plotting the cells

rm(glam_stats_M2)
rm(glam_stats_M4)
rm(glam_stats_M6)

# Only include cells from mouse 2 and mouse 4, separately
glam_stats_M2 <- data_monocle$Present[which(data_monocle$mice == "M2")] %>% as.data.frame()
colnames(glam_stats_M2) <- "Present"
# Add other columns from data_monocle
glam_stats_M2$State <- data_monocle$State[which(data_monocle$mice == "M2")]
glam_stats_M2$combined_cdr3_nt <- data_monocle$combined_cdr3_nt[which(data_monocle$mice == "M2")]
glam_stats_M2$cellID <- rownames(data_monocle@phenoData)[which(data_monocle$mice == "M2")]
# Remove any rows with NA (cells without a motif score, incluing cells that don't meet definition of a clone)
glam_stats_M2 <- glam_stats_M2[!is.na(glam_stats_M2$Present), ]

glam_stats_M4 <- data_monocle$Present[which(data_monocle$mice == "M4")] %>% as.data.frame()
colnames(glam_stats_M4) <- "Present"
# Add other columns from data_monocle
glam_stats_M4$State <- data_monocle$State[which(data_monocle$mice == "M4")]
glam_stats_M4$combined_cdr3_nt <- data_monocle$combined_cdr3_nt[which(data_monocle$mice == "M4")]
glam_stats_M4$cellID <- rownames(data_monocle@phenoData)[which(data_monocle$mice == "M4")]
# Remove any rows with NA (cells without a motif score, incluing cells that don't meet definition of a clone)
glam_stats_M4 <- glam_stats_M4[!is.na(glam_stats_M4$Present), ]

glam_stats_M6 <- data_monocle$Present[which(data_monocle$mice == "M6")] %>% as.data.frame()
colnames(glam_stats_M6) <- "Present"
# Add other columns from data_monocle
glam_stats_M6$State <- data_monocle$State[which(data_monocle$mice == "M6")]
glam_stats_M6$combined_cdr3_nt <- data_monocle$combined_cdr3_nt[which(data_monocle$mice == "M6")]
glam_stats_M6$cellID <- rownames(data_monocle@phenoData)[which(data_monocle$mice == "M6")]
# Remove any rows with NA (cells without a motif score, incluing cells that don't meet definition of a clone)
glam_stats_M6 <- glam_stats_M6[!is.na(glam_stats_M6$Present), ]



#Make NA values false
pData(data_monocle)$Present[is.na(pData(data_monocle)$Present)] <- FALSE
plot_obj <- plot_complex_cell_trajectory(data_monocle, color_by = "Present")
plot_obj$data <- plot_obj$data[order(plot_obj$data$Present),]
plot_obj + scale_color_manual(values=c("grey", "red")) + ggtitle("TRA_Th1_motif4_30_across_M2M4M6'\n Score > 30 \n ")

png(file = "summary_6mice/TRA_TH1_motif1_M2M4M6_cln4.png", width = 8, height = 6, units = "in", res = 300)
plot_obj + scale_color_manual(values=c("grey", "red")) + ggtitle("TRA_TH1_motif6_M2M4M6.png'\n Score > 37 \n 'P value '")
#dev.off()

#Create a table detailing the number of cells in each monocle state that have a good motif match
# The line below was from Robert's analysis, I'm replacing it because there was an error
# glam_score_table <- as.data.frame(table(pData(data_no_6_mono_cds)$State, pData(data_no_6_mono_cds)$Present))
rm(glam_score_table_M2)
rm(glam_score_table_M4)
rm(glam_score_table_M6)


glam_score_table_M2 <- table(glam_stats_M2$State, glam_stats_M2$Present) %>% as.data.frame()
#Combine true and false rows
glam_score_table_M2 <- cbind(glam_score_table_M2[1:5,], glam_score_table_M2[6:10,])
#Create names that make more sense
colnames(glam_score_table_M2) <- c("State", "F", "Freq_F", "State2", "T", "Freq_T")

#Remove unneeded columns
glam_score_table_M2 <- glam_score_table_M2[,c(1,3,6)]
#Create a percent column (percent of cells within each state with a good motif match)
glam_score_table_M2$pct <- glam_score_table_M2$Freq_T / (glam_score_table_M2$Freq_F + glam_score_table_M2$Freq_T)
#write.csv(glam_score_table_M2, "Motif/train/TFH/score_file/TRA_TFH_T65_motif1_M2_cln4_cells_across_M2.csv")
write.csv(glam_score_table_M2, "Motif_5mice/Train/Th1/Score/TRA_Th1_T65_motif4_M2_cln4_cells_across_M2_cells.csv")
getwd()
glam_score_table_M2$Freq_F %>% sum () # 738
glam_score_table_M2$Freq_T %>% sum() # 125

#Create a table detailing the number of cells in each monocle state that have a good motif match
# The line below was from Robert's analysis, I'm replacing it because there was an error
# glam_score_table <- as.data.frame(table(pData(data_no_6_mono_cds)$State, pData(data_no_6_mono_cds)$Present))

glam_score_table_M4 <- table(glam_stats_M4$State, glam_stats_M4$Present) %>% as.data.frame()
#Combine true and false rows
glam_score_table_M4 <- cbind(glam_score_table_M4[1:5,], glam_score_table_M4[6:10,]) ## this # depends on the # of monocle states
#Create names that make more sense
colnames(glam_score_table_M4) <- c("State", "F", "Freq_F", "State2", "T", "Freq_T")

#Remove unneeded columns
glam_score_table_M4 <- glam_score_table_M4[,c(1,3,6)]
#Create a percent column (percent of cells within each state with a good motif match)
glam_score_table_M4$pct <- glam_score_table_M4$Freq_T / (glam_score_table_M4$Freq_F + glam_score_table_M4$Freq_T)
#write.csv(glam_score_table_M4, "Motif/train/TFH/score_file/TRA_TFH_T65_motif1_M4_cln4_cells_across_M4.csv")
write.csv(glam_score_table_M4, "Motif_5mice/Train/Th1/Score/TRA_Th1_T65_motif4_M4_cln4_cells_across_M4_cells.csv")
getwd()
glam_score_table_M4$Freq_F %>% sum () # 2667
glam_score_table_M4$Freq_T %>% sum() # 551


#Create a table detailing the number of cells in each monocle state that have a good motif match
# The line below was from Robert's analysis, I'm replacing it because there was an error
# glam_score_table <- as.data.frame(table(pData(data_no_6_mono_cds)$State, pData(data_no_6_mono_cds)$Present))

glam_score_table_M6 <- table(glam_stats_M6$State, glam_stats_M6$Present) %>% as.data.frame()
#Combine true and false rows
glam_score_table_M6 <- cbind(glam_score_table_M6[1:5,], glam_score_table_M6[6:10,]) ## this # depends on the # of monocle states
#Create names that make more sense
colnames(glam_score_table_M6) <- c("State", "F", "Freq_F", "State2", "T", "Freq_T")

#Remove unneeded columns
glam_score_table_M6 <- glam_score_table_M6[,c(1,3,6)]
#Create a percent column (percent of cells within each state with a good motif match)
glam_score_table_M6$pct <- glam_score_table_M6$Freq_T / (glam_score_table_M6$Freq_F + glam_score_table_M6$Freq_T)
#write.csv(glam_score_table_M4, "Motif/train/TFH/score_file/TRA_TFH_T65_motif1_M4_cln4_cells_across_M4.csv")
write.csv(glam_score_table_M6, "Motif_5mice/Train/Th1/Score/TRA_Th1_T65_motif4_M6_cln4_cells_across_M6_cells.csv")
getwd()
glam_score_table_M6$Freq_F %>% sum () # 2257
glam_score_table_M6$Freq_T %>% sum() # 67


# Perform clone-level statistical analysis

glam_stats_M2$combined_cdr3_nt[which(glam_stats_M2$Present == TRUE)] %>% table()


#         TRA:TGCTCAGCAAGTATGACTGGAGGCTATAAAGTGGTCTTT_TRB:TGTGCCAGCAGCAAGGTAAACACAGAAGTCTTCTTT 
 #                                                                                           4 
#TRA:TGTGCAGCAAGTCTCGATGGAGGCTATAAAGTGGTCTTT_TRB:TGTGCTAGCAGTTTATCCGGGACAGGGGACGCTGAGCAGTTCTTC 
    #                                                                                       30 
   #TRA:TGTGCAGTGAGCGGGACTGGAGGCTATAAAGTGGTCTTT_TRB:TGTGCCAGCAGGACTGGGGGGGCAGGGGACACCCAGTACTTT 
    #                                                                                       13 
     # TRA:TGTGCTCTGGGCCTGACTGGAGGCTATAAAGTGGTCTTT_TRB:TGCAGTGCAAGGGGGGCGGACACCGGGCAGCTCTACTTT 
       #                                                                                    21 

grep(pattern = "TRA:TGTGCAGCAAGTTGGGGGACTGGAGGCTATAAAGTGGTCTTT_TRB:TGTGCTAGCAGTTTTGCTGGGGGGGCTTCCTATGAACAGTACTTC", x = data_clones_by_mouse_AA_TRA_TRB_NOV$combined_cdr3_nt)
# 3
as.data.frame(data_clones_by_mouse_AA_TRA_TRB_NOV[1, ])
# 79, 0, 21
getwd()
glam_stats_M4$combined_cdr3_nt[which(glam_stats_M4$Present == TRUE)] %>% table()

        #  TRA:TGTGCAGCAAGCATGAGAACAGGAAACTACAAATACGTCTTT_TRB:TGTGCCAGCAGCGCCCAAACGCTGTATTTT 
         #                                                                                      5 
      #TRA:TGTGCAGCAAGTCTTAATACAGGAAACTACAAATACGTCTTT_TRB:TGTGCCTGGAATGGGGGGGGCTATGCTGAGCAGTTCTTC 
       #                                                                                       12 
#TRA:TGTGCAGCAAGTTGGGGGACTGGAGGCTATAAAGTGGTCTTT_TRB:TGTGCTAGCAGTTTTGCTGGGGGGGCTTCCTATGAACAGTACTTC 
 #                              

```
```{r}
# Th1 motif 5

#data_monocle <- readRDS("summary_5M_CCA_V2/data_monocle_2cell.RDS")
# Th1 motif 5
rm(glam2_scores)
glam2_scores <- read.table(file = "/media/runa/Elements/Jem_revision/Seurat/Motif_5mice/Train/Th1/Motif5/Th1_Motif5.glam2scan/glam2scan_scores_Motif5.txt", header = FALSE)
#We're only interested in the cell name and score columns
glam2_scores <- glam2_scores[,c(1,6)]
#Take the top score for each cell
glam2_scores <- aggregate(.~V1, glam2_scores, FUN=max) # here multiple cells can have
#more than one score, with aggregate command we are saying in those cases just keep the max
#score
rownames(glam2_scores) <- glam2_scores$V1
# to match cell names with monocle data add 1 in cell name
glam2_scores$V1 <-
  paste0(glam2_scores$V1, "1")
rownames(glam2_scores) <- glam2_scores$V1

#Add glam score to each cell within monocle pData
pData(data_monocle)$glam_score <- sapply(rownames(pData(data_monocle)), FUN =
function(x) glam2_scores[x,]$V6)

View(pData(data_monocle))


#Observe the glam score distribution, find a high score for the cells 40 looks good here; this
#scoring will vary from motif to motif
plot(density(pData(data_monocle)$glam_score[!is.na(pData(data_monocle)$glam_score)]),
main = "Motif score distribution_TRA_Th1_Motif5", xlab = "")
abline(v = 42, col = "Red")

#Make a true/false filtering column based on the glam score for plotting 
pData(data_monocle)$Present <- pData(data_monocle)$glam_score > 42

# Make a new dataframe to use for statistical analysis but not plotting the cells

rm(glam_stats_M2)
rm(glam_stats_M4)
rm(glam_stats_M6)

# Only include cells from mouse 2 and mouse 4, separately
glam_stats_M2 <- data_monocle$Present[which(data_monocle$mice == "M2")] %>% as.data.frame()
colnames(glam_stats_M2) <- "Present"
# Add other columns from data_monocle
glam_stats_M2$State <- data_monocle$State[which(data_monocle$mice == "M2")]
glam_stats_M2$combined_cdr3_nt <- data_monocle$combined_cdr3_nt[which(data_monocle$mice == "M2")]
glam_stats_M2$cellID <- rownames(data_monocle@phenoData)[which(data_monocle$mice == "M2")]
# Remove any rows with NA (cells without a motif score, incluing cells that don't meet definition of a clone)
glam_stats_M2 <- glam_stats_M2[!is.na(glam_stats_M2$Present), ]

glam_stats_M4 <- data_monocle$Present[which(data_monocle$mice == "M4")] %>% as.data.frame()
colnames(glam_stats_M4) <- "Present"
# Add other columns from data_monocle
glam_stats_M4$State <- data_monocle$State[which(data_monocle$mice == "M4")]
glam_stats_M4$combined_cdr3_nt <- data_monocle$combined_cdr3_nt[which(data_monocle$mice == "M4")]
glam_stats_M4$cellID <- rownames(data_monocle@phenoData)[which(data_monocle$mice == "M4")]
# Remove any rows with NA (cells without a motif score, incluing cells that don't meet definition of a clone)
glam_stats_M4 <- glam_stats_M4[!is.na(glam_stats_M4$Present), ]

glam_stats_M6 <- data_monocle$Present[which(data_monocle$mice == "M6")] %>% as.data.frame()
colnames(glam_stats_M6) <- "Present"
# Add other columns from data_monocle
glam_stats_M6$State <- data_monocle$State[which(data_monocle$mice == "M6")]
glam_stats_M6$combined_cdr3_nt <- data_monocle$combined_cdr3_nt[which(data_monocle$mice == "M6")]
glam_stats_M6$cellID <- rownames(data_monocle@phenoData)[which(data_monocle$mice == "M6")]
# Remove any rows with NA (cells without a motif score, incluing cells that don't meet definition of a clone)
glam_stats_M6 <- glam_stats_M6[!is.na(glam_stats_M6$Present), ]



#Make NA values false
pData(data_monocle)$Present[is.na(pData(data_monocle)$Present)] <- FALSE
plot_obj <- plot_complex_cell_trajectory(data_monocle, color_by = "Present")
plot_obj$data <- plot_obj$data[order(plot_obj$data$Present),]
plot_obj + scale_color_manual(values=c("grey", "red")) + ggtitle("TRA_Th1_motif5_42_across_M2M4M6'\n Score > 42 \n ")

png(file = "summary_6mice/TRA_TH1_motif1_M2M4M6_cln4.png", width = 8, height = 6, units = "in", res = 300)
plot_obj + scale_color_manual(values=c("grey", "red")) + ggtitle("TRA_TH1_motif6_M2M4M6.png'\n Score > 37 \n 'P value '")
#dev.off()

#Create a table detailing the number of cells in each monocle state that have a good motif match
# The line below was from Robert's analysis, I'm replacing it because there was an error
# glam_score_table <- as.data.frame(table(pData(data_no_6_mono_cds)$State, pData(data_no_6_mono_cds)$Present))
rm(glam_score_table_M2)
rm(glam_score_table_M4)
rm(glam_score_table_M6)


glam_score_table_M2 <- table(glam_stats_M2$State, glam_stats_M2$Present) %>% as.data.frame()
#Combine true and false rows
glam_score_table_M2 <- cbind(glam_score_table_M2[1:5,], glam_score_table_M2[6:10,])
#Create names that make more sense
colnames(glam_score_table_M2) <- c("State", "F", "Freq_F", "State2", "T", "Freq_T")

#Remove unneeded columns
glam_score_table_M2 <- glam_score_table_M2[,c(1,3,6)]
#Create a percent column (percent of cells within each state with a good motif match)
glam_score_table_M2$pct <- glam_score_table_M2$Freq_T / (glam_score_table_M2$Freq_F + glam_score_table_M2$Freq_T)
#write.csv(glam_score_table_M2, "Motif/train/TFH/score_file/TRA_TFH_T65_motif1_M2_cln4_cells_across_M2.csv")
write.csv(glam_score_table_M2, "Motif_5mice/Train/Th1/Score/TRA_Th1_T65_motif5_M2_cln4_cells_across_M2_cells.csv")
getwd()
glam_score_table_M2$Freq_F %>% sum () # 848
glam_score_table_M2$Freq_T %>% sum() # 15

#Create a table detailing the number of cells in each monocle state that have a good motif match
# The line below was from Robert's analysis, I'm replacing it because there was an error
# glam_score_table <- as.data.frame(table(pData(data_no_6_mono_cds)$State, pData(data_no_6_mono_cds)$Present))

glam_score_table_M4 <- table(glam_stats_M4$State, glam_stats_M4$Present) %>% as.data.frame()
#Combine true and false rows
glam_score_table_M4 <- cbind(glam_score_table_M4[1:5,], glam_score_table_M4[6:10,]) ## this # depends on the # of monocle states
#Create names that make more sense
colnames(glam_score_table_M4) <- c("State", "F", "Freq_F", "State2", "T", "Freq_T")

#Remove unneeded columns
glam_score_table_M4 <- glam_score_table_M4[,c(1,3,6)]
#Create a percent column (percent of cells within each state with a good motif match)
glam_score_table_M4$pct <- glam_score_table_M4$Freq_T / (glam_score_table_M4$Freq_F + glam_score_table_M4$Freq_T)
#write.csv(glam_score_table_M4, "Motif/train/TFH/score_file/TRA_TFH_T65_motif1_M4_cln4_cells_across_M4.csv")
write.csv(glam_score_table_M4, "Motif_5mice/Train/Th1/Score/TRA_Th1_T65_motif5_M4_cln4_cells_across_M4_cells.csv")
getwd()
glam_score_table_M4$Freq_F %>% sum () # 2824
glam_score_table_M4$Freq_T %>% sum() # 394


#Create a table detailing the number of cells in each monocle state that have a good motif match
# The line below was from Robert's analysis, I'm replacing it because there was an error
# glam_score_table <- as.data.frame(table(pData(data_no_6_mono_cds)$State, pData(data_no_6_mono_cds)$Present))

glam_score_table_M6 <- table(glam_stats_M6$State, glam_stats_M6$Present) %>% as.data.frame()
#Combine true and false rows
glam_score_table_M6 <- cbind(glam_score_table_M6[1:5,], glam_score_table_M6[6:10,]) ## this # depends on the # of monocle states
#Create names that make more sense
colnames(glam_score_table_M6) <- c("State", "F", "Freq_F", "State2", "T", "Freq_T")

#Remove unneeded columns
glam_score_table_M6 <- glam_score_table_M6[,c(1,3,6)]
#Create a percent column (percent of cells within each state with a good motif match)
glam_score_table_M6$pct <- glam_score_table_M6$Freq_T / (glam_score_table_M6$Freq_F + glam_score_table_M6$Freq_T)
#write.csv(glam_score_table_M4, "Motif/train/TFH/score_file/TRA_TFH_T65_motif1_M4_cln4_cells_across_M4.csv")
write.csv(glam_score_table_M6, "Motif_5mice/Train/Th1/Score/TRA_Th1_T65_motif5_M6_cln4_cells_across_M6_cells.csv")
getwd()
glam_score_table_M6$Freq_F %>% sum () # 2299
glam_score_table_M6$Freq_T %>% sum() # 25


# Perform clone-level statistical analysis

glam_stats_M2$combined_cdr3_nt[which(glam_stats_M2$Present == TRUE)] %>% table()


#         TRA:TGCTCAGCAAGTATGACTGGAGGCTATAAAGTGGTCTTT_TRB:TGTGCCAGCAGCAAGGTAAACACAGAAGTCTTCTTT 
 #                                                                                           4 
#TRA:TGTGCAGCAAGTCTCGATGGAGGCTATAAAGTGGTCTTT_TRB:TGTGCTAGCAGTTTATCCGGGACAGGGGACGCTGAGCAGTTCTTC 
    #                                                                                       30 
   #TRA:TGTGCAGTGAGCGGGACTGGAGGCTATAAAGTGGTCTTT_TRB:TGTGCCAGCAGGACTGGGGGGGCAGGGGACACCCAGTACTTT 
    #                                                                                       13 
     # TRA:TGTGCTCTGGGCCTGACTGGAGGCTATAAAGTGGTCTTT_TRB:TGCAGTGCAAGGGGGGCGGACACCGGGCAGCTCTACTTT 
       #                                                                                    21 

grep(pattern = "TRA:TGTGCAGCAAGTTGGGGGACTGGAGGCTATAAAGTGGTCTTT_TRB:TGTGCTAGCAGTTTTGCTGGGGGGGCTTCCTATGAACAGTACTTC", x = data_clones_by_mouse_AA_TRA_TRB_NOV$combined_cdr3_nt)
# 3
as.data.frame(data_clones_by_mouse_AA_TRA_TRB_NOV[1, ])
# 79, 0, 21
getwd()
glam_stats_M4$combined_cdr3_nt[which(glam_stats_M4$Present == TRUE)] %>% table()

        #  TRA:TGTGCAGCAAGCATGAGAACAGGAAACTACAAATACGTCTTT_TRB:TGTGCCAGCAGCGCCCAAACGCTGTATTTT 
         #                                                                                      5 
      #TRA:TGTGCAGCAAGTCTTAATACAGGAAACTACAAATACGTCTTT_TRB:TGTGCCTGGAATGGGGGGGGCTATGCTGAGCAGTTCTTC 
       #                                                                                       12 
#TRA:TGTGCAGCAAGTTGGGGGACTGGAGGCTATAAAGTGGTCTTT_TRB:TGTGCTAGCAGTTTTGCTGGGGGGGCTTCCTATGAACAGTACTTC 
 #                              

```


```{r}

# Th1 motif 6

data_monocle <- readRDS("summary_5M_CCA_V2/data_monocle_2cell.RDS")
# Th1 motif 6
rm(glam2_scores)
glam2_scores <- read.table(file = "/media/runa/Elements/Jem_revision/Seurat/Motif_5mice/Train/Th1/Motif6/Th1_Motif6.glam2scan/glam2scan_scores_Motif6.txt", header = FALSE)
#We're only interested in the cell name and score columns
glam2_scores <- glam2_scores[,c(1,6)]
#Take the top score for each cell
glam2_scores <- aggregate(.~V1, glam2_scores, FUN=max) # here multiple cells can have
#more than one score, with aggregate command we are saying in those cases just keep the max
#score
rownames(glam2_scores) <- glam2_scores$V1
# to match cell names with monocle data add 1 in cell name
glam2_scores$V1 <-
  paste0(glam2_scores$V1, "1")
rownames(glam2_scores) <- glam2_scores$V1

#Add glam score to each cell within monocle pData
pData(data_monocle)$glam_score <- sapply(rownames(pData(data_monocle)), FUN =
function(x) glam2_scores[x,]$V6)

View(pData(data_monocle))


#Observe the glam score distribution, find a high score for the cells 40 looks good here; this
#scoring will vary from motif to motif

pdf("summary_5mice/Figure4/Th1_Motif6_Train_mice_score.pdf", width = 8, height = 6)
plot(density(pData(data_monocle)$glam_score[!is.na(pData(data_monocle)$glam_score)]),
main = "Motif score distribution_TRA_Th1_Motif6", xlab = "")
abline(v = 30, col = "Red")
dev.off()

#Make a true/false filtering column based on the glam score for plotting 
pData(data_monocle)$Present <- pData(data_monocle)$glam_score > 30

# Make a new dataframe to use for statistical analysis but not plotting the cells

rm(glam_stats_M2)
rm(glam_stats_M4)
rm(glam_stats_M6)

# Only include cells from mouse 2 and mouse 4, separately
glam_stats_M2 <- data_monocle$Present[which(data_monocle$mice == "M2")] %>% as.data.frame()
colnames(glam_stats_M2) <- "Present"
# Add other columns from data_monocle
glam_stats_M2$State <- data_monocle$State[which(data_monocle$mice == "M2")]
glam_stats_M2$combined_cdr3_nt <- data_monocle$combined_cdr3_nt[which(data_monocle$mice == "M2")]
glam_stats_M2$cellID <- rownames(data_monocle@phenoData)[which(data_monocle$mice == "M2")]
# Remove any rows with NA (cells without a motif score, incluing cells that don't meet definition of a clone)
glam_stats_M2 <- glam_stats_M2[!is.na(glam_stats_M2$Present), ]

glam_stats_M4 <- data_monocle$Present[which(data_monocle$mice == "M4")] %>% as.data.frame()
colnames(glam_stats_M4) <- "Present"
# Add other columns from data_monocle
glam_stats_M4$State <- data_monocle$State[which(data_monocle$mice == "M4")]
glam_stats_M4$combined_cdr3_nt <- data_monocle$combined_cdr3_nt[which(data_monocle$mice == "M4")]
glam_stats_M4$cellID <- rownames(data_monocle@phenoData)[which(data_monocle$mice == "M4")]
# Remove any rows with NA (cells without a motif score, incluing cells that don't meet definition of a clone)
glam_stats_M4 <- glam_stats_M4[!is.na(glam_stats_M4$Present), ]

glam_stats_M6 <- data_monocle$Present[which(data_monocle$mice == "M6")] %>% as.data.frame()
colnames(glam_stats_M6) <- "Present"
# Add other columns from data_monocle
glam_stats_M6$State <- data_monocle$State[which(data_monocle$mice == "M6")]
glam_stats_M6$combined_cdr3_nt <- data_monocle$combined_cdr3_nt[which(data_monocle$mice == "M6")]
glam_stats_M6$cellID <- rownames(data_monocle@phenoData)[which(data_monocle$mice == "M6")]
# Remove any rows with NA (cells without a motif score, incluing cells that don't meet definition of a clone)
glam_stats_M6 <- glam_stats_M6[!is.na(glam_stats_M6$Present), ]



#Make NA values false
pData(data_monocle)$Present[is.na(pData(data_monocle)$Present)] <- FALSE
plot_obj <- plot_complex_cell_trajectory(data_monocle, color_by = "Present", cell_size = 2)
plot_obj$data <- plot_obj$data[order(plot_obj$data$Present),]
plot_obj + scale_color_manual(values=c("grey", "red")) + ggtitle("TRA_Th1_motif6_30_across_M2M4M6'\n Score > 30 \n ")

pdf("summary_5mice/Figure4/Th1_Motif6_Train_mice_TREE.pdf", width = 8, height = 6, useDingbats = FALSE)
plot_obj + scale_color_manual(values=c("grey", "red")) + ggtitle("TRA_TH1_motif6_M2M4M6.png'\n Score > 30 \n 'P value 0.0007'")
dev.off()

#Create a table detailing the number of cells in each monocle state that have a good motif match
# The line below was from Robert's analysis, I'm replacing it because there was an error
# glam_score_table <- as.data.frame(table(pData(data_no_6_mono_cds)$State, pData(data_no_6_mono_cds)$Present))
rm(glam_score_table_M2)
rm(glam_score_table_M4)
rm(glam_score_table_M6)


glam_score_table_M2 <- table(glam_stats_M2$State, glam_stats_M2$Present) %>% as.data.frame()
#Combine true and false rows
glam_score_table_M2 <- cbind(glam_score_table_M2[1:5,], glam_score_table_M2[6:10,])
#Create names that make more sense
colnames(glam_score_table_M2) <- c("State", "F", "Freq_F", "State2", "T", "Freq_T")

#Remove unneeded columns
glam_score_table_M2 <- glam_score_table_M2[,c(1,3,6)]
#Create a percent column (percent of cells within each state with a good motif match)
glam_score_table_M2$pct <- glam_score_table_M2$Freq_T / (glam_score_table_M2$Freq_F + glam_score_table_M2$Freq_T)
#write.csv(glam_score_table_M2, "Motif/train/TFH/score_file/TRA_TFH_T65_motif1_M2_cln4_cells_across_M2.csv")
write.csv(glam_score_table_M2, "Motif_5mice/Train/Th1/Score/TRA_Th1_T65_motif6_M2_cln4_cells_across_M2_cells.csv")
getwd()
glam_score_table_M2$Freq_F %>% sum () # 830
glam_score_table_M2$Freq_T %>% sum() # 33

glam_stats_M2$combined_cdr3_nt[which(glam_stats_M2$Present == TRUE)] %>% table()

#TRA:TGCGCAGTCTCTTCTAATACAGGAAACTACAAATACGTCTTT_TRB:TGTGCCAGCAGCCACGGGCAGCTCTACTTT TRA:TGTGCTCTGAGTGATCACAATACCAACAAAGTCGTCTTT_TRB:TGTGCCAGCTCGCTACACTCCGACTACACCTTC 
 #                                                                              15                                                                                11 
       #  TRA:TGTGCTGCAAGTGAGAGGGGGGGAAACTACAAATACGTCTTT_TRB:TGTGCCAGCAGCAACTCCTTT 
        #                                                                        7 

#Create a table detailing the number of cells in each monocle state that have a good motif match
# The line below was from Robert's analysis, I'm replacing it because there was an error
# glam_score_table <- as.data.frame(table(pData(data_no_6_mono_cds)$State, pData(data_no_6_mono_cds)$Present))

glam_score_table_M4 <- table(glam_stats_M4$State, glam_stats_M4$Present) %>% as.data.frame()
#Combine true and false rows
glam_score_table_M4 <- cbind(glam_score_table_M4[1:5,], glam_score_table_M4[6:10,]) ## this # depends on the # of monocle states
#Create names that make more sense
colnames(glam_score_table_M4) <- c("State", "F", "Freq_F", "State2", "T", "Freq_T")

#Remove unneeded columns
glam_score_table_M4 <- glam_score_table_M4[,c(1,3,6)]
#Create a percent column (percent of cells within each state with a good motif match)
glam_score_table_M4$pct <- glam_score_table_M4$Freq_T / (glam_score_table_M4$Freq_F + glam_score_table_M4$Freq_T)
#write.csv(glam_score_table_M4, "Motif/train/TFH/score_file/TRA_TFH_T65_motif1_M4_cln4_cells_across_M4.csv")
write.csv(glam_score_table_M4, "Motif_5mice/Train/Th1/Score/TRA_Th1_T65_motif6_M4_cln4_cells_across_M4_cells.csv")
getwd()
glam_score_table_M4$Freq_F %>% sum () # 3126
glam_score_table_M4$Freq_T %>% sum() # 92

glam_stats_M4$combined_cdr3_nt[which(glam_stats_M4$Present == TRUE)] %>% table()

#TRA:TGTGCAGCAAGCATAGGAAACTACAAATACGTCTTT_TRB:TGTGCCAGCGGTGATGCAGGGACTGGGGGGGATGAACAGTACTTC    TRA:TGTGCAGCAAGCATAGGAAACTACAAATACGTCTTT_TRB:TGTGCCAGCGGTGATGCGGGGGACCAAGACACCCAGTACTTT 
 #                                                                                       60                                                                                         15 
  #       TRA:TGTGCAGCAAGCATGAGAACAGGAAACTACAAATACGTCTTT_TRB:TGTGCCAGCAGCGCCCAAACGCTGTATTTT TRA:TGTGCAGCAAGTCTTAATACAGGAAACTACAAATACGTCTTT_TRB:TGTGCCTGGAATGGGGGGGGCTATGCTGAGCAGTTCTTC 
   #                                                                                      5                                                                                         12 



#Create a table detailing the number of cells in each monocle state that have a good motif match
# The line below was from Robert's analysis, I'm replacing it because there was an error
# glam_score_table <- as.data.frame(table(pData(data_no_6_mono_cds)$State, pData(data_no_6_mono_cds)$Present))

glam_score_table_M6 <- table(glam_stats_M6$State, glam_stats_M6$Present) %>% as.data.frame()
#Combine true and false rows
glam_score_table_M6 <- cbind(glam_score_table_M6[1:5,], glam_score_table_M6[6:10,]) ## this # depends on the # of monocle states
#Create names that make more sense
colnames(glam_score_table_M6) <- c("State", "F", "Freq_F", "State2", "T", "Freq_T")

#Remove unneeded columns
glam_score_table_M6 <- glam_score_table_M6[,c(1,3,6)]
#Create a percent column (percent of cells within each state with a good motif match)
glam_score_table_M6$pct <- glam_score_table_M6$Freq_T / (glam_score_table_M6$Freq_F + glam_score_table_M6$Freq_T)
#write.csv(glam_score_table_M4, "Motif/train/TFH/score_file/TRA_TFH_T65_motif1_M4_cln4_cells_across_M4.csv")
write.csv(glam_score_table_M6, "Motif_5mice/Train/Th1/Score/TRA_Th1_T65_motif6_M6_cln4_cells_across_M6_cells.csv")
getwd()
glam_score_table_M6$Freq_F %>% sum () # 2311
glam_score_table_M6$Freq_T %>% sum() # 13


glam_stats_M6$combined_cdr3_nt[which(glam_stats_M6$Present == TRUE)] %>% table()

#  TRA:TGTGCAGCAAGCATGAGGACAGGAAACTACAAATACGTCTTT_TRB:TGTGCCTGGAGTCAAGGGGCCGAAAGATTATTTTTC 
  #                                                                                          9 
#TRA:TGTGCAGCAAGTGAGGATACAGGAAACTACAAATACGTCTTT_TRB:TGTGCCAGCAGTAGGGGGGACAGGGGTACTGAGCAGTTCTTC 
 #                                                                                           4 

```

```{r}
# Th1 motif 7

#data_monocle <- readRDS("summary_5M_CCA_V2/data_monocle_2cell.RDS")
# Th1 motif 7
rm(glam2_scores)
glam2_scores <- read.table(file = "/media/runa/Elements/Jem_revision/Seurat/Motif_5mice/Train/Th1/Motif7/Th1_Motif7.glam2scan/glam2scan_scores_Motif7.txt", header = FALSE)
#We're only interested in the cell name and score columns
glam2_scores <- glam2_scores[,c(1,6)]
#Take the top score for each cell
glam2_scores <- aggregate(.~V1, glam2_scores, FUN=max) # here multiple cells can have
#more than one score, with aggregate command we are saying in those cases just keep the max
#score
rownames(glam2_scores) <- glam2_scores$V1
# to match cell names with monocle data add 1 in cell name
glam2_scores$V1 <-
  paste0(glam2_scores$V1, "1")
rownames(glam2_scores) <- glam2_scores$V1

#Add glam score to each cell within monocle pData
pData(data_monocle)$glam_score <- sapply(rownames(pData(data_monocle)), FUN =
function(x) glam2_scores[x,]$V6)

View(pData(data_monocle))


#Observe the glam score distribution, find a high score for the cells 40 looks good here; this
#scoring will vary from motif to motif

pdf("summary_5mice/Figure4/Th1_Motif7_Train_mice_score.pdf", width = 8, height = 6)
plot(density(pData(data_monocle)$glam_score[!is.na(pData(data_monocle)$glam_score)]),
main = "Motif score distribution_TRA_Th1_Motif7", xlab = "")
abline(v = 30, col = "Red")
dev.off()

#Make a true/false filtering column based on the glam score for plotting 
pData(data_monocle)$Present <- pData(data_monocle)$glam_score > 30

# Make a new dataframe to use for statistical analysis but not plotting the cells

rm(glam_stats_M2)
rm(glam_stats_M4)
rm(glam_stats_M6)

# Only include cells from mouse 2 and mouse 4, separately
glam_stats_M2 <- data_monocle$Present[which(data_monocle$mice == "M2")] %>% as.data.frame()
colnames(glam_stats_M2) <- "Present"
# Add other columns from data_monocle
glam_stats_M2$State <- data_monocle$State[which(data_monocle$mice == "M2")]
glam_stats_M2$combined_cdr3_nt <- data_monocle$combined_cdr3_nt[which(data_monocle$mice == "M2")]
glam_stats_M2$cellID <- rownames(data_monocle@phenoData)[which(data_monocle$mice == "M2")]
# Remove any rows with NA (cells without a motif score, incluing cells that don't meet definition of a clone)
glam_stats_M2 <- glam_stats_M2[!is.na(glam_stats_M2$Present), ]

glam_stats_M4 <- data_monocle$Present[which(data_monocle$mice == "M4")] %>% as.data.frame()
colnames(glam_stats_M4) <- "Present"
# Add other columns from data_monocle
glam_stats_M4$State <- data_monocle$State[which(data_monocle$mice == "M4")]
glam_stats_M4$combined_cdr3_nt <- data_monocle$combined_cdr3_nt[which(data_monocle$mice == "M4")]
glam_stats_M4$cellID <- rownames(data_monocle@phenoData)[which(data_monocle$mice == "M4")]
# Remove any rows with NA (cells without a motif score, incluing cells that don't meet definition of a clone)
glam_stats_M4 <- glam_stats_M4[!is.na(glam_stats_M4$Present), ]

glam_stats_M6 <- data_monocle$Present[which(data_monocle$mice == "M6")] %>% as.data.frame()
colnames(glam_stats_M6) <- "Present"
# Add other columns from data_monocle
glam_stats_M6$State <- data_monocle$State[which(data_monocle$mice == "M6")]
glam_stats_M6$combined_cdr3_nt <- data_monocle$combined_cdr3_nt[which(data_monocle$mice == "M6")]
glam_stats_M6$cellID <- rownames(data_monocle@phenoData)[which(data_monocle$mice == "M6")]
# Remove any rows with NA (cells without a motif score, incluing cells that don't meet definition of a clone)
glam_stats_M6 <- glam_stats_M6[!is.na(glam_stats_M6$Present), ]



#Make NA values false
pData(data_monocle)$Present[is.na(pData(data_monocle)$Present)] <- FALSE
plot_obj <- plot_complex_cell_trajectory(data_monocle, color_by = "Present", cell_size = 2)
plot_obj$data <- plot_obj$data[order(plot_obj$data$Present),]
plot_obj + scale_color_manual(values=c("grey", "red")) + ggtitle("TRA_Th1_motif7_30_across_M2M4M6'\n Score > 30 \n ")

pdf("summary_5mice/Figure4/Th1_Motif7_Train_mice_TREE.pdf", width = 8, height = 6, useDingbats = FALSE)
plot_obj + scale_color_manual(values=c("grey", "red")) + ggtitle("TRA_TH1_motif7_M2M4M6.png'\n Score > 30 \n 'P value 0.04'")
dev.off()

#Create a table detailing the number of cells in each monocle state that have a good motif match
# The line below was from Robert's analysis, I'm replacing it because there was an error
# glam_score_table <- as.data.frame(table(pData(data_no_6_mono_cds)$State, pData(data_no_6_mono_cds)$Present))
rm(glam_score_table_M2)
rm(glam_score_table_M4)
rm(glam_score_table_M6)


glam_score_table_M2 <- table(glam_stats_M2$State, glam_stats_M2$Present) %>% as.data.frame()
#Combine true and false rows
glam_score_table_M2 <- cbind(glam_score_table_M2[1:5,], glam_score_table_M2[6:10,])
#Create names that make more sense
colnames(glam_score_table_M2) <- c("State", "F", "Freq_F", "State2", "T", "Freq_T")

#Remove unneeded columns
glam_score_table_M2 <- glam_score_table_M2[,c(1,3,6)]
#Create a percent column (percent of cells within each state with a good motif match)
glam_score_table_M2$pct <- glam_score_table_M2$Freq_T / (glam_score_table_M2$Freq_F + glam_score_table_M2$Freq_T)
#write.csv(glam_score_table_M2, "Motif/train/TFH/score_file/TRA_TFH_T65_motif1_M2_cln4_cells_across_M2.csv")
write.csv(glam_score_table_M2, "Motif_5mice/Train/Th1/Score/TRA_Th1_T65_motif7_M2_cln4_cells_across_M2_cells.csv")
getwd()
glam_score_table_M2$Freq_F %>% sum () # 858
glam_score_table_M2$Freq_T %>% sum() # 5

glam_stats_M2$combined_cdr3_nt[which(glam_stats_M2$Present == TRUE)] %>% table()

#TRA:TGTGCTGCTGAGAATAGCAACTATCAGTTGATCTGG_TRB:TGTGCCTGGATGGGGGTGAACACCCAGTACTTT 
 #                                                                            5

#Create a table detailing the number of cells in each monocle state that have a good motif match
# The line below was from Robert's analysis, I'm replacing it because there was an error
# glam_score_table <- as.data.frame(table(pData(data_no_6_mono_cds)$State, pData(data_no_6_mono_cds)$Present))

glam_score_table_M4 <- table(glam_stats_M4$State, glam_stats_M4$Present) %>% as.data.frame()
#Combine true and false rows
glam_score_table_M4 <- cbind(glam_score_table_M4[1:5,], glam_score_table_M4[6:10,]) ## this # depends on the # of monocle states
#Create names that make more sense
colnames(glam_score_table_M4) <- c("State", "F", "Freq_F", "State2", "T", "Freq_T")

#Remove unneeded columns
glam_score_table_M4 <- glam_score_table_M4[,c(1,3,6)]
#Create a percent column (percent of cells within each state with a good motif match)
glam_score_table_M4$pct <- glam_score_table_M4$Freq_T / (glam_score_table_M4$Freq_F + glam_score_table_M4$Freq_T)
#write.csv(glam_score_table_M4, "Motif/train/TFH/score_file/TRA_TFH_T65_motif1_M4_cln4_cells_across_M4.csv")
write.csv(glam_score_table_M4, "Motif_5mice/Train/Th1/Score/TRA_Th1_T65_motif7_M4_cln4_cells_across_M4_cells.csv")
getwd()
glam_score_table_M4$Freq_F %>% sum () # 3218
glam_score_table_M4$Freq_T %>% sum() # NA


#Create a table detailing the number of cells in each monocle state that have a good motif match
# The line below was from Robert's analysis, I'm replacing it because there was an error
# glam_score_table <- as.data.frame(table(pData(data_no_6_mono_cds)$State, pData(data_no_6_mono_cds)$Present))

glam_score_table_M6 <- table(glam_stats_M6$State, glam_stats_M6$Present) %>% as.data.frame()
#Combine true and false rows
glam_score_table_M6 <- cbind(glam_score_table_M6[1:5,], glam_score_table_M6[6:10,]) ## this # depends on the # of monocle states
#Create names that make more sense
colnames(glam_score_table_M6) <- c("State", "F", "Freq_F", "State2", "T", "Freq_T")

#Remove unneeded columns
glam_score_table_M6 <- glam_score_table_M6[,c(1,3,6)]
#Create a percent column (percent of cells within each state with a good motif match)
glam_score_table_M6$pct <- glam_score_table_M6$Freq_T / (glam_score_table_M6$Freq_F + glam_score_table_M6$Freq_T)
#write.csv(glam_score_table_M4, "Motif/train/TFH/score_file/TRA_TFH_T65_motif1_M4_cln4_cells_across_M4.csv")
write.csv(glam_score_table_M6, "Motif_5mice/Train/Th1/Score/TRA_Th1_T65_motif7_M6_cln4_cells_across_M6_cells.csv")
getwd()
glam_score_table_M6$Freq_F %>% sum () # 2315
glam_score_table_M6$Freq_T %>% sum() # 9


glam_stats_M6$combined_cdr3_nt[which(glam_stats_M6$Present == TRUE)] %>% table()

#TRA:TGTGCAGCAAGCTATAGCAACTATCAGTTGATCTGG_TRB:TGTGCCTGGAGTCTAGCCAACAATGCAAACACAGAAGTCTTCTTT 
 #                                                                                        9 
```

```{r}
# Th1 motif 9

#data_monocle <- readRDS("summary_5M_CCA_V2/data_monocle_2cell.RDS")
# Th1 motif 9
rm(glam2_scores)
glam2_scores <- read.table(file = "/media/runa/Elements/Jem_revision/Seurat/Motif_5mice/Train/Th1/Motif9/Th1_Motif9.glam2scan/glam2scan_scores_Motif9.txt", header = FALSE)
#We're only interested in the cell name and score columns
glam2_scores <- glam2_scores[,c(1,6)]
#Take the top score for each cell
glam2_scores <- aggregate(.~V1, glam2_scores, FUN=max) # here multiple cells can have
#more than one score, with aggregate command we are saying in those cases just keep the max
#score
rownames(glam2_scores) <- glam2_scores$V1
# to match cell names with monocle data add 1 in cell name
glam2_scores$V1 <-
  paste0(glam2_scores$V1, "1")
rownames(glam2_scores) <- glam2_scores$V1

#Add glam score to each cell within monocle pData
pData(data_monocle)$glam_score <- sapply(rownames(pData(data_monocle)), FUN =
function(x) glam2_scores[x,]$V6)

View(pData(data_monocle))


#Observe the glam score distribution, find a high score for the cells 40 looks good here; this
#scoring will vary from motif to motif
plot(density(pData(data_monocle)$glam_score[!is.na(pData(data_monocle)$glam_score)]),
main = "Motif score distribution_TRA_Th1_Motif9", xlab = "")
abline(v = 40, col = "Red")

#Make a true/false filtering column based on the glam score for plotting 
pData(data_monocle)$Present <- pData(data_monocle)$glam_score > 40

# Make a new dataframe to use for statistical analysis but not plotting the cells

rm(glam_stats_M2)
rm(glam_stats_M4)
rm(glam_stats_M6)

# Only include cells from mouse 2 and mouse 4, separately
glam_stats_M2 <- data_monocle$Present[which(data_monocle$mice == "M2")] %>% as.data.frame()
colnames(glam_stats_M2) <- "Present"
# Add other columns from data_monocle
glam_stats_M2$State <- data_monocle$State[which(data_monocle$mice == "M2")]
glam_stats_M2$combined_cdr3_nt <- data_monocle$combined_cdr3_nt[which(data_monocle$mice == "M2")]
glam_stats_M2$cellID <- rownames(data_monocle@phenoData)[which(data_monocle$mice == "M2")]
# Remove any rows with NA (cells without a motif score, incluing cells that don't meet definition of a clone)
glam_stats_M2 <- glam_stats_M2[!is.na(glam_stats_M2$Present), ]

glam_stats_M4 <- data_monocle$Present[which(data_monocle$mice == "M4")] %>% as.data.frame()
colnames(glam_stats_M4) <- "Present"
# Add other columns from data_monocle
glam_stats_M4$State <- data_monocle$State[which(data_monocle$mice == "M4")]
glam_stats_M4$combined_cdr3_nt <- data_monocle$combined_cdr3_nt[which(data_monocle$mice == "M4")]
glam_stats_M4$cellID <- rownames(data_monocle@phenoData)[which(data_monocle$mice == "M4")]
# Remove any rows with NA (cells without a motif score, incluing cells that don't meet definition of a clone)
glam_stats_M4 <- glam_stats_M4[!is.na(glam_stats_M4$Present), ]

glam_stats_M6 <- data_monocle$Present[which(data_monocle$mice == "M6")] %>% as.data.frame()
colnames(glam_stats_M6) <- "Present"
# Add other columns from data_monocle
glam_stats_M6$State <- data_monocle$State[which(data_monocle$mice == "M6")]
glam_stats_M6$combined_cdr3_nt <- data_monocle$combined_cdr3_nt[which(data_monocle$mice == "M6")]
glam_stats_M6$cellID <- rownames(data_monocle@phenoData)[which(data_monocle$mice == "M6")]
# Remove any rows with NA (cells without a motif score, incluing cells that don't meet definition of a clone)
glam_stats_M6 <- glam_stats_M6[!is.na(glam_stats_M6$Present), ]



#Make NA values false
pData(data_monocle)$Present[is.na(pData(data_monocle)$Present)] <- FALSE
plot_obj <- plot_complex_cell_trajectory(data_monocle, color_by = "Present")
plot_obj$data <- plot_obj$data[order(plot_obj$data$Present),]
plot_obj + scale_color_manual(values=c("grey", "red")) + ggtitle("TRA_Th1_motif9_40_across_M2M4M6'\n Score > 40 \n ")

png(file = "summary_6mice/TRA_TH1_motif1_M2M4M6_cln4.png", width = 8, height = 6, units = "in", res = 300)
plot_obj + scale_color_manual(values=c("grey", "red")) + ggtitle("TRA_TH1_motif6_M2M4M6.png'\n Score > 30 \n 'P value '")
#dev.off()

#Create a table detailing the number of cells in each monocle state that have a good motif match
# The line below was from Robert's analysis, I'm replacing it because there was an error
# glam_score_table <- as.data.frame(table(pData(data_no_6_mono_cds)$State, pData(data_no_6_mono_cds)$Present))
rm(glam_score_table_M2)
rm(glam_score_table_M4)
rm(glam_score_table_M6)


glam_score_table_M2 <- table(glam_stats_M2$State, glam_stats_M2$Present) %>% as.data.frame()
#Combine true and false rows
glam_score_table_M2 <- cbind(glam_score_table_M2[1:5,], glam_score_table_M2[6:10,])
#Create names that make more sense
colnames(glam_score_table_M2) <- c("State", "F", "Freq_F", "State2", "T", "Freq_T")

#Remove unneeded columns
glam_score_table_M2 <- glam_score_table_M2[,c(1,3,6)]
#Create a percent column (percent of cells within each state with a good motif match)
glam_score_table_M2$pct <- glam_score_table_M2$Freq_T / (glam_score_table_M2$Freq_F + glam_score_table_M2$Freq_T)
#write.csv(glam_score_table_M2, "Motif/train/TFH/score_file/TRA_TFH_T65_motif1_M2_cln4_cells_across_M2.csv")
write.csv(glam_score_table_M2, "Motif_5mice/Train/Th1/Score/TRA_Th1_T65_motif9_M2_cln4_cells_across_M2_cells.csv")
getwd()
glam_score_table_M2$Freq_F %>% sum () # 857
glam_score_table_M2$Freq_T %>% sum() # 6

#Create a table detailing the number of cells in each monocle state that have a good motif match
# The line below was from Robert's analysis, I'm replacing it because there was an error
# glam_score_table <- as.data.frame(table(pData(data_no_6_mono_cds)$State, pData(data_no_6_mono_cds)$Present))

glam_score_table_M4 <- table(glam_stats_M4$State, glam_stats_M4$Present) %>% as.data.frame()
#Combine true and false rows
glam_score_table_M4 <- cbind(glam_score_table_M4[1:5,], glam_score_table_M4[6:10,]) ## this # depends on the # of monocle states
#Create names that make more sense
colnames(glam_score_table_M4) <- c("State", "F", "Freq_F", "State2", "T", "Freq_T")

#Remove unneeded columns
glam_score_table_M4 <- glam_score_table_M4[,c(1,3,6)]
#Create a percent column (percent of cells within each state with a good motif match)
glam_score_table_M4$pct <- glam_score_table_M4$Freq_T / (glam_score_table_M4$Freq_F + glam_score_table_M4$Freq_T)
#write.csv(glam_score_table_M4, "Motif/train/TFH/score_file/TRA_TFH_T65_motif1_M4_cln4_cells_across_M4.csv")
write.csv(glam_score_table_M4, "Motif_5mice/Train/Th1/Score/TRA_Th1_T65_motif9_M4_cln4_cells_across_M4_cells.csv")
getwd()
glam_score_table_M4$Freq_F %>% sum () # 2981
glam_score_table_M4$Freq_T %>% sum() # 237


#Create a table detailing the number of cells in each monocle state that have a good motif match
# The line below was from Robert's analysis, I'm replacing it because there was an error
# glam_score_table <- as.data.frame(table(pData(data_no_6_mono_cds)$State, pData(data_no_6_mono_cds)$Present))

glam_score_table_M6 <- table(glam_stats_M6$State, glam_stats_M6$Present) %>% as.data.frame()
#Combine true and false rows
glam_score_table_M6 <- cbind(glam_score_table_M6[1:5,], glam_score_table_M6[6:10,]) ## this # depends on the # of monocle states
#Create names that make more sense
colnames(glam_score_table_M6) <- c("State", "F", "Freq_F", "State2", "T", "Freq_T")

#Remove unneeded columns
glam_score_table_M6 <- glam_score_table_M6[,c(1,3,6)]
#Create a percent column (percent of cells within each state with a good motif match)
glam_score_table_M6$pct <- glam_score_table_M6$Freq_T / (glam_score_table_M6$Freq_F + glam_score_table_M6$Freq_T)
#write.csv(glam_score_table_M4, "Motif/train/TFH/score_file/TRA_TFH_T65_motif1_M4_cln4_cells_across_M4.csv")
write.csv(glam_score_table_M6, "Motif_5mice/Train/Th1/Score/TRA_Th1_T65_motif9_M6_cln4_cells_across_M6_cells.csv")
getwd()
glam_score_table_M6$Freq_F %>% sum () # 2324
glam_score_table_M6$Freq_T %>% sum() # NA


# Perform clone-level statistical analysis

glam_stats_M2$combined_cdr3_nt[which(glam_stats_M2$Present == TRUE)] %>% table()


#         TRA:TGCTCAGCAAGTATGACTGGAGGCTATAAAGTGGTCTTT_TRB:TGTGCCAGCAGCAAGGTAAACACAGAAGTCTTCTTT 
 #                                                                                           4 
#TRA:TGTGCAGCAAGTCTCGATGGAGGCTATAAAGTGGTCTTT_TRB:TGTGCTAGCAGTTTATCCGGGACAGGGGACGCTGAGCAGTTCTTC 
    #                                                                                       30 
   #TRA:TGTGCAGTGAGCGGGACTGGAGGCTATAAAGTGGTCTTT_TRB:TGTGCCAGCAGGACTGGGGGGGCAGGGGACACCCAGTACTTT 
    #                                                                                       13 
     # TRA:TGTGCTCTGGGCCTGACTGGAGGCTATAAAGTGGTCTTT_TRB:TGCAGTGCAAGGGGGGCGGACACCGGGCAGCTCTACTTT 
       #                                                                                    21 

grep(pattern = "TRA:TGTGCAGCAAGTTGGGGGACTGGAGGCTATAAAGTGGTCTTT_TRB:TGTGCTAGCAGTTTTGCTGGGGGGGCTTCCTATGAACAGTACTTC", x = data_clones_by_mouse_AA_TRA_TRB_NOV$combined_cdr3_nt)
# 3
as.data.frame(data_clones_by_mouse_AA_TRA_TRB_NOV[1, ])
# 79, 0, 21
getwd()
glam_stats_M4$combined_cdr3_nt[which(glam_stats_M4$Present == TRUE)] %>% table()

        #  TRA:TGTGCAGCAAGCATGAGAACAGGAAACTACAAATACGTCTTT_TRB:TGTGCCAGCAGCGCCCAAACGCTGTATTTT 
         #                                                                                      5 
      #TRA:TGTGCAGCAAGTCTTAATACAGGAAACTACAAATACGTCTTT_TRB:TGTGCCTGGAATGGGGGGGGCTATGCTGAGCAGTTCTTC 
       #                                                                                       12 
#TRA:TGTGCAGCAAGTTGGGGGACTGGAGGCTATAAAGTGGTCTTT_TRB:TGTGCTAGCAGTTTTGCTGGGGGGGCTTCCTATGAACAGTACTTC 
 #                         

```

```{r}
# Th1 motif 10

#data_monocle <- readRDS("summary_5M_CCA_V2/data_monocle_2cell.RDS")
# Th1 motif 10
rm(glam2_scores)
glam2_scores <- read.table(file = "/media/runa/Elements/Jem_revision/Seurat/Motif_5mice/Train/Th1/Motif10/Th1_Motif10.glam2scan/glam2scan_scores_Motif10.txt", header = FALSE)
#We're only interested in the cell name and score columns
glam2_scores <- glam2_scores[,c(1,6)]
#Take the top score for each cell
glam2_scores <- aggregate(.~V1, glam2_scores, FUN=max) # here multiple cells can have
#more than one score, with aggregate command we are saying in those cases just keep the max
#score
rownames(glam2_scores) <- glam2_scores$V1
# to match cell names with monocle data add 1 in cell name
glam2_scores$V1 <-
  paste0(glam2_scores$V1, "1")
rownames(glam2_scores) <- glam2_scores$V1

#Add glam score to each cell within monocle pData
pData(data_monocle)$glam_score <- sapply(rownames(pData(data_monocle)), FUN =
function(x) glam2_scores[x,]$V6)

View(pData(data_monocle))


#Observe the glam score distribution, find a high score for the cells 40 looks good here; this
#scoring will vary from motif to motif
plot(density(pData(data_monocle)$glam_score[!is.na(pData(data_monocle)$glam_score)]),
main = "Motif score distribution_TRA_Th1_Motif10", xlab = "")
abline(v = 40, col = "Red")

#Make a true/false filtering column based on the glam score for plotting 
pData(data_monocle)$Present <- pData(data_monocle)$glam_score > 40

# Make a new dataframe to use for statistical analysis but not plotting the cells

rm(glam_stats_M2)
rm(glam_stats_M4)
rm(glam_stats_M6)

# Only include cells from mouse 2 and mouse 4, separately
glam_stats_M2 <- data_monocle$Present[which(data_monocle$mice == "M2")] %>% as.data.frame()
colnames(glam_stats_M2) <- "Present"
# Add other columns from data_monocle
glam_stats_M2$State <- data_monocle$State[which(data_monocle$mice == "M2")]
glam_stats_M2$combined_cdr3_nt <- data_monocle$combined_cdr3_nt[which(data_monocle$mice == "M2")]
glam_stats_M2$cellID <- rownames(data_monocle@phenoData)[which(data_monocle$mice == "M2")]
# Remove any rows with NA (cells without a motif score, incluing cells that don't meet definition of a clone)
glam_stats_M2 <- glam_stats_M2[!is.na(glam_stats_M2$Present), ]

glam_stats_M4 <- data_monocle$Present[which(data_monocle$mice == "M4")] %>% as.data.frame()
colnames(glam_stats_M4) <- "Present"
# Add other columns from data_monocle
glam_stats_M4$State <- data_monocle$State[which(data_monocle$mice == "M4")]
glam_stats_M4$combined_cdr3_nt <- data_monocle$combined_cdr3_nt[which(data_monocle$mice == "M4")]
glam_stats_M4$cellID <- rownames(data_monocle@phenoData)[which(data_monocle$mice == "M4")]
# Remove any rows with NA (cells without a motif score, incluing cells that don't meet definition of a clone)
glam_stats_M4 <- glam_stats_M4[!is.na(glam_stats_M4$Present), ]

glam_stats_M6 <- data_monocle$Present[which(data_monocle$mice == "M6")] %>% as.data.frame()
colnames(glam_stats_M6) <- "Present"
# Add other columns from data_monocle
glam_stats_M6$State <- data_monocle$State[which(data_monocle$mice == "M6")]
glam_stats_M6$combined_cdr3_nt <- data_monocle$combined_cdr3_nt[which(data_monocle$mice == "M6")]
glam_stats_M6$cellID <- rownames(data_monocle@phenoData)[which(data_monocle$mice == "M6")]
# Remove any rows with NA (cells without a motif score, incluing cells that don't meet definition of a clone)
glam_stats_M6 <- glam_stats_M6[!is.na(glam_stats_M6$Present), ]



#Make NA values false
pData(data_monocle)$Present[is.na(pData(data_monocle)$Present)] <- FALSE
plot_obj <- plot_complex_cell_trajectory(data_monocle, color_by = "Present")
plot_obj$data <- plot_obj$data[order(plot_obj$data$Present),]
plot_obj + scale_color_manual(values=c("grey", "red")) + ggtitle("TRA_Th1_motif10_40_across_M2M4M6'\n Score > 40 \n ")

png(file = "summary_6mice/TRA_TH1_motif1_M2M4M6_cln4.png", width = 8, height = 6, units = "in", res = 300)
plot_obj + scale_color_manual(values=c("grey", "red")) + ggtitle("TRA_TH1_motif6_M2M4M6.png'\n Score > 30 \n 'P value '")
#dev.off()

#Create a table detailing the number of cells in each monocle state that have a good motif match
# The line below was from Robert's analysis, I'm replacing it because there was an error
# glam_score_table <- as.data.frame(table(pData(data_no_6_mono_cds)$State, pData(data_no_6_mono_cds)$Present))
rm(glam_score_table_M2)
rm(glam_score_table_M4)
rm(glam_score_table_M6)


glam_score_table_M2 <- table(glam_stats_M2$State, glam_stats_M2$Present) %>% as.data.frame()
#Combine true and false rows
glam_score_table_M2 <- cbind(glam_score_table_M2[1:5,], glam_score_table_M2[6:10,])
#Create names that make more sense
colnames(glam_score_table_M2) <- c("State", "F", "Freq_F", "State2", "T", "Freq_T")

#Remove unneeded columns
glam_score_table_M2 <- glam_score_table_M2[,c(1,3,6)]
#Create a percent column (percent of cells within each state with a good motif match)
glam_score_table_M2$pct <- glam_score_table_M2$Freq_T / (glam_score_table_M2$Freq_F + glam_score_table_M2$Freq_T)
#write.csv(glam_score_table_M2, "Motif/train/TFH/score_file/TRA_TFH_T65_motif1_M2_cln4_cells_across_M2.csv")
write.csv(glam_score_table_M2, "Motif_5mice/Train/Th1/Score/TRA_Th1_T65_motif10_M2_cln4_cells_across_M2_cells.csv")
getwd()
glam_score_table_M2$Freq_F %>% sum () # 1018
glam_score_table_M2$Freq_T %>% sum() # 6

#Create a table detailing the number of cells in each monocle state that have a good motif match
# The line below was from Robert's analysis, I'm replacing it because there was an error
# glam_score_table <- as.data.frame(table(pData(data_no_6_mono_cds)$State, pData(data_no_6_mono_cds)$Present))

glam_score_table_M4 <- table(glam_stats_M4$State, glam_stats_M4$Present) %>% as.data.frame()
#Combine true and false rows
glam_score_table_M4 <- cbind(glam_score_table_M4[1:5,], glam_score_table_M4[6:10,]) ## this # depends on the # of monocle states
#Create names that make more sense
colnames(glam_score_table_M4) <- c("State", "F", "Freq_F", "State2", "T", "Freq_T")

#Remove unneeded columns
glam_score_table_M4 <- glam_score_table_M4[,c(1,3,6)]
#Create a percent column (percent of cells within each state with a good motif match)
glam_score_table_M4$pct <- glam_score_table_M4$Freq_T / (glam_score_table_M4$Freq_F + glam_score_table_M4$Freq_T)
#write.csv(glam_score_table_M4, "Motif/train/TFH/score_file/TRA_TFH_T65_motif1_M4_cln4_cells_across_M4.csv")
write.csv(glam_score_table_M4, "Motif_5mice/Train/Th1/Score/TRA_Th1_T65_motif10_M4_cln4_cells_across_M4_cells.csv")
getwd()
glam_score_table_M4$Freq_F %>% sum () # 3176
glam_score_table_M4$Freq_T %>% sum() # 237


#Create a table detailing the number of cells in each monocle state that have a good motif match
# The line below was from Robert's analysis, I'm replacing it because there was an error
# glam_score_table <- as.data.frame(table(pData(data_no_6_mono_cds)$State, pData(data_no_6_mono_cds)$Present))

glam_score_table_M6 <- table(glam_stats_M6$State, glam_stats_M6$Present) %>% as.data.frame()
#Combine true and false rows
glam_score_table_M6 <- cbind(glam_score_table_M6[1:5,], glam_score_table_M6[6:10,]) ## this # depends on the # of monocle states
#Create names that make more sense
colnames(glam_score_table_M6) <- c("State", "F", "Freq_F", "State2", "T", "Freq_T")

#Remove unneeded columns
glam_score_table_M6 <- glam_score_table_M6[,c(1,3,6)]
#Create a percent column (percent of cells within each state with a good motif match)
glam_score_table_M6$pct <- glam_score_table_M6$Freq_T / (glam_score_table_M6$Freq_F + glam_score_table_M6$Freq_T)
#write.csv(glam_score_table_M4, "Motif/train/TFH/score_file/TRA_TFH_T65_motif1_M4_cln4_cells_across_M4.csv")
write.csv(glam_score_table_M6, "Motif_5mice/Train/Th1/Score/TRA_Th1_T65_motif10_M6_cln4_cells_across_M6_cells.csv")
getwd()
glam_score_table_M6$Freq_F %>% sum () # 2462
glam_score_table_M6$Freq_T %>% sum() # NA


# Perform clone-level statistical analysis

glam_stats_M2$combined_cdr3_nt[which(glam_stats_M2$Present == TRUE)] %>% table()


#         TRA:TGCTCAGCAAGTATGACTGGAGGCTATAAAGTGGTCTTT_TRB:TGTGCCAGCAGCAAGGTAAACACAGAAGTCTTCTTT 
 #                                                                                           4 
#TRA:TGTGCAGCAAGTCTCGATGGAGGCTATAAAGTGGTCTTT_TRB:TGTGCTAGCAGTTTATCCGGGACAGGGGACGCTGAGCAGTTCTTC 
    #                                                                                       30 
   #TRA:TGTGCAGTGAGCGGGACTGGAGGCTATAAAGTGGTCTTT_TRB:TGTGCCAGCAGGACTGGGGGGGCAGGGGACACCCAGTACTTT 
    #                                                                                       13 
     # TRA:TGTGCTCTGGGCCTGACTGGAGGCTATAAAGTGGTCTTT_TRB:TGCAGTGCAAGGGGGGCGGACACCGGGCAGCTCTACTTT 
       #                                                                                    21 

grep(pattern = "TRA:TGTGCAGCAAGTTGGGGGACTGGAGGCTATAAAGTGGTCTTT_TRB:TGTGCTAGCAGTTTTGCTGGGGGGGCTTCCTATGAACAGTACTTC", x = data_clones_by_mouse_AA_TRA_TRB_NOV$combined_cdr3_nt)
# 3
as.data.frame(data_clones_by_mouse_AA_TRA_TRB_NOV[1, ])
# 79, 0, 21
getwd()
glam_stats_M4$combined_cdr3_nt[which(glam_stats_M4$Present == TRUE)] %>% table()

        #  TRA:TGTGCAGCAAGCATGAGAACAGGAAACTACAAATACGTCTTT_TRB:TGTGCCAGCAGCGCCCAAACGCTGTATTTT 
         #                                                                                      5 
      #TRA:TGTGCAGCAAGTCTTAATACAGGAAACTACAAATACGTCTTT_TRB:TGTGCCTGGAATGGGGGGGGCTATGCTGAGCAGTTCTTC 
       #                                                                                       12 
#TRA:TGTGCAGCAAGTTGGGGGACTGGAGGCTATAAAGTGGTCTTT_TRB:TGTGCTAGCAGTTTTGCTGGGGGGGCTTCCTATGAACAGTACTTC 
 #                         

```

```{r}
# Th1 motif 11

#data_monocle <- readRDS("summary_5M_CCA_V2/data_monocle_2cell.RDS")
# Th1 motif 11
rm(glam2_scores)
glam2_scores <- read.table(file = "/media/runa/Elements/Jem_revision/Seurat/Motif_5mice/Train/Th1/Motif11/Th1_Motif11.glam2scan/glam2scan_scores_Motif11.txt", header = FALSE)
#We're only interested in the cell name and score columns
glam2_scores <- glam2_scores[,c(1,6)]
#Take the top score for each cell
glam2_scores <- aggregate(.~V1, glam2_scores, FUN=max) # here multiple cells can have
#more than one score, with aggregate command we are saying in those cases just keep the max
#score
rownames(glam2_scores) <- glam2_scores$V1
# to match cell names with monocle data add 1 in cell name
glam2_scores$V1 <-
  paste0(glam2_scores$V1, "1")
rownames(glam2_scores) <- glam2_scores$V1

#Add glam score to each cell within monocle pData
pData(data_monocle)$glam_score <- sapply(rownames(pData(data_monocle)), FUN =
function(x) glam2_scores[x,]$V6)

View(pData(data_monocle))


#Observe the glam score distribution, find a high score for the cells 40 looks good here; this
#scoring will vary from motif to motif
plot(density(pData(data_monocle)$glam_score[!is.na(pData(data_monocle)$glam_score)]),
main = "Motif score distribution_TRA_Th1_Motif11", xlab = "")
abline(v = 35, col = "Red")

#Make a true/false filtering column based on the glam score for plotting 
pData(data_monocle)$Present <- pData(data_monocle)$glam_score > 35

# Make a new dataframe to use for statistical analysis but not plotting the cells

rm(glam_stats_M2)
rm(glam_stats_M4)
rm(glam_stats_M6)

# Only include cells from mouse 2 and mouse 4, separately
glam_stats_M2 <- data_monocle$Present[which(data_monocle$mice == "M2")] %>% as.data.frame()
colnames(glam_stats_M2) <- "Present"
# Add other columns from data_monocle
glam_stats_M2$State <- data_monocle$State[which(data_monocle$mice == "M2")]
glam_stats_M2$combined_cdr3_nt <- data_monocle$combined_cdr3_nt[which(data_monocle$mice == "M2")]
glam_stats_M2$cellID <- rownames(data_monocle@phenoData)[which(data_monocle$mice == "M2")]
# Remove any rows with NA (cells without a motif score, incluing cells that don't meet definition of a clone)
glam_stats_M2 <- glam_stats_M2[!is.na(glam_stats_M2$Present), ]

glam_stats_M4 <- data_monocle$Present[which(data_monocle$mice == "M4")] %>% as.data.frame()
colnames(glam_stats_M4) <- "Present"
# Add other columns from data_monocle
glam_stats_M4$State <- data_monocle$State[which(data_monocle$mice == "M4")]
glam_stats_M4$combined_cdr3_nt <- data_monocle$combined_cdr3_nt[which(data_monocle$mice == "M4")]
glam_stats_M4$cellID <- rownames(data_monocle@phenoData)[which(data_monocle$mice == "M4")]
# Remove any rows with NA (cells without a motif score, incluing cells that don't meet definition of a clone)
glam_stats_M4 <- glam_stats_M4[!is.na(glam_stats_M4$Present), ]

glam_stats_M6 <- data_monocle$Present[which(data_monocle$mice == "M6")] %>% as.data.frame()
colnames(glam_stats_M6) <- "Present"
# Add other columns from data_monocle
glam_stats_M6$State <- data_monocle$State[which(data_monocle$mice == "M6")]
glam_stats_M6$combined_cdr3_nt <- data_monocle$combined_cdr3_nt[which(data_monocle$mice == "M6")]
glam_stats_M6$cellID <- rownames(data_monocle@phenoData)[which(data_monocle$mice == "M6")]
# Remove any rows with NA (cells without a motif score, incluing cells that don't meet definition of a clone)
glam_stats_M6 <- glam_stats_M6[!is.na(glam_stats_M6$Present), ]



#Make NA values false
pData(data_monocle)$Present[is.na(pData(data_monocle)$Present)] <- FALSE
plot_obj <- plot_complex_cell_trajectory(data_monocle, color_by = "Present")
plot_obj$data <- plot_obj$data[order(plot_obj$data$Present),]
plot_obj + scale_color_manual(values=c("grey", "red")) + ggtitle("TRA_Th1_motif11_35_across_M2M4M6'\n Score > 35 \n ")

png(file = "summary_6mice/TRA_TH1_motif1_M2M4M6_cln4.png", width = 8, height = 6, units = "in", res = 300)
plot_obj + scale_color_manual(values=c("grey", "red")) + ggtitle("TRA_TH1_motif6_M2M4M6.png'\n Score > 30 \n 'P value '")
#dev.off()

#Create a table detailing the number of cells in each monocle state that have a good motif match
# The line below was from Robert's analysis, I'm replacing it because there was an error
# glam_score_table <- as.data.frame(table(pData(data_no_6_mono_cds)$State, pData(data_no_6_mono_cds)$Present))
rm(glam_score_table_M2)
rm(glam_score_table_M4)
rm(glam_score_table_M6)


glam_score_table_M2 <- table(glam_stats_M2$State, glam_stats_M2$Present) %>% as.data.frame()
#Combine true and false rows
glam_score_table_M2 <- cbind(glam_score_table_M2[1:5,], glam_score_table_M2[6:10,])
#Create names that make more sense
colnames(glam_score_table_M2) <- c("State", "F", "Freq_F", "State2", "T", "Freq_T")

#Remove unneeded columns
glam_score_table_M2 <- glam_score_table_M2[,c(1,3,6)]
#Create a percent column (percent of cells within each state with a good motif match)
glam_score_table_M2$pct <- glam_score_table_M2$Freq_T / (glam_score_table_M2$Freq_F + glam_score_table_M2$Freq_T)
#write.csv(glam_score_table_M2, "Motif/train/TFH/score_file/TRA_TFH_T65_motif1_M2_cln4_cells_across_M2.csv")
write.csv(glam_score_table_M2, "Motif_5mice/Train/Th1/Score/TRA_Th1_T65_motif11_M2_cln4_cells_across_M2_cells.csv")
getwd()
glam_score_table_M2$Freq_F %>% sum () # 863
glam_score_table_M2$Freq_T %>% sum() # NA

#Create a table detailing the number of cells in each monocle state that have a good motif match
# The line below was from Robert's analysis, I'm replacing it because there was an error
# glam_score_table <- as.data.frame(table(pData(data_no_6_mono_cds)$State, pData(data_no_6_mono_cds)$Present))

glam_score_table_M4 <- table(glam_stats_M4$State, glam_stats_M4$Present) %>% as.data.frame()
#Combine true and false rows
glam_score_table_M4 <- cbind(glam_score_table_M4[1:5,], glam_score_table_M4[6:10,]) ## this # depends on the # of monocle states
#Create names that make more sense
colnames(glam_score_table_M4) <- c("State", "F", "Freq_F", "State2", "T", "Freq_T")

#Remove unneeded columns
glam_score_table_M4 <- glam_score_table_M4[,c(1,3,6)]
#Create a percent column (percent of cells within each state with a good motif match)
glam_score_table_M4$pct <- glam_score_table_M4$Freq_T / (glam_score_table_M4$Freq_F + glam_score_table_M4$Freq_T)
#write.csv(glam_score_table_M4, "Motif/train/TFH/score_file/TRA_TFH_T65_motif1_M4_cln4_cells_across_M4.csv")
write.csv(glam_score_table_M4, "Motif_5mice/Train/Th1/Score/TRA_Th1_T65_motif11_M4_cln4_cells_across_M4_cells.csv")
getwd()
glam_score_table_M4$Freq_F %>% sum () # 3218
glam_score_table_M4$Freq_T %>% sum() # NA


#Create a table detailing the number of cells in each monocle state that have a good motif match
# The line below was from Robert's analysis, I'm replacing it because there was an error
# glam_score_table <- as.data.frame(table(pData(data_no_6_mono_cds)$State, pData(data_no_6_mono_cds)$Present))

glam_score_table_M6 <- table(glam_stats_M6$State, glam_stats_M6$Present) %>% as.data.frame()
#Combine true and false rows
glam_score_table_M6 <- cbind(glam_score_table_M6[1:5,], glam_score_table_M6[6:10,]) ## this # depends on the # of monocle states
#Create names that make more sense
colnames(glam_score_table_M6) <- c("State", "F", "Freq_F", "State2", "T", "Freq_T")

#Remove unneeded columns
glam_score_table_M6 <- glam_score_table_M6[,c(1,3,6)]
#Create a percent column (percent of cells within each state with a good motif match)
glam_score_table_M6$pct <- glam_score_table_M6$Freq_T / (glam_score_table_M6$Freq_F + glam_score_table_M6$Freq_T)
#write.csv(glam_score_table_M4, "Motif/train/TFH/score_file/TRA_TFH_T65_motif1_M4_cln4_cells_across_M4.csv")
write.csv(glam_score_table_M6, "Motif_5mice/Train/Th1/Score/TRA_Th1_T65_motif11_M6_cln4_cells_across_M6_cells.csv")
getwd()
glam_score_table_M6$Freq_F %>% sum () # 2268
glam_score_table_M6$Freq_T %>% sum() # 56


# Perform clone-level statistical analysis

glam_stats_M2$combined_cdr3_nt[which(glam_stats_M2$Present == TRUE)] %>% table()


#         TRA:TGCTCAGCAAGTATGACTGGAGGCTATAAAGTGGTCTTT_TRB:TGTGCCAGCAGCAAGGTAAACACAGAAGTCTTCTTT 
 #                                                                                           4 
#TRA:TGTGCAGCAAGTCTCGATGGAGGCTATAAAGTGGTCTTT_TRB:TGTGCTAGCAGTTTATCCGGGACAGGGGACGCTGAGCAGTTCTTC 
    #                                                                                       30 
   #TRA:TGTGCAGTGAGCGGGACTGGAGGCTATAAAGTGGTCTTT_TRB:TGTGCCAGCAGGACTGGGGGGGCAGGGGACACCCAGTACTTT 
    #                                                                                       13 
     # TRA:TGTGCTCTGGGCCTGACTGGAGGCTATAAAGTGGTCTTT_TRB:TGCAGTGCAAGGGGGGCGGACACCGGGCAGCTCTACTTT 
       #                                                                                    21 

grep(pattern = "TRA:TGTGCAGCAAGTTGGGGGACTGGAGGCTATAAAGTGGTCTTT_TRB:TGTGCTAGCAGTTTTGCTGGGGGGGCTTCCTATGAACAGTACTTC", x = data_clones_by_mouse_AA_TRA_TRB_NOV$combined_cdr3_nt)
# 3
as.data.frame(data_clones_by_mouse_AA_TRA_TRB_NOV[1, ])
# 79, 0, 21
getwd()
glam_stats_M4$combined_cdr3_nt[which(glam_stats_M4$Present == TRUE)] %>% table()

        #  TRA:TGTGCAGCAAGCATGAGAACAGGAAACTACAAATACGTCTTT_TRB:TGTGCCAGCAGCGCCCAAACGCTGTATTTT 
         #                                                                                      5 
      #TRA:TGTGCAGCAAGTCTTAATACAGGAAACTACAAATACGTCTTT_TRB:TGTGCCTGGAATGGGGGGGGCTATGCTGAGCAGTTCTTC 
       #                                                                                       12 
#TRA:TGTGCAGCAAGTTGGGGGACTGGAGGCTATAAAGTGGTCTTT_TRB:TGTGCTAGCAGTTTTGCTGGGGGGGCTTCCTATGAACAGTACTTC 
 #                         

```


```{r validate Th1 significant motifs 6 and 7 in Test mice M1 and M3}

# Th1 motif 6 in test mice

data_monocle <- readRDS("summary_5M_CCA_V2/data_monocle_2cell.RDS")
# Th1 motif 6
rm(glam2_scores)
glam2_scores <- read.table(file = "/media/runa/Elements/Jem_revision/Seurat/Motif_5mice/Test/Th1/Motif6/Th1_Motif6.glam2scan/glam2scan_scores_Motif6.txt", header = FALSE)
#We're only interested in the cell name and score columns
glam2_scores <- glam2_scores[,c(1,6)]
#Take the top score for each cell
glam2_scores <- aggregate(.~V1, glam2_scores, FUN=max) # here multiple cells can have
#more than one score, with aggregate command we are saying in those cases just keep the max
#score
rownames(glam2_scores) <- glam2_scores$V1
# to match cell names with monocle data add 1 in cell name
glam2_scores$V1 <-
  paste0(glam2_scores$V1, "1")
rownames(glam2_scores) <- glam2_scores$V1

#Add glam score to each cell within monocle pData
pData(data_monocle)$glam_score <- sapply(rownames(pData(data_monocle)), FUN =
function(x) glam2_scores[x,]$V6)

View(pData(data_monocle))


#Observe the glam score distribution, find a high score for the cells 40 looks good here; this
#scoring will vary from motif to motif

pdf("summary_5mice/Figure4/Th1_Motif6_Test_mice_score.pdf", width = 8, height = 6)
plot(density(pData(data_monocle)$glam_score[!is.na(pData(data_monocle)$glam_score)]),
main = "Motif score distribution_TRA_Th1_Motif6 in Test mice", xlab = "")
abline(v = 30, col = "Red")
dev.off()

#Make a true/false filtering column based on the glam score for plotting 
pData(data_monocle)$Present <- pData(data_monocle)$glam_score > 30

# Make a new dataframe to use for statistical analysis but not plotting the cells

rm(glam_stats_M2)
rm(glam_stats_M4)
rm(glam_stats_M6)

# Only include cells from mouse 2 and mouse 4, separately
glam_stats_M1 <- data_monocle$Present[which(data_monocle$mice == "M1")] %>% as.data.frame()
colnames(glam_stats_M1) <- "Present"
# Add other columns from data_monocle
glam_stats_M1$State <- data_monocle$State[which(data_monocle$mice == "M1")]
glam_stats_M1$combined_cdr3_nt <- data_monocle$combined_cdr3_nt[which(data_monocle$mice == "M1")]
glam_stats_M1$cellID <- rownames(data_monocle@phenoData)[which(data_monocle$mice == "M1")]
# Remove any rows with NA (cells without a motif score, incluing cells that don't meet definition of a clone)
glam_stats_M1 <- glam_stats_M1[!is.na(glam_stats_M1$Present), ]

glam_stats_M3 <- data_monocle$Present[which(data_monocle$mice == "M3")] %>% as.data.frame()
colnames(glam_stats_M3) <- "Present"
# Add other columns from data_monocle
glam_stats_M3$State <- data_monocle$State[which(data_monocle$mice == "M3")]
glam_stats_M3$combined_cdr3_nt <- data_monocle$combined_cdr3_nt[which(data_monocle$mice == "M3")]
glam_stats_M3$cellID <- rownames(data_monocle@phenoData)[which(data_monocle$mice == "M3")]
# Remove any rows with NA (cells without a motif score, incluing cells that don't meet definition of a clone)
glam_stats_M3 <- glam_stats_M3[!is.na(glam_stats_M3$Present), ]



#Make NA values false
pData(data_monocle)$Present[is.na(pData(data_monocle)$Present)] <- FALSE
plot_obj <- plot_complex_cell_trajectory(data_monocle, color_by = "Present", cell_size = 3)
plot_obj$data <- plot_obj$data[order(plot_obj$data$Present),]
plot_obj + scale_color_manual(values=c("grey", "red")) + ggtitle("TRA_Th1_motif6_30_across_M1M3'\n Score > 30 \n ")


pdf("summary_5mice/Figure4/Th1_Motif6_Test_mice_TREE.pdf", width = 8, height = 6, useDingbats = FALSE)
plot_obj + scale_color_manual(values=c("grey", "red")) + ggtitle("TRA_TH1_motif6_M1M3.pdf'\n Score > 30 \n 'P value 1.0'")
dev.off()

#Create a table detailing the number of cells in each monocle state that have a good motif match
# The line below was from Robert's analysis, I'm replacing it because there was an error
# glam_score_table <- as.data.frame(table(pData(data_no_6_mono_cds)$State, pData(data_no_6_mono_cds)$Present))
rm(glam_score_table_M1)
rm(glam_score_table_M3)



glam_score_table_M1 <- table(glam_stats_M1$State, glam_stats_M1$Present) %>% as.data.frame()
#Combine true and false rows
glam_score_table_M1 <- cbind(glam_score_table_M1[1:5,], glam_score_table_M1[6:10,])
#Create names that make more sense
colnames(glam_score_table_M1) <- c("State", "F", "Freq_F", "State2", "T", "Freq_T")

#Remove unneeded columns
glam_score_table_M1 <- glam_score_table_M1[,c(1,3,6)]
#Create a percent column (percent of cells within each state with a good motif match)
glam_score_table_M1$pct <- glam_score_table_M1$Freq_T / (glam_score_table_M1$Freq_F + glam_score_table_M1$Freq_T)
#write.csv(glam_score_table_M2, "Motif/train/TFH/score_file/TRA_TFH_T65_motif1_M2_cln4_cells_across_M2.csv")
write.csv(glam_score_table_M1, "Motif_5mice/Test/Th1/Score/TRA_Th1_T65_motif6_M1_cln4_cells_across_M1_cells.csv")
getwd()
glam_score_table_M1$Freq_F %>% sum () # 871
glam_score_table_M1$Freq_T %>% sum() # 7

glam_stats_M1$combined_cdr3_nt[which(glam_stats_M1$Present == TRUE)] %>% table()

#TRA:TGTGCAGCAAGCATGAGGGCAGGAAACTACAAATACGTCTTT_TRB:TGTGCCAGCAGCTCAGAAACGCTGTATTTT 
 #                                                                               7 

#Create a table detailing the number of cells in each monocle state that have a good motif match
# The line below was from Robert's analysis, I'm replacing it because there was an error
# glam_score_table <- as.data.frame(table(pData(data_no_6_mono_cds)$State, pData(data_no_6_mono_cds)$Present))

glam_score_table_M3 <- table(glam_stats_M3$State, glam_stats_M3$Present) %>% as.data.frame()
#Combine true and false rows
glam_score_table_M3 <- cbind(glam_score_table_M3[1:5,], glam_score_table_M3[6:10,]) ## this # depends on the # of monocle states
#Create names that make more sense
colnames(glam_score_table_M3) <- c("State", "F", "Freq_F", "State2", "T", "Freq_T")

#Remove unneeded columns
glam_score_table_M3 <- glam_score_table_M3[,c(1,3,6)]
#Create a percent column (percent of cells within each state with a good motif match)
glam_score_table_M3$pct <- glam_score_table_M3$Freq_T / (glam_score_table_M3$Freq_F + glam_score_table_M3$Freq_T)
#write.csv(glam_score_table_M4, "Motif/train/TFH/score_file/TRA_TFH_T65_motif1_M4_cln4_cells_across_M4.csv")
write.csv(glam_score_table_M3, "Motif_5mice/Test/Th1/Score/TRA_Th1_T65_motif6_M3_cln4_cells_across_M3_cells.csv")
getwd()
glam_score_table_M3$Freq_F %>% sum () # 2410
glam_score_table_M3$Freq_T %>% sum() # 7

glam_stats_M3$combined_cdr3_nt[which(glam_stats_M3$Present == TRUE)] %>% table()

#TRA:TGTGCAGCAAGTGACTCCAATACCAACAAAGTCGTCTTT_TRB:TGTGCCAGCGGTGATGTACTGGGGGGGAAGGGGAGCTCCTATGAACAGTACTTC 
 #                                                                                                    7 


```
```{r}
# Th1 motif 7 in test mice

data_monocle <- readRDS("summary_5M_CCA_V2/data_monocle_2cell.RDS")
# Th1 motif 7
rm(glam2_scores)
glam2_scores <- read.table(file = "/media/runa/Elements/Jem_revision/Seurat/Motif_5mice/Test/Th1/Motif7/Th1_Motif7.glam2scan/glam2scan_scores_Motif7.txt", header = FALSE)
#We're only interested in the cell name and score columns
glam2_scores <- glam2_scores[,c(1,6)]
#Take the top score for each cell
glam2_scores <- aggregate(.~V1, glam2_scores, FUN=max) # here multiple cells can have
#more than one score, with aggregate command we are saying in those cases just keep the max
#score
rownames(glam2_scores) <- glam2_scores$V1
# to match cell names with monocle data add 1 in cell name
glam2_scores$V1 <-
  paste0(glam2_scores$V1, "1")
rownames(glam2_scores) <- glam2_scores$V1

#Add glam score to each cell within monocle pData
pData(data_monocle)$glam_score <- sapply(rownames(pData(data_monocle)), FUN =
function(x) glam2_scores[x,]$V6)

View(pData(data_monocle))


#Observe the glam score distribution, find a high score for the cells 40 looks good here; this
#scoring will vary from motif to motif

pdf("summary_5mice/Figure4/Th1_Motif7_Test_mice_score.pdf", width = 8, height = 6)
plot(density(pData(data_monocle)$glam_score[!is.na(pData(data_monocle)$glam_score)]),
main = "Motif score distribution_TRA_Th1_Motif7 in Test mice", xlab = "")
abline(v = 30, col = "Red")
dev.off()

#Make a true/false filtering column based on the glam score for plotting 
pData(data_monocle)$Present <- pData(data_monocle)$glam_score > 30

# Make a new dataframe to use for statistical analysis but not plotting the cells

rm(glam_stats_M1)
rm(glam_stats_M3)


# Only include cells from mouse 2 and mouse 4, separately
glam_stats_M1 <- data_monocle$Present[which(data_monocle$mice == "M1")] %>% as.data.frame()
colnames(glam_stats_M1) <- "Present"
# Add other columns from data_monocle
glam_stats_M1$State <- data_monocle$State[which(data_monocle$mice == "M1")]
glam_stats_M1$combined_cdr3_nt <- data_monocle$combined_cdr3_nt[which(data_monocle$mice == "M1")]
glam_stats_M1$cellID <- rownames(data_monocle@phenoData)[which(data_monocle$mice == "M1")]
# Remove any rows with NA (cells without a motif score, incluing cells that don't meet definition of a clone)
glam_stats_M1 <- glam_stats_M1[!is.na(glam_stats_M1$Present), ]

glam_stats_M3 <- data_monocle$Present[which(data_monocle$mice == "M3")] %>% as.data.frame()
colnames(glam_stats_M3) <- "Present"
# Add other columns from data_monocle
glam_stats_M3$State <- data_monocle$State[which(data_monocle$mice == "M3")]
glam_stats_M3$combined_cdr3_nt <- data_monocle$combined_cdr3_nt[which(data_monocle$mice == "M3")]
glam_stats_M3$cellID <- rownames(data_monocle@phenoData)[which(data_monocle$mice == "M3")]
# Remove any rows with NA (cells without a motif score, incluing cells that don't meet definition of a clone)
glam_stats_M3 <- glam_stats_M3[!is.na(glam_stats_M3$Present), ]



#Make NA values false
pData(data_monocle)$Present[is.na(pData(data_monocle)$Present)] <- FALSE
plot_obj <- plot_complex_cell_trajectory(data_monocle, color_by = "Present", cell_size = 2)
plot_obj$data <- plot_obj$data[order(plot_obj$data$Present),]
plot_obj + scale_color_manual(values=c("grey", "red")) + ggtitle("TRA_Th1_motif7_30_across_M1M3'\n Score > 30 \n ")

pdf("summary_5mice/Figure4/Th1_Motif7_Test_mice_TREE.pdf", width = 8, height = 6, useDingbats = FALSE)
plot_obj + scale_color_manual(values=c("grey", "red")) + ggtitle("TRA_TH1_motif7_M1M3.pdf'\n Score > 30 \n 'P value 0.1'")
dev.off()

#Create a table detailing the number of cells in each monocle state that have a good motif match
# The line below was from Robert's analysis, I'm replacing it because there was an error
# glam_score_table <- as.data.frame(table(pData(data_no_6_mono_cds)$State, pData(data_no_6_mono_cds)$Present))
rm(glam_score_table_M1)
rm(glam_score_table_M3)

glam_score_table_M1 <- table(glam_stats_M1$State, glam_stats_M1$Present) %>% as.data.frame()
#Combine true and false rows
glam_score_table_M1 <- cbind(glam_score_table_M1[1:5,], glam_score_table_M1[6:10,])
#Create names that make more sense
colnames(glam_score_table_M1) <- c("State", "F", "Freq_F", "State2", "T", "Freq_T")

#Remove unneeded columns
glam_score_table_M1 <- glam_score_table_M1[,c(1,3,6)]
#Create a percent column (percent of cells within each state with a good motif match)
glam_score_table_M1$pct <- glam_score_table_M1$Freq_T / (glam_score_table_M1$Freq_F + glam_score_table_M1$Freq_T)
#write.csv(glam_score_table_M2, "Motif/train/TFH/score_file/TRA_TFH_T65_motif1_M2_cln4_cells_across_M2.csv")
write.csv(glam_score_table_M1, "Motif_5mice/Test/Th1/Score/TRA_Th1_T65_motif7_M1_cln4_cells_across_M1_cells.csv")
getwd()
glam_score_table_M1$Freq_F %>% sum () # 835
glam_score_table_M1$Freq_T %>% sum() # 43

glam_stats_M1$combined_cdr3_nt[which(glam_stats_M1$Present == TRUE)] %>% table()

#TRA:TGTGCAGCAAGTTACAGCAACTATCAGTTGATCTGG_TRB:TGTGCCAGCAGTTTCCCGGACAGGGGAGACACCCAGTACTTT 
#                                                                                     43 

#Create a table detailing the number of cells in each monocle state that have a good motif match
# The line below was from Robert's analysis, I'm replacing it because there was an error
# glam_score_table <- as.data.frame(table(pData(data_no_6_mono_cds)$State, pData(data_no_6_mono_cds)$Present))

glam_score_table_M3 <- table(glam_stats_M3$State, glam_stats_M3$Present) %>% as.data.frame()
#Combine true and false rows
glam_score_table_M3 <- cbind(glam_score_table_M3[1:5,], glam_score_table_M3[6:10,]) ## this # depends on the # of monocle states
#Create names that make more sense
colnames(glam_score_table_M3) <- c("State", "F", "Freq_F", "State2", "T", "Freq_T")

#Remove unneeded columns
glam_score_table_M3 <- glam_score_table_M3[,c(1,3,6)]
#Create a percent column (percent of cells within each state with a good motif match)
glam_score_table_M3$pct <- glam_score_table_M3$Freq_T / (glam_score_table_M3$Freq_F + glam_score_table_M3$Freq_T)
#write.csv(glam_score_table_M4, "Motif/train/TFH/score_file/TRA_TFH_T65_motif1_M4_cln4_cells_across_M4.csv")
write.csv(glam_score_table_M3, "Motif_5mice/Test/Th1/Score/TRA_Th1_T65_motif7_M3_cln4_cells_across_M3_cells.csv")
getwd()
glam_score_table_M3$Freq_F %>% sum () # 2417
glam_score_table_M3$Freq_T %>% sum() # NA 


```

```{r make heatmap to show clonal dist using all clones with at least 2 cells from 5 mice}
## including all clones having >=2 cells with the overlapped ones
rm(list=ls())


data_clones_by_mouse_TRA_TRB <- read.csv("summary_5M_CCA_V2/data_clones_by_mouse_TRA_TRB_5mice.csv")

## add a column named CDR3nt_Ident with orig.ident and cdr3nt information to keep tracking of overlapped 4 clones
data_clones_by_mouse_TRA_TRB$CDR3nt_Ident <- paste0(
  data_clones_by_mouse_TRA_TRB$combined_cdr3_nt,
  data_clones_by_mouse_TRA_TRB$mice,
  sep = "-"
)

write.csv(data_clones_by_mouse_TRA_TRB, file = "summary_5M_CCA_V2/data_clones_by_mouse_TRA_TRB_5mice.csv")

length(unique(data_clones_by_mouse_TRA_TRB$CDR3nt_Ident)) # 673 w/ considering overlapped 4 clones

length(unique(data_clones_by_mouse_TRA_TRB$combined_cdr3_nt)) # 669 w/0 considering overlapped 4 clones

data_clones_by_mouse_TRA_TRB_2cell <- data_clones_by_mouse_TRA_TRB[data_clones_by_mouse_TRA_TRB$Freq >= 2, ]
# 673 true clones with >= 2 cells


length(unique(data_clones_by_mouse_TRA_TRB_2cell$CDR3nt_Ident)) # 673 clones if including overlapped clones
length(unique(data_clones_by_mouse_TRA_TRB_2cell$combined_cdr3_nt)) # 669 if considering overlapped clones collapsed
write.csv(data_clones_by_mouse_TRA_TRB_2cell, file = "data_clones_by_mouse_TRA_TRB_2cell.csv")



# data_clones DF to use for heatmap ###########################################
clone3Lin2 <- data_clones_by_mouse_TRA_TRB %>% filter (Freq >= 2) %>% select(Monocle_Th1, Monocle_Tcmp, Monocle_TFH) %>% as.matrix() # 673 clones

# Make it so all positions in the matrix with more than 8 cells count as having 8 cells
# THis is done to simplify the heatmap and matches the heatmap used in Figure 1h of the Zhang paper (doi:10.1038/s41586-018-0694-x)
clones3Lin2Cutoff <- clone3Lin2
clones3Lin2Cutoff[clones3Lin2Cutoff >= 8] <- 8
clones3Lin2Cutoff

# Filter clones to only keep clones with at least 4 cells
clones3Lin2Cutoff <- clones3Lin2Cutoff[which(rowSums(clones3Lin2Cutoff) >= 2), ]

        
# How many clones in each lineage have at least 1 one cell?
colSums(clone3Lin2 > 0)

 #Monocle_Th1 Monocle_Tcmp  Monocle_TFH 
  #   499          292          567 

rm(heatmapS7)
rm(heatmapS7_ord)
# Make heatmap in sections, one for each "block" lineages (e.g. only TFH, TFH and Tcmp, all 3, etc.)
# 7 sections total
heatmapS1 <- clones3Lin2Cutoff[
  intersect(
    which(clones3Lin2Cutoff[, 1] > 0),
    intersect(
      which(clones3Lin2Cutoff[, 2] > 0),
      which(clones3Lin2Cutoff[, 3] > 0)
    )
  )
, ]

heatmapS2 <- clones3Lin2Cutoff[
  intersect(
    which(clones3Lin2Cutoff[, 1] == 0),
    intersect(
      which(clones3Lin2Cutoff[, 2] > 0),
      which(clones3Lin2Cutoff[, 3] > 0)
    )
  )
, ]

heatmapS3 <- clones3Lin2Cutoff[
  intersect(
    which(clones3Lin2Cutoff[, 1] > 0),
    intersect(
      which(clones3Lin2Cutoff[, 2] == 0),
      which(clones3Lin2Cutoff[, 3] > 0)
    )
  )
, ]

heatmapS4 <- clones3Lin2Cutoff[
  intersect(
    which(clones3Lin2Cutoff[, 1] > 0),
    intersect(
      which(clones3Lin2Cutoff[, 2] > 0),
      which(clones3Lin2Cutoff[, 3] == 0)
    )
  )
, ] # this has one clone only

heatmapS5 <- clones3Lin2Cutoff[
  intersect(
    which(clones3Lin2Cutoff[, 1] > 0),
    intersect(
      which(clones3Lin2Cutoff[, 2] == 0),
      which(clones3Lin2Cutoff[, 3] == 0)
    )
  )
, ]

heatmapS6 <- clones3Lin2Cutoff[
  intersect(
    which(clones3Lin2Cutoff[, 1] == 0),
    intersect(
      which(clones3Lin2Cutoff[, 2] > 0),
      which(clones3Lin2Cutoff[, 3] == 0)
    )
  )
, ] # none 

heatmapS7 <- clones3Lin2Cutoff[
  intersect(
    which(clones3Lin2Cutoff[, 1] == 0),
    intersect(
      which(clones3Lin2Cutoff[, 2] == 0),
      which(clones3Lin2Cutoff[, 3] > 0)
    )
  )
, ]

heatmapS1_ord <- heatmapS1[order(heatmapS1[, 1], heatmapS1[, 3], heatmapS1[, 2], decreasing = T),]
heatmapS2_ord <- heatmapS2[order(heatmapS2[, 3], heatmapS2[, 2], decreasing = T),]
heatmapS3_ord <- heatmapS3[order(heatmapS3[, 1], heatmapS3[, 3], decreasing = T),]
heatmapS4_ord <- heatmapS4[order(heatmapS4[, 1], heatmapS4[, 2], decreasing = T),]
heatmapS5_ord <- heatmapS5[order(heatmapS5[, 1], decreasing = T),]
heatmapS6_ord <- heatmapS6[order(heatmapS6[, 2], decreasing = T),]
heatmapS7_ord <- heatmapS7[order(heatmapS7[, 3], decreasing = T),]

rm(Concat3Lin)
# add t() for which you get only one clone
Concat3Lin <- rbind(heatmapS1_ord, heatmapS2_ord, heatmapS3_ord, heatmapS4_ord, heatmapS5_ord, heatmapS6_ord, heatmapS7_ord)
rownames(Concat3Lin) <- 1:nrow(Concat3Lin)

rm(heatmap_annot)
# Make data frame of annotations to show on side
heatmap_annot <- data.frame(Lineages = rep("", nrow(clone3Lin2)))
heatmap_annot$Lineages <- as.character(heatmap_annot$Lineages)

#heatmap_annot <- Concat3Lin
#heatmap_annot <- as.data.frame(heatmap_annot)
#heatmap_annot$Tcmp <- NULL # why did I do this?
#heatmap_annot$Th1 <- NULL
#colnames(heatmap_annot) <- c("Lineages")
heatmap_annot$Lineages[1:nrow(heatmapS1)] <- "Th1, Tcmp, and TFH"
heatmap_annot$Lineages[(1+nrow(heatmapS1)):(nrow(heatmapS1)+nrow(heatmapS2))] <- "Tcmp and TFH"
heatmap_annot$Lineages[(1+nrow(heatmapS1)+nrow(heatmapS2)):(nrow(heatmapS1)+nrow(heatmapS2)+nrow(heatmapS3))] <- "Th1 and TFH"
heatmap_annot$Lineages[(1+nrow(heatmapS1)+nrow(heatmapS2)+nrow(heatmapS3)):(nrow(heatmapS1)+nrow(heatmapS2)+nrow(heatmapS3)+nrow(heatmapS4))] <- "Th1 and Tcmp"
heatmap_annot$Lineages[(1+nrow(heatmapS1)+nrow(heatmapS2)+nrow(heatmapS3)+nrow(heatmapS4)):(nrow(heatmapS1)+nrow(heatmapS2)+nrow(heatmapS3)+nrow(heatmapS4)+nrow(heatmapS5))] <- "Th1 only"
heatmap_annot$Lineages[(1+nrow(heatmapS1)+nrow(heatmapS2)+nrow(heatmapS3)+nrow(heatmapS4)+nrow(heatmapS5)):(1+nrow(heatmapS1)+nrow(heatmapS2)+nrow(heatmapS3)+nrow(heatmapS4)+nrow(heatmapS5)+nrow(heatmapS6))] <- "Tcmp only"
heatmap_annot$Lineages[(2+nrow(heatmapS1)+nrow(heatmapS2)+nrow(heatmapS3)+nrow(heatmapS4)+nrow(heatmapS5)+nrow(heatmapS6)):(0+nrow(heatmapS1)+nrow(heatmapS2)+nrow(heatmapS3)+nrow(heatmapS4)+nrow(heatmapS5)+nrow(heatmapS6)+nrow(heatmapS7))] <- "TFH only"

heatmap_annot$Lineages <- factor(heatmap_annot$Lineages)

# Make it so all positions in the matrix with more than 10 cells count as having 10 cells
# THis is done to simplify the heatmap
Concat3Lin[Concat3Lin >= 8] <- 8

#Colors_concat <- inferno(11)
Colors_concat <- rev(heat.colors(9)) # these lines are from the previous script to make scale yellow and red
Colors_concat[1] <- "gray"


# Make new colors for lineage groups so colors stay consistent in later graphs
annotat_colors <- c("black", "red3", "gray40", "darkorange3", "darkred", "lightblue", "darkgreen")
names(annotat_colors) <- unique(heatmap_annot$Lineages)
annotat_colors <- list(Lineages = annotat_colors)

# Heatmap
#undebug(pheatmap)

library(pheatmap)

pheatmap(Concat3Lin, decreasing = T, show_rownames = F, show_colnames = T, cluster_rows = F, cluster_cols = F, breaks = c(-1:8), color = Colors_concat, annotation_row = heatmap_annot, border_color = NA, annotation_colors = annotat_colors, angle_col = 0)

pdf("summary_5mice/Figure3/Heatmap_2cell_clone_5mice.pdf", width = 10, height = 8)
pheatmap(Concat3Lin, decreasing = T, show_rownames = F, show_colnames = T, cluster_rows = F, cluster_cols = F, breaks = c(-1:8), color = Colors_concat, annotation_row = heatmap_annot, border_color = NA, annotation_colors = annotat_colors, angle_col = 0)
dev.off()

# Percentages of each Lineage block - 4 cells
# All 3 - 203/673 = 
# Tcmp and TFH - 60/673 = 
# Th1 and TFH - 190/673 = 
# Tcmp and Th1 - 29/673 = 
# Th1 - 77/673 = 
# Tcmp - 0/673 = 
# Tfh - 114/673 = 

# Multiple fate = 203+ 60+ 190+ 29 = 482*100/673 = 71.6%
# one fate = 77+ 114 = 191*100/673 = 28.2% 
# so this is better than before 

# Tcmp to Multiple fate = 203+ 60+ 29 = 2922*100/673 = ~44%
# one fate or TFH and Th1 fated = 77+ 114 + 190 = 381*100/673 = ~56%
# so this is better than before 
```


```{r clone size distribution using clone3Lin2 DF}
# for the clone size distribution we will use data clones by mouse TRA TRB with the overlapped clones DF that has the overlapped clones 

rm(list=ls())
colnames(data_clones_by_mouse_TRA_TRB)

data_clones_by_mouse_TRA_TRB <- read.csv("summary_5M_CCA_V2/data_clones_by_mouse_TRA_TRB_5mice.csv")
colnames(data_clones_by_mouse_TRA_TRB)

# I want to include combined CDR3nt info as well with CDR3nt_Ident
clone3Lin5Df <- data_clones_by_mouse_TRA_TRB %>% filter(Freq >= 4) %>% select(combined_cdr3_nt, CDR3nt_Ident, mice, Monocle_Th1, Monocle_Tcmp, Monocle_TFH, Percent_Monocle_Th1, Percent_Monocle_Tcmp, Percent_Monocle_TFH) %>% as.data.frame()
# 352 clones


clone3Lin5Df$Total_cells <- rowSums(clone3Lin5Df[, 4:6])

# Make a new column that specifies cell fate (Th1, TFH, Tcmp, multifate)
clone3Lin5Df$Fate <- ""

for (i in 1:nrow(clone3Lin5Df)) {
  if(clone3Lin5Df$Percent_Monocle_Th1[i] >= 65) {
    clone3Lin5Df$Fate[i] <- "Th1"
  }
  if(clone3Lin5Df$Percent_Monocle_TFH[i] >= 65) {
    clone3Lin5Df$Fate[i] <- "TFH"
  }
  if(clone3Lin5Df$Percent_Monocle_Tcmp[i] >= 65) {
    clone3Lin5Df$Fate[i] <- "Tcmp"
  }    
}

clone3Lin5Df$Fate[which(clone3Lin5Df$Fate == "")] <- "Multifate"
clone3Lin5Df$Fate <- as.factor(clone3Lin5Df$Fate)
write.csv (clone3Lin5Df, file = "summary_5mice/Figure3/clone3Lin5Df_65.csv")

clone3Lin5Df$clone_number <- 1:nrow(clone3Lin5Df)

clone3Lin5Df$Fate %>% table()  ## here we can Tcmp baised clone but not seeing anything like this in heatmap cause this biasness is counted by 65% of cells in a clone

# Multifate      Tcmp     TFH        Th1 
 #     160         1       128        63 

### plot legend are always in alphabetical order; MPC used to come before Multifate, but now Tcmp comes after Multifate; so just switch the color in order in labels in scale_color_manual ()#####

png(file = "summary_5mice/clone_size_distribution_with_65.png", width = 10, height = 8, units = "in", res = 300)
ggplot(clone3Lin5Df, aes(x = reorder(clone_number, Total_cells), y = Total_cells )) +
 geom_bar(stat = "identity", aes(fill = Fate)) +
  theme_minimal() +
  scale_color_manual(values = c("darkgray", "darkblue", "darkgreen", "darkred"), labels = c("Multifate", "Tcmp", expression(T[FH]), "Th1"), aesthetics = "fill") +
  ggtitle("65% Threshold") +
  theme(axis.text.x = element_blank(), panel.grid = element_blank(), text = element_text(size = 20), plot.title = element_text(hjust = 0.5)) +
  guides(fill = guide_legend(label.hjust = 0)) +
  xlab("Clones") +
  ylab("Total cells")
dev.off()

pdf(file = "summary_5mice/Figure3/clone_size_distribution_with_65.pdf", width = 8, height = 6)
ggplot(clone3Lin5Df, aes(x = reorder(clone_number, Total_cells), y = Total_cells )) +
 geom_bar(stat = "identity", aes(fill = Fate)) +
  theme_minimal() +
  scale_color_manual(values = c("darkgray", "darkblue", "darkgreen", "darkred"), labels = c("Multifate", "Tcmp", expression(T[FH]), "Th1"), aesthetics = "fill") +
  ggtitle("65% Threshold") +
  theme(axis.text.x = element_blank(), panel.grid = element_blank(), text = element_text(size = 20), plot.title = element_text(hjust = 0.5)) +
  guides(fill = guide_legend(label.hjust = 0)) +
  xlab("Clones") +
  ylab("Total cells")
dev.off()


# check with 55% and 75% thresholds

# 55%
rm(clone3Lin5Df)

clone3Lin5Df <- data_clones_by_mouse_TRA_TRB %>% filter(Freq >= 4) %>% select(combined_cdr3_nt, CDR3nt_Ident, mice, Monocle_Th1, Monocle_Tcmp, Monocle_TFH, Percent_Monocle_Th1, Percent_Monocle_Tcmp, Percent_Monocle_TFH) %>% as.data.frame()

clone3Lin5Df$Total_cells <- rowSums(clone3Lin5Df[, 4:6])

# Make a new column that specifies cell fate (Th1, TFH, Tcmp, multifate)
clone3Lin5Df$Fate <- ""
for (i in 1:nrow(clone3Lin5Df)) {
  if (clone3Lin5Df$Percent_Monocle_Th1[i] >= 55) {
    clone3Lin5Df$Fate[i] <- "Th1"
  }
  if (clone3Lin5Df$Percent_Monocle_TFH[i] >= 55) {
    clone3Lin5Df$Fate[i] <- "TFH"
  }
  if (clone3Lin5Df$Percent_Monocle_Tcmp[i] >= 55) {
    clone3Lin5Df$Fate[i] <- "Tcmp"
  }
}
clone3Lin5Df$Fate[which(clone3Lin5Df$Fate == "")] <- "Multifate"
clone3Lin5Df$Fate <- as.factor(clone3Lin5Df$Fate)


clone3Lin5Df$Fate %>% table()

#Multifate      Tcmp       TFH       Th1 
 #      94         2       168        88 


clone3Lin5Df$clone_number <- 1:nrow(clone3Lin5Df)

write.csv (clone3Lin5Df, file = "summary_5mice/Figure3/clone3Lin5Df_4cell_55.csv")

### plot legend are always in alphabetical order; MPC used to come before Multifate, but now Tcmp comes after Multifate; so just switch the color in order in labels in scale_color_manual ()#####

png(file = "summary_5mice/Figure3/clone_size_distribution_with_55.png", width = 8, height = 6, units = "in", res = 300)
ggplot(clone3Lin5Df, aes(x = reorder(clone_number, Total_cells), y = Total_cells )) +
 geom_bar(stat = "identity", aes(fill = Fate)) +
  theme_minimal() +
  scale_color_manual(values = c("darkgray","lightblue", "darkgreen", "darkred"), labels = c("Multifate", "Tcmp", expression(T[FH]), "Th1"), aesthetics = "fill") +
  ggtitle("55% Threshold") +
  theme(axis.text.x = element_blank(), panel.grid = element_blank(), text = element_text(size = 20), plot.title = element_text(hjust = 0.5)) +
  guides(fill = guide_legend(label.hjust = 0)) +
  xlab("Clones") +
  ylab("Total cells")
dev.off()

pdf(file = "summary_5mice/Figure3/clone_size_distribution_with_55.pdf", width = 8, height = 6)
ggplot(clone3Lin5Df, aes(x = reorder(clone_number, Total_cells), y = Total_cells )) +
 geom_bar(stat = "identity", aes(fill = Fate)) +
  theme_minimal() +
  scale_color_manual(values = c("darkgray","lightblue", "darkgreen", "darkred"), labels = c("Multifate", "Tcmp", expression(T[FH]), "Th1"), aesthetics = "fill") +
  ggtitle("55% Threshold") +
  theme(axis.text.x = element_blank(), panel.grid = element_blank(), text = element_text(size = 20), plot.title = element_text(hjust = 0.5)) +
  guides(fill = guide_legend(label.hjust = 0)) +
  xlab("Clones") +
  ylab("Total cells")
  ylab("Total cells")
dev.off()

# 75%
rm(clone3Lin5Df)

clone3Lin5Df <- data_clones_by_mouse_TRA_TRB %>% filter(Freq >= 4) %>% select(combined_cdr3_nt, CDR3nt_Ident, mice, Monocle_Th1, Monocle_Tcmp, Monocle_TFH, Percent_Monocle_Th1, Percent_Monocle_Tcmp, Percent_Monocle_TFH) %>% as.data.frame()

clone3Lin5Df$Total_cells <- rowSums(clone3Lin5Df[, 4:6])

# Make a new column that specifies cell fate (Th1, TFH, Tcmp, multifate)
clone3Lin5Df$Fate <- ""
for (i in 1:nrow(clone3Lin5Df)) {
  if (clone3Lin5Df$Percent_Monocle_Th1[i] >= 75) {
    clone3Lin5Df$Fate[i] <- "Th1"
  }
  if (clone3Lin5Df$Percent_Monocle_TFH[i] >= 75) {
    clone3Lin5Df$Fate[i] <- "TFH"
  }
  if (clone3Lin5Df$Percent_Monocle_Tcmp[i] >= 75) {
    clone3Lin5Df$Fate[i] <- "Tcmp"
  }
}
clone3Lin5Df$Fate[which(clone3Lin5Df$Fate == "")] <- "Multifate"
clone3Lin5Df$Fate <- as.factor(clone3Lin5Df$Fate)


clone3Lin5Df$Fate %>% table()
# Multifate       TFH       Th1 
 #     218        92        42 


clone3Lin5Df$clone_number <- 1:nrow(clone3Lin5Df)

write.csv (clone3Lin5Df, file = "summary_5mice/Figure3/clone3Lin5Df_4cell_75.csv")

### plot legend are always in alphabetical order; MPC used to come before Multifate, but now Tcmp comes after Multifate; so just switch the color in order in labels in scale_color_manual ()#####

png(file = "summary_5mice/Figure3/clone_size_distribution_with_75.png", width = 8, height = 6, units = "in", res = 300)
ggplot(clone3Lin5Df, aes(x = reorder(clone_number, Total_cells), y = Total_cells )) +
 geom_bar(stat = "identity", aes(fill = Fate)) +
  theme_minimal() +
  scale_color_manual(values = c("darkgray",  "darkgreen", "darkred"), labels = c("Multifate", expression(T[FH]), "Th1"), aesthetics = "fill") +
  ggtitle("75% Threshold") +
  theme(axis.text.x = element_blank(), panel.grid = element_blank(), text = element_text(size = 20), plot.title = element_text(hjust = 0.5)) +
  guides(fill = guide_legend(label.hjust = 0)) +
  xlab("Clones") +
  ylab("Total cells")
dev.off()

pdf(file = "summary_5mice/Figure3/clone_size_distribution_with_75.pdf", width = 8, height = 6)
ggplot(clone3Lin5Df, aes(x = reorder(clone_number, Total_cells), y = Total_cells )) +
 geom_bar(stat = "identity", aes(fill = Fate)) +
  theme_minimal() +
  scale_color_manual(values = c("darkgray",  "darkgreen", "darkred"), labels = c("Multifate", expression(T[FH]), "Th1"), aesthetics = "fill") +
  ggtitle("75% Threshold") +
  theme(axis.text.x = element_blank(), panel.grid = element_blank(), text = element_text(size = 20), plot.title = element_text(hjust = 0.5)) +
  guides(fill = guide_legend(label.hjust = 0)) +
  xlab("Clones") +
  ylab("Total cells")
dev.off()
```

```{r r Startrac transition box plots for 5 mice}

# load seurat and monocle object to make DF with all mouse data to be used for Motif, heatmap and clone size dist
rm(list=ls())
## load seurat and monocle object to make DF with all mouse data to be used for Motif, heatmap and clone size dist
#rm(immune.combined_edit2_2cell)
immune.combined_edit3_2cell <- readRDS("summary_5M_CCA_V2/immune.combined_edit3_2cell.RDS")
TSNEPlot(immune.combined_edit3_2cell)

data_monocle <- readRDS("summary_5M_CCA_V2/data_monocle_2cell.RDS")

## state 1 Th1; 2,3 Tcmp; 4,5 TFH


### add cell ident to monocle branches
data_monocle$Branch_3subset <- data_monocle$State %>% as.character()
data_monocle$Branch_3subset <- gsub(pattern = "4|5", replacement = "TFH", x = data_monocle$Branch_3subset)
data_monocle$Branch_3subset <- gsub(pattern = "2|3", replacement = "Tcmp", x = data_monocle$Branch_3subset)
data_monocle$Branch_3subset <- gsub(pattern = "1", replacement = "Th1", x = data_monocle$Branch_3subset)
data_monocle$Branch_3subset %>% unique()
#[1] "TFH"  "Th1"  "Tcmp"

## input for startrac
inputStartrac_3Lin <- data.frame(
  Cell_Name = rownames(data_monocle@phenoData@data), 
  clone.id = data_monocle$combined_cdr3_nt,
  clone.status = "Clonal",
  patient = data_monocle$mice,
  sampleType = "Sample",
  stype = "CD4",
  majorCluster = data_monocle$Branch_3subset, # State uses states calculated using Monocle
  loc = rep("Mouse", nrow(data_monocle@phenoData@data)),
  stringsAsFactors = F
  )

# Check to see what type of vector each column is
head(inputStartrac_3Lin)

# majorCluster is a factor, convert to character
inputStartrac_3Lin$majorCluster <- inputStartrac_3Lin$majorCluster %>% as.character()

# Try step by step approach from vignette page
rm(outputStartrac)
outputStartrac_3Lin <- new("Startrac", inputStartrac_3Lin, aid = "CRC")
outputStartrac_3Lin <- calIndex(outputStartrac_3Lin)

# Rearrange factor levels so x axis is arranged as Th1, Tcmp, TFH
# To be used for plots later


# Transition Plot # this seems more applicable cause ours is based on Monocle
ggplot(data = outputStartrac_3Lin@cluster.data, aes(x = majorCluster, y = tran, fill = majorCluster)) +
  geom_bar(stat = "identity") +
  geom_text(aes(x = majorCluster, y = tran + 0.03, label = round(tran, digits = 3))) +
  xlab("Lineage") +
  ylab("STARTRAC Transition Score") +
  scale_fill_discrete(
    name = "Lineage",
    labels = c(
      expression(Tcmp),
      expression("T"[FH]),
      expression(Th1)
  )) +
  scale_x_discrete(name = "Lineage", labels = c(
    expression(Tcmp),
      expression("T"[FH]),
      expression(Th1) 
    )) +
    theme(legend.text.align = 0)

```

```{r startrac by mouse for 5 mice}
outputStartrac_3Lin_M1 <- new("Startrac", inputStartrac_3Lin[which(inputStartrac_3Lin$patient == "M1"), ], aid = "CRC")
outputStartrac_3Lin_M1 <- calIndex(outputStartrac_3Lin_M1)

outputStartrac_3Lin_M2 <- new("Startrac", inputStartrac_3Lin[which(inputStartrac_3Lin$patient == "M2"), ], aid = "CRC")
outputStartrac_3Lin_M2 <- calIndex(outputStartrac_3Lin_M2)

outputStartrac_3Lin_M3 <- new("Startrac", inputStartrac_3Lin[which(inputStartrac_3Lin$patient == "M3"), ], aid = "CRC")
outputStartrac_3Lin_M3 <- calIndex(outputStartrac_3Lin_M3)

outputStartrac_3Lin_M4 <- new("Startrac", inputStartrac_3Lin[which(inputStartrac_3Lin$patient == "M4"), ], aid = "CRC")
outputStartrac_3Lin_M4 <- calIndex(outputStartrac_3Lin_M4)

outputStartrac_3Lin_M6 <- new("Startrac", inputStartrac_3Lin[which(inputStartrac_3Lin$patient == "M6"), ], aid = "CRC")
outputStartrac_3Lin_M6 <- calIndex(outputStartrac_3Lin_M6)

outputStartrac_3Lin_bymouse <- rbind(outputStartrac_3Lin_M1@cluster.data,
                                     outputStartrac_3Lin_M2@cluster.data, outputStartrac_3Lin_M3@cluster.data, outputStartrac_3Lin_M4@cluster.data,
                                     outputStartrac_3Lin_M6@cluster.data)

outputStartrac_3Lin_bymouse$mice <- c(rep("M1", 3), rep("M2", 3), rep("M3", 3), rep("M4", 3), rep("M6", 3))

# Rearrange factor levels so x axis is arranged as Th1, Tcmp, TFH
# To be used for plots later
outputStartrac_3Lin_bymouse$Cluster_ord <- factor(
  outputStartrac_3Lin_bymouse$majorCluster,
  levels = levels(factor(outputStartrac_3Lin_bymouse$majorCluster))[c(3, 1, 2)])

# cluster_ord to use to maintain the order
# Transition boxplot by mouse
pdf(file = "summary_5mice/Figure3/Transition_boxplot_5mice.pdf", width = 10, height = 8, useDingbats = FALSE)
ggplot(
  data = outputStartrac_3Lin_bymouse,
  aes(
    x = Cluster_ord,
    y = tran,
    fill = Cluster_ord)) +
  geom_boxplot() +
  geom_dotplot(mapping = aes(color = mice), binaxis = "y", stackdir = "center", dotsize = 1, stroke = 4) +
  xlab("Lineage") +
  ylab("LPI Score") +
  scale_fill_manual(
    name = "Lineage",
    labels = c(
      expression(Th1),
      expression(Tcmp),
      expression("T"[FH])),
    values = c("darkred", "lightblue", "darkgreen")) +
  scale_color_manual(
    name = "Mouse",
    labels = c("M1", "M2", "M3", "M4", "M6"),
    values = c("#1B9E77", "#D95F02", "#7570B3", "#E7298A", "#66A61E")) +
  scale_x_discrete(name = "Lineage", labels = c(
    expression(Th1),
      expression(Tcmp),
      expression("T"[FH]) 
    )) +
  theme_minimal() +
  theme(
     element_blank())
dev.off()
#ggsave("summary_5mice/Figure3/Startrac_Transition_Mouse.png", dpi = 300)
ggsave("summary_5mice/Figure3/Startrac_Transition_Mouse.pdf", dpi = 300, useDingbats = FALSE)


## save file upon which transition plot based on

write.csv(outputStartrac_3Lin_bymouse, file = "summary_5mice/Figure3/LPI_score_by_mouse.csv")

```

```{r Add Th1 and TFH module score to clones having >= 4 cells and defined by 65% threshold}
rm(list = ls())
immune.combined_edit3_2cell <- readRDS("summary_5M_CCA_V2/immune.combined_edit3_2cell.RDS")



clone3Lin5Df_65 <- read.csv("summary_5mice/Figure3/clone3Lin6Df_65.csv", row.names = 1)


clone3Lin5Df_65 <- read.csv("summary_5mice/Figure3/clone_65thresh_4cell_352_09262020.csv", row.names = 1)



# Add columns for module scores to clone DF
clone3Lin5Df_65$Th1_module_score <- 0
clone3Lin5Df_65$TFH_module_score <- 0

# Module score for each clone will be the average module score for all cells in that clone
### Need to account for both combined cdr3 nt sequence and orig.ident because of overlapping clones
for (i in 1:nrow(clone3Lin5Df_65)) {
  
  clone3Lin5Df_65$Th1_module_score[i] <- 
    mean(
      immune.combined_edit3_2cell@meta.data$Geneset_Th11[
        which(immune.combined_edit3_2cell@meta.data$combined_cdr3_nt == clone3Lin5Df_65$combined_cdr3_nt[i] &
                immune.combined_edit3_2cell@meta.data$mice == clone3Lin5Df_65$mice[i])])
  
  clone3Lin5Df_65$TFH_module_score[i] <- 
    mean(
      immune.combined_edit3_2cell@meta.data$Geneset_Tfh1[
        which(immune.combined_edit3_2cell@meta.data$combined_cdr3_nt == clone3Lin5Df_65$combined_cdr3_nt[i] &
                immune.combined_edit3_2cell@meta.data$mice == clone3Lin5Df_65$mice[i])])
}

### Check a couple examples manually to make sure it worked

clone3Lin5Df_65$combined_cdr3_nt[300]
## TRA:TGTGCAGCAATCTATGGGAGCAGTGGCAACAAGCTCATCTTT_TRB:TGTGCCTGGAGTCTATCGGGTTATGAACAGTACTTC
#grep("TRA:TGTGCAGCAAAAGGGTCTAATTACAACGTGCTTTACTTC_TRB:TGTGCCTGGAGTCTACTGGGGGATGAACAGTACTTC", clone3Lin5Df_65$combined_cdr3_nt)
#mean(
   #immune.combined_edit3_2cell@meta.data$Geneset_Tfh1[
  #  which( immune.combined_edit3_2cell@meta.data$combined_cdr3_nt == "TRA:TGTGCAGCAAAAGGGTCTAATTACAACGTGCTTTACTTC_TRB:TGTGCCTGGAGTCTACTGGGGGATGAACAGTACTTC")])
## 0.1990
clone3Lin5Df_65$Geneset_TFH[300]
## 0.1990

write.csv(clone3Lin5Df_65, file = "summary_5mice/Figure3/cellsubset_module_score_65thresh_4cell_352.csv")

```

```{r}
### Do logistic regression to validate pre-determined cell fate and Th1/Tfh module score at the clone level

clone3Lin5Df_65$Th1 <- 0
clone3Lin5Df_65$Th1[which(clone3Lin5Df_65$Fate == "Th1")] <- 1
clone3Lin5Df_65$Th1 <- factor(clone3Lin5Df_65$Th1)

clone3Lin5Df_65$TFH <- 0
clone3Lin5Df_65$TFH[which(clone3Lin5Df_65$Fate == "TFH")] <- 1
clone3Lin5Df_65$TFH <- factor(clone3Lin5Df_65$TFH)

summary(glm(clone3Lin5Df_65$Th1 ~ clone3Lin5Df_65$Th1_module_score, family = "binomial"))

fit_Th1 <- summary(glm(clone3Lin5Df_65$Th1 ~ clone3Lin5Df_65$Th1_module_score, family = "binomial")) # this shows exact p value if not shown
fit_Th1$coefficients
# exact p value 4.21e-13

summary(glm(clone3Lin5Df_65$TFH ~ clone3Lin5Df_65$TFH_module_score, family = "binomial"))

fit_TFH <- summary(glm(clone3Lin5Df_65$TFH ~ clone3Lin5Df_65$TFH_module_score, family = "binomial")) # this shows exact p value if not shown
fit_TFH$coefficients
# exact p value 3.411503e-18

# Th1 logistic regression plot
ggplot(clone3Lin5Df_65, aes(x = Geneset_Th1, y = as.numeric(Th1) - 1, color = Fate)) +
  geom_point(size = 2) +
  geom_smooth(method = "glm", 
              method.args = list(family = "binomial"), 
              se = FALSE,
              color = "black",
              linetype = "solid") +
  theme_minimal() +
  xlab("Th1 module score at clonal level") +
  ylab("Th1-biased clone") +
  theme(panel.grid = element_blank(), legend.text.align = 0) +
 scale_y_continuous(labels = c("No", "Yes"), breaks = c(0, 1)) +
  scale_color_manual(
    values = c("darkgray", "lightblue", "darkgreen", "darkred"),
    labels = c("Multifate", "Tcmp", expression(T[FH]), "Th1"))


ggsave("summary_5mice/Figure3/validation_Th1clones_bymodulescore_65_352clones.png", dpi = 300)
ggsave("summary_5mice/Figure3/validation_Th1clones_bymodulescore_65_352clones.pdf", dpi = 300)

# 4.21e-13


# TFH logistic regression plot
ggplot(clone3Lin5Df_65, aes(x = Geneset_TFH, y = as.numeric(TFH) - 1, color = Fate)) +
  geom_point(size = 2) +
  geom_smooth(method = "glm", 
              method.args = list(family = "binomial"), 
              se = FALSE,
              color = "black",
              linetype = "solid") +
  theme_minimal() +
  xlab("TFH module score at the clonal level") +
  ylab("TFH-biased clone") +
  theme(panel.grid = element_blank(), legend.text.align = 0) +
  scale_y_continuous(labels = c("No", "Yes"), breaks = c(0, 1)) +
  scale_color_manual(
    values = c("darkgray", "lightblue", "darkgreen", "darkred"),
    labels = c("Multifate", "Tcmp", expression(T[FH]), "Th1"))




ggsave("summary_5mice/Figure3/validation_TFHclones_bymodulescore_65_352clones.png", dpi = 300)
ggsave("summary_5mice/Figure3/validation_TFHclones_bymodulescore_65_352clones.pdf", dpi = 300)

# exact p value 3.411503e-18


rm(Acute_6_mice_edit5_2cell)
### Make a percentile rank for the module scores to compare **between** Th1 and TFH scores
### Percentile ranks based on all module scores in the data, not the averaged clone scores
## we need these to validate to significant Th1 or TFH motif speicifc clones cause we need
## to compare two scores (th1 and tfh on same clone) and they have different scale or range
## so percentile is good way to compare
immune.combined_edit3_2cell@meta.data$Th1_modulescore_percentile <- (immune.combined_edit3_2cell@meta.data$Geneset_Th11 - min(immune.combined_edit3_2cell@meta.data$Geneset_Th11)) /
  (max(immune.combined_edit3_2cell@meta.data$Geneset_Th11) - min(immune.combined_edit3_2cell@meta.data$Geneset_Th11))

immune.combined_edit3_2cell@meta.data$TFH_modulescore_percentile <- (immune.combined_edit3_2cell@meta.data$Geneset_Tfh1 - min(immune.combined_edit3_2cell@meta.data$Geneset_Tfh1)) /
  (max(immune.combined_edit3_2cell@meta.data$Geneset_Tfh1) - min(immune.combined_edit3_2cell@meta.data$Geneset_Tfh1))

# Add to the dataframe
# Percentile module score for each clone will be the average percentile module score for all cells in that clone
### Need to account for both combined cdr3 nt sequence and orig.ident because of overlapping clones
clone3Lin5Df_65$Th1_modulescore_percentile <- 0
clone3Lin5Df_65$TFH_modulescore_percentile <- 0

for (i in 1:nrow(clone3Lin5Df_65)) {
  
  clone3Lin5Df_65$Th1_modulescore_percentile[i] <- 
    mean(
      immune.combined_edit3_2cell@meta.data$Th1_modulescore_percentile[
        which(immune.combined_edit3_2cell@meta.data$combined_cdr3_nt == clone3Lin5Df_65$combined_cdr3_nt[i] &
                immune.combined_edit3_2cell@meta.data$mice == clone3Lin5Df_65$mice[i])])
  
  clone3Lin5Df_65$TFH_modulescore_percentile[i] <- 
    mean(
      immune.combined_edit3_2cell@meta.data$TFH_modulescore_percentile[
        which(immune.combined_edit3_2cell@meta.data$combined_cdr3_nt == clone3Lin5Df_65$combined_cdr3_nt[i] &
                immune.combined_edit3_2cell@meta.data$mice == clone3Lin5Df_65$mice[i])])
}

clone3Lin5Df_65 %>% filter(Th1 == 1) %>% pull(Th1_modulescore_percentile) %>% summary() # 0.63
#    Min. 1st Qu.  Median  Mean   3rd Qu.    Max. 
# 0.5297  0.5746  0.6207  0.6304  0.6715  0.8324 
clone3Lin5Df_65 %>% filter(Th1 == 1) %>% pull(TFH_modulescore_percentile) %>% summary()
#    Min. 1st Qu.  Median  Mean   3rd Qu.    Max. 
# 0.1530  0.2741  0.3267  0.3147  0.3630  0.4355 

clone3Lin5Df_65 %>% filter(TFH == 1) %>% pull(TFH_modulescore_percentile) %>% summary() # Mean 0.5 (~ .47)
#    Min. 1st Qu.  Median   Mean  3rd Qu.    Max. 
# 0.2796  0.4327  0.4726  0.4797  0.5278  0.6903 
clone3Lin5Df_65 %>% filter(TFH == 1) %>% pull(Th1_modulescore_percentile) %>% summary()
#    Min. 1st Qu.  Median  Mean    3rd Qu.    Max. 
# 0.1080  0.3239  0.3533  0.3551  0.4032  0.4860 


## check mean and range for Th1 and TFH modulescore score

clone3Lin5Df_65 %>% pull(Th1_module_score) %>% summary()
 #   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
 #0.1293  0.4883  0.6029  0.6072  0.7196  1.0916 

clone3Lin5Df_65 %>% pull(TFH_module_score) %>% summary()

#    Min.   1st Qu.   Median     Mean  3rd Qu.     Max. 
# -0.01191  0.23674  0.30355  0.30417  0.37302  0.64078 


### change colnames in clone3Lin5DF_65 (Geneset_Th1/TFH  to Th1/TFH Module score)

#colnames(clone3Lin5Df_65)[colnames(clone3Lin5Df_65) == "Geneset_TFH"] <- "TFH_Module_Score"
#colnames(clone3Lin5Df_65)[colnames(clone3Lin5Df_65) == "Geneset_Th1"] <- "Th1_Module_Score"

clone3Lin5Df_65$Th1 <- NULL
clone3Lin5Df_65$TFH <- NULL

## change M6 to M5

clone3Lin5Df_65$mice <- lapply(
  X = clone3Lin5Df_65$mice,
  FUN = function(x) gsub("M6", "M5", x)
)



clone3Lin5Df_65_TFH <- clone3Lin5Df_65[which(clone3Lin5Df_65$Fate == "TFH"), ]

TFH_0.4 <- clone3Lin5Df_65_TFH[which(clone3Lin5Df_65_TFH$TFH_modulescore_percentile > 0.4), ] 

TFH_0.4[which(TFH_0.4$Th1_modulescore_percentile >= 0.4), ] %>% length() # 15

TFH_0.3 <- clone3Lin5Df_65_TFH[which(clone3Lin5Df_65_TFH$TFH_modulescore_percentile > 0.3), ]
TFH_0.3[which(TFH_0.3$Th1_modulescore_percentile >= 0.3), ] %>% length() # 15 

# based on percentile module score, 15 clones out of 128 has a little higher Th1 modulescore_percentile than TFH modulescore percentile

clone3Lin5Df_65 <- apply(clone3Lin5Df_65, 2, as.character) # this is to avoid error "unimplemented type 'list' in 'EncodeElement'" # "2" does not mean column #
write.csv(clone3Lin5Df_65, file = "summary_5mice/Figure3/clone_65thresh_4cell_352_09262020.csv") # this DF has T cell subset sp module scores plus percentile scores as well

```

#Th1 and TFH module score is calculated at the clonal level by taking the average module score for all cells in that clone.


#Two module scores have different distribution with different range and different means, so they can't be compared directly. In order to compare them directly, we calculated percentile within the distribution of each module score for each cell or clone. Later percentile module scores were assigned at the clonal level by taking the average percentile module scores for all cells in that clone


```{r UMAP for the abstract}

immune.combined_edit2_2cell <- readRDS("summary_5M_CCA_V2/immune.combined_edit2_2cell.RDS")
data_clones_by_mouse_TRA_TRB <- read.csv("summary_5mice/data_clones_by_mouse_TRA_TRB_5mice.csv")



pdf(file = "summary_5mice/Figure3/Multifate_clone1.pdf", width = 8, height = 6)
TSNEPlot(
  immune.combined_edit3_2cell,
  ncol = 2,
  cells.highlight = rownames(immune.combined_edit3_2cell@meta.data)[
    which(immune.combined_edit3_2cell@meta.data$combined_cdr3_nt == data_clones_by_mouse_TRA_TRB$combined_cdr3_nt[1])], # Change clone number here
  cols.highlight = "grey", # Change color here
  pt.size = 0.5,
  sizes.highlight = 1.75, colors.use = c("grey", "darkblue")) #+
  #NoLegend()
dev.off()

```


```{r make DF for TFH motif sp significan clones in train mice M2 M4 M6}


#write.csv(clone3Lin6Df_65, file = "summary_6mice/clone3Lin6Df_65_wpercentile.csv")
clone3Lin5Df_65 <- read.csv("summary_5mice/Figure3/clone_65thresh_4cell_352.csv")

## change M6 to M5 first

clone3Lin5Df_65$mice <- lapply( 
               X = clone3Lin5Df_65$mice,
               FUN = function(x) gsub("M6", "M5", x))

clone3Lin5Df_65$CDR3nt_Ident <- lapply( 
               X = clone3Lin5Df_65$CDR3nt_Ident,
               FUN = function(x) gsub("M6", "M5", x))
## we need to filter this clone3Lin6Df DF to get the clones and percetile scores for the clones found significnt for
## TFH specific motif 1, 2, 5, 10, 16 in train mice M2,M4,M6
## lets first separate out those clones for motif 6 and motif 10 in test mice


# TFH Motif 1 in M2

# TRA:TGTGCAGCAAGTCTCGATGGAGGCTATAAAGTGGTCTTT_TRB:TGTGCTAGCAGTTTATCCGGGACAGGGGACGCTGAGCAGTTCTTC 
#                                                                                           31 
#   TRA:TGTGCAGTGAGCGGGACTGGAGGCTATAAAGTGGTCTTT_TRB:TGTGCCAGCAGGACTGGGGGGGCAGGGGACACCCAGTACTTT 
#                                                                                           13 
#      TRA:TGTGCTCTGGGCCTGACTGGAGGCTATAAAGTGGTCTTT_TRB:TGCAGTGCAAGGGGGGCGGACACCGGGCAGCTCTACTTT 
 #                                                                                          24 


grep(pattern = "TRA:TGTGCAGCAAGTCTCGATGGAGGCTATAAAGTGGTCTTT_TRB:TGTGCTAGCAGTTTATCCGGGACAGGGGACGCTGAGCAGTTCTTC", x= clone3Lin5Df_65$combined_cdr3_nt) # 52

grep(pattern = "TRA:TGTGCAGTGAGCGGGACTGGAGGCTATAAAGTGGTCTTT_TRB:TGTGCCAGCAGGACTGGGGGGGCAGGGGACACCCAGTACTTT", x= clone3Lin5Df_65$combined_cdr3_nt) # 128

grep(pattern = "TRA:TGTGCTCTGGGCCTGACTGGAGGCTATAAAGTGGTCTTT_TRB:TGCAGTGCAAGGGGGGCGGACACCGGGCAGCTCTACTTT", x= clone3Lin5Df_65$combined_cdr3_nt) # 63

# TFH Motif 1 in M4

# TRA:TGCTCAGCCCCCGGGACTGGAGGCTATAAAGTGGTCTTT_TRB:TGTGCCAGCAGCCAACAAAACACAGAAGTCTTCTTT 
  #                                                                                              5 
 # TRA:TGTGCAGCAAGTTGGGGGACTGGAGGCTATAAAGTGGTCTTT_TRB:TGTGCTAGCAGTTTTGCTGGGGGGGCTTCCTATGAACAGTACTTC 
  #                                                                                            805 

grep(pattern = "TRA:TGCTCAGCCCCCGGGACTGGAGGCTATAAAGTGGTCTTT_TRB:TGTGCCAGCAGCCAACAAAACACAGAAGTCTTCTTT", x= clone3Lin5Df_65$combined_cdr3_nt) # 286

grep(pattern = "TRA:TGTGCAGCAAGTTGGGGGACTGGAGGCTATAAAGTGGTCTTT_TRB:TGTGCTAGCAGTTTTGCTGGGGGGGCTTCCTATGAACAGTACTTC", x= clone3Lin5Df_65$combined_cdr3_nt) # 1


## make a DF with clone # 1, 52, 63, 128, 286  TFH Motif 1 in Train mice M2M4M6

TFH_Motif1_train_DF <- as.data.frame(clone3Lin5Df_65[c(1,52, 63, 128, 286), c("mice", "CDR3nt_Ident", "Percent_Monocle_Th1", "Percent_Monocle_Tcmp","Percent_Monocle_TFH", "Total_cells", "Fate", "Th1_modulescore_percentile", "TFH_modulescore_percentile")])

write.csv(TFH_Motif1_train_DF, file = "summary_5mice/Figure4/TFH_Motif1_Train_mice_DF.csv")

TFH_Motif1_train_DF$Motif <- paste0("TFH-Motif1")


# TFH Motif 2 in train mice M2 M4 M6 #####################################

# M2

# TRA:TGTGCAGCAAGTCCTGGAGGCAATAATAAGCTGACTTTT_TRB:TGTGCCTGGGTGGGTAGTGCAGAAACGCTGTATTTT 
#                                                                                              30 
#               TRA:TGTGCTCTGAGCCCCAACATGGGCTACAAACTTACCTTC_TRB:TGTGCCAGCAGCCAAGAGAGGGAACAGTACTTC 
#                                                                                               4 
# TRA:TGTGCTCTGGGGAATTCTGGAGGAAGCAATGCAAAGCTAACCTTC_TRB:TGTGCCTGGAGCGGGATGGGGGGGCAGGACACCCAGTACTTT 
 #                                                                                             26 

grep(pattern = "TRA:TGTGCAGCAAGTCCTGGAGGCAATAATAAGCTGACTTTT_TRB:TGTGCCTGGGTGGGTAGTGCAGAAACGCTGTATTTT", x = clone3Lin5Df_65$combined_cdr3_nt) # 55

grep(pattern = "TRA:TGTGCTCTGAGCCCCAACATGGGCTACAAACTTACCTTC_TRB:TGTGCCAGCAGCCAAGAGAGGGAACAGTACTTC", x = clone3Lin5Df_65$combined_cdr3_nt) # 321

grep(pattern = "TRA:TGTGCTCTGGGGAATTCTGGAGGAAGCAATGCAAAGCTAACCTTC_TRB:TGTGCCTGGAGCGGGATGGGGGGGCAGGACACCCAGTACTTT", x = clone3Lin5Df_65$combined_cdr3_nt) # 61

# M4 

 # TRA:TGTGCAGCAAGTCCTGGAGGCAATAATAAGCTGACTTTT_TRB:TGTGCCTGGAGTCTGGGGGCAGAAACGCTGTATTTT 
    #                                                                                           69 
    #TRA:TGTGCAGCAAGTCCTGGAGGCAATAATAAGCTGACTTTT_TRB:TGTGCTAGCAGTAGAGCGGGACTCGCAAACTCCGACTACACCTTC 
     #                                                                                          16 
 #TRA:TGTGCTCTGACTAACAGTGCAGGGAACAAGCTAACTTTT_TRB:TGTGCCAGCGGTGATTTGGGGGGGGCGAACCAAGACACCCAGTACTTT 
  #                                                                                              4 

grep(pattern = "TRA:TGTGCAGCAAGTCCTGGAGGCAATAATAAGCTGACTTTT_TRB:TGTGCCTGGAGTCTGGGGGCAGAAACGCTGTATTTT", x = clone3Lin5Df_65$combined_cdr3_nt) # 25

grep(pattern = "TRA:TGTGCAGCAAGTCCTGGAGGCAATAATAAGCTGACTTTT_TRB:TGTGCTAGCAGTAGAGCGGGACTCGCAAACTCCGACTACACCTTC", x = clone3Lin5Df_65$combined_cdr3_nt) # 110

grep(pattern = "TRA:TGTGCTCTGACTAACAGTGCAGGGAACAAGCTAACTTTT_TRB:TGTGCCAGCGGTGATTTGGGGGGGGCGAACCAAGACACCCAGTACTTT", x = clone3Lin5Df_65$combined_cdr3_nt) # 333

# M6

  #TRA:TGTGCAGCAAGTGGGGACTCGGGATACAACAAACTCACTTTT_TRB:TGTGCCAGCGGTGATGTGGGACAGGGTAAAGACACCCAGTACTTT 
   #                                                                                              8 
  #   TRA:TGTGCAGCCCGTGGGGACTCGGGATACAACAAACTCACTTTT_TRB:TGTGCCTGGAGTCGGGACAATAACCAAGACACCCAGTACTTT 
   #                                                                                              7 
  #TRA:TGTGCTCTGAGTGACTCGGGATACAACAAACTCACTTTT_TRB:TGTGCCAGCGGTGATGCGGGGGGGCCTAGTGCAGAAACGCTGTATTTT 
   #                                                                                             34 
  #   TRA:TGTGCTCTGAGTGACTCGGGATACAACAAACTCACTTTT_TRB:TGTGCCAGCGGTGATTATGGGGCTAGTCAAAACACCTTGTACTTT 
   #                                                                                             23 
  #TRA:TGTGCTCTTACTAACAGTGCAGGGAACAAGCTAACTTTT_TRB:TGTGCCAGCGGTGATGTGGGGGGGCGGGGCCAAGACACCCAGTACTTT 
   #                                                                                              4 

grep(pattern = "TRA:TGTGCAGCAAGTGGGGACTCGGGATACAACAAACTCACTTTT_TRB:TGTGCCAGCGGTGATGTGGGACAGGGTAAAGACACCCAGTACTTT", x = clone3Lin5Df_65$combined_cdr3_nt) # 199

grep(pattern = "TRA:TGTGCAGCCCGTGGGGACTCGGGATACAACAAACTCACTTTT_TRB:TGTGCCTGGAGTCGGGACAATAACCAAGACACCCAGTACTTT", x = clone3Lin5Df_65$combined_cdr3_nt) # 223

grep(pattern = "TRA:TGTGCTCTGAGTGACTCGGGATACAACAAACTCACTTTT_TRB:TGTGCCAGCGGTGATGCGGGGGGGCCTAGTGCAGAAACGCTGTATTTT", x = clone3Lin5Df_65$combined_cdr3_nt) # 47

grep(pattern = "TRA:TGTGCTCTGAGTGACTCGGGATACAACAAACTCACTTTT_TRB:TGTGCCAGCGGTGATTATGGGGCTAGTCAAAACACCTTGTACTTT", x = clone3Lin5Df_65$combined_cdr3_nt) # 73

grep(pattern = "TRA:TGTGCTCTTACTAACAGTGCAGGGAACAAGCTAACTTTT_TRB:TGTGCCAGCGGTGATGTGGGGGGGCGGGGCCAAGACACCCAGTACTTT", x = clone3Lin5Df_65$combined_cdr3_nt) # 350


## make a DF with clone # 55, 321, 61, 25, 110, 333, 199, 223, 47, 73, 350 for TFH Motif 2 in Train mice M2M4M6

TFH_Motif2_train_DF <- as.data.frame(clone3Lin5Df_65[c(55, 321, 61, 25, 110, 333, 199, 223, 47, 73, 350), c("mice", "CDR3nt_Ident", "Percent_Monocle_Th1", "Percent_Monocle_Tcmp","Percent_Monocle_TFH", "Total_cells", "Fate", "Th1_modulescore_percentile", "TFH_modulescore_percentile")])

write.csv(TFH_Motif2_train_DF, file = "summary_5mice/Figure4/TFH_Motif2_Train_mice_DF.csv")

TFH_Motif2_train_DF$Motif <- paste0("TFH-Motif2")

# TFH Motif 5 in train mice M2 M4 M6 #####################################

# M2

#TRA:TGTGCAGCAGGGGTTAATAACAACAATGCCCCACGATTT_TRB:TGTGCCAGCAGTGATCCGGGGGGGCAGGGTCAAAACACCTTGTACTTT 
 #                                                                                             23 

grep(pattern = "TRA:TGTGCAGCAGGGGTTAATAACAACAATGCCCCACGATTT_TRB:TGTGCCAGCAGTGATCCGGGGGGGCAGGGTCAAAACACCTTGTACTTT", x = clone3Lin5Df_65$combined_cdr3_nt) # 70


# M4

# TRA:TGTGCAGCAAGTGGCAACAACAATGCCCCACGATTT_TRB:TGTGCCAGCAGTGAACATCAAGACACCCAGTACTTT 
   #                                                                                   30 
   #    TRA:TGTGCAGCAAGTGGGAACAACAATGCCCCACGATTT_TRB:TGTGCCAGCAGTGACCATCAAGACACCCAGTACTTT 
    #                                                                                   8 
 #TRA:TGTGCAGCAGGCTATAACAACAATGCCCCACGATTT_TRB:TGTGCCAGCGGTGATGCTGGGGACCAAGACACCCAGTACTTT 
  #                                                                                    38 
 #TRA:TGTGCAGCAGGGTATAACAACAATGCCCCACGATTT_TRB:TGTGCCAGCGGTGATGCGGGGGACCAAGACACCCAGTACTTT 
  #                                                                                   101

grep(pattern = "TRA:TGTGCAGCAAGTGGCAACAACAATGCCCCACGATTT_TRB:TGTGCCAGCAGTGAACATCAAGACACCCAGTACTTT", x = clone3Lin5Df_65$combined_cdr3_nt) # 56

grep(pattern = "TRA:TGTGCAGCAAGTGGGAACAACAATGCCCCACGATTT_TRB:TGTGCCAGCAGTGACCATCAAGACACCCAGTACTTT", x = clone3Lin5Df_65$combined_cdr3_nt) # 170, 197

grep(pattern = "TRA:TGTGCAGCAGGCTATAACAACAATGCCCCACGATTT_TRB:TGTGCCAGCGGTGATGCTGGGGACCAAGACACCCAGTACTTT", x = clone3Lin5Df_65$combined_cdr3_nt) # 42

grep(pattern = "TRA:TGTGCAGCAGGGTATAACAACAATGCCCCACGATTT_TRB:TGTGCCAGCGGTGATGCGGGGGACCAAGACACCCAGTACTTT", x = clone3Lin5Df_65$combined_cdr3_nt) # 17


#  TRA:TGTGCAGCAGCGTATAACAACAATGCCCCACGATTT_TRB:TGTGCCAGCAGTGACTGGGGGTATGAACAGTACTTC 
    #                                                                                      60 
  #TRA:TGTGCAGCAGGTTATAACAACAATGCCCCACGATTT_TRB:TGTGCCAGCGGTGATGCGGGACAGGGGAAGGATGAGCAGTTCTTC 
   #                                                                                       19 

grep(pattern = "TRA:TGTGCAGCAGCGTATAACAACAATGCCCCACGATTT_TRB:TGTGCCAGCAGTGACTGGGGGTATGAACAGTACTTC", x = clone3Lin5Df_65$combined_cdr3_nt) # 30

grep(pattern = "TRA:TGTGCAGCAGGTTATAACAACAATGCCCCACGATTT_TRB:TGTGCCAGCGGTGATGCGGGACAGGGGAAGGATGAGCAGTTCTTC", x = clone3Lin5Df_65$combined_cdr3_nt) # 91


## make a DF with clone # 70, 56, 170, 197, 42, 17, 30, 91 for TFH Motif 5 in Train mice M2M4M6

TFH_Motif5_train_DF <- as.data.frame(clone3Lin5Df_65[c(70, 56, 197, 42, 17, 30, 91), c("mice", "CDR3nt_Ident", "Percent_Monocle_Th1", "Percent_Monocle_Tcmp","Percent_Monocle_TFH", "Total_cells", "Fate", "Th1_modulescore_percentile", "TFH_modulescore_percentile")])

write.csv(TFH_Motif5_train_DF, file = "summary_5mice/Figure4/TFH_Motif5_Train_mice_DF.csv")

TFH_Motif5_train_DF$Motif <- paste0("TFH-Motif5")

# TFH Motif 10 in train mice M2 M4 M6 #####################################

# M2

#TRA:TGTGCAGCATTGTATGGGAGCAGTGGCAACAAGCTCATCTTT_TRB:TGTGCCTGGAGTCCGGGACAGCAAGACACCCAGTACTTT TRA:TGTGCTTTGGAAGGGGATGGGAGCAGTGGCAACAAGCTCATCTTT_TRB:TGTGCCTGGAGTCTAGAACTGTATGAACAGTACTTC 
 #                                                                                       43                                                                                          7 

grep(pattern = "TRA:TGTGCAGCATTGTATGGGAGCAGTGGCAACAAGCTCATCTTT_TRB:TGTGCCTGGAGTCCGGGACAGCAAGACACCCAGTACTTT", x = clone3Lin5Df_65$combined_cdr3_nt) # 37

grep(pattern = "TRA:TGTGCTTTGGAAGGGGATGGGAGCAGTGGCAACAAGCTCATCTTT_TRB:TGTGCCTGGAGTCTAGAACTGTATGAACAGTACTTC", x = clone3Lin5Df_65$combined_cdr3_nt) # 214


# M4

#TRA:TGTGCAGCAATCTATGGGAGCAGTGGCAACAAGCTCATCTTT_TRB:TGTACTTCTGTGCCAGCAGTGATTGGGGGGGGGTATAGTGCAGAAACGCTGTATTTT 
  #                                                                                                        233 
   #             TRA:TGTGCAGCAATCTATGGGAGCAGTGGCAACAAGCTCATCTTT_TRB:TGTACTTCTGTGCCAGGGGACAGCTTCTGTGAACAGTACTTC 
    #                                                                                                      173 
     #  TRA:TGTGCAGCAATCTATGGGAGCAGTGGCAACAAGCTCATCTTT_TRB:TGTGCCAGCAGTGATTGGGGGGGGGGGTATAGTGCAGAAACGCTGTATTTT 
      #                                                                                                     19 
       #   TRA:TGTGCAGCAATCTATGGGAGCAGTGGCAACAAGCTCATCTTT_TRB:TGTGCCAGCAGTGATTGGGGGGGGTATAGTGCAGAAACGCTGTATTTT 
        #                                                                                                    5 
         #             TRA:TGTGCAGCAATCTATGGGAGCAGTGGCAACAAGCTCATCTTT_TRB:TGTGCCTGGAGTCTATCGGGTTATGAACAGTACTTC 
          #                                                                                                  6 
           #  TRA:TGTGCAGCAATCTATGGGAGCAGTGGCAACAAGCTCATCTTT_TRB:TGTGCTAGCAGTTTTGCTGGGGGGGCTTCCTATGAACAGTACTTC 
            #                                                                                                6 

grep(pattern = "TRA:TGTGCAGCAATCTATGGGAGCAGTGGCAACAAGCTCATCTTT_TRB:TGTACTTCTGTGCCAGCAGTGATTGGGGGGGGGTATAGTGCAGAAACGCTGTATTTT", x = clone3Lin5Df_65$combined_cdr3_nt) # 6

grep(pattern = "TRA:TGTGCAGCAATCTATGGGAGCAGTGGCAACAAGCTCATCTTT_TRB:TGTACTTCTGTGCCAGGGGACAGCTTCTGTGAACAGTACTTC", x = clone3Lin5Df_65$combined_cdr3_nt) # 12

grep(pattern = "TRA:TGTGCAGCAATCTATGGGAGCAGTGGCAACAAGCTCATCTTT_TRB:TGTGCCAGCAGTGATTGGGGGGGGGGGTATAGTGCAGAAACGCTGTATTTT", x = clone3Lin5Df_65$combined_cdr3_nt) # 87

grep(pattern = "TRA:TGTGCAGCAATCTATGGGAGCAGTGGCAACAAGCTCATCTTT_TRB:TGTGCCAGCAGTGATTGGGGGGGGTATAGTGCAGAAACGCTGTATTTT", x = clone3Lin5Df_65$combined_cdr3_nt) # 288

grep(pattern = "TRA:TGTGCAGCAATCTATGGGAGCAGTGGCAACAAGCTCATCTTT_TRB:TGTGCCTGGAGTCTATCGGGTTATGAACAGTACTTC", x = clone3Lin5Df_65$combined_cdr3_nt) # 245

grep(pattern = "TRA:TGTGCAGCAATCTATGGGAGCAGTGGCAACAAGCTCATCTTT_TRB:TGTGCTAGCAGTTTTGCTGGGGGGGCTTCCTATGAACAGTACTTC", x = clone3Lin5Df_65$combined_cdr3_nt) # 246


 # M6

  #TRA:TGTGCAGCAACCTATGGGAGCAGTGGCAACAAGCTCATCTTT_TRB:TGTGCCAGCGGTGATCACAACTCTGGAAATACGCTCTATTTT 
             #                                                                                                          9 
              #          TRA:TGTGCAGCAACCTATGGGAGCAGTGGCAACAAGCTCATCTTT_TRB:TGTGCCAGCGGTGATTATGGGGCTAGTCAAAACACCTTGTACTTT 
               #                                                                                                        6 
                #                 TRA:TGTGCAGCAAGCTATGGGAGCAGTGGCAACAAGCTCATCTTT_TRB:TGTGCCTGGAGCTGGGGTGCAGAAACGCTGTATTTT 
                 #                                                                                                      8 
                  #         TRA:TGTGCAGCACTCTATGGGAGCAGTGGCAACAAGCTCATCTTT_TRB:TGTGCCTGGAGTCGTACTGGAAACCAAGACACCCAGTACTTT 
                   #                                                                                                    5 
            #TRA:TGTGCAGCATCGAATTATGGGAGCAGTGGCAACAAGCTCATCTTT_TRB:TGTGCCAGCAGTGATGTCCTGGGGGGGCGCTGGAGTGCAGAAACGCTGTATTTT 
             #                                                                                                          8 

grep(pattern = "TRA:TGTGCAGCAACCTATGGGAGCAGTGGCAACAAGCTCATCTTT_TRB:TGTGCCAGCGGTGATCACAACTCTGGAAATACGCTCTATTTT", x = clone3Lin5Df_65$combined_cdr3_nt) # 179

grep(pattern = "TRA:TGTGCAGCAACCTATGGGAGCAGTGGCAACAAGCTCATCTTT_TRB:TGTGCCAGCGGTGATTATGGGGCTAGTCAAAACACCTTGTACTTT", x = clone3Lin5Df_65$combined_cdr3_nt) # 251

grep(pattern = "TRA:TGTGCAGCAAGCTATGGGAGCAGTGGCAACAAGCTCATCTTT_TRB:TGTGCCTGGAGCTGGGGTGCAGAAACGCTGTATTTT", x = clone3Lin5Df_65$combined_cdr3_nt) # 198

grep(pattern = "TRA:TGTGCAGCACTCTATGGGAGCAGTGGCAACAAGCTCATCTTT_TRB:TGTGCCTGGAGTCGTACTGGAAACCAAGACACCCAGTACTTT", x = clone3Lin5Df_65$combined_cdr3_nt) # 292

grep(pattern = "TRA:TGTGCAGCATCGAATTATGGGAGCAGTGGCAACAAGCTCATCTTT_TRB:TGTGCCAGCAGTGATGTCCTGGGGGGGCGCTGGAGTGCAGAAACGCTGTATTTT", x = clone3Lin5Df_65$combined_cdr3_nt) # 201

## make a DF with clone # 37, 214, 6, 12, 87, 288, 245, 246, 179, 251, 198, 292, 201 for TFH Motif 10 in Train mice M2M4M6

TFH_Motif10_train_DF <- as.data.frame(clone3Lin5Df_65[c(37, 214, 6, 12, 87, 288, 245, 246, 179, 251, 198, 292, 201), c("mice", "CDR3nt_Ident", "Percent_Monocle_Th1", "Percent_Monocle_Tcmp","Percent_Monocle_TFH", "Total_cells", "Fate", "Th1_modulescore_percentile", "TFH_modulescore_percentile")])

write.csv(TFH_Motif10_train_DF, file = "summary_5mice/Figure4/TFH_Motif10_Train_mice_DF.csv")

TFH_Motif10_train_DF$Motif <- paste0("TFH-Motif10")

# TFH Motif 16 in train mice M2 M4 M6 #####################################


# M4

#TRA:TGTGCAGCAAATCAAGGAGGGTCTGCGAAGCTCATCTTT_TRB:TGTGCCTGGAGTTCCGACAGTTCTGGAAATACGCTCTATTTT TRA:TGTGCAGCGAATCAAGGAGGGTCTGCGAAGCTCATCTTT_TRB:TGTGCCTGGAGTTCCGACAGTTCTGGAAATACGCTCTATTTT 
 #                                                                                       13                                                                                         34 


grep(pattern = "TRA:TGTGCAGCAAATCAAGGAGGGTCTGCGAAGCTCATCTTT_TRB:TGTGCCTGGAGTTCCGACAGTTCTGGAAATACGCTCTATTTT", x = clone3Lin5Df_65$combined_cdr3_nt) # 134

grep(pattern = "TRA:TGTGCAGCGAATCAAGGAGGGTCTGCGAAGCTCATCTTT_TRB:TGTGCCTGGAGTTCCGACAGTTCTGGAAATACGCTCTATTTT", x = clone3Lin5Df_65$combined_cdr3_nt) # 46


# M6

#TRA:TGTGCAGCAAATCAAGGAGGGTCTGCGAAGCTCATCTTT_TRB:TGTGCCTGGAGTTCGGACAATTCTGGAAATACGCTCTATTTT 
  #                                                                                        6 

grep(pattern = "TRA:TGTGCAGCAAATCAAGGAGGGTCTGCGAAGCTCATCTTT_TRB:TGTGCCTGGAGTTCGGACAATTCTGGAAATACGCTCTATTTT", x = clone3Lin5Df_65$combined_cdr3_nt) # 250


## make a DF with clone # 46, 134, 250 for TFH Motif 16 in Train mice M2M4M6

TFH_Motif16_train_DF <- as.data.frame(clone3Lin5Df_65[c(46, 134, 250), c("mice", "CDR3nt_Ident", "Percent_Monocle_Th1", "Percent_Monocle_Tcmp","Percent_Monocle_TFH", "Total_cells", "Fate", "Th1_modulescore_percentile", "TFH_modulescore_percentile")])

write.csv(TFH_Motif16_train_DF, file = "summary_5mice/Figure4/TFH_Motif16_Train_mice_DF.csv")

TFH_Motif16_train_DF$Motif <- paste0("TFH-Motif16")

# combine all TFH Train DFs 



TFH_Train_Motif_Based_DF <- rbind(TFH_Motif1_train_DF, TFH_Motif2_train_DF, TFH_Motif5_train_DF, TFH_Motif10_train_DF, TFH_Motif16_train_DF)

write.csv(TFH_Train_Motif_Based_DF, file = "summary_5mice/Figure4/Train/TFH_Train_Motif_Based_DF.csv")
```

```{r make DF for TFH motif sp significan clones in test mice M1 M3}

## TFH Motif 1 in test mice

# M1

# TRA:TGCTCAGCATCTAGGGGAGGAGGCTATAAAGTGGTCTTT_TRB:TGTGCCAGCAGCACGACAGCCACAGAAGTCTTCTTT 
 #                                                                                        7 
#   TRA:TGTGCAGCAAGCATGAGGACTGGAGGCTATAAAGTGGTCTTT_TRB:TGTGCCAGCAGCCTCGGAGGAAACACCTTGTACTTT 
 #                                                                                        5 
#TRA:TGTGCAGCCGGGACTGGAGGCTATAAAGTGGTCTTT_TRB:TGTGCCTGGAGTCCGGACAGGGGGCGGAACACAGAAGTCTTCTTT 
 #                                                                                       11 
#TRA:TGTGCAGCCGGGACTGGAGGCTATAAAGTGGTCTTT_TRB:TGTGCTAGCAGTCCCGCCGGGACAGGGGACGAAAGATTATTTTTC 
    #                                                                                    13 


grep(pattern = "TRA:TGCTCAGCATCTAGGGGAGGAGGCTATAAAGTGGTCTTT_TRB:TGTGCCAGCAGCACGACAGCCACAGAAGTCTTCTTT", x = clone3Lin5Df_65$combined_cdr3_nt) # 204

grep(pattern = "TRA:TGTGCAGCAAGCATGAGGACTGGAGGCTATAAAGTGGTCTTT_TRB:TGTGCCAGCAGCCTCGGAGGAAACACCTTGTACTTT", x = clone3Lin5Df_65$combined_cdr3_nt) # 264

grep(pattern = "TRA:TGTGCAGCCGGGACTGGAGGCTATAAAGTGGTCTTT_TRB:TGTGCCTGGAGTCCGGACAGGGGGCGGAACACAGAAGTCTTCTTT", x = clone3Lin5Df_65$combined_cdr3_nt) # 144

grep(pattern = "TRA:TGTGCAGCCGGGACTGGAGGCTATAAAGTGGTCTTT_TRB:TGTGCTAGCAGTCCCGCCGGGACAGGGGACGAAAGATTATTTTTC", x = clone3Lin5Df_65$combined_cdr3_nt) # 125


# M3

#TRA:TGTGCAGCAAGTGGCACAAATGCTTACAAAGTCATCTTT_TRB:TGTGCCTGGGGACAGGGAGGAGAAGTCTTCTTT
     #                                                                               9 


grep(pattern = "TRA:TGTGCAGCAAGTGGCACAAATGCTTACAAAGTCATCTTT_TRB:TGTGCCTGGGGACAGGGAGGAGAAGTCTTCTTT", x = clone3Lin5Df_65$combined_cdr3_nt) # 169

## make a DF with clone # 204, 264, 144, 125, 169 for TFH Motif 1 in Test mice M1M3

TFH_Motif1_test_DF <- as.data.frame(clone3Lin5Df_65[c(204, 264, 144, 125, 169), c("mice", "CDR3nt_Ident", "Percent_Monocle_Th1", "Percent_Monocle_Tcmp","Percent_Monocle_TFH", "Total_cells", "Fate", "Th1_modulescore_percentile", "TFH_modulescore_percentile")])

write.csv(TFH_Motif1_test_DF, file = "summary_5mice/Figure4/TFH_Motif1_Test_mice_DF.csv")

TFH_Motif1_test_DF$Motif <- paste0("TFH-Motif1")


## TFH Motif 2 in test mice

# M1

     #TRA:TGTGCAGCAAGCGGGTCTGGAGGAAGCAATGCAAAGCTAACCTTC_TRB:TGTGCCAGCGGTGATGGCGGGCTTGAACAGTACTTC 
 #                                                                                        6 


grep(pattern = "TRA:TGTGCAGCAAGCGGGTCTGGAGGAAGCAATGCAAAGCTAACCTTC_TRB:TGTGCCAGCGGTGATGGCGGGCTTGAACAGTACTTC", x = clone3Lin5Df_65$combined_cdr3_nt) # 238


## just 1 clone for motif 2


## TFH Motif 5 in test mice


# M1

#TRA:TGTGCAGCAAGCTATAACAACAATGCCCCACGATTT_TRB:TGTAACTATGCTGAGCAGTTCTTC 
 #                                                                  69

grep(pattern = "TRA:TGTGCAGCAAGCTATAACAACAATGCCCCACGATTT_TRB:TGTAACTATGCTGAGCAGTTCTTC", x = clone3Lin5Df_65$combined_cdr3_nt) # 24



# M3

 #TRA:TGTGCAGCAAGTACTAACAACAATGCCCCACGATTT_TRB:TGTGCCAGCAGTGAGGGGACAGGGGCAAACACCGGGCAGCTCTACTTT 
  #                                                                                          91 
 #            TRA:TGTGCAGCAAGTGGGAACAACAATGCCCCACGATTT_TRB:TGTGCCAGCAGTGACCATCAAGACACCCAGTACTTT 
  #                                                                                           9 
 #            TRA:TGTGCAGCAAGTGGTAACAACAATGCCCCACGATTT_TRB:TGTGCCAGCAGTGAACACCAAGACACCCAGTACTTT 
  #                                                                                          10 


grep(pattern = "TRA:TGTGCAGCAAGTACTAACAACAATGCCCCACGATTT_TRB:TGTGCCAGCAGTGAGGGGACAGGGGCAAACACCGGGCAGCTCTACTTT", x = clone3Lin5Df_65$combined_cdr3_nt) # 19

grep(pattern = "TRA:TGTGCAGCAAGTGGGAACAACAATGCCCCACGATTT_TRB:TGTGCCAGCAGTGACCATCAAGACACCCAGTACTTT", x = clone3Lin5Df_65$combined_cdr3_nt) # 170, 197

grep(pattern = "TRA:TGTGCAGCAAGTGGTAACAACAATGCCCCACGATTT_TRB:TGTGCCAGCAGTGAACACCAAGACACCCAGTACTTT", x = clone3Lin5Df_65$combined_cdr3_nt) # 158


## make a DF with clone # 24, 19, 170, 197, 158 for TFH Motif 5 in Test mice M1M3, add 238 number clone here for Motif2 since it has just one clone

TFH_Motif5_test_DF <- as.data.frame(clone3Lin5Df_65[c(238,24, 19, 170, 158), c("mice", "CDR3nt_Ident", "Percent_Monocle_Th1", "Percent_Monocle_Tcmp","Percent_Monocle_TFH", "Total_cells", "Fate", "Th1_modulescore_percentile", "TFH_modulescore_percentile")])

write.csv(TFH_Motif5_test_DF, file = "summary_5mice/Figure4/TFH_Motif5_Test_mice_DF.csv")

TFH_Motif5_test_DF$Motif <- paste0("TFH-Motif5")


## TFH Motif 10 in test mice

# M1

#TRA:TGTGCAGCAACTTATGGGAGCAGTGGCAACAAGCTCATCTTT_TRB:TGCCAGCAGTCTGATCAGGGATATGAACAGTACTTC 
 #                                                                                          31 
#TRA:TGTGCAGCAAGTTATGGGAGCAGTGGCAACAAGCTCATCTTT_TRB:TGTGCCTGGAGTCCGACTGGGGGGGAAGACACCCAGTACTTT 
 #                                                                                           5 
#TRA:TGTGCAGCCCTTTATGGGAGCAGTGGCAACAAGCTCATCTTT_TRB:TGTGCCTGGAGTCCCACTGGGGGGCAAGACACCCAGTACTTT 
 #                                                                                          13 

grep(pattern = "TRA:TGTGCAGCAACTTATGGGAGCAGTGGCAACAAGCTCATCTTT_TRB:TGCCAGCAGTCTGATCAGGGATATGAACAGTACTTC", x = clone3Lin5Df_65$combined_cdr3_nt) # 50

grep(pattern = "TRA:TGTGCAGCAAGTTATGGGAGCAGTGGCAACAAGCTCATCTTT_TRB:TGTGCCTGGAGTCCGACTGGGGGGGAAGACACCCAGTACTTT", x = clone3Lin5Df_65$combined_cdr3_nt) # 266

grep(pattern = "TRA:TGTGCAGCCCTTTATGGGAGCAGTGGCAACAAGCTCATCTTT_TRB:TGTGCCTGGAGTCCCACTGGGGGGCAAGACACCCAGTACTTT", x = clone3Lin5Df_65$combined_cdr3_nt) # 124


# M3

 #TRA:TGTGCAGCAAACTATGGGAGCAGTGGCAACAAGCTCATCTTT_TRB:TGTGCCTGGAGCCCCGGGACAGGGGGCGAGGAACAGTACTTC 
  #                                                                                                 4 
 #      TRA:TGTGCAGCAACTTATGGGAGCAGTGGCAACAAGCTCATCTTT_TRB:TGTGCCAGCAGTGATATGGGGGCTGACGAAAGATTATTTTTC 
  #                                                                                               271 
 #TRA:TGTGCAGCAAGTTGGGGGAGCAGTGGCAACAAGCTCATCTTT_TRB:TGTGCTAGCAGTTTCCCCGGGACAACTAGTGCAGAAACGCTGTATTTT 
  #                                                                                               206 
   #       TRA:TGTGCAGCACTTTATGGGAGCAGTGGCAACAAGCTCATCTTT_TRB:TGTGCCTGGGCCCCGGGACAAACAGACACCCAGTACTTT 
    #                                                                                              55 
    #TRA:TGTGCAGCCACTTATGGGAGCAGTGGCAACAAGCTCATCTTT_TRB:TGTGCCAGCGGTGAACCGGGACAGGGGGCAGACACCCAGTACTTT 
        #                                                                                          14 
         #    TRA:TGTGCAGCCTCTTATGGGAGCAGTGGCAACAAGCTCATCTTT_TRB:TGTGCCTGGACTGGGGGGTTGGACACCCAGTACTTT 
          #                                                                                         5 


grep(pattern = "TRA:TGTGCAGCAAACTATGGGAGCAGTGGCAACAAGCTCATCTTT_TRB:TGTGCCTGGAGCCCCGGGACAGGGGGCGAGGAACAGTACTTC", x = clone3Lin5Df_65$combined_cdr3_nt) # 324

grep(pattern = "TRA:TGTGCAGCAACTTATGGGAGCAGTGGCAACAAGCTCATCTTT_TRB:TGTGCCAGCAGTGATATGGGGGCTGACGAAAGATTATTTTTC", x = clone3Lin5Df_65$combined_cdr3_nt) # 5

grep(pattern = "TRA:TGTGCAGCAAGTTGGGGGAGCAGTGGCAACAAGCTCATCTTT_TRB:TGTGCTAGCAGTTTCCCCGGGACAACTAGTGCAGAAACGCTGTATTTT", x = clone3Lin5Df_65$combined_cdr3_nt) # 11

grep(pattern = "TRA:TGTGCAGCACTTTATGGGAGCAGTGGCAACAAGCTCATCTTT_TRB:TGTGCCTGGGCCCCGGGACAAACAGACACCCAGTACTTT", x = clone3Lin5Df_65$combined_cdr3_nt) # 31

grep(pattern = "TRA:TGTGCAGCCACTTATGGGAGCAGTGGCAACAAGCTCATCTTT_TRB:TGTGCCAGCGGTGAACCGGGACAGGGGGCAGACACCCAGTACTTT", x = clone3Lin5Df_65$combined_cdr3_nt) # 119

grep(pattern = "TRA:TGTGCAGCCTCTTATGGGAGCAGTGGCAACAAGCTCATCTTT_TRB:TGTGCCTGGACTGGGGGGTTGGACACCCAGTACTTT", x = clone3Lin5Df_65$combined_cdr3_nt) # 281



## make a DF with clone # 50, 226, 124, 5, 11, 31, 119, 281 for TFH Motif 10 in Test mice M1M3 ## 

TFH_Motif10_test_DF <- as.data.frame(clone3Lin5Df_65[c(50, 226, 124, 5, 11, 31, 119, 281), c("mice", "CDR3nt_Ident", "Percent_Monocle_Th1", "Percent_Monocle_Tcmp","Percent_Monocle_TFH", "Total_cells", "Fate", "Th1_modulescore_percentile", "TFH_modulescore_percentile")])

write.csv(TFH_Motif10_test_DF, file = "summary_5mice/Figure4/TFH_Motif10_Test_mice_DF.csv")

TFH_Motif10_test_DF$Motif <- paste0("TFH-Motif10")


# combine all TFH Test DFs 



TFH_Test_Motif_Based_DF <- rbind(TFH_Motif1_test_DF, TFH_Motif5_test_DF, TFH_Motif10_test_DF)

write.csv(TFH_Test_Motif_Based_DF, file = "summary_5mice/Figure4/Test/TFH_Test_Motif_Based_DF.csv")


```


```{r make Th1 motif based DF in train mice M2M4M6}

# Th1 Motif 6 in train mice

# M2

#TRA:TGCGCAGTCTCTTCTAATACAGGAAACTACAAATACGTCTTT_TRB:TGTGCCAGCAGCCACGGGCAGCTCTACTTT TRA:TGTGCTCTGAGTGATCACAATACCAACAAAGTCGTCTTT_TRB:TGTGCCAGCTCGCTACACTCCGACTACACCTTC 
              #                                                                 15                                                                                11 
       #  TRA:TGTGCTGCAAGTGAGAGGGGGGGAAACTACAAATACGTCTTT_TRB:TGTGCCAGCAGCAACTCCTTT 
                        #                                                        7 

grep(pattern = "TRA:TGCGCAGTCTCTTCTAATACAGGAAACTACAAATACGTCTTT_TRB:TGTGCCAGCAGCCACGGGCAGCTCTACTTT", x = clone3Lin5Df_65$combined_cdr3_nt) # 113

grep(pattern = "TRA:TGTGCTCTGAGTGATCACAATACCAACAAAGTCGTCTTT_TRB:TGTGCCAGCTCGCTACACTCCGACTACACCTTC", x = clone3Lin5Df_65$combined_cdr3_nt) # 146

grep(pattern = "TRA:TGTGCTGCAAGTGAGAGGGGGGGAAACTACAAATACGTCTTT_TRB:TGTGCCAGCAGCAACTCCTTT", x = clone3Lin5Df_65$combined_cdr3_nt) # 213

# M4

#TRA:TGTGCAGCAAGCATAGGAAACTACAAATACGTCTTT_TRB:TGTGCCAGCGGTGATGCAGGGACTGGGGGGGATGAACAGTACTTC    TRA:TGTGCAGCAAGCATAGGAAACTACAAATACGTCTTT_TRB:TGTGCCAGCGGTGATGCGGGGGACCAAGACACCCAGTACTTT 
         #                                                                                       60                                                                                         15 
          #       TRA:TGTGCAGCAAGCATGAGAACAGGAAACTACAAATACGTCTTT_TRB:TGTGCCAGCAGCGCCCAAACGCTGTATTTT TRA:TGTGCAGCAAGTCTTAATACAGGAAACTACAAATACGTCTTT_TRB:TGTGCCTGGAATGGGGGGGGCTATGCTGAGCAGTTCTTC 
           #                                                                                      5                                                                                         12 


grep(pattern = "TRA:TGTGCAGCAAGCATAGGAAACTACAAATACGTCTTT_TRB:TGTGCCAGCGGTGATGCAGGGACTGGGGGGGATGAACAGTACTTC", x = clone3Lin5Df_65$combined_cdr3_nt) # 29

grep(pattern = "TRA:TGTGCAGCAAGCATAGGAAACTACAAATACGTCTTT_TRB:TGTGCCAGCGGTGATGCGGGGGACCAAGACACCCAGTACTTT", x = clone3Lin5Df_65$combined_cdr3_nt) # 116

grep(pattern = "TRA:TGTGCAGCAAGCATGAGAACAGGAAACTACAAATACGTCTTT_TRB:TGTGCCAGCAGCGCCCAAACGCTGTATTTT", x = clone3Lin5Df_65$combined_cdr3_nt) # 287

grep(pattern = "TRA:TGTGCAGCAAGTCTTAATACAGGAAACTACAAATACGTCTTT_TRB:TGTGCCTGGAATGGGGGGGGCTATGCTGAGCAGTTCTTC", x = clone3Lin5Df_65$combined_cdr3_nt) # 143

# M6

   #  TRA:TGTGCAGCAAGCATGAGGACAGGAAACTACAAATACGTCTTT_TRB:TGTGCCTGGAGTCAAGGGGCCGAAAGATTATTTTTC 
             #                                                                                          9 
           #TRA:TGTGCAGCAAGTGAGGATACAGGAAACTACAAATACGTCTTT_TRB:TGTGCCAGCAGTAGGGGGGACAGGGGTACTGAGCAGTTCTTC 
            #                                                                                           4 
grep(pattern = "TRA:TGTGCAGCAAGCATGAGGACAGGAAACTACAAATACGTCTTT_TRB:TGTGCCTGGAGTCAAGGGGCCGAAAGATTATTTTTC", x = clone3Lin5Df_65$combined_cdr3_nt) # 180

grep(pattern = "TRA:TGTGCAGCAAGTGAGGATACAGGAAACTACAAATACGTCTTT_TRB:TGTGCCAGCAGTAGGGGGGACAGGGGTACTGAGCAGTTCTTC", x = clone3Lin5Df_65$combined_cdr3_nt) # 338


## make a DF with clone # 113, 146, 213, 29, 116, 287, 143, 180, 338 for TFH Motif 6 in Train mice M2M4M6 

Th1_Motif6_train_DF <- as.data.frame(clone3Lin5Df_65[c(29, 113, 116, 143, 146, 180, 213, 287, 338), c("mice", "CDR3nt_Ident", "Percent_Monocle_Th1", "Percent_Monocle_Tcmp","Percent_Monocle_TFH", "Total_cells", "Fate", "Th1_modulescore_percentile", "TFH_modulescore_percentile")])


write.csv(Th1_Motif6_train_DF, file = "summary_5mice/Figure4/Th1_Motif6_train_DF.csv")

Th1_Motif6_train_DF$Motif <- paste0("Th1-Motif6")



# Th1 Motif 7 in train mice


# M2
#TRA:TGTGCTGCTGAGAATAGCAACTATCAGTTGATCTGG_TRB:TGTGCCTGGATGGGGGTGAACACCCAGTACTTT 
 #                                                                            5

grep(pattern = "TRA:TGTGCTGCTGAGAATAGCAACTATCAGTTGATCTGG_TRB:TGTGCCTGGATGGGGGTGAACACCCAGTACTTT", x = clone3Lin5Df_65$combined_cdr3_nt) # 277

# M6

# M6
 #TRA:TGTGCAGCAAGCTATAGCAACTATCAGTTGATCTGG_TRB:TGTGCCTGGAGTCTAGCCAACAATGCAAACACAGAAGTCTTCTTT 
  #                                                                                        9 


grep(pattern = "TRA:TGTGCAGCAAGCTATAGCAACTATCAGTTGATCTGG_TRB:TGTGCCTGGAGTCTAGCCAACAATGCAAACACAGAAGTCTTCTTT", x = clone3Lin5Df_65$combined_cdr3_nt) # 182


## make a DF with clone # 277, 182 for TFH Motif 7 in Train mice M2M4M6 

Th1_Motif7_train_DF <- as.data.frame(clone3Lin5Df_65[c(277, 182), c("mice", "CDR3nt_Ident", "Percent_Monocle_Th1", "Percent_Monocle_Tcmp","Percent_Monocle_TFH", "Total_cells", "Fate", "Th1_modulescore_percentile", "TFH_modulescore_percentile")])



write.csv(Th1_Motif7_train_DF, file = "summary_5mice/Figure4/Th1_Motif7_train_DF.csv")

Th1_Motif7_train_DF$Motif <- paste0("Th1-Motif7")



# combine all TFH Train DFs 



Th1_Train_Motif_Based_DF <- rbind(Th1_Motif6_train_DF, Th1_Motif7_train_DF)

Th1_Train_Motif_Based_DF <- apply(Th1_Train_Motif_Based_DF, 2, as.character)
write.csv(Th1_Train_Motif_Based_DF, file = "summary_5mice/Figure4/Train/Th1_Train_Motif_Based_DF.csv")


```

```{r make Th1 motif based DF in test mice M1M3}

## Motif 6 in Test mice

# M1

#TRA:TGTGCAGCAAGCATGAGGGCAGGAAACTACAAATACGTCTTT_TRB:TGTGCCAGCAGCTCAGAAACGCTGTATTTT 
 #                                                                               7 


grep(pattern = "TRA:TGTGCAGCAAGCATGAGGGCAGGAAACTACAAATACGTCTTT_TRB:TGTGCCAGCAGCTCAGAAACGCTGTATTTT", x = clone3Lin5Df_65$combined_cdr3_nt) # 205

# M3

#TRA:TGTGCAGCAAGTGACTCCAATACCAACAAAGTCGTCTTT_TRB:TGTGCCAGCGGTGATGTACTGGGGGGGAAGGGGAGCTCCTATGAACAGTACTTC 
  #                                                                                                    7 

grep(pattern = "TRA:TGTGCAGCAAGTGACTCCAATACCAACAAAGTCGTCTTT_TRB:TGTGCCAGCGGTGATGTACTGGGGGGGAAGGGGAGCTCCTATGAACAGTACTTC", x = clone3Lin5Df_65$combined_cdr3_nt) # 215



## Motif 7 in Test mice

#M1
#TRA:TGTGCAGCAAGTTACAGCAACTATCAGTTGATCTGG_TRB:TGTGCCAGCAGTTTCCCGGACAGGGGAGACACCCAGTACTTT 
#                                                                                     43 

grep(pattern = "TRA:TGTGCAGCAAGTTACAGCAACTATCAGTTGATCTGG_TRB:TGTGCCAGCAGTTTCCCGGACAGGGGAGACACCCAGTACTTT", x = clone3Lin5Df_65$combined_cdr3_nt) # 36


## make a DF with clone # 205, 215, 36 for TFH Motif 6 nad 7 in Test mice M1 M3 

Th1_test_motif_based_DF <- as.data.frame(clone3Lin5Df_65[c(205, 215, 36), c("mice", "CDR3nt_Ident", "Percent_Monocle_Th1", "Percent_Monocle_Tcmp","Percent_Monocle_TFH", "Total_cells", "Fate", "Th1_modulescore_percentile", "TFH_modulescore_percentile")])

write.csv(Th1_test_motif_based_DF, file = "summary_5mice/Figure4/Test/Th1_test_motif_based_DF.csv")


 
```


```{r upset plot to check clonal overlap across mice with including treg cells here}
### Change file and object names as needed

rm(list=ls())
immune.combined_edit2_2cell <- readRDS("summary_5M_CCA_V2/immune.combined_edit2_2cell.RDS")
# immune.combined_edit2_2cell <- readRDS("immune.combined_edit2_2cell.RDS")

TSNEPlot(immune.combined_edit2_2cell, do.label = T)

Acute_CD4_5mice_clones_by_mouse <- immune.combined_edit2_2cell@meta.data %>% select(combined_cdr3_nt, mice) %>% table() %>% as.data.frame()
# Remove "No CDR3" rows
# Acute_CD4_6mice_clones_by_mouse <- Acute_CD4_6mice_clones_by_mouse[-which(Acute_CD4_6mice_clones_by_mouse$combined_cdr3_nt == "No CDR3"), ]
# table() double counted some cells but left the counts as 0; remove these
Acute_CD4_5mice_clones_by_mouse <- Acute_CD4_5mice_clones_by_mouse[which(Acute_CD4_5mice_clones_by_mouse$Freq > 0), ]
# Order in decreasing order
Acute_CD4_5mice_clones_by_mouse <- Acute_CD4_5mice_clones_by_mouse[order(Acute_CD4_5mice_clones_by_mouse$Freq, decreasing = T), ]

Acute_CD4_5mice_clones_by_mouse <- Acute_CD4_5mice_clones_by_mouse[which(Acute_CD4_5mice_clones_by_mouse$Freq > 1), ]

# Convert to wide format data
Acute_CD4_5mice_clones_by_mouse_wide <- Acute_CD4_5mice_clones_by_mouse %>% spread(mice, Freq, fill = 0)



# Check overlap
Acute_CD4_5mice_clones_by_mouse_wide_binary <- Acute_CD4_5mice_clones_by_mouse_wide
for (i in 2:ncol(Acute_CD4_5mice_clones_by_mouse_wide_binary)) {
  Acute_CD4_5mice_clones_by_mouse_wide_binary[, i][which(Acute_CD4_5mice_clones_by_mouse_wide_binary[, i] > 0)] <- 1
}


which(rowSums(Acute_CD4_5mice_clones_by_mouse_wide_binary[, -1]) > 1) %>% length() # 4
Acute_CD4_5mice_clones_by_mouse_wide_binary[which(rowSums(Acute_CD4_5mice_clones_by_mouse_wide_binary[, -1]) > 1), ]

#                                                                           combined_cdr3_nt    M1 M2 M3 M4 M6
#136          TRA:TGTGCAGCAAGCATCTATGCCCAGGGATTAACCTTC_TRB:TGTGCCAGCGGTGATAACCAAGACACCCAGTACTTT  0  0  0  1  1
#155    TRA:TGTGCAGCAAGCCCCAATACCAACAAAGTCGTCTTT_TRB:TGTGCCAGCGGTGATTGGGGGGTCCAAGACACCCAGTACTTT  0  0  0  1  1
#188 TRA:TGTGCAGCAAGTAATAACTATGCCCAGGGATTAACCTTC_TRB:TGTGCCAGCAGTGACAGGGGGCCCAACGAAAGATTATTTTTC  1  0  1  0  0
#273          TRA:TGTGCAGCAAGTGGGAACAACAATGCCCCACGATTT_TRB:TGTGCCAGCAGTGACCATCAAGACACCCAGTACTTT  0  0  1  1  0


# Use the binarized clones_by_mouse DF for the UpSet clonal overlap plot
# First convert to long format data
Acute_CD4_5mice_clones_by_mouse_long_binary <- Acute_CD4_5mice_clones_by_mouse_wide_binary %>%
  pivot_longer(cols = -combined_cdr3_nt, names_to = "Mouse")


# Make a new DF with a single column for Mouse that is a list of mice with that clone
Acute_CD4_5mice_clones_by_mouse_upset_binary <- Acute_CD4_5mice_clones_by_mouse_wide_binary %>% 
  select(combined_cdr3_nt) %>% 
  mutate(Mouse = NA)

for (i in 1:nrow(Acute_CD4_5mice_clones_by_mouse_upset_binary)) {
  Acute_CD4_5mice_clones_by_mouse_upset_binary$Mouse[i] <-
    list(
      # unique(
      Acute_CD4_5mice_clones_by_mouse_long_binary$Mouse[
        which(Acute_CD4_5mice_clones_by_mouse_long_binary$combined_cdr3_nt == Acute_CD4_5mice_clones_by_mouse_upset_binary$combined_cdr3_nt[i] & Acute_CD4_5mice_clones_by_mouse_long_binary$value == 1)]
      # )
    )
}

Acute_CD4_5mice_clones_by_mouse_upset_binary 

# test <- Acute_CD4_5mice_clones_by_mouse_upset_binary

Acute_CD4_5mice_clones_by_mouse_upset_binary$Mouse <- lapply(
  X = Acute_CD4_5mice_clones_by_mouse_upset_binary$Mouse,
  FUN = function(x) gsub("M6", "M5", x))

write.csv(Acute_CD4_5mice_clones_by_mouse_upset_binary, file = "summary_5mice/Acute_CD4_5mice_clones_by_mouse_upset_binary.csv")



# Make plot
pdf(file = "summary_5mice/Figure1/clonal_overlap_2cellclone_5mice.pdf", width = 8, height = 6, useDingbats = FALSE)
ggplot(Acute_CD4_5mice_clones_by_mouse_upset_binary, aes(x = Mouse)) +
  geom_bar() +
  scale_x_upset() +
  axis_combmatrix(levels = paste0("M", 1:5)) +
  theme_minimal() +
  theme(
    panel.grid.major.x = element_blank(),
    panel.grid.minor.x = element_blank(),
    axis.title.x = element_blank(),
    plot.title = element_text(hjust = 0.5)) +
  ylab("Number of clones") +
  ggtitle("TCR Alpha and Beta")
dev.off()

### check clonal overlap based on alpha and beta chain individually ################################

Acute_CD4_5mice_clones_by_mouse_Alpha <- immune.combined_edit2_2cell@meta.data %>% select(cdr3_nt_TRA, mice) %>% table() %>% as.data.frame()
# Remove "No CDR3" rows
# Acute_CD4_6mice_clones_by_mouse <- Acute_CD4_6mice_clones_by_mouse[-which(Acute_CD4_6mice_clones_by_mouse$combined_cdr3_nt == "No CDR3"), ]
# table() double counted some cells but left the counts as 0; remove these
Acute_CD4_5mice_clones_by_mouse_Alpha <- Acute_CD4_5mice_clones_by_mouse_Alpha[which(Acute_CD4_5mice_clones_by_mouse_Alpha$Freq > 0), ]
# Order in decreasing order
Acute_CD4_5mice_clones_by_mouse_Alpha <- Acute_CD4_5mice_clones_by_mouse_Alpha[order(Acute_CD4_5mice_clones_by_mouse_Alpha$Freq, decreasing = T), ]
# 606 clones

Acute_CD4_5mice_clones_by_mouse_Alpha <- Acute_CD4_5mice_clones_by_mouse_Alpha[which(Acute_CD4_5mice_clones_by_mouse_Alpha$Freq > 1), ] # 605 clones

# Convert to wide format data
Acute_CD4_5mice_clones_by_mouse_wide_Alpha <- Acute_CD4_5mice_clones_by_mouse_Alpha %>% spread(mice, Freq, fill = 0)

# Check overlap
Acute_CD4_5mice_clones_by_mouse_wide_Alpha_binary <- Acute_CD4_5mice_clones_by_mouse_wide_Alpha
for (i in 2:ncol(Acute_CD4_5mice_clones_by_mouse_wide_Alpha_binary)) {
  Acute_CD4_5mice_clones_by_mouse_wide_Alpha_binary[, i][which(Acute_CD4_5mice_clones_by_mouse_wide_Alpha_binary[, i] > 0)] <- 1
}


which(rowSums(Acute_CD4_5mice_clones_by_mouse_wide_Alpha_binary[, -1]) > 1) %>% length() # 38
Acute_CD4_5mice_clones_by_mouse_wide_Alpha_binary[which(rowSums(Acute_CD4_5mice_clones_by_mouse_wide_Alpha_binary[, -1]) > 1), ]

#                                         cdr3_nt_TRA M1 M2 M3 M4 M6
#42           TGCTCAGCCCCCGGGACTGGAGGCTATAAAGTGGTCTTT  0  1  0  1  0
#81              TGTGCAGCAAACTATGGAAATGAGAAAATAACTTTT  0  0  1  0  1
#85           TGTGCAGCAAATCAAGGAGGGTCTGCGAAGCTCATCTTT  0  0  0  1  1
#92        TGTGCAGCAACCTATGGGAGCAGTGGCAACAAGCTCATCTTT  0  0  0  1  1
#99        TGTGCAGCAACTTATGGGAGCAGTGGCAACAAGCTCATCTTT  1  0  1  0  0
#114             TGTGCAGCAAGCATCGGCAATAACAGAATCTTCTTT  1  0  0  0  1
#116             TGTGCAGCAAGCATCTATGCCCAGGGATTAACCTTC  0  0  0  1  1
#128             TGTGCAGCAAGCCCCAATACCAACAAAGTCGTCTTT  0  1  1  1  1
#138          TGTGCAGCAAGCTATAACTATGCCCAGGGATTAACCTTC  1  1  0  1  1
#142             TGTGCAGCAAGCTGGAATTACAACGTGCTTTACTTC  1  0  1  1  0
#148             TGTGCAGCAAGGTACAGCAACAACAGACTTACTTTG  0  0  1  0  1
#151          TGTGCAGCAAGTAATAACTATGCCCAGGGATTAACCTTC  1  0  1  0  0
#156          TGTGCAGCAAGTAGGAACTATGCCCAGGGATTAACCTTC  0  0  1  0  1
#166             TGTGCAGCAAGTCCGGGTTACCAGAACTTCTATTTT  0  1  0  1  0
#167          TGTGCAGCAAGTCCTGGAGGCAATAATAAGCTGACTTTT  0  1  0  1  0
#174          TGTGCAGCAAGTGACACAAATGCTTACAAAGTCATCTTT  0  1  0  1  1
#175          TGTGCAGCAAGTGACACCAATACAGGCAAATTAACCTTT  0  0  0  1  1
#217             TGTGCAGCAAGTGGGAACAACAATGCCCCACGATTT  0  0  1  1  0
#220       TGTGCAGCAAGTGGGGACTATGGAAATGAGAAAATAACTTTT  0  0  0  1  1
#230          TGTGCAGCAAGTGTGGCCAATACCAACAAAGTCGTCTTT  1  0  0  0  1
#234             TGTGCAGCAAGTTATAACAACAATGCCCCACGATTT  0  1  1  1  0
#235          TGTGCAGCAAGTTATAACTATGCCCAGGGATTAACCTTC  0  1  0  1  1
#238       TGTGCAGCAAGTTATGGGAGCAGTGGCAACAAGCTCATCTTT  1  1  0  0  0
#242             TGTGCAGCAAGTTCTAATTACAACGTGCTTTACTTC  1  1  0  0  0
#246       TGTGCAGCAAGTTGGGGGACTGGAGGCTATAAAGTGGTCTTT  1  0  0  1  0
#250          TGTGCAGCAAGTTTTAACTATGCCCAGGGATTAACCTTC  1  1  0  1  0
#260          TGTGCAGCACCCAACTATGGAAATGAGAAAATAACTTTT  1  1  1  1  0
#268          TGTGCAGCAGATAACACCAATACAGGCAAATTAACCTTT  1  1  0  0  0
#282             TGTGCAGCAGGGTCTAATTACAACGTGCTTTACTTC  1  0  0  1  0
#300          TGTGCAGCCCCCAACTATGGAAATGAGAAAATAACTTTT  0  1  0  0  1
#377             TGTGCTACTGACACAAATGCTTACAAAGTCATCTTT  1  1  0  0  1
#400 TGTGCTATGAGAGAGGGAGGAGGTTCAGCCTTAGGGAGGCTGCATTTT  0  1  1  0  0
#406          TGTGCTATGGAACGAGGAAGCAATGCAAAGCTAACCTTC  0  1  1  0  1
#408          TGTGCTATGGAACGGGGAAGCAATGCAAAGCTAACCTTC  1  0  0  0  1
#423          TGTGCTCTGAGCAATTATAACCAGGGGAAGCTTATCTTT  0  0  1  0  1
#452          TGTGCTCTGAGTGATGATAGCAACTATCAGTTGATCTGG  0  0  1  0  1
#508             TGTGCTGCTGAGGATAGCAACTATCAGTTGATCTGG  1  1  0  1  0
#533             TGTGCTTCTGACACAAATGCTTACAAAGTCATCTTT  0  1  0  0  1


# Use the binarized clones_by_mouse DF for the UpSet clonal overlap plot
# First convert to long format data
Acute_CD4_5mice_clones_by_mouse_long_binary <- Acute_CD4_5mice_clones_by_mouse_wide_Alpha_binary %>%
  pivot_longer(cols = -cdr3_nt_TRA, names_to = "Mouse")


# Make a new DF with a single column for Mouse that is a list of mice with that clone
Acute_CD4_5mice_clones_by_mouse_upset_binary <- Acute_CD4_5mice_clones_by_mouse_wide_Alpha_binary %>% 
  select(cdr3_nt_TRA) %>% 
  mutate(Mouse = NA)

for (i in 1:nrow(Acute_CD4_5mice_clones_by_mouse_upset_binary)) {
  Acute_CD4_5mice_clones_by_mouse_upset_binary$Mouse[i] <-
    list(
      # unique(
      Acute_CD4_5mice_clones_by_mouse_long_binary$Mouse[
        which(Acute_CD4_5mice_clones_by_mouse_long_binary$cdr3_nt_TRA == Acute_CD4_5mice_clones_by_mouse_upset_binary$cdr3_nt_TRA[i] & Acute_CD4_5mice_clones_by_mouse_long_binary$value == 1)]
      # )
    )
}

Acute_CD4_5mice_clones_by_mouse_upset_binary$Mouse <- lapply(
  X = Acute_CD4_5mice_clones_by_mouse_upset_binary$Mouse,
  FUN = function(x) gsub("M6", "M5", x))

# Make plot # 38 overlapped clones
pdf(file = "summary_5mice/Figure1/clonal_overlap_2cellclone_6mice_TCR_Alpha_chain.pdf", width = 8, height = 6, useDingbats = FALSE)
ggplot(Acute_CD4_5mice_clones_by_mouse_upset_binary, aes(x = Mouse)) +
  geom_bar() +
  scale_x_upset() +
  axis_combmatrix(levels = paste0("M", 1:5)) +
  theme_minimal() +
  theme(
    panel.grid.major.x = element_blank(),
    panel.grid.minor.x = element_blank(),
    axis.title.x = element_blank(),
    plot.title = element_text(hjust = 0.5)) +
  ylab("Number of clones") +
  ggtitle("TCR Alpha")
dev.off()

# Although there 38 but showing 19 cause there are two overlapped events 

### check clonal overlap based on beta chain individually ################################

Acute_CD4_5mice_clones_by_mouse_Beta <- immune.combined_edit2_2cell@meta.data %>% select(cdr3_nt_TRB, mice) %>% table() %>% as.data.frame()
# Remove "No CDR3" rows
# Acute_CD4_6mice_clones_by_mouse <- Acute_CD4_6mice_clones_by_mouse[-which(Acute_CD4_6mice_clones_by_mouse$combined_cdr3_nt == "No CDR3"), ]
# table() double counted some cells but left the counts as 0; remove these
Acute_CD4_5mice_clones_by_mouse_Beta <- Acute_CD4_5mice_clones_by_mouse_Beta[which(Acute_CD4_5mice_clones_by_mouse_Beta$Freq > 0), ]
# Order in decreasing order
Acute_CD4_5mice_clones_by_mouse_Beta <- Acute_CD4_5mice_clones_by_mouse_Beta[order(Acute_CD4_5mice_clones_by_mouse_Beta$Freq, decreasing = T), ] 

Acute_CD4_5mice_clones_by_mouse_Beta <- Acute_CD4_5mice_clones_by_mouse_Beta[which(Acute_CD4_5mice_clones_by_mouse_Beta$Freq > 1), ] # 586 clones

# Convert to wide format data
Acute_CD4_5mice_clones_by_mouse_wide_Beta <- Acute_CD4_5mice_clones_by_mouse_Beta %>% spread(mice, Freq, fill = 0)
# 756 clones cause there are overlaps between mice so from 771 to 756 clones

# Check overlap
Acute_CD4_5mice_clones_by_mouse_wide_Beta_binary <- Acute_CD4_5mice_clones_by_mouse_wide_Beta
for (i in 2:ncol(Acute_CD4_5mice_clones_by_mouse_wide_Beta_binary)) {
  Acute_CD4_5mice_clones_by_mouse_wide_Beta_binary[, i][which(Acute_CD4_5mice_clones_by_mouse_wide_Beta_binary[, i] > 0)] <- 1
}


which(rowSums(Acute_CD4_5mice_clones_by_mouse_wide_Beta_binary[, -1]) > 1) %>% length() # 11
Acute_CD4_5mice_clones_by_mouse_wide_Beta_binary[which(rowSums(Acute_CD4_5mice_clones_by_mouse_wide_Beta_binary[, -1]) > 1), ]

#                                   cdr3_nt_TRB M1 M2 M3 M4 M6
#13                    TGTAACTATGCTGAGCAGTTCTTC  1  1  0  0  0
#85        TGTGCCAGCAGCCCCGACAGGGATGAACAGTACTTC  0  0  1  1  0
#176       TGTGCCAGCAGTCGGGACAACTATGAACAGTACTTC  1  1  0  0  0
#177       TGTGCCAGCAGTCGGGACAGATATGAACAGTACTTC  0  0  0  1  1
#193 TGTGCCAGCAGTGACAGGGGGCCCAACGAAAGATTATTTTTC  1  0  1  0  0
#199       TGTGCCAGCAGTGACCATCAAGACACCCAGTACTTT  0  0  1  1  0
#224 TGTGCCAGCAGTGATCACAATTCTGGAAATACGCTCTATTTT  0  1  1  0  0
#359       TGTGCCAGCGGTGATAACCAAGACACCCAGTACTTT  0  0  0  1  1
#403 TGTGCCAGCGGTGATTGGGGGGTCCAAGACACCCAGTACTTT  0  0  0  1  1
#425    TGTGCCTGGAATGGGGGGGGCTATGCTGAGCAGTTCTTC  0  0  0  1  1
#517 TGTGCCTGGAGTTCCGACAGTTCTGGAAATACGCTCTATTTT  1  0  0  1  0


# Use the binarized clones_by_mouse DF for the UpSet clonal overlap plot
# First convert to long format data
Acute_CD4_5mice_clones_by_mouse_long_binary <- Acute_CD4_5mice_clones_by_mouse_wide_Beta_binary %>%
  pivot_longer(cols = -cdr3_nt_TRB, names_to = "Mouse")


# Make a new DF with a single column for Mouse that is a list of mice with that clone
Acute_CD4_5mice_clones_by_mouse_upset_binary <- Acute_CD4_5mice_clones_by_mouse_wide_Beta_binary %>% 
  select(cdr3_nt_TRB) %>% 
  mutate(Mouse = NA)

for (i in 1:nrow(Acute_CD4_5mice_clones_by_mouse_upset_binary)) {
  Acute_CD4_5mice_clones_by_mouse_upset_binary$Mouse[i] <-
    list(
      # unique(
      Acute_CD4_5mice_clones_by_mouse_long_binary$Mouse[
        which(Acute_CD4_5mice_clones_by_mouse_long_binary$cdr3_nt_TRB == Acute_CD4_5mice_clones_by_mouse_upset_binary$cdr3_nt_TRB[i] & Acute_CD4_5mice_clones_by_mouse_long_binary$value == 1)]
      # )
    )
}

## we want change M6 to M5 in mouse column
Acute_CD4_5mice_clones_by_mouse_upset_binary$Mouse <- lapply(
  X = Acute_CD4_5mice_clones_by_mouse_upset_binary$Mouse,
  FUN = function(x) gsub("M6", "M5", x))

# Make plot # 11 overlapped clones
pdf(file = "summary_5mice/clonal_overlap_2cellclone_6mice_TCR_Beta_chain.pdf", width = 8, height = 6, useDingbats = FALSE)
ggplot(Acute_CD4_5mice_clones_by_mouse_upset_binary, aes(x = Mouse)) +
  geom_bar() +
  scale_x_upset() +
  axis_combmatrix(levels = paste0("M", 1:5)) +
  theme_minimal() +
  theme(
    panel.grid.major.x = element_blank(),
    panel.grid.minor.x = element_blank(),
    axis.title.x = element_blank(),
    plot.title = element_text(hjust = 0.5)) +
  ylab("Number of clones") +
  ggtitle("TCR Beta")
dev.off()
# Although there 11 but showing 6 cause there are two overlapped events 

```



```{r need to find some TCMP dominated clones in UMAP which shows progression }

## check data_clones_by_mouse_TRA_TRB and clones6fDF_65 DF

rm(list=ls())

immune.combined_edit2_2cell <- readRDS("summary_5M_CCA_V2/immune.combined_edit2_2cell.RDS")

data_clones_by_mouse_TRA_TRB <- read.csv("summary_5mice/data_clones_by_mouse_TRA_TRB_5mice.csv")

#clone3Lin6Df_65 <- read.csv("summary_5mice/Figure3/clone_65thresh_4cell_352.csv")



#data_clones_by_mouse_TRA_TRB$X <- NULL
#data_clones_by_mouse_TRA_TRB$X.2 <- NULL

TCMP_100percent_TSNE <- data_clones_by_mouse_TRA_TRB[which(data_clones_by_mouse_TRA_TRB$Percent_0_Tcmp == 100), ] # 27

## change M6 to M5

TCMP_100percent_TSNE$mice <- lapply(
  X = TCMP_100percent_TSNE$mice,
  FUN = function(x) gsub("M6", "M5", x)
)

sapply(TCMP_100percent_TSNE, class) ## check the class of each columns

unlist(TCMP_100percent_TSNE)

TCMP_100percent_TSNE %>% as.matrix()

TCMP_100percent_TSNE <- apply(TCMP_100percent_TSNE, 2, as.character) # do this to get rid of error  unimplemented type 'list' in 'EncodeElement'

colnames(TCMP_100percent_TSNE)

#TCMP_100percentcells_TSNE <- TCMP_100percent_TSNE %>% select(mice, combined_cdr3_nt, Freq, Percent_0_Tcmp, Percent_1_Ly6cHi_Th1, Percent_2_Pre_TFH, Percent_3_Lag3hi_Th1, Percent_4_TFH1, Percent_5_GC_TFH2, Percent_Monocle_Tcmp, Percent_Monocle_TFH, Percent_Monocle_Th1,  Clone_number )

write.csv(TCMP_100percent_TSNE, file = "summary_5mice/Figure3/Tcmp_100percent_TSNE.csv") 


# this time we won't be making any Monocle tree trajectory plot to show TCMP sp clones in TSNE that already made progression
# towards either Th1 or TFH fates; In stead we will show this table of 27 clones that has 100% cells from TCMP in TSNE but Monocle TCMP % shows their progression towards other subsets


```


```{r highlight cells on either UMAP for figure 3A or for Monocle plots with Motif sp significant cell sp clones}

# Use clone3Lin6DF_65 to highlight following clones using Acute_6_mice_edit4_2cell DF
# multifated clones 1,2,3
# Th1 clones 67, 161, 109, 118
# TFH clones 42, 68, 101


immune.combined_edit2_2cell <- readRDS("summary_5M_CCA_V2/immune.combined_edit2_2cell.RDS")
data_clones_by_mouse_TRA_TRB <- read.csv("summary_5mice/data_clones_by_mouse_TRA_TRB_5mice.csv")



png(file = "summary_5mice/Figure3/Multifate_clone1.png", width = 8, height = 6, units = "in", res = 300)
TSNEPlot(
  immune.combined_edit2_2cell,
  ncol = 2,
  cells.highlight = rownames(immune.combined_edit2_2cell@meta.data)[
    which(immune.combined_edit2_2cell@meta.data$combined_cdr3_nt == data_clones_by_mouse_TRA_TRB$combined_cdr3_nt[1])], # Change clone number here
  cols.highlight = "darkblue", # Change color here
  pt.size = 0.5,
  sizes.highlight = 1.75, colors.use = c("grey", "darkblue")) #+
  #NoLegend()
dev.off()

png(file = "summary_5mice/Figure3/Multifate_clone2.png", width = 8, height = 6, units = "in", res = 300)
TSNEPlot(
  immune.combined_edit2_2cell,
  ncol = 2,
  cells.highlight = rownames(immune.combined_edit2_2cell@meta.data)[
    which(immune.combined_edit2_2cell@meta.data$combined_cdr3_nt == data_clones_by_mouse_TRA_TRB$combined_cdr3_nt[2])], # Change clone number here
  cols.highlight = "darkblue", # Change color here
  pt.size = 0.5,
  sizes.highlight = 1.75, colors.use = c("grey", "darkblue")) #+
  #NoLegend()
dev.off()

png(file = "summary_5mice/Figure3/Multifate_clone3.png", width = 8, height = 6, units = "in", res = 300)
TSNEPlot(
  immune.combined_edit2_2cell,
  ncol = 2,
  cells.highlight = rownames(immune.combined_edit2_2cell@meta.data)[
    which(immune.combined_edit2_2cell@meta.data$combined_cdr3_nt == data_clones_by_mouse_TRA_TRB$combined_cdr3_nt[3])], # Change clone number here
  cols.highlight = "darkblue", # Change color here
  pt.size = 0.5,
  sizes.highlight = 1.75, colors.use = c("grey", "darkblue")) #+
  #NoLegend()
dev.off()

## TFH clones  68, 120, 144  ###################################

png(file = "summary_5mice/Figure3/TFH_clone68.png", width = 8, height = 6, units = "in", res = 300)
TSNEPlot(
  immune.combined_edit2_2cell,
  ncol = 2,
  cells.highlight = rownames(immune.combined_edit2_2cell@meta.data)[
    which(immune.combined_edit2_2cell@meta.data$combined_cdr3_nt == data_clones_by_mouse_TRA_TRB$combined_cdr3_nt[110])], # Change clone number here
  cols.highlight = "darkgreen", # Change color here
  pt.size = 0.5,
  sizes.highlight = 3.0, colors.use = c("grey", "darkgreen")) #+
  #NoLegend()
dev.off()

png(file = "summary_5mice/Figure3/TFH_clone120.png", width = 8, height = 6, units = "in", res = 300)
TSNEPlot(
  immune.combined_edit2_2cell,
  ncol = 2,
  cells.highlight = rownames(immune.combined_edit2_2cell@meta.data)[
    which(immune.combined_edit2_2cell@meta.data$combined_cdr3_nt == data_clones_by_mouse_TRA_TRB$combined_cdr3_nt[120])], # Change clone number here
  cols.highlight = "darkgreen", # Change color here
  pt.size = 0.5,
  sizes.highlight = 3.0, colors.use = c("grey", "darkgreen")) #+
  #NoLegend()
dev.off()

png(file = "summary_5mice/Figure3/TFH_clone144.png", width = 8, height = 6, units = "in", res = 300)
TSNEPlot(
  immune.combined_edit2_2cell,
  ncol = 2,
  cells.highlight = rownames(immune.combined_edit2_2cell@meta.data)[
    which(immune.combined_edit2_2cell@meta.data$combined_cdr3_nt == data_clones_by_mouse_TRA_TRB$combined_cdr3_nt[144])], # Change clone number here
  cols.highlight = "darkgreen", # Change color here
  pt.size = 0.5,
  sizes.highlight = 3.0, colors.use = c("grey", "darkgreen")) #+
  #NoLegend()
dev.off()


## Th1 clones 67, 109, 118, 161 ###################################

png(file = "summary_5mice/Figure3/Th1_clone67.png", width = 8, height = 6, units = "in", res = 300)
TSNEPlot(
  immune.combined_edit2_2cell,
  ncol = 2,
  cells.highlight = rownames(immune.combined_edit2_2cell@meta.data)[
    which(immune.combined_edit2_2cell@meta.data$combined_cdr3_nt == data_clones_by_mouse_TRA_TRB$combined_cdr3_nt[67])], # Change clone number here
  cols.highlight = "darkred", # Change color here
  pt.size = 0.5,
  sizes.highlight = 3.0, colors.use = c("grey", "darkred")) #+
  #NoLegend()
dev.off()

png(file = "summary_5mice/Figure3/Th1_clone118.png", width = 8, height = 6, units = "in", res = 300)
TSNEPlot(
  immune.combined_edit2_2cell,
  ncol = 2,
  cells.highlight = rownames(immune.combined_edit2_2cell@meta.data)[
    which(immune.combined_edit2_2cell@meta.data$combined_cdr3_nt == data_clones_by_mouse_TRA_TRB$combined_cdr3_nt[118])], # Change clone number here
  cols.highlight = "darkred", # Change color here
  pt.size = 0.5,
  sizes.highlight = 3.0, colors.use = c("grey", "darkred")) #+
  #NoLegend()
dev.off()

png(file = "summary_5mice/Figure3/Th1_clone161.png", width = 8, height = 6, units = "in", res = 300)
TSNEPlot(
  immune.combined_edit2_2cell,
  ncol = 2,
  cells.highlight = rownames(immune.combined_edit2_2cell@meta.data)[
    which(immune.combined_edit2_2cell@meta.data$combined_cdr3_nt == data_clones_by_mouse_TRA_TRB$combined_cdr3_nt[161])], # Change clone number here
  cols.highlight = "darkred", # Change color here
  pt.size = 0.5,
  sizes.highlight = 3.0, colors.use = c("grey", "darkred")) #+
  #NoLegend()
dev.off()


```

```{r make a DF for Treg clone infromation}

immune.combined_edit2_2cell <- readRDS("summary_5M_CCA_V2/immune.combined_edit2_2cell.RDS")

table(immune.combined_edit2_2cell@meta.data$mice)

#M1   M2   M3   M4   M6 
#1049 1031 2516 3427 2469

table(immune.combined_edit2_2cell@meta.data$cluster_ident) 
#  0_Tcmp 1_Ly6cHi_Th1    2_Pre_TFH 3_Lag3hi_Th1       4_TFH1    5_GC_TFH2       6_Treg 
# 2379        1880         1967         1806           1370         1047           43

# total 10492
# Th1 3686
# 

table(immune.combined_edit2_2cell@meta.data$mice, immune.combined_edit2_2cell@meta.data$cluster_ident) %>% prop.table(margin = 2) * 100

 #       0_Tcmp 1_Ly6cHi_Th1 2_Pre_TFH 3_Lag3hi_Th1    4_TFH1 5_GC_TFH2    6_Treg
#  M1 10.298445    13.457447  7.625826     9.025471  9.489051  9.646609 16.279070
#  M2 10.802858    12.021277  9.049314     7.918051  9.197080  8.978032 16.279070
#  M3 19.125683    33.085106 19.115404    28.959025 21.970803 22.063037 18.604651
 # M4 38.335435    23.989362 31.875953    30.620155 32.335766 40.783190 32.558140
#  M6 21.437579    17.446809 32.333503    23.477298 27.007299 18.529131 16.279070


colnames(immune.combined_edit2_2cell@meta.data)



# Make new DF that includes mouse identity
data_clones_by_mouse <- immune.combined_edit2_2cell@meta.data %>% select(combined_cdr3_nt, mice) %>% table() %>% as.data.frame()


# table() double counted some cells but left the counts as 0; remove these
data_clones_by_mouse <- data_clones_by_mouse[which(data_clones_by_mouse$Freq > 0), ]

data_clones_by_mouse <- data_clones_by_mouse[which(data_clones_by_mouse$Freq > 1), ]
# Order in decreasing order
data_clones_by_mouse <- data_clones_by_mouse[order(data_clones_by_mouse$Freq, decreasing = T), ]



length(unique(data_clones_by_mouse$combined_cdr3_nt)) # 682 clones, but showing 686 clones with 4 overlapped clones

# Convert to wide format data
data_clones_by_mouse_wide <- data_clones_by_mouse %>% spread(mice, Freq, fill = 0)

# Check overlap
data_clones_by_mouse_wide_binary <- data_clones_by_mouse_wide
for (i in 2:ncol(data_clones_by_mouse_wide_binary)) {
  data_clones_by_mouse_wide_binary[, i][which(data_clones_by_mouse_wide_binary[, i] > 0)] <- 1
}

which(rowSums(data_clones_by_mouse_wide_binary[, -1]) > 1) %>% length() # 4
data_clones_by_mouse_wide_binary[which(rowSums(data_clones_by_mouse_wide_binary[, -1]) > 1), ]
# 4 overlaps with both chains

#                                                                              combined_cdr3_nt M1 M2 M3 M4 M6
#136          TRA:TGTGCAGCAAGCATCTATGCCCAGGGATTAACCTTC_TRB:TGTGCCAGCGGTGATAACCAAGACACCCAGTACTTT  0  0  0  1  1
#155    TRA:TGTGCAGCAAGCCCCAATACCAACAAAGTCGTCTTT_TRB:TGTGCCAGCGGTGATTGGGGGGTCCAAGACACCCAGTACTTT  0  0  0  1  1
#188 TRA:TGTGCAGCAAGTAATAACTATGCCCAGGGATTAACCTTC_TRB:TGTGCCAGCAGTGACAGGGGGCCCAACGAAAGATTATTTTTC  1  0  1  0  0
#273          TRA:TGTGCAGCAAGTGGGAACAACAATGCCCCACGATTT_TRB:TGTGCCAGCAGTGACCATCAAGACACCCAGTACTTT  0  0  1  1  0


# Subset the first data frame to only include clones with both TRA and TRB
rm(data_clones_by_mouse_TRA_TRB)
data_clones_by_mouse_TRA_TRB <- data_clones_by_mouse[grep(pattern = "_", x = data_clones_by_mouse$combined_cdr3_nt), ]

length(unique(immune.combined_edit2_2cell@meta.data$combined_cdr3_nt)) # 682
length(unique(data_clones_by_mouse_TRA_TRB$combined_cdr3_nt)) # 682 


table(data_clones_by_mouse_TRA_TRB$mice)
# M1  M2  M3  M4  M6 
#134 140 102 160 150 



### trying this cuase now the idents have changed cluster names so no need to have paste0 option
data_clones_by_mouse_TRA_TRB[, levels(immune.combined_edit2_2cell@ident)] <- 0
for (i in 1:nrow(immune.combined_edit2_2cell@meta.data)) {
  if (grepl(pattern = "_", x = immune.combined_edit2_2cell@meta.data$combined_cdr3_nt[i])) {
    data_clones_by_mouse_TRA_TRB [
      which(data_clones_by_mouse_TRA_TRB$combined_cdr3_nt == immune.combined_edit2_2cell@meta.data$combined_cdr3_nt[i] & data_clones_by_mouse_TRA_TRB$mice == immune.combined_edit2_2cell@meta.data$mice[i]),
      as.character(immune.combined_edit2_2cell@ident[i])] <-
      data_clones_by_mouse_TRA_TRB [
      which(data_clones_by_mouse_TRA_TRB$combined_cdr3_nt == immune.combined_edit2_2cell@meta.data$combined_cdr3_nt[i] & data_clones_by_mouse_TRA_TRB$mice == immune.combined_edit2_2cell@meta.data$mice[i]),
       as.character(immune.combined_edit2_2cell@ident[i])] + 1
  }
}

 data_clones_by_mouse_TRA_TRB_Treg <-  data_clones_by_mouse_TRA_TRB[data_clones_by_mouse_TRA_TRB$`6_Treg` > 0, ]

write.csv(data_clones_by_mouse_TRA_TRB_Treg, file = "summary_5mice/Figure3/Treg_clones_info.csv")


```

```{r check clonal preference for clusters in tSNE based on %}

rm(list=ls())
data_clones_by_mouse_TRA_TRB <- read.csv("summary_5mice/data_clones_by_mouse_TRA_TRB_5mice.csv")

table(data_clones_by_mouse_TRA_TRB$mice)


# M1  M2  M3  M4  M6 
#131 138 100 155 149 

table(data_clones_by_mouse_TRA_TRB$Freq, data_clones_by_mouse_TRA_TRB$mice)

data_clones_by_mouse_TRA_TRB$Freq %>% sum() # 10447

## change M6 to M5 in this DF

data_clones_by_mouse_TRA_TRB$mice <- lapply(
  X= data_clones_by_mouse_TRA_TRB$mice, 
  FUN = function(x) gsub("M6", "M5", x)
)

data_clones_by_mouse_TRA_TRB$CDR3nt_Ident <- lapply(
  X= data_clones_by_mouse_TRA_TRB$CDR3nt_Ident, 
  FUN = function(x) gsub("M6", "M5", x)
)

data_clones_by_mouse_TRA_TRB$X.2 <- NULL
data_clones_by_mouse_TRA_TRB$X <- NULL

data_clones_by_mouse_TRA_TRB <- apply(data_clones_by_mouse_TRA_TRB, 2, as.character)
write.csv(data_clones_by_mouse_TRA_TRB, file = "summary_5mice/Figure3/DFs/data_clones_by_mouse_TRA_TRB_5mice.csv")

#TCMP_100percent_TSNE <- apply(TCMP_100percent_TSNE, 2, as.character) # do this to get rid of error  unimplemented type 'list' in 'EncodeElement'

#colnames(TCMP_100percent_TSNE)

#TCMP_100percentcells_TSNE <- TCMP_100percent_TSNE %>% select(mice, combined_cdr3_nt, Freq, Percent_0_Tcmp, Percent_1_Ly6cHi_Th1, Percent_2_Pre_TFH, Percent_3_Lag3hi_Th1, Percent_4_TFH1, Percent_5_GC_TFH2, Percent_Monocle_Tcmp, Percent_Monocle_TFH, Percent_Monocle_Th1,  Clone_number )

#write.csv(TCMP_100percent_TSNE, file = "summary_5mice/Figure3/Tcmp_100percent_TSNE.csv") 

Tcmp_DF_pref <- data_clones_by_mouse_TRA_TRB[which(data_clones_by_mouse_TRA_TRB$Percent_0_Tcmp == 100), ]  # 27
#write.csv(TCMP_100percent_TSNE, file = "summary_5mice/Figure3/Tcmp_100percent_TSNE.csv") 
Tcmp_DF_No_pref <- data_clones_by_mouse_TRA_TRB[which(data_clones_by_mouse_TRA_TRB$Percent_0_Tcmp == 0), ]  # 271
Tcmp_DF_No_pref  <- apply(Tcmp_DF_No_pref, 2, as.character)
write.csv(Tcmp_DF_No_pref, file = "summary_5mice/Figure3/DFs/Tcmp_DF_No_pref.csv") 


Ly6cHi_DF_preferred <- data_clones_by_mouse_TRA_TRB[which(data_clones_by_mouse_TRA_TRB$Percent_1_Ly6cHi_Th1 == 100), ]  # 24
Ly6cHi_DF_preferred  <- apply(Ly6cHi_DF_preferred, 2, as.character)
write.csv(Ly6cHi_DF_preferred, file = "summary_5mice/Figure3/DFs/Ly6cHi_DF_preferred.csv")

Ly6cHi_DF_No_pref <- data_clones_by_mouse_TRA_TRB[which(data_clones_by_mouse_TRA_TRB$Percent_1_Ly6cHi_Th1 == 0), ]  # 306
Ly6cHi_DF_No_pref <- apply(Ly6cHi_DF_No_pref, 2, as.character)
write.csv(Ly6cHi_DF_No_pref, file = "summary_5mice/Figure3/DFs/Ly6cHi_DF_No_pref.csv")

Pre_TFH_DF_preferred <- data_clones_by_mouse_TRA_TRB[which(data_clones_by_mouse_TRA_TRB$Percent_2_Pre_TFH == 100), ]  #5
Pre_TFH_DF_preferred   <- apply(Pre_TFH_DF_preferred , 2, as.character)
write.csv(Pre_TFH_DF_preferred, file = "summary_5mice/Figure3/DFs/Pre_TFH_DF_preferred.csv")

Pre_TFH_DF_No_pref <- data_clones_by_mouse_TRA_TRB[which(data_clones_by_mouse_TRA_TRB$Percent_2_Pre_TFH == 0), ]  # 322
Pre_TFH_DF_No_pref <- apply(Pre_TFH_DF_No_pref, 2, as.character)
write.csv(Pre_TFH_DF_No_pref, file = "summary_5mice/Figure3/DFs/Pre_TFH_DF_No_pref.csv")


Lag3hi_DF_preferred <- data_clones_by_mouse_TRA_TRB[which(data_clones_by_mouse_TRA_TRB$Percent_3_Lag3hi_Th1 == 100), ]  # 17
Lag3hi_DF_preferred <- apply(Lag3hi_DF_preferred, 2, as.character)
write.csv(Lag3hi_DF_preferred, file = "summary_5mice/Figure3/DFs/Lag3hi_DF_preferred.csv")

Lag3hi_DF_No_pref <- data_clones_by_mouse_TRA_TRB[which(data_clones_by_mouse_TRA_TRB$Percent_3_Lag3hi_Th1 == 0), ]  # 319
Lag3hi_DF_No_pref <- apply(Lag3hi_DF_No_pref, 2, as.character)
write.csv(Lag3hi_DF_No_pref, file = "summary_5mice/Figure3/DFs/Lag3hi_DF_No_pref.csv")


TFH1_DF_preferred <- data_clones_by_mouse_TRA_TRB[which(data_clones_by_mouse_TRA_TRB$Percent_4_TFH1 == 100), ]  #6
TFH1_DF_preferred <- apply(TFH1_DF_preferred, 2, as.character)
write.csv(TFH1_DF_preferred, file = "summary_5mice/Figure3/DFs/TFH1_DF_preferred.csv")


TFH1_DF_No_pref <- data_clones_by_mouse_TRA_TRB[which(data_clones_by_mouse_TRA_TRB$Percent_4_TFH1 == 0), ]  # 346
TFH1_DF_No_pref <- apply(TFH1_DF_No_pref, 2, as.character)
write.csv(TFH1_DF_No_pref, file = "summary_5mice/Figure3/DFs/TFH1_DF_No_pref.csv")


GC_TFH2_DF_preferred <- data_clones_by_mouse_TRA_TRB[which(data_clones_by_mouse_TRA_TRB$Percent_5_GC_TFH2 == 100), ]  #4
GC_TFH2_DF_preferred <- apply(GC_TFH2_DF_preferred, 2, as.character)
write.csv(GC_TFH2_DF_preferred, file = "summary_5mice/Figure3/DFs/GC_TFH2_DF_preferred.csv")

GC_TFH2_DF_No_pref <- data_clones_by_mouse_TRA_TRB[which(data_clones_by_mouse_TRA_TRB$Percent_5_GC_TFH2 == 0), ]  # 430
GC_TFH2_DF_No_pref <- apply(GC_TFH2_DF_No_pref, 2, as.character)
write.csv(GC_TFH2_DF_No_pref, file = "summary_5mice/Figure3/DFs/GC_TFH2_DF_No_pref.csv")

### change M6 to M5

tableS4 <- read.csv("TableS4.csv")

tableS4$mice <- lapply(
  X = tableS4$mice,
  FUN = function(x) gsub("M6", "M5", x)
)
tableS4$X <- NULL

tableS4$CDR3nt_Ident <- lapply(
  X = tableS4$CDR3nt_Ident,
  FUN = function(x) gsub("M6", "M5", x)
)


tableS4  <- apply(tableS4, 2, as.character)


write.csv(tableS4, file = "summary_5mice/tableS4.csv")

```

